---
title: "Phyloseq_2021_2019_MiSeq"
author: "Sarah Tucker"
date: "9/27/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown


Okay more info on how to reclassify using the PR2
###https://www.protocols.io/private/C0F9404AB3DAEC96683F142351CEF59C?step=13




```{r}
rm(list=ls())

# load libraries
library("ggplot2") 
library("phyloseq") 
library(ape)
library("knitr")
#library(dada2)
library(methods)
library(plotly)
library(tidyverse)
library(ggpubr)
library(compositions)
library(reshape2)
library(fpc)
library(pracma)
library(cluster)
library(ade4)
library(vegan)
library(DESeq2)
library(RColorBrewer)
library(randomcoloR)
library(purrr)
library(lomb)
library(lubridate)
library(tidyverse)
library(mgcv)



###this is from the R tutorial document"https://github.com/joey711/phyloseq/files/1795523/QIIME2_to_Phyloseq.pdf"


setwd("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq") 

getwd()
```




```{r}

# read in otu table
otu_table_FCM = read.csv("FCM_OTU_TABLE.csv", sep=",", row.names=1) 
otu_table_FCM = as.matrix(otu_table_FCM)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

#note this was the original taxonomy, before using Pr2
taxonomy_FCM = read.csv("FCM_Taxa_Table.csv", sep=",", row.names=1) 


taxonomy_FCM = as.matrix(taxonomy_FCM)

#poop=subset(taxonomy_KBytNERR, Order=="D_3__Chloroplast")

#read in metadata
##note that there is not seq data for all metadata samples 
###when analyzing metadata strictly, need to edit

no_dup_metadata_FCM=read.csv("v2_nodups_woAprFP_metadata_2019_2021_11252021.csv", header=TRUE, fill=TRUE)

rownames(no_dup_metadata_FCM) <- no_dup_metadata_FCM$SampleID
no_dup_metadata_FCM$SampleID
META_no_dup_FCM = sample_data(no_dup_metadata_FCM)
nsamples(META_no_dup_FCM)

# import as phyloseq objects
OTU_FCM = otu_table(otu_table_FCM, taxa_are_rows = TRUE) 
TAX_FCM = tax_table(taxonomy_FCM)


physeq_FCM = phyloseq(OTU_FCM, TAX_FCM, META_no_dup_FCM)

ntaxa(physeq_FCM)


```




```{r, echo=FALSE, warning=FALSE, message=FALSE}

###now that you are in the right format you can bring things in as phyloseq objects

# read in otu table
otu_table_KBytNERR = read.csv("2021_2019_MiSeq_feature-table_Sep272021.csv", sep=",", row.names=1) 
otu_table_KBytNERR = as.matrix(otu_table_KBytNERR)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

#note this was the original taxonomy, before using Pr2
taxonomy_KBytNERR = read.csv("2021_2019_MiSeq_taxonomy_Sep272021.csv", sep=",", row.names=1) 


taxonomy_KBytNERR = as.matrix(taxonomy_KBytNERR)

#poop=subset(taxonomy_KBytNERR, Order=="D_3__Chloroplast")

#read in metadata
##note that there is not seq data for all metadata samples 
###when analyzing metadata strictly, need to edit
metadata_KBytNERR=read.csv("metadata_2019_2021_10222021.csv", header=TRUE, fill=TRUE)

rownames(metadata_KBytNERR) <- metadata_KBytNERR$SampleID
metadata_KBytNERR$SampleID
META_KBytNERR = sample_data(metadata_KBytNERR)

#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# import as phyloseq objects
OTU_KBytNERR = otu_table(otu_table_KBytNERR, taxa_are_rows = TRUE) 
TAX_KBytNERR = tax_table(taxonomy_KBytNERR)


physeq_KBytNERR = phyloseq(OTU_KBytNERR, TAX_KBytNERR, phy_treeKBytNERR)

physeq_KBytNERR=merge_phyloseq(physeq_KBytNERR, META_KBytNERR)
ntaxa(physeq_KBytNERR)
#4462

write.csv(tax_table(physeq_KBytNERR), "tax_table_physeq_KByTNERR.csv")

```




######
CLASSIFYING CHLOROPLAST & EUKARYOTIC SEQUENCES
######


```{r}
test=subset_taxa(physeq_KBytNERR, Domain=="D_0__Eukaryota")
ntaxa(test)

head(tax_table(physeq_KBytNERR))

chloro_euks_unassigned=subset_taxa(physeq_KBytNERR, Order=="D_3__Chloroplast" | Domain=="D_0__Eukaryota"| Domain=="Unassigned")
ntaxa(chloro_euks_unassigned)
test=as.data.frame(tax_table(chloro_euks_unassigned))
tail(tax_table(chloro_euks_unassigned))

write.csv(taxa_names(chloro_euks_unassigned), "chloro_euks_unassigned.csv")

```


```{bash}
cd /Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq
grep -A 1 -f <(cut -d ',' -f 2 chloro_euks_unassigned.csv | tr -d "\"") 2021_2019_merged-dna-sequences.fasta | sed '/^--$/d' >> chloro_euk_unassigned_ASV.fasta



###get the seqeunces into tab form
awk 'BEGIN{RS=">"}{print "#"$1"\t"$2;}' chloro_euk_unassigned_ASV.fasta | tail -n+2 > chloro_euk_unassigned_ASV_reformat.txt

```


##The threshold of 60% is recommended at the default confidence threshold.


```{r}
#if (!requireNamespace("BiocManager", quietly=TRUE))
#   install.packages("BiocManager")
#BiocManager::install("DECIPHER")


#install.packages(devtools)
#devtools::install_github("pr2database/pr2database")

library(DECIPHER)
library(pr2database)


# Read training set 
trainingSet <- readRDS("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq/pr2_version_4.14.0_SSU.decipher.trained.rds")

# Read sequences to assign
seq <- readDNAStringSet("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq/chloro_euk_unassigned_ASV.fasta")

# Get the taxonomy from the training set

  ids <- IdTaxa(seq,
                trainingSet,
                type="extended",
                strand="top",
                threshold=60)
#ids threshold is 60 now, before did not set a threshold
  
  #n_seq <- length(ids)
  #df_rows <- list()
taxo_levels <- c("kingdom", "supergroup", "division", "class", "order", "family", "genus", "species")

  
#for(i in 1:n_seq){
 # seq_name <- names(ids[i])
 # taxonomy=paste(x$taxon,collapse=";")
 # confidence <- ids[[i]]$confidence
 # df_rows[[i]] = data.frame(seq_name, taxonomy, confidence, taxo_level=c("Root", taxo_levels))
#}


#df <- reduce(df_rows, bind_rows) %>% 
 # filter(taxo_level !="Root") %>% 
 # pivot_wider(names_from = taxo_level, values_from = c(taxonomy, confidence))

###Code I used 
output <- sapply(ids,
function (id) {
seq_name <- names(ids)
paste(id$taxon,
";",
round(id$confidence, digits=1),
sep="",
collapse="; ")
})
tail(output)

idk=as.data.frame(output)
write.csv(idk, "pr2_assigment_chloro_euk_unassigned.csv")

pr2_taxonomy=read.csv("pr2_assigment_chloro_euk_unassigned_edited.csv")


silva_taxonomy=read.csv("tax_table_physeq_KByTNERR_edited.csv")


combined_taxonomy=left_join(silva_taxonomy, pr2_taxonomy, by="OTUID")

write.csv(combined_taxonomy, "all_taxonomy_pr2_euk_chloro_unassigned.csv")


  
```






######
BRINGING IN REASSIGNMENT OF TAXONOMY OF CHLOROPLAST SEQUENCES
######

```{r}
taxonomy_Phyto3 = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_v4.csv", sep=",") 

bleh=read.csv("distinct_phyto_kmeans3.csv")

tog=left_join(taxonomy_Phyto3, bleh, by="ASV")

#write.csv(tog, "all_taxonomy_pr2_euk_chloro_unassigned_correct_v5.csv")

```




```{r}

taxonomy_Phyto = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_v5.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)
ntaxa(TAX_Phyto)

physeq_Dav = phyloseq(OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav=merge_phyloseq(physeq_Dav, META_KBytNERR)
ntaxa(physeq_Dav)
#4462


##Removing unwanted sequences

table(tax_table(physeq_Dav)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_clean <- subset_taxa(physeq_Dav, !is.na(Domain) & !Domain %in% c("", "unclassified_Root"))
table(tax_table(physeq_Dav_clean)[, "Domain"], exclude = NULL)

##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_clean)
ntaxa(physeq_Dav_clean)
#4343

#remove all of the uncharacterized phylum
#table(tax_table(physeq_Dav_clean)[, "Phylum"], exclude = NULL)

#physeq_Dav_clean<- subset_taxa(physeq_Dav_clean, !is.na(Phylum) & !Phylum %in% c("", "Unassigned"))
#nsamples(physeq_Dav_clean)
#ntaxa(physeq_Dav_clean)




```



######
CLEANING UP POORLY CLASSIFIED SAMPLES
######

######
ASSESSING AND REMOVING DUPLICATE SAMPLES
######

```{r}

Dav.real.abun <- transform_sample_counts(physeq_Dav_clean, function(OTU) OTU/sum(OTU))
ntaxa(Dav.real.abun)


test=subset_samples(Dav.real.abun, Duplicate=="Yes")
nsamples(test)

melt=psmelt(test)
melt$Date1 <- as.Date(melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=melt, aes(x=SampleID, y=Abundance, fill=Order))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Replicates of MiSeq") +theme(legend.position = "none")
ggsave("Replicated_Miseq.pdf")
#+ scale_fill_manual(values=cbPalette8 )


AR=subset_samples(Dav.real.abun, Site=="AR")
nsamples(AR)

AR_melt=psmelt(AR)
#ALLSAR11_melt$Date1 <- as.Date(ALLSAR11_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=AR_melt, aes(x=SampleID, y=Abundance, fill=Order))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("AR MiSeq") +theme(legend.position = "none")
ggsave("AR.pdf")


NB=subset_samples(Dav.real.abun, Site=="NB")
nsamples(NB)

NB_melt=psmelt(NB)
#ALLSAR11_melt$Date1 <- as.Date(ALLSAR11_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=NB_melt, aes(x=SampleID, y=Abundance, fill=Order))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("NB of MiSeq") +theme(legend.position = "none")
ggsave("NB.pdf")




SR2=subset_samples(Dav.real.abun, Site=="SR2")
nsamples(SR2)

SR2_melt=psmelt(SR2)
#ALLSAR11_melt$Date1 <- as.Date(ALLSAR11_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=SR2_melt, aes(x=SampleID, y=Abundance, fill=Order))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("NB of MiSeq") +theme(legend.position = "none")
ggsave("SR2.pdf")


```



####
Removed duplicate samples from phyloseq object 
Also remove KAHO & WAI2 4.16.2021 sample to keep just quarterly sampling events, this is because poor sequencing quality 
####


```{r}






no_dup_metadata_KBytNERR=read.csv("v4_nodups_wo_AprFP_2019_2021_03142022.csv", header=TRUE, fill=TRUE)

rownames(no_dup_metadata_KBytNERR) <- no_dup_metadata_KBytNERR$SampleID
no_dup_metadata_KBytNERR$SampleID
META_no_dup_KBytNERR = sample_data(no_dup_metadata_KBytNERR)
nsamples(META_no_dup_KBytNERR)

# read in otu table
no_dup_otu_table_KBytNERR = read.csv("no_duplicates_2021_2019_MiSeq_feature-table_Sep282021.csv", sep=",", row.names=1) 
no_dup_otu_table_KBytNERR = as.matrix(no_dup_otu_table_KBytNERR)

# import as phyloseq objects
no_dup_OTU_KBytNERR = otu_table(no_dup_otu_table_KBytNERR, taxa_are_rows = TRUE) 


physeq_Dav_no_dup = phyloseq(no_dup_OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav_no_dup=merge_phyloseq(physeq_Dav_no_dup, META_no_dup_KBytNERR)
ntaxa(physeq_Dav_no_dup)
#4462


##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean <- subset_taxa(physeq_Dav_no_dup, !is.na(Domain) & !Domain %in% c("", "unclassified_Root"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean)
ntaxa(physeq_Dav_no_dup_clean)
#4343
table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)


#check to see all of the uncharacterized phylum
table(tax_table(physeq_Dav_no_dup_clean)[, "Phylum"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))

###make sure the right otut able is in here rel abun without chloro, cell # 
Euks_18S = subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota")
ntaxa(Euks_18S)
Euks_18S_rel.abun= subset_taxa(rel.abun,Domain == "Eukaryota")


ntaxa(Euks_18S_rel.abun)


library(randomcoloR)
n <- 23
palette2 <- distinctColorPalette(n)

Euk_melt=psmelt(Euks_18S_rel.abun)
Euk_melt$Date1 <- as.Date(Euk_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Euk_melt, aes(x=Sampling.Order, y=Abundance, fill=Phylum))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Euk 18S") +theme(legend.position = "right")+ scale_fill_manual(values=palette2)
ggsave("Euk18S.pdf", width=10, height=10)



Archaea_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Archaea")
Arc_melt=psmelt(Archaea_rel.abun)
Arc_melt$Date1 <- as.Date(Arc_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Arc_melt, aes(x=Sampling.Order, y=Abundance, fill=Genus))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Euk 18S") +theme(legend.position = "right")+ scale_fill_manual(values=palette2)
ggsave("arc.pdf", width=10, height=10)



Unassigned_Phylum=subset_taxa(rel.abun, Phylum=="")
ntaxa(Unassigned_Phylum)


Unassigned_Phylum_melt=psmelt(Unassigned_Phylum)
Unassigned_Phylum_melt$Date1 <- as.Date(Unassigned_Phylum_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Unassigned_Phylum_melt, aes(x=Date1, y=Abundance, fill=Phylum))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Unassigned_Phylum Plastid& 16S") +theme(legend.position = "right")
ggsave("Unassigned_Phylum_Plastid and 16S.pdf")




#I will not remove taxa that cannot be assigned at the phylum level
#physeq_Dav_no_dup_clean<- subset_taxa(physeq_Dav_no_dup_clean, !is.na(Phylum) & !Phylum %in% c("", "Unassigned"))
#nsamples(physeq_Dav_no_dup_clean)
#ntaxa(physeq_Dav_no_dup_clean)
#4108

```

###evaluate if Euk 18S sequences can be used, appears that they are at such low abudnances that this does not need to be part of the story 

```{r}

all_melt=psmelt(rel.abun)

Supergroup4_sum=all_melt %>% dplyr::group_by(Supergroup4,SampleID,Site)%>%dplyr::summarise(sum=sum(Abundance)*100)


Supergroup4_means=Supergroup4_sum %>% dplyr::group_by(Supergroup4,Site)%>%dplyr::summarise(Mean=round(mean(sum),2), sd=round(sd(sum),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

f3=Supergroup4_means %>%
  ggplot( aes(x=Site, y=Mean, group=Supergroup4, color=Supergroup4)) +
    geom_line()+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))

f3
ggsave("line_major_groups.pdf")

Supergroup4_sum$Supergroup4

Supergroup4_means$Supergroup4=factor(Supergroup4_means$Supergroup4, levels=c("D_0__Archaea", "Eukaryota", "Eukaryota_plas", "Cyanobacteria","D_0__Bacteria"))

p <- ggplot(data=Supergroup4_means, aes(x=Site, y=Mean, fill=Supergroup4))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("All Taxa")

all_melt$Date1 <- as.Date(all_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=all_melt, aes(x=Date1, y=Abundance, fill=Supergroup4))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("All Taxa")  + scale_fill_manual(values=palette2)


```



```{r}
##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean_no18S <- subset_taxa(physeq_Dav_no_dup_clean, !is.na(Domain) & !Domain %in% c("", "Eukaryota"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean_no18S)
ntaxa(physeq_Dav_no_dup_clean_no18S)
#3987
table(tax_table(physeq_Dav_no_dup_clean_no18S)[, "Domain"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))
ntaxa(rel.abun)
#3987
nsamples(rel.abun)
#366

write.csv(otu_table(physeq_Dav_no_dup_clean_no18S), "Count_data_physeq_Dav_no_dup_clean_no18S.csv")
write.csv(tax_table(physeq_Dav_no_dup_clean_no18S), "Taxonomy_Count_data_physeq_Dav_no_dup_clean_no18S.csv")

All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")

write.csv(otu_table(All_phyto), "Count_data_All_phyto.csv")
write.csv(tax_table(All_phyto), "Taxonomy_Count_data_All_phyto.csv")


```



```{r}

#Bac_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Bac_16S= subset_taxa(physeq_Dav_no_dup_clean_no18S,Domain == "D_0__Bacteria")

rel2Bac_16S <- transform_sample_counts(Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


Just_Het_Bac_16S= subset_taxa(Bac_16S,Phylum != "D_1__Cyanobacteria")
ntaxa(Just_Het_Bac_16S)

rel2_Het_Bac_16S <- transform_sample_counts(Just_Het_Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 
ntaxa(rel2_Het_Bac_16S)



All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


All_phyto_Rel = subset_taxa(rel.abun,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto_Rel)
table(tax_table(All_phyto)[, "Phylum"], exclude = NULL)

```







######
EVALUATING SEQUENCING DEPTH
######



```{r}

#### note here we are still including 18S 
##########
############
#############


##not using cleaned, cleaned removes sequences based on our poor taxonomic assignments not based on well it was sequenced

sdt = data.frame(as(sample_data(physeq_Dav_no_dup), "data.frame"),
                 TotalReads = sample_sums(physeq_Dav_no_dup), keep.rownames = TRUE)
sdt

write.csv(sdt, "sdt.csv")

pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth +facet_wrap(~Season)

pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth +facet_wrap(~Environment)


pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth 


summary(sdt$TotalReads)

sd(sdt$TotalReads)

sdt2 <-sdt  %>% group_by(Enviro2)%>% summarise(average=mean(TotalReads), SD=sd(TotalReads))

sdt3 <-sdt  %>% group_by(Site)%>% summarise(average=mean(TotalReads), SD=sd(TotalReads))



write.csv(sdt3, "test.csv")

ggqqplot(sdt$TotalReads)


##sdt3=subset(sdt, SampleID != "NBJa0121" & SampleID != "CBJn0218")


library("multcomp")

sdt$Environment=as.factor(sdt$Environment)
sdt$Season=as.factor(sdt$Season)

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Environment, data = sdt)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Environment= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = sdt)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
coastal=subset(sdt, Environment=="Coastal")
transition=subset(sdt,Environment=="Estuary")
offshore=subset(sdt, Environment=="Offshore")

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = coastal)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = transition)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
    ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = offshore)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  ??multicomp
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Site, data = sdt)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Site= "Tukey"))
  summary(g, test = adjusted("holm"))
  
```


#######
nMDS of Phytoplankton Taxa and all Taxa 
#######


https://github.com/WeitzGroup/analyzing_microbiome_timeseries/blob/master/module_periodicity/periodicity_tutorial.r

https://github.com/shu251/RAIN_analysis_example





```{r}

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)
#3987
nsamples(rel.abun)
#366


write.csv(as.data.frame(tax_table(All_phyto)), "All_phyto_taxonomy.csv")

rr=as.data.frame(otu_table(All_phyto))
library(data.table)
rs=setDT(rr, keep.rownames = TRUE)[]

pp=as.data.frame(tax_table(All_phyto))
ps=setDT(pp, keep.rownames = TRUE)[]
tog=left_join(ps,rs, by="rn")

write.csv(tog, "OTU_Table_All_Phyto.csv")


GP.ord <- ordinate(All_phyto, "NMDS", "bray")
p2 = plot_ordination(All_phyto, GP.ord, type="samples", color="Environment", label="SampleID") 
p2
ggsave("All_phytoplankton_nmds.pdf")



GP.ord <- ordinate(All_phyto, "NMDS", "bray")
p2 = plot_ordination(All_phyto, GP.ord, type="samples", color="Clustering", label="SampleID") 
p2
ggsave("All_phytoplankton_nmds_clustering.pdf")



GP.ord <- ordinate(rel.abun, "NMDS", "bray")
p2 = plot_ordination(All_phyto, GP.ord, type="samples", color="Clustering", label="SampleID") 
p2
ggsave("All_taxa_nmds_clustering.pdf")



coastal_phyto=subset_samples(All_phyto, Enviro2=="Coastal")
GP.ord <- ordinate(coastal_phyto, "NMDS", "bray")
p2 = plot_ordination(All_phyto, GP.ord, type="samples", color="Enviro3", label="SampleID", title="Coastal Phytoplankton Taxa") 
p2
ggsave("Coastal_Phytoplankton_taxa_nMDS.pdf")



GP.ord <- ordinate(rel.abun, "NMDS", "bray")
p2 = plot_ordination(rel.abun, GP.ord, type="samples", color="Enviro3", label="SampleID", title="All Taxa") 
p2
ggsave("Coastal_taxa_nMDS.pdf")




#print(p2)
#p2 + geom_polygon(aes(fill=Enviro2)) + geom_point(size=5) + ggtitle("samples")

```




#######
Bargraphs over time and space and diff taxa
#######



```{r}

All_phyto
rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


Bacillariophyta= subset_taxa(All_phyto, Class == "Bacillariophyta")
rel2All_Bacillariophyta <- transform_sample_counts(Bacillariophyta, function(OTU) OTU/sum(OTU))



ntaxa(Bacillariophyta)
taxa_names(Bacillariophyta)


test=psmelt(rel2All_Bacillariophyta)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Family))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=cbPalette8)
ggsave("Bacillariophyta_100_Famile.pdf")




###SAR11
library(randomcoloR)
n <- 18
palette2 <- distinctColorPalette(n)


test=psmelt(SAR11)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

test$Sampling.Order=as.factor(test$Sampling.Order)
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=Genus))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  +ggtitle("SAR11 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "Dark2")  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black", fill = NA,size = 1))
ggsave("SAR11_byGenus.pdf")






Bacillariophyta= subset_taxa(rel2All_phyto, Class == "Bacillariophyta")
Bacillariophyta_melt=psmelt(Bacillariophyta)

Supergroup4_sum=Bacillariophyta_melt %>% dplyr::group_by(Curated,SampleID,Site)%>%dplyr::summarise(sum=sum(Abundance)*100)
Supergroup4_means

Supergroup4_means=Supergroup4_sum %>% dplyr::group_by(Curated,Site)%>%dplyr::summarise(Mean=round(mean(sum),2), sd=round(sd(sum),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

p <- ggplot(data=Supergroup4_means, aes(x=Site, y=Mean, fill=Curated))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Diatom Taxa Relative to all Phytoplankton")


ggsave("Average_Diatom_16S.pdf")






###make sure the right otut able is in here rel abun without chloro, cell # 
Euks_18S = subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota")
ntaxa(Euks_18S)
Euks_18S_rel.abun= subset_taxa(rel.abun,Domain == "Eukaryota")



Bacillariophyta= subset_taxa(Euks_18S, Class == "Bacillariophyta")
rel2All_Bacillariophyta_18S <- transform_sample_counts(Bacillariophyta, function(OTU) OTU/sum(OTU))



nr2=subset_samples(Euks_18S_rel.abun, Site=="NR2")
test=psmelt(nr2)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Order))
p  + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=palette2)

library(randomcoloR)
n <- 14
palette2 <- distinctColorPalette(n)


ntaxa(Bacillariophyta)
taxa_names(Bacillariophyta)


test=psmelt(rel2All_Bacillariophyta_18S)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=custom_final28)
ggsave("Bacillariophyta_100_18S.pdf")



rel2All_18S <- transform_sample_counts(Euks_18S, function(OTU) OTU/sum(OTU))
Diatoms_rel_18S= subset_taxa(rel2All_18S,Class == "Bacillariophyta")
Diatoms_rel_18S=subset_samples(Diatoms_rel_18S, SampleID != "NRJa0362")
nsamples(Diatoms_rel_18S)


Bacillariophyta_melt=psmelt(Diatoms_rel_18S)

Supergroup4_sum=Bacillariophyta_melt %>% dplyr::group_by(Curated,SampleID,Site)%>%dplyr::summarise(sum=sum(Abundance)*100)
Supergroup4_means

Supergroup4_means=Supergroup4_sum %>% dplyr::group_by(Curated,Site)%>%dplyr::summarise(Mean=round(mean(sum),2), sd=round(sd(sum),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

library(randomcoloR)
n <- 40
palette3 <- distinctColorPalette(n)




p <- ggplot(data=Supergroup4_means, aes(x=Site, y=Mean, fill=Curated))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Diatom Taxa Relative to all Euks")+ scale_fill_manual(values=palette3)


ggsave("Average_Diatom_16S.pdf")



```



```{r}



Syn= subset_taxa(rel.abun, Genus == "D_5__Synechococcus_CC9902")
ntaxa(Syn)

library(RColorBrewer)
display.brewer.all()


test=psmelt(Syn)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=ASV))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack")+theme(axis.text.x = element_text(angle = 90)) +ggtitle("") + scale_fill_manual(values=palette2) +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("Syn_by_ASV.pdf")



Pro= subset_taxa(rel.abun, Genus == "D_5__Prochlorococcus_MIT9313")
ntaxa(Syn)

library(RColorBrewer)
display.brewer.all()


test=psmelt(Pro)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=ASV))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack")+theme(axis.text.x = element_text(angle = 90)) +ggtitle("") + scale_fill_manual(values=palette2) +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("Pro_by_ASV.pdf")



```



#######
Evaluate Most abundant Orders and ASV
#######



```{r}
###make sure the right otu table is in here rel abun without chloro, cell # 
Abun_By_Order=tax_glom(rel.abun, "Order")

top15<- names(sort(taxa_sums(Abun_By_Order), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, Abun_By_Order)
tax_table(ps.top15)
plot_bar(ps.top15, fill="Order") + facet_wrap(~Site, scales="free_x")
ggsave("Top15byOrder_allTaxa_Barplots.pdf")

df_Order_total <- Abun_By_Order %>% psmelt() %>% group_by(OTU,Order) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 

df_Order_total$percent_total_mean=df_Order_total$mean*100
df_Order_total$percent_total_SD=df_Order_total$sd*100


test=df_Order_total[df_Order_total$percent_total_mean %in% tail(sort(df_Order_total$percent_total_mean),15),]
write.csv(test, "Mean_allsamples_Order.csv")


df_Order_enviro <- Abun_By_Order %>% psmelt() %>% group_by(OTU,Order,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


df_Order_enviro_mean=dcast(df_Order_enviro, OTU+Order ~ Enviro2, value.var ="mean")
df_Order_enviro_sd=dcast(df_Order_enviro , OTU+Order ~ Enviro2, value.var = "sd")

write.csv(df_Order_enviro_mean, "df_Order_enviro_mean.csv")
write.csv(df_Order_enviro_sd, "df_Order_enviro_sd.csv")


###horizontal figure


df_Order_enviro <- Abun_By_Order %>% psmelt() %>% group_by(OTU,Order,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


df_Order_enviro_top15=subset(df_Order_enviro, Order=="D_3__Synechococcales" | Order=="D_3__SAR11_clade"| Order=="D_3__Flavobacteriales" |  Order== "D_3__Rhodobacterales" | Order== "D_3__Puniceispirillales" | Order == "Mamiellales" | Order == "D_3__SAR86_clade" | Order == "Bacillariophyta_X" | Order =="D_3__Cellvibrionales" | Order == "D_3__Rhodospirillales"  | Order == "D_3__Actinomarinales" | Order == "Phaeocystales" | Order == "Cryptomonadales" | Order== "D_3__Oceanospirillales" | Order== "D_3__Rickettsiales")


df_Order_enviro_top15$Order <- factor(df_Order_enviro_top15$Order,levels=c("D_3__Synechococcales", "D_3__SAR11_clade", "D_3__Flavobacteriales","D_3__Rhodobacterales", "D_3__Puniceispirillales", "Mamiellales", "D_3__SAR86_clade" , "Bacillariophyta_X" ,"D_3__Cellvibrionales" , "D_3__Rhodospirillales" , "D_3__Actinomarinales" , "Phaeocystales" , "Cryptomonadales",  "D_3__Oceanospirillales", "D_3__Rickettsiales"))

df_Order_enviro_top15$Enviro2

df_Order_enviro_top15$Enviro2 <- factor(df_Order_enviro_top15$Enviro2, levels=c("Offshore", "Transition", "Coastal", "WAI2"))

bp=ggplot(df_Order_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Order), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right") +
    scale_fill_gradient(low = "white", high = "red")  + scale_y_discrete(name="Order", position="right") + labs(fill = "Average Relative Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust=-0.25))
bp + coord_flip()

ggsave("Top_15_all_taxa.pdf", width=12)





```





```{r}

Phyto_Abun_By_Order=tax_glom(All_phyto, "Order")

top15<- names(sort(taxa_sums(Phyto_Abun_By_Order), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, Phyto_Abun_By_Order)
plot_bar(ps.top15, fill="Order") + facet_wrap(~Site, scales="free_x")
ggsave("Top15byOrder_Phyto_Barplots.pdf")



Phyto_Abun_By_Phylum=tax_glom(All_phyto, "Phylum")

top15<- names(sort(taxa_sums(Phyto_Abun_By_Phylum), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, Phyto_Abun_By_Phylum)
plot_bar(ps.top15, fill="Phylum") + facet_wrap(~Site, scales="free_x")
ggsave("Top15byPhylum_Phyto_Barplots.pdf")

Phyto_Abun_By_Genus=tax_glom(All_phyto, "Genus")

top15<- names(sort(taxa_sums(Phyto_Abun_By_Genus), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, Phyto_Abun_By_Genus)
plot_bar(ps.top15, fill="Genus") + facet_wrap(~Site, scales="free_x")
ggsave("Top15byGenus_Phyto_Barplots.pdf")





Phyto_Genus_total <- Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 

Phyto_Genus_total$percent_total_mean=Phyto_Genus_total$mean*100
Phyto_Genus_total$percent_total_SD=Phyto_Genus_total$sd*100


test=Phyto_Genus_total[Phyto_Genus_total$percent_total_mean %in% tail(sort(Phyto_Genus_total$percent_total_mean),15),]
write.csv(test, "Phyto_Genus_total.csv")



Phyto_Abun_By_Genus_enviro <- Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


Phyto_Abun_By_Genus_enviro_mean=dcast(Phyto_Abun_By_Genus_enviro, OTU+Genus ~ Enviro2, value.var ="mean")
Phyto_Abun_By_Genus_enviro_sd=dcast(Phyto_Abun_By_Genus_enviro , OTU+Genus ~ Enviro2, value.var = "sd")

write.csv(Phyto_Abun_By_Genus_enviro_mean, "Phyto_Abun_By_Genus_enviro_mean.csv")
write.csv(Phyto_Abun_By_Genus_enviro_sd, "Phyto_Abun_By_Genus_enviro_sd.csv")


###horizontal figure


Phyto_Genus_enviro <- Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


Phyto_Genus_enviro_top15=subset(Phyto_Genus_enviro, Genus=="D_5__Synechococcus_CC9902" | Genus=="D_5__Prochlorococcus_MIT9313"| Genus=="Micromonas" |  Genus== "D_5__Cyanobium_PCC-6307" | Genus== "Thalassiosira" | Genus == "Phaeocystis" | Genus == "Ostreococcus" | Genus == "Mantoniella" | Genus =="Aureococcus" | Genus == "Chaetoceros"  | Genus == "Teleaulax" | Genus == "Chloropicon" | Genus == "Mamiellaceae_X" | Genus== "Pyramimonas" | Genus== "Arcocellulus")


Phyto_Genus_enviro_top15$Genus <- factor(Phyto_Genus_enviro_top15$Genus,levels=c("D_5__Synechococcus_CC9902", "D_5__Prochlorococcus_MIT9313","Micromonas" ,"D_5__Cyanobium_PCC-6307","Thalassiosira", "Phaeocystis" , "Ostreococcus" , "Mantoniella" ,"Aureococcus", "Chaetoceros","Teleaulax" ,"Chloropicon" , "Mamiellaceae_X" ,"Pyramimonas" , "Arcocellulus"))



Phyto_Genus_enviro_top15$Enviro2 <- factor(Phyto_Genus_enviro_top15$Enviro2, levels=c("Offshore", "Transition", "Coastal", "WAI2"))

library(viridis)
bp=ggplot(Phyto_Genus_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right") +
    scale_fill_viridis(option="turbo")  + scale_y_discrete(name="Genus", position="right") + labs(fill = "Average Relative Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust=-0.25))
bp + coord_flip()

ggsave("Top_15_Phyto_Genus.pdf", width=12)


#####MUCH BETTER VERSION



Phyto_Genus_enviro_top15$Genus <- factor(Phyto_Genus_enviro_top15$Genus,levels=c("Arcocellulus", "Pyramimonas" ,"Mamiellaceae_X" ,"Chloropicon" , "Teleaulax" ,"Chaetoceros","Aureococcus", "Mantoniella" ,"Ostreococcus" ,"Phaeocystis" ,  "Thalassiosira", "D_5__Cyanobium_PCC-6307","Micromonas" ,"D_5__Prochlorococcus_MIT9313","D_5__Synechococcus_CC9902" ))



library(viridis)
bp=ggplot(Phyto_Genus_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right") +
    scale_fill_viridis(option="turbo")  + scale_y_discrete(name="Genus", position="right") + labs(fill = "Average Relative Abundance") 
bp 

ggsave("Top_15_Phyto_Genus.pdf", width=9)



```



#########
LETS TRY WITH RELATIVE TO AMOUNT OF PHYTOPLANKTON IN A SAMPLE



```{r}


rel218S_Abun_By_Genus=tax_glom(rel2All_18S, "Class")

top15<- names(sort(taxa_sums(rel2Phyto_Abun_By_Genus), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, rel2Phyto_Abun_By_Genus)
plot_bar(ps.top15, fill="Genus") + facet_wrap(~Site, scales="free_x")
ggsave("rel2Phyto_Abun_By_Genus_Phyto_Barplots.pdf")





Dinoflagellata = subset_taxa(rel2All_18S, Phylum== "Dinoflagellata")
ntaxa(Dinoflagellata)
taxa_names(Dinoflagellata)
test=psmelt(Dinoflagellata)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Order))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Dinoflagellata") + scale_fill_manual(values=palette) + ylab("Relative to all 18S seq")
ggsave("Dinoflagellata.pdf")


library(randomcoloR)
n <- 98
palette <- distinctColorPalette(n)


test=psmelt(rel2All_18S)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Phylum))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Mamiellophyceae") + scale_fill_manual(values=palette)


```



```{r}



rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 





rel2Phyto_Abun_By_Genus=tax_glom(rel2All_phyto, "Genus")

top15<- names(sort(taxa_sums(rel2Phyto_Abun_By_Genus), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, rel2Phyto_Abun_By_Genus)
plot_bar(ps.top15, fill="Genus") + facet_wrap(~Site, scales="free_x")
ggsave("rel2Phyto_Abun_By_Genus_Phyto_Barplots.pdf")



rel2Phyto_Abun_By_Order=tax_glom(rel2All_phyto, "Order")

top15<- names(sort(taxa_sums(rel2Phyto_Abun_By_Order), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, rel2Phyto_Abun_By_Order)
plot_bar(ps.top15, fill="Order") + facet_wrap(~Site, scales="free_x")
ggsave("rel2Phyto_Abun_By_Order_Phyto_Barplots.pdf")


rel2Phyto_Abun_By_Class=tax_glom(rel2All_phyto, "Class")

top15<- names(sort(taxa_sums(rel2Phyto_Abun_By_Class), decreasing=TRUE))[1:15]
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, rel2Phyto_Abun_By_Class)
plot_bar(ps.top15, fill="Class") + facet_wrap(~Site, scales="free_x")
ggsave("rel2Phyto_Abun_By_Class_Phyto_Barplots.pdf")

write.csv(as.data.frame(tax_table(ps.top15)), "Top15_Phytoplankton_Class.csv")

tax_table(ps.top15)




rel2Phyto_Genus_total <- rel2Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 

Phyto_Genus_total$percent_total_mean=Phyto_Genus_total$mean*100
Phyto_Genus_total$percent_total_SD=Phyto_Genus_total$sd*100


test=Phyto_Genus_total[Phyto_Genus_total$percent_total_mean %in% tail(sort(Phyto_Genus_total$percent_total_mean),15),]
write.csv(test, "Phyto_Genus_total.csv")



Phyto_Abun_By_Genus_enviro <- Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


Phyto_Abun_By_Genus_enviro_mean=dcast(Phyto_Abun_By_Genus_enviro, OTU+Genus ~ Enviro2, value.var ="mean")
Phyto_Abun_By_Genus_enviro_sd=dcast(Phyto_Abun_By_Genus_enviro , OTU+Genus ~ Enviro2, value.var = "sd")

write.csv(Phyto_Abun_By_Genus_enviro_mean, "Phyto_Abun_By_Genus_enviro_mean.csv")
write.csv(Phyto_Abun_By_Genus_enviro_sd, "Phyto_Abun_By_Genus_enviro_sd.csv")


###horizontal figure


rel2Phyto_Genus_enviro <- rel2Phyto_Abun_By_Genus %>% psmelt() %>% group_by(OTU,Genus,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


rel2Phyto_Genus_enviro_top15=subset(rel2Phyto_Genus_enviro, Genus=="D_5__Synechococcus_CC9902" | Genus=="D_5__Prochlorococcus_MIT9313"| Genus=="Micromonas" |  Genus== "D_5__Cyanobium_PCC-6307" | Genus== "Thalassiosira" | Genus == "Phaeocystis" | Genus == "Ostreococcus" | Genus == "Mantoniella" | Genus =="Aureococcus" | Genus == "Chaetoceros"  | Genus == "Teleaulax" | Genus == "Chloropicon" | Genus == "Mamiellaceae_X" | Genus== "Pyramimonas" | Genus== "Arcocellulus")


rel2Phyto_Genus_enviro_top15$Genus <- factor(rel2Phyto_Genus_enviro_top15$Genus,levels=c("D_5__Synechococcus_CC9902", "D_5__Prochlorococcus_MIT9313","Micromonas" ,"D_5__Cyanobium_PCC-6307","Thalassiosira", "Phaeocystis" , "Ostreococcus" , "Mantoniella" ,"Aureococcus", "Chaetoceros","Teleaulax" ,"Chloropicon" , "Mamiellaceae_X" ,"Pyramimonas" , "Arcocellulus"))



rel2Phyto_Genus_enviro_top15$Enviro2 <- factor(rel2Phyto_Genus_enviro_top15$Enviro2, levels=c("Offshore","Coastal","Estuary"))

library(viridis)
bp=ggplot(rel2Phyto_Genus_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right") +
    scale_fill_viridis(option="turbo")  + scale_y_discrete(name="Genus", position="right") + labs(fill = "Relative Abundance as Percent of Phytoplankton Abundance") +
  theme(axis.text.x = element_text(angle = 45, hjust=-0.25))
bp + coord_flip()

ggsave("Top_15_Phyto_Genus.pdf", width=12)


#####MUCH BETTER VERSION



rel2Phyto_Genus_enviro_top15$Genus <- factor(rel2Phyto_Genus_enviro_top15$Genus,levels=c( "Teleaulax" , "Phaeocystis" ,"Aureococcus","Pyramimonas","Chloropicon" ,"Mamiellaceae_X" ,"Mantoniella" ,"Ostreococcus" ,"Micromonas", "Arcocellulus" ,"Chaetoceros", "Thalassiosira", "D_5__Cyanobium_PCC-6307" ,"D_5__Prochlorococcus_MIT9313","D_5__Synechococcus_CC9902" ))

rel2Phyto_Genus_enviro_top15$Enviro2 <- factor(rel2Phyto_Genus_enviro_top15$Enviro2, levels=c("Estuary","Coastal","Offshore"))

library(viridis)
library(scales)
bp=ggplot(rel2Phyto_Genus_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right", trans = log10_trans()) +
  scale_fill_viridis(option="turbo")  + scale_y_discrete(name="Genus", position="right") + labs(fill = "Relative Abundance as Percent of Phytoplankton Abundance") 
bp 

ggsave("rel2Phyto_Genus_byTaxonomy.pdf", width=10)




bp=ggplot(rel2Phyto_Genus_enviro_top15, aes(x=as.factor(Enviro2), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right", trans = log10_trans()) +
  scale_fill_viridis(option="turbo", trans = 'log10')  + labs(fill = "Relative Abundance as Percent of Phytoplankton Abundance") + scale_y_discrete(name="Genus", position="right")
bp 

ggsave("rel2Phyto_Genus_byTaxonomy_log.pdf", width=10)


```



```{r}



All_phyto = subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota_plas")


rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 



library(randomcoloR)
n <- 98
palette <- distinctColorPalette(n)



custom_bar=c("white","violetred4", "violetred3", "maroon1", "palevioletred", "lightcyan3", "orange","gold" ,"khaki1", "forestgreen", "darkolivegreen4", "darkolivegreen3", "darkseagreen", "limegreen", "mediumorchid", "navy", "mediumblue", "dodgerblue3", "royalblue1", "lightblue2","lightskyblue", "deepskyblue")


#green
coloresbarplot=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="forestgreen",
"Archaeplastida:Bathycoccus"="darkolivegreen4", 
"Archaeplastida:Ostreococcus"="darkolivegreen3",
"Archaeplastida:Micromonas"="darkseagreen",
"Archaeplastida:Pedinophyceae"="limegreen",
"Archaeplastida:Other"="chartreuse4", 
"Hacrobia:Other"="thistle2",
"Hacrobia:Cryptomonadales"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiales"="red",
"Stramenopiles:Diatoms"="gold",
"Stramenopiles:Dictyochophyceae"="khaki1",
"Stramenopiles:Pelagophyceae"="darkgoldenrod3",
'Eukaryota:Other'="gray44",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")













test=psmelt(rel2All_phyto)
test$Curated_v3=factor(test$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))


helpful$Curated_v3=factor(helpful$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))



test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3)))
p + facet_wrap(~kmeans3, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("All Phytoplankton") + scale_fill_manual(values=coloresbarplot)
ggsave("All_Phytoplankton_curated.pdf", width=13)


test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=order.euc.dend, y=Abundance, fill=as.factor(Curated_v3)))
p + facet_wrap(~Clustering, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("All Phytoplankton") + scale_fill_manual(values=coloresbarplot)
ggsave("All_Phytoplankton_curated_by_Clusters.pdf", width=17)


Supergroup4_sum=Bacillariophyta_melt %>% dplyr::group_by(Curated,SampleID,Site)%>%dplyr::summarise(sum=sum(Abundance)*100)
Supergroup4_means

Supergroup4_means=Supergroup4_sum %>% dplyr::group_by(Curated,Site)%>%dplyr::summarise(Mean=round(mean(sum),2), sd=round(sd(sum),2))
Supergroup4_means



helpful <- rel2All_phyto %>% psmelt() %>% dplyr::group_by(Curated_v3,SampleID,Sampling.Order, kmeans3) %>%dplyr::summarise(sum=sum(Abundance)*100)

helpful= helpful%>% dplyr::group_by(Curated_v3,Sampling.Order,kmeans3)%>%dplyr::summarise(Mean=round(mean(sum),2))

head(helpful)

test=test %>% drop_na(kmeans3)

helpful <- test %>% group_by(Curated_v3,kmeans3) %>% summarise(mean = mean(Abundance)*100) 


helpful <- test %>% group_by(Curated_v3,kmeans3) %>% summarise(mean = mean(sum)) 

helpful$Site=factor(helpful$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "CB", "SB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))
ballet


bottom <- ggplot(data=helpful, aes(x=Sampling.Order, y=Mean, fill=as.factor(Curated_v3)))
bottom=bottom+ facet_wrap(~kmeans3, scales="free_x")  + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("") + ylab("Average Relative Abundance (%)") + xlab("Cluster")+ scale_fill_manual(values=coloresbarplot)+theme(axis.text=element_text(size=24), legend.text=element_text(size=24), axis.title=element_text(size=24,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +theme(legend.position = "none") 
bottom

ggsave("bottom.pdf")

ggsave("barchart_RA.svg", width=28, height=18)

ggsave("barchart_RA.SVG", width=20, height=18)




no_dup_metadata_KBytNERR$Site=factor(no_dup_metadata_KBytNERR$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "CB", "SB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))



    geom_violin(color="black", fill="gray53") 

library(rstatix)

library(scales)
chla <- no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L) %>% group_by(Site) %>% summarise(mean = mean(chla.ug.per.L)) 

top <- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=chla.ug.per.L))+geom_boxplot(color="black", fill="gray53") 
top=top+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(axis.text=element_text(size=24),axis.title=element_text(size=24,face="bold"), legend.text=element_text(size=24), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
top


library(cowplot)

plot_grid(top, bottom, labels = c('', ''), align = 'h',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
ggsave("chla_barchart.svg", width=13, height=24)



helpful2=reshape2::dcast(helpful, Curated_v3 ~ Site, value.var ="mean")




ntaxa(rel2All_phyto)






```



###trying to map chla on top of all barcharts 


```{r}





test=psmelt(rel2All_phyto)
test$Curated_v3=factor(test$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))




test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

HP1_t=subset(test, Site=="HP1")
p_HP1_t <- ggplot(data=HP1_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance")
p_HP1_t

AR_t=subset(test, Site=="AR")
p_AR_t <- ggplot(data=AR_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("AR") + ylab("Relative abundance")
p_AR_t


CB_t=subset(test, Site=="CB")
p_CB_t <- ggplot(data=CB_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("CB") + ylab("Relative abundance")
p_CB_t



SR8_t=subset(test, Site=="SR8")
p_SR8_t <- ggplot(data=SR8_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("SR8") + ylab("Relative abundance")
p_SR8_t

SB_t=subset(test, Site=="SB")
p_SB_t <- ggplot(data=SB_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("SB") + ylab("Relative abundance")
p_SB_t



NB_t=subset(test, Site=="NB")
p_NB_t <- ggplot(data=NB_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("NB") + ylab("Relative abundance")
p_NB_t



SR2_t=subset(test, Site=="SR2")
p_SR2_t <- ggplot(data=SR2_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("SR2") + ylab("Relative abundance")
p_SR2_t



NR2_t=subset(test, Site=="NR2")
p_NR2_t <- ggplot(data=NR2_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("NR2") + ylab("Relative abundance")
p_NR2_t



STO1_t=subset(test, Site=="STO1")
p_STO1_t <- ggplot(data=STO1_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("STO1") + ylab("Relative abundance")
p_STO1_t


NTO1_t=subset(test, Site=="NTO1")
p_NTO1_t <- ggplot(data=NTO1_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("NTO1") + ylab("Relative abundance")
p_NTO1_t



library(rstatix)

library(scales)
library(cowplot)
chla <- no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L) 
summary(chla)


HP1_c=subset(chla, Site=="HP1")

p_HP1_c <- ggplot(data=HP1_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_HP1_c



AR_c=subset(chla, Site=="AR")

p_AR_c <- ggplot(data=AR_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_AR_c


CB_c=subset(chla, Site=="CB")

p_CB_c <- ggplot(data=CB_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_CB_c



SR8_c=subset(chla, Site=="SR8")

p_SR8_c <- ggplot(data=SR8_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_SR8_c



SB_c=subset(chla, Site=="SB")

p_SB_c <- ggplot(data=SB_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_SB_c


NB_c=subset(chla, Site=="NB")

p_NB_c <- ggplot(data=NB_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_NB_c

SR2_c=subset(chla, Site=="SR2")

p_SR2_c <- ggplot(data=SR2_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_SR2_c

NR2_c=subset(chla, Site=="NR2")

p_NR2_c <- ggplot(data=NR2_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_NR2_c

STO1_c=subset(chla, Site=="STO1")
p_STO1_c <- ggplot(data=STO1_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_STO1_c

NTO1_c=subset(chla, Site=="NTO1")
p_NTO1_c <- ggplot(data=NTO1_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_NTO1_c



#################

tog_HP1=plot_grid(p_HP1_c, p_HP1_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2), rel_widths = c(1, -0.1, 1))
tog_HP1

tog_AR=plot_grid(p_AR_c, p_AR_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))

tog_CB=plot_grid(p_CB_c, p_CB_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_CB

tog_SR8=plot_grid(p_SR8_c, p_SR8_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_SR8

tog_SB=plot_grid(p_SB_c, p_SB_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_SB

tog_NB=plot_grid(p_NB_c, p_NB_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_SR8

tog_SR2=plot_grid(p_SR2_c, p_SR2_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_SR8

tog_NR2=plot_grid(p_NR2_c, p_NR2_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_SR8

tog_STO1=plot_grid(p_STO1_c, p_STO1_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_STO1

tog_NTO1=plot_grid(p_NTO1_c, p_NTO1_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_NTO1


plot_grid(tog_HP1, tog_AR, tog_CB,tog_SR8,tog_SB,tog_NB,tog_SR2,tog_NR2,tog_STO1,tog_NTO1, align="hv", ncol=3, nrow=4)

#plot_grid(p_HP1_c, p_AR_c, p_HP1_t, p_AR_t, labels = c('', '', "", ""), align = 'vh',label_size = 12, ncol = 2, rel_heights = c(0.5,2))


ggsave("chla_barchart_all.pdf", width=20, height=24)



```






```{r}

iDist <- distance(All_phyto, method="bray")
    # Calculate ordination
    iMDS  <- ordinate(All_phyto, "MDS", distance="bray")
    ## Make plot
    # Don't carry over previous plot (if error, p will be blank)
    p <- NULL
    # Create plot, store as temp variable, p
    p <- plot_ordination(All_phyto, iMDS, color="Enviro2")
   p
   
   plot_heatmap(All_phyto, "NMDS", "jaccard")
   
   

```







```{r}


custom_col42 = c("#781156","#A51876","#D21E96","#E43FAD","#EA6CC0","#F098D3","#114578","#185EA5","#1E78D2","#3F91E4","#6CABEA","#98C4F0","#117878","#18A5A5","#3FE4E4","#6CEAEA","#98F0F0", "#117845","#18A55E","#1ED278","#3FE491","#6CEAAB","#98F0C4","#787811","#A5A518","#D2D21E","#E4E43F","#EAEA6C","#F0F098","#F7F7C5","#784511","#A55E18","#D2781E","#E4913F","#EAAB6C","#F0C498","#781122","#A5182F","#D21E2C","#E43F5B","#EA6C81","#F098A7")

cbPalette8 <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

custom_final28 = c("#771155", "#AA4488", "#EA6CC0", "#CC99BB", "#114477", "#4477AA","#1E78D2", "#77AADD", "#117777", "#44AAAA", "#3FE4E4", "#77CCCC", "#117744","#44AA77", "#1ED278", "#88CCAA", "#771122", "#AA4455", "#D21E2C","#DD7788","#777711", "#AAAA44", "#D2D21E", "#DDDD77","#774411", "#AA7744", "#D2781E", "#DDAA77")

custom_col21 = c("#771155","#AA4488", "#CC99BB", "#114477", "#4477AA", "#77AADD", "#117777", "#44AAAA", "#77CCCC", "#117744", "#44AA77", "#88CCAA", "#777711", "#AAAA44", "#DDDD77", "#774411", "#AA7744", "#DDAA77", "#771122", "#AA4455", "#DD7788")

custom_test=c("#781156","#A51876","#D21E96","#E43FAD","#EA6CC0","#F098D3","#114578","#185EA5","#1E78D2","#3F91E4","#6CABEA","#98C4F0","#117878","#18A5A5","#3FE4E4","#6CEAEA","#98F0F0", "#117845","#18A55E","#1ED278","#3FE491","#6CEAAB","#98F0C4","#787811","#A5A518","#D2D21E","#E4E43F","#EAEA6C","#F0F098","#F7F7C5","#784511","#A55E18","#D2781E","#E4913F","#EAAB6C","#F0C498","#781122","#A5182F","#D21E2C","#E43F5B","#EA6C81","#F098A7", "#771155", "#AA4488", "#EA6CC0", "#CC99BB", "#114477", "#4477AA","#1E78D2", "#77AADD", "#117777", "#44AAAA", "#3FE4E4", "#77CCCC", "#117744","#44AA77", "#1ED278", "#88CCAA", "#771122", "#AA4455", "#D21E2C","#DD7788","#777711", "#AAAA44", "#D2D21E", "#DDDD77","#774411", "#AA7744", "#D2781E", "#DDAA77", "#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

```



```{r}


Mamiellophyceae = subset_taxa(rel2All_phyto, Class == "Mamiellophyceae")
ntaxa(Mamiellophyceae)
taxa_names(Mamiellophyceae)
```







```{r}


Mamiellophyceae = subset_taxa(rel2All_phyto, Class == "Mamiellophyceae")
ntaxa(Mamiellophyceae)
taxa_names(Mamiellophyceae)


test=psmelt(Mamiellophyceae)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Mamiellophyceae") + scale_fill_manual(values=cbPalette8)


Prymnesiophyceae= subset_taxa(rel2All_phyto, Class == "Prymnesiophyceae")
ntaxa(Prymnesiophyceae)
tax_table(Prymnesiophyceae)


test=psmelt(Prymnesiophyceae)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Prymnesiophyceae") + scale_fill_manual(values=cbPalette8)
ggsave("Prymnesiophyceae.pdf")





Cryptophyceae= subset_taxa(rel2All_phyto, Class == "Cryptophyceae")
ntaxa(Cryptophyceae)
taxa_names(Cryptophyceae)


test=psmelt(Cryptophyceae)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Cryptophyceae") + scale_fill_manual(values=custom_col21)
ggsave("Cryptophyceae.pdf")





Archaeplastida= subset_taxa(rel2All_phyto, Supergroup == "Archaeplastida")

ntaxa(Cryptophyceae)
taxa_names(Cryptophyceae)


test=psmelt(Archaeplastida)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Cryptophyceae") + scale_fill_manual(values=custom_col42)+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("Archaeplastida.pdf")


helpful <- test %>% group_by(Curated,Enviro2) %>% summarise(sum = sum(Abundance)) 

helpful2=reshape2::dcast(helpful, Curated ~ Enviro2, value.var ="sum")




Hacrobia= subset_taxa(rel2All_phyto, Supergroup == "Hacrobia")

ntaxa(Cryptophyceae)
taxa_names(Cryptophyceae)


test=psmelt(Hacrobia)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Cryptophyceae") + scale_fill_manual(values=custom_col42)
ggsave("Archaeplastida.pdf")


helpful <- test %>% group_by(Curated,Enviro2) %>% summarise(sum = sum(Abundance)) 

helpful2=reshape2::dcast(helpful, Curated ~ Enviro2, value.var ="sum")






Bacillariophyta= subset_taxa(rel2All_phyto, Class == "Bacillariophyta")
ntaxa(Bacillariophyta)
taxa_names(Bacillariophyta)


test=psmelt(Bacillariophyta)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=custom_final28)
ggsave("Bacillariophyta.pdf")


p <- ggplot(data=test, aes(x=Site, y=Abundance, fill=Genus))
p + facet_wrap(~Enviro2, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=custom_final28)
ggsave("Bacillariophyta_enviro.pdf")







```



```{r}
#Make one model that has all the predictors,
Modelling <- read.csv('models_10252021.csv', row.names = 1)
#dont inlcude Ammonia, many NAs
wNutrients=subset(Modelling, Phosphate !="NA")
#coastal=subset(wNutrients, Site= "WAI2" | Site== "KAHO" | Site=="HP1")
#wNutrients=coastal[,3:17]

sapply(wNutrients, class)
check_cor=wNutrients %>% mutate_if(is.factor,as.numeric)
check_cor=wNutrients%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
#check_cor2=check_cor[,-1]

cormat <- round(cor(check_cor, method = "pearson"),2)
cormat
class(cormat)





library("dendextend")
??ggdendrogram
?hclust

hc <- hclust(dist(cormat), "ave")
hclust <- as.dendrogram(hc)
plot(hclust)



library(corrplot)
svg("correlation_Phytoplankton.svg", width=12, height=12)
corrplot(cormat, type = "lower",
         tl.col = "black", col=col1(10), tl.cex=2, cl.cex = 2)
dev.off()

corrplot(cormat, type = "lower",
         tl.col = "black", col=col1(10))

col1 = colorRampPalette(c('#00007F', 'blue',  '#007FFF', 'cyan', 'white',
                           'yellow', '#FF7F00', 'red', '#7F0000'))


colorRampPalette(c("blue","white","red"))(200)




```
```{r}

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)
write.csv(t(otu_table((SAR11w5.rel_NoChloroNoMito))),"SAR11w5_rel_nochloro.csv") 
rex=as.data.frame(sample_data(SAR11w5.rel_NoChloroNoMito))
write.csv(rex, file="rex2_noChloro.csv")



SAR11Edited=read.csv(file = "ORDERED_otu_heat_New_Seasons_nochloro.csv", header= TRUE, row.names=1)

reltable=otu_table(SAR11Edited, taxa_are_rows = T)

reltable=as.data.frame(reltable)

poo=t(reltable)

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

#OG playing with 70
mat_breaks <- quantile_breaks(poo, n =10000)

mat_breaks

mat_breaks2 <- seq(min(poo), max(poo), length.out = 100)



colorBlindGrey8   <- c("#0072B2", "#56B4E9", "#E69F00","#999999", "#009E73","#F0E442", "#D55E00", "#CC79A7")

Tol=c("#332288", "#88CCEE", "#117733", "#999933", "#DDCC77", "#CC6677", "#882255", "#AA4499")


##TO TRY OUR LINEAR 


#write.csv(flip_sar11, "test.csv")

nameSAR11 <- read.csv(file = "taxa_name_ordered3.csv", header = TRUE, row.names=1) 
annotation_row=nameSAR11
annotation_row=as.data.frame(annotation_row)

labelsrow <- read.csv(file = "labels_row.csv", header = TRUE, row.names=1) 
labelsrow=as.data.frame(labelsrow)
labelsrow


#Note:this is with more of the metadata variables
#meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4.csv", header = TRUE, row.names = 1) 

meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4A.csv", header = TRUE, row.names = 1) 

annotation_col=as.data.frame(meta_site)
annotation_col

col.pal <- RColorBrewer::brewer.pal(9, "Reds")

library(viridis)
aka4 = list(
  Season = c(Spring="#f7943e",Summer = "#E7665E", Fall= "#2ab0b9", Winter="#9D77FC"))

aka5 = list(
  Season = c(Winter="#ababff", Spring="#ffffab",Summer = "#ffabab", Fall= "#abffab"),
Subgroups=c(Ia="#0072B2", Ib="#56B4E9", IIa="#E69F00",IIb= "#999999",IIIa= "#009E73",IV= "#F0E442", Va= "#D55E00",Vb= "#CC79A7") )


library(pals)
#https://cran.r-project.org/web/packages/pals/vignettes/pals_examples.html

##cubicl works best with lots of breaks 10,000 but not 100,000
##tol.rainbow also works but only with 10,000 breaks



labels_row = c("E   S", "E   S", "E", "E", "E", "E", "E", "E", "     S", "     S", "", "", "", "", "", 
    "", "", "", "", "", "", "", "E", "E", "E", "E", "E   S", "E", "E", "E", "E", "", "", "", "", 
    "", "", "", "", "", "", "E   S", "E", "E", "E", "E", "E", "E", "E   S", "E   S", "E", "E", "", "", "", 
    "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", 
    "", "", "", "", "", "", "", "", "", "E   S", "E   S", "E", "E", "", "", "", "E", "", "", "", 
    "E   S", "E", "", "E", "E   S", "E", 
    "E   S", "", "", "", "E   S")

pdf(file = "SAR11_all_8222020_nochloro_newseasons_cubicl.pdf")



pheatmap(reltable, 
          color = cubicl(length(mat_breaks)-1),
        breaks=mat_breaks,
                   cluster_row = F,
                   cluster_cols = F,
        gaps_row = c(22,41,82,84,91,95,98),
         main = "",
                   annotation_col = annotation_col,
                   annotation_row = annotation_row,
        labels_row = labels_row,
                   fontsize = 12,
                   fontsize_row=4, 
                   fontsize_col = 2,
                border_color=NA,
annotation_colors = aka5,
        gaps_col=c(120,160))



```



```{r}

SAR11Edited=read.csv(file = "ORDERED_otu_heat_New_Seasons_nochloro.csv", header= TRUE, row.names=1)

reltable=otu_table(pos_chla, taxa_are_rows = T)

install.packages("pheatmap")
library(pheatmap)

pdf("test.pdf", width=20)
pheatmap(reltable, 
                   cluster_row = F,
                   cluster_cols = F,
         main = "",
                   fontsize = 12,
                   fontsize_row=4, 
                   fontsize_col = 2,
                border_color=NA)


color = cubicl(length(mat_breaks)-1),
        breaks=mat_breaks,

```





```{r}


bloom=subset_samples(rel2All_phyto, Date1=="2/12/21")

library(randomcoloR)
n <- 98
palette <- distinctColorPalette(n)



test=psmelt(bloom)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Site, y=Abundance, fill=Curated))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=palette)



test=psmelt(sigcor_ASV)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Bacillariophyta") + scale_fill_manual(values=custom_final28)



sigcor_ASV=subset_taxa(rel2All_phyto, ASV=="3d868cb8c63d0b8dd74adbc12f6dd154"|ASV=="4c19a0eb78e4c7dfd400538085248690"|ASV=="385805d1a4796f8258d4533beae0a62b"|ASV=="26c4bf94339bbcabcf2b9e16f24427c7"|ASV=="e6719aefdda6941be9f597a88fc3384e"|ASV=="23880d5ecb51a362e9282e5cc82b6c8e"|ASV=="10768a5c1a314cd9f73baee744c7df10"|ASV=="1bff589486f68e77b4476d6837244cd6"|ASV=="d392a3faa6434bd04030546f428dcffd"|ASV=="9107af9e4ab0766c43fd15901c926f3d"|ASV=="41d08288e2062e3578ac2e95f799a2bf"|ASV=="4e43d2b5baa0082c7e42921ce45e4dde"|ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|ASV=="558d9745881b8379f4553fc42b83cccd"|ASV=="44095b9d0c379f3ca649b693174bbdfc"|ASV=="efd9be6e5ad49f28be30f42932f882e8"|ASV=="a908b00f4edf992ef1db8de70b4d7615"|ASV=="2b5da926477e5820dfdc9bac8d2ce40b"|ASV=="83bd0cb9b9e313ae5a863dcb031a23ac"|ASV=="b33930f5c458850ac8e382fe9bba4001"|ASV=="c357a7d4eade7ed830cc23b54139c249")

test=psmelt(sigcor_ASV)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("sigcor_ASV") + scale_fill_manual(values=custom_final28)


sigcor_ASV_0.3=subset_taxa(rel2All_phyto, ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|ASV=="558d9745881b8379f4553fc42b83cccd"|ASV=="44095b9d0c379f3ca649b693174bbdfc"|ASV=="efd9be6e5ad49f28be30f42932f882e8"|ASV=="a908b00f4edf992ef1db8de70b4d7615"|ASV=="2b5da926477e5820dfdc9bac8d2ce40b"|ASV=="83bd0cb9b9e313ae5a863dcb031a23ac"|ASV=="b33930f5c458850ac8e382fe9bba4001"|ASV=="c357a7d4eade7ed830cc23b54139c249")
test=psmelt(sigcor_ASV_0.3)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("sigcor_ASV") + scale_fill_manual(values=custom_final28)



samp=as.data.frame(otu_table(sigcor_ASV_0.3))
write.csv(samp, "sigcor_ASV_03.csv")
kk=read.csv("sigcor_ASV_03.csv")
KK=rename(kk,OTUID=X)

tax=as.data.frame(tax_table(sigcor_ASV_0.3))
write.csv(as.data.frame(tax_table(sigcor_ASV_0.3)),"tax_sigcor_ASV_03.CSV")
tt=read.csv("tax_sigcor_ASV_03.CSV")
TT=rename(tt,OTUID=X)

pop=left_join(TT, KK, by="OTUID")

write.csv(pop, "maybe_v3.csv")



Dinoflagellata = subset_taxa(rel2All_18S, Phylum== "Dinoflagellata")

samp=as.data.frame(otu_table(Dinoflagellata))
write.csv(samp, "Dinoflagellata_otu.csv")
kk=read.csv("Dinoflagellata_otu.csv")
KK=rename(kk,OTUID=X)


tax=as.data.frame(tax_table(Dinoflagellata))
write.csv(as.data.frame(tax_table(Dinoflagellata)),"DINO_tax.CSV")
tt=read.csv("DINO_tax.CSV")
TT=rename(tt,OTUID=X)

pop=left_join(TT, KK, by="OTUID")

write.csv(pop, "maybe_v4.csv")




Chla=subset_taxa(rel2All_phyto, ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|
ASV=="406bb14741ded383380219611b8564d9"|
ASV=="5e7894007198ccd65ab8b06f349d8ee3"|
ASV=="44095b9d0c379f3ca649b693174bbdfc"|
ASV=="297678da5cdd4560b181be93e40d8e3e"|
ASV=="13f195313f60ce69ff1d3f24e072cef5"|
ASV=="0a474686e6615f230a75dc3b07db3cea"|
ASV=="26c4bf94339bbcabcf2b9e16f24427c7"|
ASV=="2b3c3d1f367f316274c94349372be56f"|
ASV=="e5a35d824858b0a8c85ae863af9caed0"|
ASV=="84db962c29105bd1d38d0ff8779ea923"|
ASV=="42cd59d539b4edfa4ca8128418cca49f"|
ASV=="0488881eda9d6a04dc611c4a4095d328"|
ASV=="738271e245688a6eecd5db6d29abd188"|
ASV=="7f9b32ae7899c455086a089a6ad7c7bd"|
ASV=="efd9be6e5ad49f28be30f42932f882e8"|
ASV=="6ce8db6d8ee1fdaa3e32fbf15415a65b"|
ASV=="b0dba0147194c2bf5465c38eaa04e559"|
ASV=="1b1742d585f5f1ace45e5ba9a92ee07d"|
ASV=="32769a9cdf9984d0acd50a3b3cb4b5a1"|
ASV=="2625290899b205f7d8a3a6a36a4b4430"|
ASV=="913910085577b60c29fc5676fc4e6e8c"|
ASV=="e6510d6615987f47739507a9d5d83d95"|
ASV=="ecfbb028e04b03f207bba70c7554b75e"|
ASV=="d0d06719369a3471cd8c6a6335e3386f"|
ASV=="211f2e171afc061e6745fed9d0d33583"|
ASV=="e6719aefdda6941be9f597a88fc3384e"|
ASV=="0f44d43fd24f140fe66b089f27dfb4ad"|
ASV=="23880d5ecb51a362e9282e5cc82b6c8e"|
ASV=="1bff589486f68e77b4476d6837244cd6"|
ASV=="a908b00f4edf992ef1db8de70b4d7615"|
ASV=="2b5da926477e5820dfdc9bac8d2ce40b"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="622994d1a3b21bd9df22236d03434b84"|
ASV=="bd6e2afd311cd170d79defd15fd77ad8"|
ASV=="d392a3faa6434bd04030546f428dcffd"|
ASV=="ee7d879868f6e2a04692c147ed497c88"|
ASV=="1a01dca998f93882d0290802a579c464"|
ASV=="a4c9be481e6bba68cfcc839c1b1019b2"|
ASV=="87b66d169861236181e60e3ff3edf5d0"|
ASV=="edf69e9168c8dd37c91686aeeab2d0a4"|
ASV=="c4fab2e3f3f61f3cdf840d40364f9738"|
ASV=="4dc0abe3d727666b2866e5f631512b27"|
ASV=="0b45d16cc99dcdc9b5bc12b637c5d971"|
ASV=="2533d90c79f13829fa5d0dde39a7931e"|
ASV=="9107af9e4ab0766c43fd15901c926f3d"|
ASV=="01e3d73fa64a2f6b2a0329089680ef4f"|
ASV=="82342e879c1835654963307f25e3d947"|
ASV=="0d2fef790750acb309a2cc43311050e1"|
ASV=="b95e2146b9bc1f9f38dacb16afa751c1"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="c357a7d4eade7ed830cc23b54139c249"|
ASV=="e5adae9e96a5065fb7a94a7db136332e"|
ASV=="f3ee74de7eead5fdc014bfd4d3c48295"|
ASV=="b7b54b056986d6f0aec5a03a7a837dae"|
ASV=="3334ddadd2f68f4f767c229665a55956"|
ASV=="fb2c4366cf3b10ea36e892dc28f4a97a"|
ASV=="03296caf3fa6eb808ac51270bcfb0b28"|
ASV=="df12f0cbeafde6ec7a1ee430e70035ed"|
ASV=="4e43d2b5baa0082c7e42921ce45e4dde"|
ASV=="e06f50344798d12e83c6e163975ad4cd"|
ASV=="e0bcd67e37d43165add5c7c89efb52fb")


pos_chla=subset_taxa(rel2All_phyto, c(ASV=="d0d06719369a3471cd8c6a6335e3386f"|
ASV=="406bb14741ded383380219611b8564d9"|
ASV=="211f2e171afc061e6745fed9d0d33583"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="a908b00f4edf992ef1db8de70b4d7615"|
ASV=="913910085577b60c29fc5676fc4e6e8c"|
ASV=="2533d90c79f13829fa5d0dde39a7931e"|
ASV=="e5adae9e96a5065fb7a94a7db136332e"|
ASV=="b0dba0147194c2bf5465c38eaa04e559"|
ASV=="26c4bf94339bbcabcf2b9e16f24427c7"|
ASV=="87b66d169861236181e60e3ff3edf5d0"|
ASV=="5e7894007198ccd65ab8b06f349d8ee3"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="0f44d43fd24f140fe66b089f27dfb4ad"|
ASV=="e6719aefdda6941be9f597a88fc3384e"|
ASV=="c357a7d4eade7ed830cc23b54139c249"|
ASV=="2b5da926477e5820dfdc9bac8d2ce40b"|
ASV=="e0bcd67e37d43165add5c7c89efb52fb"|
ASV=="4e43d2b5baa0082c7e42921ce45e4dde"|
ASV=="6ce8db6d8ee1fdaa3e32fbf15415a65b"|
ASV=="1a01dca998f93882d0290802a579c464"|
ASV=="2b3c3d1f367f316274c94349372be56f"|
ASV=="738271e245688a6eecd5db6d29abd188"|
ASV=="0a474686e6615f230a75dc3b07db3cea"|
ASV=="32769a9cdf9984d0acd50a3b3cb4b5a1"|
ASV=="7f9b32ae7899c455086a089a6ad7c7bd"|
ASV=="13f195313f60ce69ff1d3f24e072cef5"|
ASV=="c4fab2e3f3f61f3cdf840d40364f9738"|
ASV=="9107af9e4ab0766c43fd15901c926f3d"|
ASV=="b95e2146b9bc1f9f38dacb16afa751c1"|
ASV=="ee7d879868f6e2a04692c147ed497c88"|
ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|
ASV=="edf69e9168c8dd37c91686aeeab2d0a4"|
ASV=="44095b9d0c379f3ca649b693174bbdfc"|
ASV=="4dc0abe3d727666b2866e5f631512b27"|
ASV=="2625290899b205f7d8a3a6a36a4b4430"|
ASV=="b7b54b056986d6f0aec5a03a7a837dae"|
ASV=="0b45d16cc99dcdc9b5bc12b637c5d971"|
ASV=="0488881eda9d6a04dc611c4a4095d328"|
ASV=="d392a3faa6434bd04030546f428dcffd"|
ASV=="efd9be6e5ad49f28be30f42932f882e8"|
ASV=="82342e879c1835654963307f25e3d947"|
ASV=="01e3d73fa64a2f6b2a0329089680ef4f"|
ASV=="0d2fef790750acb309a2cc43311050e1"))

###all sig 
test$Curated=factor(test$Curated, levels=c("Bathycoccus",
"Chlorellales",
"Chloropicon",
"Mamiellaceae",
"Marsupiomonas",
"Pyramimonadales",
"Cyanobium",
"Prochlorococcus",
"Synechococcus",
"Cryptomonadales",
"Isochrysis",
"Phaeocystis",
"Prymnesiales",
"Prymnesiophyceae",
"Teleaulax",
"Chlorarachnida",
"Aureococcus",
"Pelagomonadaceae",
"Pelagophyceae",
"Dictyochophyceae",
"Chrysophyceae",
"Ochromonas",
"Ochrophyta",
"Bacillariophyta",
"Chaetoceros",
"Polar-centric-Mediophyceae",
"Radial-centric-basal-Coscinodiscophyceae",
"Raphid-pennate"))




###pos sig 
test$Curated=factor(test$Curated, levels=c("Bathycoccus",
"Chlorellales",
"Chloropicon",
"Mamiellaceae",
"Marsupiomonas",
"Pyramimonadales",
"Cyanobium",
"Synechococcus",
"Cryptomonadales",
"Isochrysis",
"Prymnesiophyceae",
"Teleaulax",
"Chlorarachnida",
"Aureococcus",
"Pelagophyceae",
"Dictyochophyceae",
"Ochromonas",
"Ochrophyta",
"Bacillariophyta",
"Chaetoceros",
"Polar-centric-Mediophyceae",
"Radial-centric-basal-Coscinodiscophyceae",
"Raphid-pennate"))






bob=as.data.frame(tax_table(Chla))

write.csv(bob, "tax_table_sigASV.csv")

test=psmelt(pos_chla)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Chla") + scale_fill_manual(values=colors_sig)





MostAbunASVs_Rel=subset_taxa(rel2All_phyto, ASV=="024dcc1ab6ba58bbb754448afd9c1127"|ASV=="03296caf3fa6eb808ac51270bcfb0b28"|ASV=="10768a5c1a314cd9f73baee744c7df10"|ASV=="1bff589486f68e77b4476d6837244cd6"|ASV=="300296cb89902c1721ff71f7cbc316b7"|ASV=="85125408ea648cf8e6317b860cd2f1ac"|ASV=="936685ee710e6e0ad190df2c0e863f75"|ASV=="c4c69ec041fd6bae7de4e690ed25d033"|ASV=="d0d06719369a3471cd8c6a6335e3386f"|ASV=="ea325221e74caca2941b3bc9db57c62d"| ASV=="0d2fef790750acb309a2cc43311050e1"|ASV=="44095b9d0c379f3ca649b693174bbdfc"|ASV=="558d9745881b8379f4553fc42b83cccd"|ASV=="9107af9e4ab0766c43fd15901c926f3d"|ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|ASV=="e5adae9e96a5065fb7a94a7db136332e"|ASV=="26c4bf94339bbcabcf2b9e16f24427c7"|ASV=="406bb14741ded383380219611b8564d9"|ASV=="a908b00f4edf992ef1db8de70b4d7615"|ASV=="edf69e9168c8dd37c91686aeeab2d0a4")

test=psmelt(MostAbunASVs_Rel)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Curated))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("sigcor_ASV") + scale_fill_manual(values=custom_final28)



idk=psmelt(rel2All_phyto)
idk$Abundance=idk$Abundance*100



samp=as.data.frame(otu_table(rel2All_phyto))
write.csv(samp, "rel2All_phyto_otu.csv")
kk=read.csv("rel2All_phyto_otu.csv")
KK=rename(kk,OTUID=X)

tax=as.data.frame(tax_table(rel2All_phyto))
write.csv(as.data.frame(tax_table(rel2All_phyto)),"tax_table_phyto_only.CSV")
tt=read.csv("tax_table_phyto_only.CSV")
TT=rename(tt,OTUID=X)

pop=left_join(TT, KK, by="OTUID")

write.csv(pop, "maybe.csv")


yas=read.csv("sig_PhytoplanktonASVs_2.csv", header=TRUE, row.names = 1)
yas=read.csv("dino_cor_first_v2.csv", header=TRUE, row.names = 1)


sapply(yas, class)
check_cor=yas %>% mutate_if(is.factor,as.numeric)
check_cor=yas%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
#check_cor2=check_cor[,-1]

cormat <- round(cor(check_cor, method = "pearson"),2)
cormat
class(cormat)





library(Hmisc)

cors <- function(df) { 
   # turn all three matrices (r, n, and P into a data frame)
   M <- Hmisc::rcorr(as.matrix(df, type=c("pearson")))
   # return the three data frames in a list return(Mdf)
   Mdf <- map(M, ~data.frame(.x))
  }

cors(check_cor) %>% first() %>% head()

cors2=cors(check_cor)%>% first()
cors2=as.matrix(check_cor)



library("dendextend")
??ggdendrogram
?hclust

hc <- hclust(dist(cormat), "ave")
hclust <- as.dendrogram(hc)
plot(dend)



library(corrplot)
pdf("correlation_Phytoplankton.pdf")
corrplot(cormat, type = "upper",  
         tl.col = "black", col=col1(10))


col1 = colorRampPalette(c(    'white',
                           'cyan',  ))

col1 = colorRampPalette(c('#00007F', 'blue',  '#007FFF', 'cyan', 'white',
                           'yellow', '#FF7F00', 'red', '#7F0000'))


colorRampPalette(c("blue","white","red"))(200)



?upper_tri
# Get lower triangle of the correlation matrix
  get_lower_tri<-function(cors2){
    cors2[upper.tri(cors2)] <- NA
    return(cors2)
  }
  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cors2){
    cors2[lower.tri(cors2)]<- NA
    return(cors2)
  }

  lower_tri <- get_lower_tri(cors2)
lower_tri
  class(lower_tri)
  
library(reshape2)
melted_cormat <- melt(lower_tri, na.rm = TRUE)

write.csv(melted_cormat, "melted_cormat.csv")


formatted_cors <- function(df){
 cors(df) %>%
 map(~rownames_to_column(.x, var="Var1")) %>%
 map(~pivot_longer(.x, -Var1, "Var2")) %>% 
 bind_rows(.id = "id") %>%
 pivot_wider(names_from = id, values_from = value) %>%
 mutate(sig_p = ifelse(P < .05, T, F), p_if_sig = ifelse(P <.05, P, NA), r_if_sig = ifelse(P <.05, r, NA)) 
}

test2=formatted_cors(cors2)
write.csv(test2,"test2.csv")

sigP=subset(test2, sig_p=="TRUE")
sigP=subset(sigP, Var1=="chla.ug.per.L")


hist(sigP$r)

bigr=sigP%>% filter( abs(r) > 0.2 )
  

write.csv(bigr, "bigr2.csv")

hist(log(yas$chla.ug.per.L))


test3=test2%>%unite("Merged", Var1:Var2, remove=FALSE)
melted_cormat2=melted_cormat %>%unite("Merged", Var1:Var2, remove=FALSE)

god=left_join(melted_cormat2, test3, by="Merged")
god


god$Var1.x<- factor(god$Var1.x,levels=c(
"PRO.mL","Ia.4.1483", "Ia.4.rs39", "Ia.3.V", "Ia.4", "Salinity", "pH", "NitrateNitrite", "Ia.3.II", "Ia.3.III", "Ia.3.VI", "chla.ug.per.L","SYN.mL", "PEUK.mL", "HBACT.mL", "Silicate","Phosphate","SW.Temperature.at.site"))

god$Var2.x<- factor(god$Var2.x,levels=c(
"PRO.mL","Ia.4.1483", "Ia.4.rs39", "Ia.3.V", "Ia.4", "Salinity", "pH", "NitrateNitrite", "Ia.3.II", "Ia.3.III", "Ia.3.VI", "chla.ug.per.L","SYN.mL", "PEUK.mL", "HBACT.mL", "Silicate","Phosphate","SW.Temperature.at.site"))

write.csv(god, "god2.csv")


god%>% 
 ggplot(aes(Var1.x, Var2.x, fill=r, label=round(r_if_sig,2))) +
 geom_tile() + 
 labs(x = NULL, y = NULL, fill = "Spearmans's\nCorrelation", subtitle="Only significant Spearmans's correlation coefficients shown") + 
 scale_fill_gradient2(mid="#FBFEF9",low="#0C6291",high="#A63446", limits=c(-1,1)) +
 geom_text() +
 theme_classic() 

ggsave("Correlations_sigPvalue.pdf", width=7)



geom_text(size=geom.text.size)+
geom.text.size = 2
ggheatmap <- ggplot(god, aes(Var1.x, Var2.x, fill=r, label=round(r_if_sig,2)))+
 geom_tile(color = "black")+ 
 scale_fill_gradient2(mid="#FBFEF9",low="#0C6291",high="#A63446", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
    name="Spearman\nCorrelation") + 
  theme_minimal()+ # minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+ 
 labs(x = NULL, y = NULL, fill = "Spearman's\nCorrelation", title="") +
 coord_fixed() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 1))
# Print the heatmap

ggheatmap
ggsave("corr_plot_labels_v2.pdf", width=11, height=7)


??DESeq2()
  
```




#######
CORRELATION ANALYSIS


```{r}

library(tidyverse)

metadata_mg_KByT_Q2Q3=read.csv("/Users/sarahtucker/Documents/Metagenome_manuscript/SAR11_Isolates_Metagenomic_Analyses/working_in_R/q2q3_normalized_cca.csv",header=TRUE)

metadata_mg_KByT_Q2Q3=metadata_mg_KByT_Q2Q3%>%dplyr::select(c("Sample","PRO.mL","Ia.4.1483", "Ia.4.rs39", "Ia.3.V", "Ia.4", "Salinity", "pH", "NitrateNitrite", "Ia.3.II", "Ia.3.III", "Ia.3.VI", "chla.ug.per.L",
                                                       "SYN.mL", 
                                                       "PEUK.mL", 
                                                       "HBACT.mL", 
                                                       "Silicate",
                                                         "Phosphate",
                                                        "SW.Temperature.at.site"))



sapply(metadata_mg_KByT_Q2Q3, class)
check_cor=metadata_mg_KByT_Q2Q3 %>% mutate_if(is.factor,as.numeric)
check_cor=metadata_mg_KByT_Q2Q3%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]

cormat <- round(cor(check_cor2, method = "spearman"),2)
cormat
class(cormat)

#install.packages("ggdendro")

library("dendextend")
??ggdendrogram
?hclust

hc <- hclust(dist(cormat), "ave")
dend <- as.dendrogram(hc)
plot(dend)



library(corrplot)
corrplot(cormat, type = "upper", order = "hclust", 
         tl.col = "black", tl.srt = 45)

flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
    )
}

res2<-rcorr(as.matrix(check_cor2))
cors2=flattenCorrMatrix(res2$r, res2$P)



library(Hmisc)

mtcars_cor <- Hmisc::rcorr(as.matrix(check_cor2), type=c("spearman"))
data.frame(mtcars_cor$r) %>% head()
R_cor=data.frame(mtcars_cor$r)
data.frame(mtcars_cor$P) %>% head()
P_cor=data.frame(mtcars_cor$P)

cors <- function(df) { 
   # turn all three matrices (r, n, and P into a data frame)
   M <- Hmisc::rcorr(as.matrix(df, type=c("spearman")))
   # return the three data frames in a list return(Mdf)
   Mdf <- map(M, ~data.frame(.x))
  }

cors(check_cor2) %>% first() %>% head()

cors2=cors(check_cor2)%>% first()
cors2=as.matrix(check_cor2)



?upper_tri
# Get lower triangle of the correlation matrix
  get_lower_tri<-function(cors2){
    cors2[upper.tri(cors2)] <- NA
    return(cors2)
  }
  # Get upper triangle of the correlation matrix
  get_upper_tri <- function(cors2){
    cors2[lower.tri(cors2)]<- NA
    return(cors2)
  }

  lower_tri <- get_lower_tri(cors2)
lower_tri
  class(lower_tri)
  
library(reshape2)
melted_cormat <- melt(lower_tri, na.rm = TRUE)

write.csv(melted_cormat, "melted_cormat.csv")

# Heatmap
library(ggplot2)
ggplot(data = melted_cormat, aes(Var1, Var2, fill = value))+
 geom_tile(color = "white")+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
   name="Spearman\nCorrelation") +
  theme_minimal()+ 
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()



formatted_cors <- function(df){
 cors(df) %>%
 map(~rownames_to_column(.x, var="Var1")) %>%
 map(~pivot_longer(.x, -Var1, "Var2")) %>% 
 bind_rows(.id = "id") %>%
 pivot_wider(names_from = id, values_from = value) %>%
 mutate(sig_p = ifelse(P < .05, T, F), p_if_sig = ifelse(P <.05, P, NA), r_if_sig = ifelse(P <.05, r, NA)) 
}

test2=formatted_cors(cors2)
write.csv(test2,"test2.csv")


test3=test2%>%unite("Merged", Var1:Var2, remove=FALSE)
melted_cormat2=melted_cormat %>%unite("Merged", Var1:Var2, remove=FALSE)

god=left_join(melted_cormat2, test3, by="Merged")
god


god$Var1.x<- factor(god$Var1.x,levels=c(
"PRO.mL","Ia.4.1483", "Ia.4.rs39", "Ia.3.V", "Ia.4", "Salinity", "pH", "NitrateNitrite", "Ia.3.II", "Ia.3.III", "Ia.3.VI", "chla.ug.per.L","SYN.mL", "PEUK.mL", "HBACT.mL", "Silicate","Phosphate","SW.Temperature.at.site"))

god$Var2.x<- factor(god$Var2.x,levels=c(
"PRO.mL","Ia.4.1483", "Ia.4.rs39", "Ia.3.V", "Ia.4", "Salinity", "pH", "NitrateNitrite", "Ia.3.II", "Ia.3.III", "Ia.3.VI", "chla.ug.per.L","SYN.mL", "PEUK.mL", "HBACT.mL", "Silicate","Phosphate","SW.Temperature.at.site"))

write.csv(god, "god2.csv")


god%>% 
 ggplot(aes(Var1.x, Var2.x, fill=r, label=round(r_if_sig,2))) +
 geom_tile() + 
 labs(x = NULL, y = NULL, fill = "Spearmans's\nCorrelation", subtitle="Only significant Spearmans's correlation coefficients shown") + 
 scale_fill_gradient2(mid="#FBFEF9",low="#0C6291",high="#A63446", limits=c(-1,1)) +
 geom_text() +
 theme_classic() 

ggsave("Correlations_sigPvalue.pdf", width=7)




geom.text.size = 2
ggheatmap <- ggplot(god, aes(Var1.x, Var2.x, fill=r, label=round(r_if_sig,2)))+
 geom_tile(color = "black")+ 
 scale_fill_gradient2(mid="#FBFEF9",low="#0C6291",high="#A63446", 
   midpoint = 0, limit = c(-1,1), space = "Lab", 
    name="Spearman\nCorrelation") + geom_text(size=geom.text.size)+
  theme_minimal()+ # minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+ 
 labs(x = NULL, y = NULL, fill = "Spearman's\nCorrelation", title="") +
 coord_fixed() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 1))
# Print the heatmap

ggheatmap
ggsave("corr_plot_labels_v2.pdf", width=11, height=7)




```






```{r}

```





```{r}

rel2Phyto_ASV_total <- rel2All_phyto %>% psmelt() %>% group_by(OTU,ASV) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 

rel2Phyto_ASV_total$percent_total_mean=rel2Phyto_ASV_total $mean*100
rel2Phyto_ASV_total$percent_total_SD=rel2Phyto_ASV_total $sd*100


test=rel2Phyto_ASV_total[rel2Phyto_ASV_total$percent_total_mean %in% tail(sort(rel2Phyto_ASV_total$percent_total_mean),15),]
write.csv(test, "rel2Phyto_ASV_total.csv")




rel2Phyto_Abund_enviro <- rel2All_phyto %>% psmelt() %>% group_by(OTU,ASV,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 

test=rel2Phyto_Abund_enviro[rel2Phyto_Abund_enviro$mean %in% tail(sort(rel2Phyto_Abund_enviro$mean ),30),]
write.csv(test, "rel2Phyto_ASV_total_by_enviro.csv")

rel2Phyto_Abund_enviro_mean=dcast(rel2Phyto_Abund_enviro, OTU+ASV ~ Enviro2, value.var ="mean")
rel2Phyto_Abund_enviro_sd=dcast(rel2Phyto_Abund_enviro , OTU+ASV ~ Enviro2, value.var = "sd")

write.csv(rel2Phyto_Abund_enviro_mean, "rel2Phyto_Abund_enviro_mean.csv")



rel2Phyto_ASV_enviro_top15=subset(rel2Phyto_Abund_enviro,ASV=="c4c69ec041fd6bae7de4e690ed25d033"|
ASV=="1bff589486f68e77b4476d6837244cd6"|
ASV=="10768a5c1a314cd9f73baee744c7df10"|
ASV=="e5adae9e96a5065fb7a94a7db136332e"|
ASV=="d0d06719369a3471cd8c6a6335e3386f"|
ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|
ASV=="558d9745881b8379f4553fc42b83cccd"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="0d2fef790750acb309a2cc43311050e1"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="44095b9d0c379f3ca649b693174bbdfc"|
ASV=="9107af9e4ab0766c43fd15901c926f3d"|
ASV=="300296cb89902c1721ff71f7cbc316b7"|
ASV=="2533d90c79f13829fa5d0dde39a7931e"|
ASV=="83bd0cb9b9e313ae5a863dcb031a23ac")



test=psmelt(Prymnesiophyceae)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=rel2Phyto_ASV_enviro_top15, aes(x=Enviro2, y=mean, fill=ASV))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))
+ggtitle("Prymnesiophyceae") 
ggsave("Prymnesiophyceae.pdf")
+ scale_fill_manual(values=cbPalette8)





Bacillariophyta= subset(rel2Phyto_Class_enviro , Class == "Bacillariophyta")
ntaxa(Bacillariophyta)
taxa_names(Bacillariophyta)






Coastal_top10=subset(rel2Phyto_Abund_enviro, ASV=="c4c69ec041fd6bae7de4e690ed25d033"|
ASV=="d0d06719369a3471cd8c6a6335e3386f"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="300296cb89902c1721ff71f7cbc316b7"|
ASV=="a908b00f4edf992ef1db8de70b4d7615"|
ASV=="03296caf3fa6eb808ac51270bcfb0b28"|
ASV=="406bb14741ded383380219611b8564d9"|
ASV=="26c4bf94339bbcabcf2b9e16f24427c7"|
ASV=="edf69e9168c8dd37c91686aeeab2d0a4")


Coastal_top10=subset(Coastal_top10,Enviro2=="Coastal")
100-sum(Coastal_top10$mean)



ggplot(Coastal_top10, aes(x="", y=mean, fill=ASV)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  
  theme_void()




Estuary_top10=subset(rel2Phyto_Abund_enviro,ASV=="c4c69ec041fd6bae7de4e690ed25d033"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="e5adae9e96a5065fb7a94a7db136332e"|
ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20"|
ASV=="558d9745881b8379f4553fc42b83cccd"|
ASV=="300296cb89902c1721ff71f7cbc316b7"|
ASV=="0d2fef790750acb309a2cc43311050e1"|
ASV=="44095b9d0c379f3ca649b693174bbdfc"|
ASV=="9107af9e4ab0766c43fd15901c926f3d")
Estuary_top10=subset(Estuary_top10,Enviro2=="Estuary")
100-sum(Estuary_top10$mean)





Offshore_top10=subset(rel2Phyto_Abund_enviro,ASV=="c4c69ec041fd6bae7de4e690ed25d033"|
ASV=="1bff589486f68e77b4476d6837244cd6"|
ASV=="10768a5c1a314cd9f73baee744c7df10"|
ASV=="300296cb89902c1721ff71f7cbc316b7"|
ASV=="85125408ea648cf8e6317b860cd2f1ac"|
ASV=="03296caf3fa6eb808ac51270bcfb0b28"|
ASV=="d0d06719369a3471cd8c6a6335e3386f"|
ASV=="936685ee710e6e0ad190df2c0e863f75"|
ASV=="ea325221e74caca2941b3bc9db57c62d"|
ASV=="024dcc1ab6ba58bbb754448afd9c1127")
Offshore_top10=subset(Offshore_top10,Enviro2=="Offshore")
100-sum(Offshore_top10$mean)

write.csv(Offshore_top10, "Offshore_top10.csv")
write.csv(Estuary_top10, "Estuary_top10.csv")
write.csv(Coastal_top10, "Coastal_top10.csv")


ugghh=read.csv("top10_ASVs.csv")
poop=reshape2::dcast(ugghh, TAXONOMY+ASV.Rename ~ Enviro2, value.var ="mean")


write.csv(poop, "poop.csv")

fart=read.csv("poop.csv", header=TRUE, row.names=1)



rel2Phyto_Class_enviro_top5$Class <- factor(rel2Phyto_Class_enviro_top5$Class,levels=c( "Cryptophyceae","Prymnesiophyceae" ,"Bacillariophyta" , "Mamiellophyceae", "D_2__Cyanobacteriia"  ))



ugghh$TAXONOMY <- factor(ugghh$TAXONOMY, levels=c("Other","Bacillariophyta_X","Polar-centric-Mediophyceae_2",
"Polar-centric-Mediophyceae_1",
"Polar-centric-Mediophyceae_3",
"Aureococcus_anophagefferens",
"Prymnesiophyceae",
"Hemiselmidaceae",
"Teleaulax",
"Micromonas_2",
"Micromonas_1",
"Mamiellaceae",
"Mamiellaceae_2",
"Ostreococcus", "Chloropicon_sp.", "Prochlorococcus_1",
"Prochlorococcus_2",
"Synechococcus_2",
"Synechococcus_1",
"Cyanobium_1",
"Cyanobium_2"
))
ugghh$Enviro2 <- factor(ugghh$Enviro2, levels=c("Estuary","Coastal","Offshore"))

p <- ggplot(data=ugghh, aes(x=Enviro2, y=mean, fill=TAXONOMY))
p  + geom_bar(aes(), stat="identity", position="stack")  +ggtitle("")+ ylab("Mean Relative Abundance\n(% of Phytoplankton Total Abundance) ") + xlab("") + scale_fill_manual(values=custom_bar)+theme(text = element_text(size=14), axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("top_10_ASVs_v2.svg", width=10, height=6)



#custom_bar = c("#FFFFFF","#781156","#A51876","#D21E96","#F48FB1" ,"#AEB6BF","#F28C28", "#FFF59D","#FDD835" ,"#117878","#18A5A5", "#117845","#18A55E","#1ED278", "#B39DDB","#114578","#185EA5","#1E78D2","#3F91E4","#6CABEA","#98C4F0")


custom_bar=c("white","violetred4", "violetred3", "maroon1", "palevioletred", "lightcyan3", "orange","gold" ,"khaki1", "forestgreen", "darkolivegreen4", "darkolivegreen3", "darkseagreen", "limegreen", "mediumorchid", "navy", "mediumblue", "dodgerblue3", "royalblue1", "lightblue2","lightskyblue", "deepskyblue")


,"#3FE491","#6CEAAB","#98F0C4","#787811","#A5A518","#D2D21E","#E4E43F","#EAEA6C","#F0F098","#F7F7C5","#784511","#A55E18","#D2781E","#E4913F","#EAAB6C","#F0C498","#781122","#A5182F","#D21E2C","#E43F5B","#EA6C81","#F098A7"



rel2Phyto_Class_enviro_top5$Enviro2 <- factor(rel2Phyto_Class_enviro_top5$Enviro2, levels=c("Estuary","Coastal","Offshore"))

bp=ggplot(rel2Phyto_Class_enviro_top5, aes(x=as.factor(Enviro2), y=as.factor(Class), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + scale_y_continuous(position = "right", trans = log10_trans()) +
  scale_fill_viridis(option="turbo", trans = 'log10')  + labs(fill = "Relative Abundance as Percent of Phytoplankton Abundance") + scale_y_discrete(name="Class", position="right") +theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
bp 

```







#######
To create venndiagrams and get counts of ASVs per environment
#######


```{r}



ntaxa(physeq_Dav_no_dup_clean)
###getting actual relative abundances for taxa means you need to transform all the data and then use that transformed data set to do a subset 

coastal_all=subset_samples(physeq_Dav_no_dup_clean, Enviro2=="Coastal")

ps1b <- prune_taxa(taxa_sums(coastal_all) > 1, coastal_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Coastal_allTaxa_allSample.csv")



mid_all=subset_samples(physeq_Dav_no_dup_clean , Enviro2=="Offshore")

ps1b <- prune_taxa(taxa_sums(mid_all) > 1, mid_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Offshore_allTaxa_allSample.csv")




wai2_all=subset_samples(physeq_Dav_no_dup_clean, Enviro2=="WAI2")

ps1b <- prune_taxa(taxa_sums(wai2_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "WAI2_allTaxa_allSample.csv")



############
#############
#For phytoplankton


All_phyto_absolute = subset_taxa(physeq_Dav_no_dup_clean, Domain == "Eukaryota" | Phylum == "D_1__Cyanobacteria")
ntaxa(All_phyto_absolute)



###getting actual relative abundances for taxa means you need to transform all the data and then use that transformed data set to do a subset 

coastal_all=subset_samples(All_phyto_absolute , Enviro2=="Coastal")

ps1b <- prune_taxa(taxa_sums(coastal_all) > 1, coastal_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Coastal_Phyto.csv")



off_all=subset_samples(All_phyto_absolute, Enviro2=="Offshore")

ps1b <- prune_taxa(taxa_sums(off_all) > 1, off_all)
ntaxa(ps1b)
tax_table(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Offshore_Phyto.csv")


Estuary_all=subset_samples(All_phyto_absolute, Enviro2=="Estuary")

ps1b <- prune_taxa(taxa_sums(Estuary_all) > 1, off_all)
ntaxa(ps1b)
tax_table(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Estuary_Phyto.csv")


KAHO_all=subset_samples(All_phyto_absolute, Site=="KAHO")

ps1b <- prune_taxa(taxa_sums(KAHO_all) > 1, off_all)
ntaxa(ps1b)
tax_table(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "KAHO_Phyto.csv")


HP1_all=subset_samples(All_phyto_absolute, Site=="HP1")

ps1b <- prune_taxa(taxa_sums(HP1_all) > 1, off_all)
ntaxa(ps1b)
tax_table(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "HP1_Phyto.csv")



WAI2_only=subset_taxa(All_phyto,ASV=="6212aa2e44acc2afdc55401c3b5579fe" | ASV=="db5225c9a18c27b63c92b9c179b14a1d" |
ASV=="82342e879c1835654963307f25e3d947" |
ASV=="b80d592a2b6c3c3fc1673dc21637ee6a" |
ASV=="cdbe25389c9e3ef1d9ef5f654f6c4fd9" |
ASV=="b33930f5c458850ac8e382fe9bba4001" |
ASV=="482103ec92be800ddcbf8db069831ba6" |
ASV=="ff30f2518828bdc3e3d77258b5729749")



WAI2_only_melt=psmelt(WAI2_only)
WAI2_only_melt$Date1 <- as.Date(WAI2_only_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=WAI2_only_melt, aes(x=Date1, y=Abundance, fill=Phylum))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("WAI2 ASVs Unique")
ggsave("all_phyto_byOrder.pdf")



#####################################
######################################
#################Only four sampling events where all sites were sampled at

all_sites=subset_samples(physeq_Dav_no_dup_clean, Date2=="9/22/20" | Date2=="12/23/20" | Date2=="2/12/21"|Date2== "5/24/21")


coastal_all=subset_samples(all_sites, Enviro2=="Coastal")
ps1b <- prune_taxa(taxa_sums(coastal_all) > 1, coastal_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Coastal_AllTAXA_QuarterlyNERR.csv")


off_all=subset_samples(all_sites, Enviro2=="Offshore")
ps1b <- prune_taxa(taxa_sums(off_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Offshore_AllTAXA_QuarterlyNERR.csv")


Estuary_all=subset_samples(all_sites, Enviro2=="Estuary")
ps1b <- prune_taxa(taxa_sums(Estuary_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Estuary_AllTAXA_QuarterlyNERR.csv")


KAHO_all=subset_samples(all_sites, Site=="KAHO")
ps1b <- prune_taxa(taxa_sums(KAHO_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "KAHO_AllTAXA_QuarterlyNERR.csv")


HP1_all=subset_samples(all_sites, Site=="HP1")
ps1b <- prune_taxa(taxa_sums(HP1_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "HP1_AllTAXA_QuarterlyNERR.csv")





All_phyto_all_sites = subset_taxa(all_sites, Domain == "Eukaryota" | Phylum == "D_1__Cyanobacteria")
ntaxa(All_phyto_all_sites)


coastal_all=subset_samples(All_phyto_all_sites , Enviro2=="Coastal")
ps1b <- prune_taxa(taxa_sums(coastal_all) > 1, coastal_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Coastal_Phyto_QuarterlyNERR.csv")


off_all=subset_samples(All_phyto_all_sites, Enviro2=="Offshore")
ps1b <- prune_taxa(taxa_sums(off_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Offshore_Phyto_QuarterlyNERR.csv")


estuary_all=subset_samples(All_phyto_all_sites, Enviro2=="Estuary")
ps1b <- prune_taxa(taxa_sums(estuary_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Estuary_Phyto_QuarterlyNERR.csv")


KAHO_all=subset_samples(All_phyto_all_sites, Site=="KAHO")
ps1b <- prune_taxa(taxa_sums(KAHO_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "KAHO_Phyto_QuarterlyNERR.csv")


HP1_all=subset_samples(All_phyto_all_sites, Site=="HP1")
ps1b <- prune_taxa(taxa_sums(HP1_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "HP1_Phyto_QuarterlyNERR.csv")


#####################################
######################################
#################Seasonal phyto all samples


Fall_all=subset_samples(All_phyto_absolute, Season=="Fall")

ps1b <- prune_taxa(taxa_sums(Fall_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Fall_Phyto.csv")



Spring_all=subset_samples(All_phyto_absolute, Season=="Spring")

ps1b <- prune_taxa(taxa_sums(Spring_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Spring_Phyto.csv")



Summer_all=subset_samples(All_phyto_absolute, Season=="Summer")

ps1b <- prune_taxa(taxa_sums(Summer_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Summer_Phyto.csv")



Winter_all=subset_samples(All_phyto_absolute, Season=="Winter")

ps1b <- prune_taxa(taxa_sums(Winter_all) > 1, off_all)
ntaxa(ps1b)
write.csv(as.data.frame(tax_table(ps1b)), "Winter_Phyto.csv")



```



```{r}
install.packages("eulerr")
library(eulerr)


wilkinson2012 <-  c(Coastal = 126, Offshore = 78, Estuary = 15,
                    "Offshore&Coastal" = 84, "Offshore&Estuary" = 0, "Coastal&Estuary" = 37, 
                    "Offshore&Coastal&Estuary" = 61)
fit3 <- euler(wilkinson2012)
svg("venn.svg")
plot(fit3, labels = list(font = 18), quantities = TRUE, shape="ellipse", fills = c( "#e9f12d","#1d78d4", "#b4145e"))
#alpha = c(0.5, 0.5)



Estuary_only = subset_taxa(rel.abun, ASV=="6212aa2e44acc2afdc55401c3b5579fe"|
ASV=="bf1869a233061a8bc520dafc6f0adf06"|
ASV=="6a181c0b74e7472cab5e0682d687d8b2"|
ASV=="db5225c9a18c27b63c92b9c179b14a1d"|
ASV=="3d2349ca0eabd01c5a188130a2f5ac9a"|
ASV=="82342e879c1835654963307f25e3d947"|
ASV=="b80d592a2b6c3c3fc1673dc21637ee6a"|
ASV=="cdbe25389c9e3ef1d9ef5f654f6c4fd9"|
ASV=="93de4e959490df5f046f91bc1cf91e8e"|
ASV=="fa81a0d7b0efc0496fb2d6a14e97a1db"|
ASV=="b33930f5c458850ac8e382fe9bba4001"|
ASV=="482103ec92be800ddcbf8db069831ba6"|
ASV=="5d2277cf994bb7d22f127b568db05139"|
ASV=="36837ef9b239a7890a6af9c74066f9b5"|
ASV=="ff30f2518828bdc3e3d77258b5729749")


pp=as.data.frame(tax_table(Estuary_only))


###############
#################
#all samples


wilkinson2012 <-  c(Coastal = 109, Offshore = 92, Estuary = 8,
                    "Offshore&Coastal" = 244, "Offshore&Estuary" = 2, "Coastal&Estuary" = 26, 
                    "Offshore&Coastal&Estuary" = 77)
fit3 <- euler(wilkinson2012)
svg("venn_allsamples.svg")
plot(fit3, labels = list(font = 18), quantities = TRUE, fills = c( "#e9f12d","#1d78d4", "#b4145e"))
#alpha = c(0.5, 0.5)




Coastal_only = subset_taxa(All_phyto, ASV=="b5f33f6f2352ad79b066d30dcf8dc7ae"|
ASV=="93bd2c0cbba3418cf0f08041c9144593"|
ASV=="933022c5d1d9304da886a0637c5b21e5"|
ASV=="bb2e00b0e1d5967e317c01fc108fbb43"|
ASV=="00261482f21b019fa05f1137e7b28a4a"|
ASV=="500b518c12bdf4fab7dd5a02887f086e"|
ASV=="ba0303cf0d7cef56a010656dc12b868a"|
ASV=="8e6d7c6198c9ed087286d5028aea9249"|
ASV=="b21ec3ddb3f3e4e226854c8856a9e300"|
ASV=="b4d3df983ed6ca0f0e6481031c3d5bf2"|
ASV=="a0c3f9813d3b118a39c64501340c80cd"|
ASV=="d5c1483d48b6c462c3f172eabe06c323"|
ASV=="c735be3664999216b6c04fac87935dae"|
ASV=="c4512359dc5d146c7e67ca40b2d1d252"|
ASV=="af0973a66b699a3476f0e6e28f5ea6a2"|
ASV=="9c171880332fed16f330332de44553e1"|
ASV=="290b822668d8d62638818cf065651b4c"|
ASV=="22c0dfd12befa56bf63e7cc3c27aa115"|
ASV=="58a1c069c326a7dc7ff115e06d1684d9"|
ASV=="23b1a5d62b4be8e531754f6ffe4ed062"|
ASV=="eea97472b770fab35022d49530026697"|
ASV=="02905a5c1f922e04f5c42da23de219cd"|
ASV=="cdcb57762735f1679431c2d079ce1a46"|
ASV=="1509f12eed755efccfdef044d68c4227"|
ASV=="9b134f429427c1bc1015aeba71f9cbaa"|
ASV=="95b7896257f7f2d6deca859a042eada7"|
ASV=="f0ab6665b40d7d0d8a7599fee13e9690"|
ASV=="cf9799eeaa9f205066b2103eb3a0e861"|
ASV=="919cc58b73d06e2156e33fdefa0d7f52"|
ASV=="094a06e82b907ef2f4864ea4a5b06984"|
ASV=="985cee7dccc74c65cc68f01222afee2c"|
ASV=="bafaeb9756a85214bf8068b32f2db120"|
ASV=="fb6279754cb58f9c830b8d2d3c748752"|
ASV=="040e0aa7eb9fab5f03a522029500bbd5"|
ASV=="c614d8943eff594f41bd33ecdbb0773c"|
ASV=="e606f4ea2cf32d2cd0fb5f7a7264e3e3"|
ASV=="28cfce8cf5995cae53edcc9da0527fc2"|
ASV=="cc478d7c19d4ccfccdbf2346a2b630d9"|
ASV=="cdd9620eadf9f02c6b7309829bc11416"|
ASV=="5496fc2322b7206a2e397820bee6fb97"|
ASV=="d6c1a903ffdadb77f28c4a9690cafbb3"|
ASV=="695aadae8a05c4f429762cecf185788a"|
ASV=="953f3b42fd9b67b6bacf14e22596996d"|
ASV=="8c3a78216e2791e6179ca9dfe1d0ecbe"|
ASV=="13f9064248de93538d3de3a967028a34"|
ASV=="8b32ed3050364bbab087b0fdb1d0f3b4"|
ASV=="150be4f2a69901475ba669b88c930961"|
ASV=="69a39578fdcdd1bf268a4b552c75ac4f"|
ASV=="467de37425e029f2c337c02e5780908d"|
ASV=="b3ff4af4acb66c941e0d31610d07c028"|
ASV=="181bdc919ac6d9dd2b726c7b7719264d"|
ASV=="e3355ea291d5e6bc8d6b031f883d1ebb"|
ASV=="71de3ad32b2d9e09240a4f60194d71d8"|
ASV=="30802c03ece8a82f1cd016021e77420c"|
ASV=="fd65e23be7d4ae1a7ba1de401de31ff6"|
ASV=="3140ae3c194588a4ac498d17da2cd345"|
ASV=="afc2b4b7aa26b279b049f2115e846b87"|
ASV=="4601afaafa3da02f5e0e972d9dfb2e6b"|
ASV=="726f59716363310b0165b87336aeed5f"|
ASV=="7e05edb03c526ce39b537f6f1c5ba346"|
ASV=="558dc7e9af12831d0d592a747930a908"|
ASV=="c064082e04ada93f057880bb68a39b93"|
ASV=="acef1e586a4199c47e627cb906b468ab"|
ASV=="49aba8ad9dc3d0ffb92f5656c7b54904"|
ASV=="d37af3c20c8a5c456b992276f585b25c"|
ASV=="d50ad8fab193dd048d7c5c5a3e33fd81"|
ASV=="1f75c067bdd39b1cbed9136d785e8f3c"|
ASV=="0aeacd83026abb3797894515cf7fd262"|
ASV=="ccde27571b9ccd5f2018270294e0ff19"|
ASV=="1fd9fbdfc5fc51e46aacfbd2f6bf4d71"|
ASV=="fdfd3ddacfa158fe5642f501a250a1b0"|
ASV=="23152b9aa8081fb6e364731e769d731a"|
ASV=="6c845bf8242484c0fb38d816d2b36d33"|
ASV=="bab1dfb73e511d391c6849a0f5b5f1d7"|
ASV=="11966f074830881c73a19d1f3c9d801d"|
ASV=="7949c90ddc2fc773c33d747f264c215a"|
ASV=="55b196c7e18e3ec11d215db5beab28a3"|
ASV=="5cc927a1c7ce5d5a458c69a5deee7931"|
ASV=="dac058bc32e0943b2a35e7e4fea8338d"|
ASV=="23e2362ca5907072d3be377145cd1bcb"|
ASV=="79d639cc91f540812817c1e732797ad9"|
ASV=="05b3ff3ffa3d33475c762422c1085887"|
ASV=="0e78f5423ef33d83c884c778a9e84fb0"|
ASV=="cd9e50f9750f81880968fcbda9089090"|
ASV=="043e84656b0027fc6a29619c435a51cb"|
ASV=="510ba4022ffc78b56b67d7577c4d2db3"|
ASV=="45db645554e7dfee51e25e2ce6b918bd"|
ASV=="a8944acd276341f91bcec37d657b0315"|
ASV=="a72ba4e06b063e1372eafe83ba28f9f7"|
ASV=="5546cfabcdd60c134bdd5581ecb1d067"|
ASV=="7ca14e0d56187e724a4dfc106676b470"|
ASV=="3368d2e981d754ddd9799870c49958aa"|
ASV=="7dab3345adb11499438c55cc8c188c09"|
ASV=="60b890405d718f1175d0131bc0d438a4"|
ASV=="c78e0de9e52a0926cc5028b7cb67db24"|
ASV=="2d03cf3d334e3f32d7eb1ed442fbd3bc"|
ASV=="71fbcd0857f5faf61533cf98906a5988"|
ASV=="1d1baa517177307db40d52c55bb3509b"|
ASV=="1d90f8d0293798c8f3ab9a39affe2d11"|
ASV=="62d304cfd73bd84f1b09c88a77f3df98"|
ASV=="abc365a36a5fadaaae5af0ee0d2fb823"|
ASV=="a64ce4c65ff5a1d46caeb87ebc28a7ae"|
ASV=="805961f867858041622ca732021cc9c9"|
ASV=="aac6bfb0726ef445f84543c10bfd3309"|
ASV=="a5f4c76a9f3f07bde93e3cf7ee9d08b3"|
ASV=="a309d13b160e1e8e0bbcfb439398e0ef"|
ASV=="fda51e54152a20ad0873e188137c3f60"|
ASV=="72572c34c692a5d0bd5c0c13f9eb7bac"|
ASV=="4040f83030d99604c9292f47ded6ab33")


pp=as.data.frame(tax_table(Coastal_only))


Coastal_only=subset_samples(Coastal_only, Enviro2=="Coastal")

Coastal_only_melt=psmelt(Coastal_only)
Coastal_only_melt$Date1 <- as.Date(Coastal_only_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Coastal_only_melt, aes(x=Date1, y=Abundance, fill=Class))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Coastal ASVs Unique")
ggsave("Coastal_unique_ASVs.pdf")




Offshore_only = subset_taxa(All_phyto,ASV=="230fe9b9f06cfb26e80419c5bda4d5ae"|
ASV=="bbc1f42b124d03523f023f7f393454ed"|
ASV=="42a71f08b0a2fb67db2960c8c4d8cf5f"|
ASV=="24c9d477287b572b34e93af58d7608ab"|
ASV=="d8e8cfafa9d41c032d2905d3b126984b"|
ASV=="d73f7b64e64e38532e3d841879160bb7"|
ASV=="3919397797fd704974cbe1db16fd2911"|
ASV=="866a80df5daeb407cdafc24ce56f6b1b"|
ASV=="bb34c69f67539a44b2dccd506d601c75"|
ASV=="e6ec87c954d7ba77caccf903e3ae6e7b"|
ASV=="df66c89ab85d1adb98a8227f09f964db"|
ASV=="30baa5aa95283e878cb3c44b14482cea"|
ASV=="c2dd9b94a60afcd26779ec0032a03ebe"|
ASV=="21cb8273f2960d6f660100312f70c90d"|
ASV=="78e8f438a96726b85a8d8c7324727339"|
ASV=="af33a84a3f08f9d24babc0dfc3dd0518"|
ASV=="53d18c6d2f6e5c67d0394a93cdccce90"|
ASV=="395ff7dce27006bc695d360f38c09224"|
ASV=="6bfe56565a740b481b5328f950267bc6"|
ASV=="33a0710c385b00f2398eae9cbc1296fb"|
ASV=="33417ecc6e666f82ad205b38b0bccc25"|
ASV=="c80020cc3806036312abc0a91be7bc84"|
ASV=="d081222ecc361b970f9d837cae20a10e"|
ASV=="a7618398706dd809d14fffaee9833339"|
ASV=="e6510d6615987f47739507a9d5d83d95"|
ASV=="957c121f10a6486973716afdaed844f3"|
ASV=="ecfbb028e04b03f207bba70c7554b75e"|
ASV=="fb938011d56121722a6326ff4640809d"|
ASV=="c944890292c1813482d05b6f6fcb9562"|
ASV=="704eec0ea57c3be583b77a9e6e63a92d"|
ASV=="1c950ec439291f32b15aa87ae3417e4c"|
ASV=="3822dc2e11b7baa2595f835c91e8e8be"|
ASV=="882d7ca2dae2a9bd6187fff63ea60dbc"|
ASV=="ee168218f7fc638cbed6e8403a9b5fdd"|
ASV=="23880d5ecb51a362e9282e5cc82b6c8e"|
ASV=="cedbe8d08df3532be01ac374ff383f65"|
ASV=="7a6c38e7a3af6fa54bd7ea028cc73118"|
ASV=="537fce5b9b68f9ea0e912bb4a01b4102"|
ASV=="a16a4c3a7c384919224460516d74a306"|
ASV=="d9f021f51de252dde3b7f7731bb7d781"|
ASV=="a2f63ec952f41e13fb49e81bc18b5d67"|
ASV=="1e746114d0b7e6cd5b871d7510534972"|
ASV=="10dcc2184bd1bfaceb21fd147dc3241a"|
ASV=="ea596669c85727a2f13e003100e7717e"|
ASV=="d4bd2a326f97e5865679b4dc064694b1"|
ASV=="5c78db514239bfcb82235eb534755f78"|
ASV=="6393eee32af2c373d3177b743ca57fe1"|
ASV=="86215bfc26ab6967d26b6adcbc011490"|
ASV=="2caaf9a033d65bb23034c52f26cfc08a"|
ASV=="60fc80b0ce9b3ef44139fc3f95d7c0e9"|
ASV=="e503f09fa03c47753777415f5c738362"|
ASV=="cbb31ec4434c6fd3883a9c807ac67c21"|
ASV=="5b656ef7325c1d1eb40a1191f8bf85da"|
ASV=="7fb7194a11e4abdd88688c8208270795"|
ASV=="b55155609dfe85a392bea62ed313eeee"|
ASV=="1da26efbe078f2c27c4413b3c86353db"|
ASV=="1a1471964993d81d996ea6169b688bc4"|
ASV=="271d57a3a0dd4eb5457907efe53d8284"|
ASV=="7f77777331ce012901bf45069d519902"|
ASV=="2677f785e5cdd2154a4912479e1c635e"|
ASV=="ed453273c9bf91109087fa3e0ac847a5"|
ASV=="4f401e73188ba91ccfeacdb5dcf7d848"|
ASV=="8f2d5dd3c02e06a586102c270fb7467d"|
ASV=="4d6bbf7f9c896f271b7b8038c8e4d8f0"|
ASV=="e9b993abe7c3b43a83b44662fa0bc5b3"|
ASV=="fe594f2e8a754c53ffbfdd69651e4efb"|
ASV=="b0432b08b95cbc2cb7e69d4ae87019f6"|
ASV=="a420a2f8b54699410048941a6ca5c0af"|
ASV=="05adbf79158704f4d69b1c6df7bb7a50"|
ASV=="7a8203f95741b7439bdf156a3635623c"|
ASV=="c5f877192bc6715960f1bd8005e2c3be"|
ASV=="46e6bfad75b1c565fbeb9a2249c31bc2"|
ASV=="3334ddadd2f68f4f767c229665a55956"|
ASV=="abf4f87e022b8dd771b92247581566d5"|
ASV=="7ea549b78d0a14ade45725fb23cfb8aa"|
ASV=="4bf08bd93610a18b5c2dd6c205979f21"|
ASV=="38734b7d550a8ef81c278df8e019a8aa"|
ASV=="8956dd1c33a387f5ee457a682ad07994"|
ASV=="efb662384eb5f1f63f1c2bfa63921ac4"|
ASV=="4c0085b7aae18f36e093169b768826bd"|
ASV=="6dfbf9ac1cab3966c2aae708bd4d3205"|
ASV=="c88170da4cd1e58aaf6f7f8d41bff9b3"|
ASV=="13a62ab4ca91db055f4f7ccea4c6a079"|
ASV=="8629372f1e4bf7d370e93547c9571d1f"|
ASV=="36843b4878e6003804119e4f4bc699ca"|
ASV=="eb057d99af2bf07ebc92adc72e8987a0"|
ASV=="a8316cc973070f049e10b6449b4208a7"|
ASV=="610e6eddabffe9cc72d6a7939f4de4ea"|
ASV=="e09f7f081957fe0a67ff71c9b2e61ca9"|
ASV=="7ac12c8f61f8c1e1d6d3d71b60f10fef"|
ASV=="bd9cc56842b3d27eed96316728b1f2eb"|
ASV=="e205b6abf77c72217bc09d8b1566844b")
ntaxa(Offshore_only)


pp=as.data.frame(tax_table(Offshore_only))


Offshore_only=subset_samples(Offshore_only, Enviro2=="Offshore")

Offshore_only_melt=psmelt(Offshore_only)
Offshore_only_melt$Date1 <- as.Date(Offshore_only_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Offshore_only_melt, aes(x=Date1, y=Abundance, fill=Genus))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Offshore ASVs Unique")
ggsave("Offshore_unique_ASVs.pdf")


Estuary_only = subset_taxa(All_phyto, ASV=="6212aa2e44acc2afdc55401c3b5579fe"|
ASV=="db5225c9a18c27b63c92b9c179b14a1d"|
ASV=="82342e879c1835654963307f25e3d947"|
ASV=="b80d592a2b6c3c3fc1673dc21637ee6a"|
ASV=="cdbe25389c9e3ef1d9ef5f654f6c4fd9"|
ASV=="b33930f5c458850ac8e382fe9bba4001"|
ASV=="482103ec92be800ddcbf8db069831ba6"|
ASV=="ff30f2518828bdc3e3d77258b5729749")


Estuary_only=subset_samples(Estuary_only, Enviro2=="Estuary")

Estuary_only_melt=psmelt(Estuary_only)
Estuary_only_melt$Date1 <- as.Date(Estuary_only_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Estuary_only_melt, aes(x=Date1, y=Abundance, fill=Class))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Estuary ASVs Unique")
ggsave("Estuary_unique_ASVs.pdf")



```




```{r}





Abun_By_Domain=tax_glom(rel.abun, "Domain")
Abun_By_Phylum=tax_glom(rel.abun, "Phylum")

Abun_Euk_glom = subset_taxa(Abun_By_Domain, Domain == "Eukaryota" )


All_Euk_SUM <- Abun_Euk_glom %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(percent = round(mean(Abundance)*100,1), sd=round(sd(Abundance)*100,1)) 
AllPhyto_enviro=dcast(AllPhyto, OTU ~ Enviro2, value.var ="sum")



Abun_Euk_glom = subset_taxa(Abun_By_Domain, Domain == "Eukaryota" )


All_Euk_SUM <- Abun_Euk_glom %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(percent = round(mean(Abundance)*100,1), sd=round(sd(Abundance)*100,1)) 
AllPhyto_enviro=dcast(AllPhyto, OTU ~ Enviro2, value.var ="sum")



All_phyto = subset_taxa(rel.abun, Domain == "Eukaryota" | Phylum == "D_1__Cyanobacteria")
ntaxa(All_phyto)


```


###############
BINARY

```{r}
AllPhyto_binary <- transform_sample_counts(All_phyto, function(abund) 1*(abund>0))
e=otu_table(AllPhyto_binary)
write.csv(e, "AllPhyto_binary.csv")

AllPhyto_abun_ASV <- All_phyto %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 
AllPhyto_abun_ASV_enviro=reshape2::dcast(AllPhyto_abun_ASV, OTU ~ Enviro2, value.var ="mean")


AllPhyto_binary <- AllPhyto_binary %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(sum = sum(Abundance)) 
AllPhyto_binary_enviro=dcast(AllPhyto_binary, OTU ~ Enviro2, value.var ="sum")




```




```{r}
Phylum == "D_1__Cyanobacteria"
table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)
All_Prokaryotes_Absolute = subset_taxa(physeq_Dav_no_dup_clean, Domain != "D_0__Eukaryota" & Domain != "Eukaryota" )
table(tax_table(All_Prokaryotes_Absolute)[, "Domain"], exclude = NULL)
table(tax_table(All_Prokaryotes_Absolute)[, "Genus"], exclude = NULL)

ntaxa(All_Prokaryotes_Absolute)


All_Prokaryotes_rel.abun <- transform_sample_counts(All_Prokaryotes_Absolute, function(OTU) OTU/sum(OTU))
ntaxa(All_Prokaryotes_rel.abun )
nsamples(All_Prokaryotes_rel.abun )
#366

Prochlorococcus= subset_taxa(All_Prokaryotes_rel.abun , Genus == "D_5__Prochlorococcus_MIT9313")
ntaxa(Prochlorococcus)



Prochlorococcus_melt=psmelt(Prochlorococcus)
Prochlorococcus_melt$Date1 <- as.Date(Prochlorococcus_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=Prochlorococcus_melt, aes(x=Sampling.Order, y=Abundance*100, fill=ASV))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Prochlorococcus 16S rRNA Gene Relative Abundance") +theme(legend.position = "right")+xlab("Time") + ylab("Prochlorococcus abundance \n relative to total prokaryotic community (%)")
ggsave("Prochlorococcus.pdf")


```




################
HIERARCHICAL CLUSTERING USING 16S BACTERIAL + PLASTID ASV IN DESEQ
should i make another with 18S sequences?
#################



```{r}

#if (!requireNamespace("BiocManager", quietly = TRUE))
  #  install.packages("BiocManager")

#BiocManager::install("DESeq2")
library(colorspace)
library(dendextend)
library(DESeq2)

All_16S_plastid=subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota_plas" |Domain == "D_0__Bacteria" |Domain == "D_0__Archaea")
table(tax_table(All_16S_plastid)[, "Phylum"], exclude = NULL)


Clustering_type <- factor(sample_data(All_16S_plastid)$Clustering, levels=c("Cluster_1","Cluster_2","Cluster_3"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_cluster <- c("coral1","burlywood1","lightblue")
col_Clustering_type <- colors_cluster[Clustering_type]





Enviro_type <- factor(sample_data(All_16S_plastid)$Environment, levels=c("Estuary", "Coastal", "Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_c <- c("firebrick3","gold", "dodgerblue1")
col_Enviro_type <- colors_c[Enviro_type]



Season_type <- factor(sample_data(All_16S_plastid)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkorange","darkturquoise", "darkorchid","darkolivegreen1" )
col_Season_type <- colors_Season[Season_type]


Year_type <- factor(sample_data(All_16S_plastid)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]




deseq_counts <- phyloseq_to_deseq2(All_16S_plastid, ~Environment)
deseq_counts_vst <- varianceStabilizingTransformation(deseq_counts)
vst_trans_count_tab <- assay(deseq_counts_vst)
euc_dist <- dist(t(vst_trans_count_tab))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 

### for the colored bars 
k234 <- cutree(d1, k = 2:4)

pdf("VST_Dendrogram_VSTenvironment.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance", horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_Year_type, col_Season_type, col_Enviro_type, col_Clustering_type), d1, rowLabels = c(paste0("k = ", 4:2),"Year", "Seasons" ,"Enviroment", "Clustering"), horiz = TRUE, y_shift=20)
dev.off()



euc_dend <- as.dendrogram(euc_clust, hang= -1,lwd = 3, lty = 3,
    sub = "")
dend_cols <- (sample_data(phyeuphminV1filt)$color)[order.dendrogram(euc_dend)]
labels_colors(euc_dend)<-dend_cols
namesdend<- (sample_data(phyeuphminV1filt)$name)[order.dendrogram
    (euc_dend)]
labels(euc_dend)<-namesdend
col_depth<-(sample_data(phyeuphminV1filt)$colordepth)
pdf("euclplot.pdf", height=11,width=12)
par(cex=0.3)
plot(euc_dend, xlab="", ylab="", main="", sub="", axes=FALSE)
#colored_bars(col_depth,euc_dend, rowLabels = "Depth", cex.rowLabels=1, y_shift = -52)

par(cex=0.2)
title("NAAMES 1 and 2 dendogram", line=1)
par(cex=0.2)
title(ylab="VST Euclidean distance")
axis(2)
dev.off()
legend("topright", legend = c("Subtropical-Spring","Subtropical-
    Winter","Subpolar-Spring","Subpolar-Winter") , fill= c("
    seagreen3","darkgreen", "steelblue2", "royalblue"), bty="n",
    cex=1.00, title="Region-Season")
legend("topleft", legend = c("5m","25m","50m", "75m", "100m") ,
    fill= c("cadetblue1", "cadetblue2", "cadetblue3","aquamarine3
    ", "cadetblue4"), bty="n", cex=.8)
dev.off()



dend_cols <- (sample_data(phyeuphminV1filt)$color)[order.dendrogram(euc_dend)]
labels_colors(euc_dend)<-dend_cols
namesdend<- (sample_data(phyeuphminV1filt)$name)[order.dendrogram
    (euc_dend)]
labels(euc_dend)<-namesdend
col_depth<-(sample_data(phyeuphminV1filt)$colordepth)


```

```{r}
All_16S_plastid=subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota_plas" |Domain == "D_0__Bacteria" |Domain == "D_0__Archaea")
table(tax_table(All_16S_plastid)[, "Phylum"], exclude = NULL)


Clustering_type <- factor(sample_data(physeq_Dav_no_dup_clean)$Clustering, levels=c("Cluster_1","Cluster_2","Cluster_3"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_cluster <- c("coral1","burlywood1","lightblue")
col_Clustering_type <- colors_cluster[Clustering_type]





Enviro_type <- factor(sample_data(physeq_Dav_no_dup_clean)$Environment, levels=c("Estuary", "Coastal", "Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_c <- c("firebrick3","gold", "dodgerblue1")
col_Enviro_type <- colors_c[Enviro_type]



Season_type <- factor(sample_data(physeq_Dav_no_dup_clean)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkorange","darkturquoise", "darkorchid","darkolivegreen1" )
col_Season_type <- colors_Season[Season_type]


Year_type <- factor(sample_data(physeq_Dav_no_dup_clean)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]




deseq_counts <- phyloseq_to_deseq2(physeq_Dav_no_dup_clean, ~Environment+Season)
deseq_counts_vst <- varianceStabilizingTransformation(deseq_counts)
vst_trans_count_tab <- assay(deseq_counts_vst)
euc_dist <- dist(t(vst_trans_count_tab))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 

### for the colored bars 
k234 <- cutree(d1, k = 2:4)

pdf("VST_Dendrogram_VSTenvironment_w18S.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance", horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_Year_type, col_Season_type, col_Enviro_type, col_Clustering_type), d1, rowLabels = c(paste0("k = ", 4:2),"Year", "Seasons" ,"Enviroment", "Clustering"), horiz = TRUE, y_shift=20)
dev.off()





```


```{r}
All_phyto = subset_taxa(physeq_Dav_no_dup_clean,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto)

Clustering_type <- factor(sample_data(All_phyto)$kmeans3, levels=c("Cluster_1","Cluster_2","Cluster_3"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_cluster <- c("coral1","burlywood1","lightblue")
col_Clustering_type <- colors_cluster[Clustering_type]





Enviro_type <- factor(sample_data(All_phyto)$Environment, levels=c("Estuary", "Coastal", "Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_c <- c("firebrick3","gold", "dodgerblue1")
col_Enviro_type <- colors_c[Enviro_type]



Season_type <- factor(sample_data(All_phyto)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkorange","darkturquoise", "darkorchid","darkolivegreen1" )
col_Season_type <- colors_Season[Season_type]


Year_type <- factor(sample_data(All_phyto)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]


## Check to see how the transformation did
within_trans_means<-apply(vst_trans_count_tab,1,mean)
within_trans_vars<-apply(vst_trans_count_tab,1,var)
# Plot:
plot(within_trans_means,within_trans_vars)
# See how now most of the variances are the same between otus? 
# This means the data are more intercomparable using a z-score transformation + detrending


transformed_detrended<-apply(vst_trans_count_tab,1,pracma::detrend)
# Now we rescale so numbers are again intercomparable and now overdispersion has been
# dealt with
trans_dt_scaled<-apply(transformed_detrended,2,scale)
## We lost rownames so tack those back on
rownames(trans_dt_scaled)<-colnames(vst_trans_count_tab)
#
## Comparing the distribution of the data along every step of the pipeline -- notice how by the time we get to the 
## z-scores we're looking more like a standard normal
hist(vst_trans_count_tab,
    main='VST Sequence Count Observations',
    xlab='# Total Observations',
    ylab='Frequency')
hist(transformed_detrended,
    main='VST+Detrended Data',
    xlab='Total Observations Accounting for Linear Trends',
    ylab='Frequency')
hist(trans_dt_scaled,
    main='VST+Detrended+Scaled Data (Z-scores)',
    xlab='Expression Level Relative to per-OTU Avg',
    ylab='Frequency')



noNA_All_phyto=subset_samples(All_phyto, kmeans3!="kmeans3")
nsamples(noNA_All_phyto)

noNA_All_phyto$kmeans3=as.factor(noNA_All_phyto$kmeans3)

deseq_counts <- phyloseq_to_deseq2(noNA_All_phyto, ~kmeans3)
deseq_counts_vst <- varianceStabilizingTransformation(deseq_counts)
vst_trans_count_tab <- assay(deseq_counts_vst)
euc_dist <- dist(t(vst_trans_count_tab))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 
plot(d1)

### for the colored bars 
k234 <- cutree(d1, k = 2:4)

pdf("VST_Dendrogram_VSTenvironment_justPhyto.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance", horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_Year_type, col_Season_type, col_Enviro_type, col_Clustering_type), d1, rowLabels = c(paste0("k = ", 4:2),"Year", "Seasons" ,"Enviroment", "kmeans3"), horiz = TRUE, y_shift=20)
dev.off()




```



```{r}

All_phyto_Dinos=subset_taxa(physeq_Dav_no_dup_clean,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas" | Phylum=="Dinoflagellata")
  

All_phyto = subset_taxa(physeq_Dav_no_dup_clean,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")




diagdds_pol = phyloseq_to_deseq2(noNA_All_phyto, ~ Season)
diagdds_pol = DESeq(diagdds_pol, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(noNA_All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_Season.csv")




diagdds_pol = phyloseq_to_deseq2(noNA_All_phyto, ~ kmeans3)
diagdds_pol = DESeq(diagdds_pol, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(noNA_All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_kmeans.csv")

par(mar=c(8,5,2,2))
pdf("boxplot.pdf")
boxplot(log10(assays(diagdds_pol)[["cooks"]]), range=0, las=2)


All_phyto
sample_data(All_phyto)

library(DESeq2)






diagdds_pol = phyloseq_to_deseq2(All_phyto, ~ chla.ug.per.L)
diagdds_pol = DESeq(diagdds_pol, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_Chla.csv")

par(mar=c(8,5,2,2))
pdf("boxplot.pdf")
boxplot(log10(assays(diagdds_pol)[["cooks"]]), range=0, las=2)


All_phyto
sample_data(All_phyto)

library(DESeq2)






##########################################
####This is to visualize the results of DeSeq
#########################################
#note this is only 2 log fold change not both negative and postive change 
#posigtab = sigtab[sigtab[, "log2FoldChange"] > 2,]
#posigtab = posigtab[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus","Species", "ASV")]
#write.csv(posigtab, "DeSeq2_results_Season.csv")



colors_sig=c("Bathycoccus"="darkolivegreen4", 
"Chlorellales"="greenyellow",
"Chloropicon"="darkolivegreen3",
"Mamiellaceae" ="forestgreen",
"Marsupiomonas"="darkseagreen",
"Pyramimonadales"="palegreen1", 
"Cyanobium"="deepskyblue",
"Prochlorococcus"="navy",
"Synechococcus"="dodgerblue3",
"Cryptomonadales"="orangered", 
"Isochrysis"="mediumvioletred",
"Phaeocystis"="thistle2",
"Prymnesiales"="red",
"Prymnesiophyceae"="orchid1",
"Teleaulax"="maroon1",
"Chlorarachnida"="purple",
"Aureococcus"="#FFFF00",
"Pelagomonadaceae"="#EFFD5F",
"Pelagophyceae"="#FFBF00",
"Dictyochophyceae"="#E4D00A",
"Chrysophyceae"="gold",
"Ochromonas"="darkkhaki",
"Ochrophyta"="lightgoldenrod1",
"Bacillariophyta"="orange4",
"Chaetoceros"="peachpuff3",
"Polar-centric-Mediophyceae"="rosybrown",
"Radial-centric-basal-Coscinodiscophyceae"="tan3",
"Raphid-pennate"="mistyrose4")


#FFBF00


sigtabgen$Curated=factor(sigtabgen$Curated, levels=c("Bathycoccus",
"Chlorellales",
"Chloropicon",
"Mamiellaceae",
"Marsupiomonas",
"Pyramimonadales",
"Cyanobium",
"Prochlorococcus",
"Synechococcus",
"Cryptomonadales",
"Isochrysis",
"Phaeocystis",
"Prymnesiales",
"Prymnesiophyceae",
"Teleaulax",
"Chlorarachnida",
"Aureococcus",
"Pelagomonadaceae",
"Pelagophyceae",
"Dictyochophyceae",
"Chrysophyceae",
"Ochromonas",
"Ochrophyta",
"Bacillariophyta",
"Chaetoceros",
"Polar-centric-Mediophyceae",
"Radial-centric-basal-Coscinodiscophyceae",
"Raphid-pennate"))

library("ggplot2")
theme_set(theme_bw())
sigtabgen = subset(sigtab, !is.na(ASV))
# Phylum order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$Curated, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$Curated = factor(as.character(sigtabgen$Curated), levels=names(x))



# Genus order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$ASV, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$ASV = factor(as.character(sigtabgen$ASV), levels=names(x))

ggplot(sigtabgen, aes(y=Curated, x=log2FoldChange, color=Curated)) + 
  geom_vline(xintercept = 0.0, color = "gray", size = 0.5) +
  geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5)) +coord_flip()+ geom_vline(xintercept=2, linetype="dashed", color = "red") +  geom_vline(xintercept=-2, linetype="dashed", color = "red") + theme(axis.text=element_text(size=24),axis.title=element_text(size=24,face="bold"), legend.text=element_text(
    size=24), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+scale_color_manual(values = colors_sig)
ggsave("differential_ASV_Clustering.pdf", height=12, width=20)



```

```{r}

theme_set(theme_bw())
sigtabgen = subset(sigtab, !is.na(Genus))
# Phylum order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$Phylum = factor(as.character(sigtabgen$Phylum), levels=names(x))
# Genus order
x = tapply(sigtabgen$log2FoldChange, sigtabgen$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtabgen$Genus = factor(as.character(sigtabgen$Genus), levels=names(x))
ggplot(sigtabgen, aes(y=Genus, x=log2FoldChange, color=Phylum)) + 
  geom_vline(xintercept = 0.0, color = "gray", size = 0.5) +
  geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))



```


#https://microsud.github.io/microbiomeutilities/articles/microbiomeutilities.html

```{r}


taxonomy_Phyto = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_vtest.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)
ntaxa(TAX_Phyto)

physeq_Dav = phyloseq(OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav=merge_phyloseq(physeq_Dav, META_KBytNERR)
ntaxa(physeq_Dav)
#4462


noNA_All_phyto=subset_samples(All_phyto, kmeans3!="kmeans3")
nsamples(noNA_All_phyto)
ralph=subset_taxa(noNA_All_phyto, DeSeq_phyto_Kmeans3=="distinct")
ntaxa(ralph)

write.csv(otu_table(ralph), "otu_table_rel2allphyto.csv")
sample_data(ra)

#optional step if required to rename taxa_names
taxa_names(ralph) <- gsub("Domain", "OTU", taxa_names(ralph))
taxa_names(ralph)

heat.sample <- plot_taxa_heatmap(ralph,
  VariableA = "kmeans3",
  heatcolors = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
  transformation = "log10"
)



?transform()

Cluster1=subset_samples(noNA_All_phyto, kmeans3=="Cluster_1")
Cluster2=subset_samples(noNA_All_phyto, kmeans3=="Cluster_2")
Cluster3=subset_samples(noNA_All_phyto, kmeans3=="Cluster_3")
nsamples(Cluster3)
nsamples(Cluster1)
nsamples(Cluster2)

p0.rel <- transform(Cluster1, "compositional")

#heat.cols <- c("#a8dadc","#457b9d", "#1d3557")
# create a gradient color palette for abundance
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
simple_heatmap(p0.rel,
               group.facet = "Season",
               group.order = NULL,
               abund.thres = 0.01,
               prev.thres = 0.01,
               level = "ASV",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange=TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "Cluster_1")




ggsave("RA_Cluster1_Seasonal_ASV.pdf")

library(microbiome)
library(viridis)
devtools::install_github("microsud/microbiomeutilities")
BiocManager::install("microbiome")
library(microbiomeutilities)

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)
write.csv(t(otu_table((SAR11w5.rel_NoChloroNoMito))),"SAR11w5_rel_nochloro.csv") 
rex=as.data.frame(sample_data(SAR11w5.rel_NoChloroNoMito))
write.csv(rex, file="rex2_noChloro.csv")



SAR11Edited=read.csv(file = "ORDERED_otu_heat_New_Seasons_nochloro.csv", header= TRUE, row.names=1)

reltable=otu_table(SAR11Edited, taxa_are_rows = T)

reltable=as.data.frame(reltable)

poo=t(reltable)

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

#OG playing with 70
mat_breaks <- quantile_breaks(poo, n =10000)

mat_breaks

mat_breaks2 <- seq(min(poo), max(poo), length.out = 100)



colorBlindGrey8   <- c("#0072B2", "#56B4E9", "#E69F00","#999999", "#009E73","#F0E442", "#D55E00", "#CC79A7")

Tol=c("#332288", "#88CCEE", "#117733", "#999933", "#DDCC77", "#CC6677", "#882255", "#AA4499")


##TO TRY OUR LINEAR 


#write.csv(flip_sar11, "test.csv")

nameSAR11 <- read.csv(file = "taxa_name_ordered3.csv", header = TRUE, row.names=1) 
annotation_row=nameSAR11
annotation_row=as.data.frame(annotation_row)

labelsrow <- read.csv(file = "labels_row.csv", header = TRUE, row.names=1) 
labelsrow=as.data.frame(labelsrow)
labelsrow


#Note:this is with more of the metadata variables
#meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4.csv", header = TRUE, row.names = 1) 

meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4A.csv", header = TRUE, row.names = 1) 

annotation_col=as.data.frame(meta_site)
annotation_col

col.pal <- RColorBrewer::brewer.pal(9, "Reds")

library(viridis)
aka4 = list(
  Season = c(Spring="#f7943e",Summer = "#E7665E", Fall= "#2ab0b9", Winter="#9D77FC"))

aka5 = list(
  Season = c(Winter="#ababff", Spring="#ffffab",Summer = "#ffabab", Fall= "#abffab"),
Subgroups=c(Ia="#0072B2", Ib="#56B4E9", IIa="#E69F00",IIb= "#999999",IIIa= "#009E73",IV= "#F0E442", Va= "#D55E00",Vb= "#CC79A7") )


library(pals)
#https://cran.r-project.org/web/packages/pals/vignettes/pals_examples.html

##cubicl works best with lots of breaks 10,000 but not 100,000
##tol.rainbow also works but only with 10,000 breaks



labels_row = c("E   S", "E   S", "E", "E", "E", "E", "E", "E", "     S", "     S", "", "", "", "", "", 
    "", "", "", "", "", "", "", "E", "E", "E", "E", "E   S", "E", "E", "E", "E", "", "", "", "", 
    "", "", "", "", "", "", "E   S", "E", "E", "E", "E", "E", "E", "E   S", "E   S", "E", "E", "", "", "", 
    "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", 
    "", "", "", "", "", "", "", "", "", "E   S", "E   S", "E", "E", "", "", "", "E", "", "", "", 
    "E   S", "E", "", "E", "E   S", "E", 
    "E   S", "", "", "", "E   S")

pdf(file = "SAR11_all_8222020_nochloro_newseasons_cubicl.pdf")



pheatmap(reltable, 
          color = cubicl(length(mat_breaks)-1),
        breaks=mat_breaks,
                   cluster_row = F,
                   cluster_cols = F,
        gaps_row = c(22,41,82,84,91,95,98),
         main = "",
                   annotation_col = annotation_col,
                   annotation_row = annotation_row,
        labels_row = labels_row,
                   fontsize = 12,
                   fontsize_row=4, 
                   fontsize_col = 2,
                border_color=NA,
annotation_colors = aka5,
        gaps_col=c(120,160))



```



```{r}

SAR11Edited=read.csv(file = "ORDERED_otu_heat_New_Seasons_nochloro.csv", header= TRUE, row.names=1)

reltable=otu_table(pos_chla, taxa_are_rows = T)

install.packages("pheatmap")
library(pheatmap)

pdf("test.pdf", width=20)
pheatmap(reltable, 
                   cluster_row = F,
                   cluster_cols = F,
         main = "",
                   fontsize = 12,
                   fontsize_row=4, 
                   fontsize_col = 2,
                border_color=NA)


```











```{r}


chla <- no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L) 
summary(chla)

chla$Clustering=as.factor(chla$Clustering)
chla$Season=as.factor(chla$Season)
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(chla.ug.per.L ~ Clustering, data = chla)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Clustering= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
    amod <- aov(chla.ug.per.L ~ Season, data = chla)
  amod
  summary(amod)
  
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
clus_1=subset(chla, Clustering=="Cluster_1")
clus_2=subset(chla, Clustering=="Cluster_2")
clus_3=subset(chla, Clustering=="Cluster_3")

clus_1$Season=as.factor(clus_1$Season)
clus_2$Season=as.factor(clus_2$Season)
clus_3$Season=as.factor(clus_3$Season)


 amod <- aov(chla.ug.per.L ~ Season, data = clus_3)
  amod
  summary(amod)
  
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  

```


##spatial

```{r}



Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>% dplyr::group_by(Site, Sampling.Order)
v3=Phosphate_means%>%select(Site, Sampling.Order)
write.csv(v3, "nut_test.csv")

```



```{r}



Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>% dplyr::group_by(Site, Sampling.Order)
v3=Phosphate_means%>%select(Site, Sampling.Order)
write.csv(v3, "nut_test.csv")

Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>% dplyr::group_by(Site)%>%dplyr::summarise(Phosphate.n = n(), Phosphate.Mean=round(mean(Phosphate),2), Phosphate.sd=round(sd(Phosphate),2))
Phosphate_means

Nitrate_means=no_dup_metadata_KBytNERR%>% drop_na(NitrateNitrite)%>% dplyr::group_by(Site)%>%dplyr::summarise(NitrateNitrite.n = n(), NitrateNitrite.Mean=round(mean(NitrateNitrite),2), NitrateNitrite.sd=round(sd(NitrateNitrite),2))
Nitrate_means

Silicate_means=no_dup_metadata_KBytNERR%>% drop_na(Silicate)%>% dplyr::group_by(Site)%>%dplyr::summarise(Silicate.n = n(), Silicate.Mean=round(mean(Silicate),2), Silicate.sd=round(sd(Silicate),2))
Silicate_means


Ammonia_means=no_dup_metadata_KBytNERR%>% drop_na(Ammonia)%>% dplyr::group_by(Site)%>%dplyr::summarise(Ammonia.n = n(), Ammonia.Mean=round(mean(Ammonia),2), Ammonia.sd=round(sd(Ammonia),2))
Ammonia_means


chla_means=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)%>% dplyr::group_by(Site)%>%dplyr::summarise(chla.n = n(), chla.Mean=round(mean(chla.ug.per.L),2), chla.sd=round(sd(chla.ug.per.L),2))
chla_means

Salinity_means=no_dup_metadata_KBytNERR%>% drop_na(Salinity)%>% dplyr::group_by(Site)%>%dplyr::summarise(Salinity.n = n(), Salinity.Mean=round(mean(Salinity),2), Salinity.sd=round(sd(Salinity),2))
Salinity_means


Temp_means=no_dup_metadata_KBytNERR%>% drop_na(SW.Temperature.at.site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Temp.n = n(), Temp.Mean=round(mean(SW.Temperature.at.site),2), Temp.sd=round(sd(SW.Temperature.at.site),2))
Temp_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>% dplyr::group_by(Site)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>% dplyr::group_by(Site)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means

PEUK_means=no_dup_metadata_KBytNERR%>% drop_na(PEUK.mL)%>% dplyr::group_by(Site)%>%dplyr::summarise(PEUK.n = n(), PEUK.Mean=round(mean(PEUK.mL),2), PEUK.sd=round(sd(PEUK.mL),2))
PEUK_means

SYN_means=no_dup_metadata_KBytNERR%>% drop_na(SYN.mL)%>% dplyr::group_by(Site)%>%dplyr::summarise(SYN.n = n(), SYN.Mean=round(mean(PEUK.mL),2), SYN.sd=round(sd(SYN.mL),2))
SYN_means

PRO_means=no_dup_metadata_KBytNERR%>% drop_na(PRO.mL)%>% dplyr::group_by(Site)%>%dplyr::summarise(PRO.n = n(), PRO.Mean=round(mean(PRO.mL),2), PRO.sd=round(sd(PRO.mL),2))
PRO_means


HBACT_means=no_dup_metadata_KBytNERR%>% drop_na(HBACT.mL)%>% dplyr::group_by(Site)%>%dplyr::summarise(HBACT.n = n(), HBACT.Mean=round(mean(HBACT.mL),2), HBACT.sd=round(sd(HBACT.mL),2))
HBACT_means

maybe=bind_cols(Temp_means, Salinity_means,pH_means,chla_means,HBACT_means, PRO_means, SYN_means, PEUK_means, Silicate_means, Phosphate_means, Nitrate_means)



```



```{r}

Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(Phosphate.n = n(), Phosphate.Mean=round(mean(Phosphate),2), Phosphate.sd=round(sd(Phosphate),2))
Phosphate_means

Nitrate_means=no_dup_metadata_KBytNERR%>% drop_na(NitrateNitrite)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(NitrateNitrite.n = n(), NitrateNitrite.Mean=round(mean(NitrateNitrite),2), NitrateNitrite.sd=round(sd(NitrateNitrite),2))
Nitrate_means

Silicate_means=no_dup_metadata_KBytNERR%>% drop_na(Silicate)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(Silicate.n = n(), Silicate.Mean=round(mean(Silicate),2), Silicate.sd=round(sd(Silicate),2))
Silicate_means


Ammonia_means=no_dup_metadata_KBytNERR%>% drop_na(Ammonia)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(Ammonia.n = n(), Ammonia.Mean=round(mean(Ammonia),2), Ammonia.sd=round(sd(Ammonia),2))
Ammonia_means


chla_means=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(chla.n = n(), chla.Mean=round(mean(chla.ug.per.L),2), chla.sd=round(sd(chla.ug.per.L),2))
chla_means

Salinity_means=no_dup_metadata_KBytNERR%>% drop_na(Salinity)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(Salinity.n = n(), Salinity.Mean=round(mean(Salinity),2), Salinity.sd=round(sd(Salinity),2))
Salinity_means


Temp_means=no_dup_metadata_KBytNERR%>% drop_na(SW.Temperature.at.site)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(Temp.n = n(), Temp.Mean=round(mean(SW.Temperature.at.site),2), Temp.sd=round(sd(SW.Temperature.at.site),2))
Temp_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means

PEUK_means=no_dup_metadata_KBytNERR%>% drop_na(PEUK.mL)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(PEUK.n = n(), PEUK.Mean=round(mean(PEUK.mL),2), PEUK.sd=round(sd(PEUK.mL),2))
PEUK_means

SYN_means=no_dup_metadata_KBytNERR%>% drop_na(SYN.mL)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(SYN.n = n(), SYN.Mean=round(mean(PEUK.mL),2), SYN.sd=round(sd(SYN.mL),2))
SYN_means

PRO_means=no_dup_metadata_KBytNERR%>% drop_na(PRO.mL)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(PRO.n = n(), PRO.Mean=round(mean(PRO.mL),2), PRO.sd=round(sd(PRO.mL),2))
PRO_means


HBACT_means=no_dup_metadata_KBytNERR%>% drop_na(HBACT.mL)%>% dplyr::group_by(Site, Season)%>%dplyr::summarise(HBACT.n = n(), HBACT.Mean=round(mean(HBACT.mL),2), HBACT.sd=round(sd(HBACT.mL),2))
HBACT_means

maybe=bind_cols(Temp_means, Salinity_means,pH_means,chla_means,HBACT_means, PRO_means, SYN_means, PEUK_means, Silicate_means, Phosphate_means, Nitrate_means)



```





###seasonality 



```{r}


Supergroup4_means=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)%>% dplyr::group_by(Site, Month, Environment)%>%dplyr::summarise(Mean=round(mean(chla.ug.per.L),2), sd=round(sd(chla.ug.per.L),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

Supergroup4_means$Month=factor(Supergroup4_means$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

Supergroup4_means

f3=Supergroup4_means %>%
  ggplot(aes(x=Month, y=Mean)) + facet_wrap(~Site, scales="free_y")+ geom_bar(stat="identity")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))

f3

nobloom=subset(no_dup_metadata_KBytNERR, Date2!="2/12/21")
Supergroup4_means=nobloom%>% drop_na(chla.ug.per.L)%>% dplyr::group_by(Site, Month)%>%dplyr::summarise(Mean=round(mean(chla.ug.per.L),2), sd=round(sd(chla.ug.per.L),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

Supergroup4_means$Month=factor(Supergroup4_means$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

Supergroup4_means


f3=Supergroup4_means %>%
  ggplot( aes(x=Month, y=Mean, group=Site, color=Site))+
    geom_line()+ geom_point()+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ scale_y_continuous(trans = log10_trans())

f3



f3=Supergroup4_means %>%
  ggplot(aes(x=Month, y=Mean)) + facet_wrap(~Site, scales="free_y")+ geom_bar(stat="identity")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))

f3





nobloom=subset(no_dup_metadata_KBytNERR, Date2!="2/12/21")
nobloom$Month=factor(nobloom$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

ggplot(nobloom, aes(Month, chla.ug.per.L)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.25, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))


no_dup_metadata_KBytNERR$Season=factor(no_dup_metadata_KBytNERR$Season, levels=c("Spring", "Summer", "Fall", "Winter"))
nobloom$Season=factor(nobloom$Season, levels=c("Spring", "Summer", "Fall", "Winter"))

nobloom=subset(no_dup_metadata_KBytNERR, Date2!="2/12/21")
ggplot(no_dup_metadata_KBytNERR, aes(Season, chla.ug.per.L)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun.data=mean_sdl, mult=1, size=0.6, geom="pointrange", color="black") 

nobloom=subset(no_dup_metadata_KBytNERR, Date2!="2/12/21")


chla_noNA=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)
ggplot(nobloom, aes(Month, chla.ug.per.L)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun.data=mean_sdl, mult=1,size=0.6, geom="pointrange", color="black")



no_dup_metadata_KBytNERR$Month=factor(no_dup_metadata_KBytNERR$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))


ggplot(no_dup_metadata_KBytNERR, aes(Month, Kaneohe.stream)) +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun.data=mean_sdl, mult=1,size=0.6, geom="pointrange", color="black")


ggplot(no_dup_metadata_KBytNERR, aes(Season, Kahaluu.stream)) +
  geom_jitter(width = 0.1, aes(colour = Season))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun.data=mean_sdl, mult=1,size=0.6, geom="pointrange", color="black")



PRO_noNA=no_dup_metadata_FCM%>% drop_na(PRO.mL)
PRO_noNA$Month=factor(PRO_noNA$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))
PRO_noNA$Season=factor(PRO_noNA$Season, levels=c("Spring", "Summer", "Fall", "Winter"))

ggplot(PRO_noNA, aes(Month, PRO.mL)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month, shape=FCM.Machine))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun=mean,size=3, geom="point", color="black")


ggplot(PRO_noNA, aes(Season, PRO.mL)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month, shape=FCM.Machine))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +   
    stat_summary(fun=mean,size=3, geom="point", color="black")





ggplot(no_dup_metadata_KBytNERR, aes(Season, Phosphate)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))


ggplot(no_dup_metadata_KBytNERR, aes(Season, Phosphate)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))



ggplot(no_dup_metadata_KBytNERR, aes(Season, Salinity)) + facet_wrap(~Site, scales="free_y") +
  geom_jitter(width = 0.1, aes(colour = Month))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))


```



```{r}
#install.packages("wql")
library(wql) 

#install.packages("lomb")
library(lomb)
library(lubridate)
library(tidyverse)

##Function lsp computes the Lomb-Scargle periodogram for unevenly sampled times series (e.g., series with missing data). P-values for the highest peak in the periodogram are computed from the exponential distribution. Alternatively, function randlsp computes a p-value for the largest peak in the periodogram by repeatedly randomising the time-series sequence. Both functions allow setting the range of frequencies to be inspected, as well as the stepsize (oversampling factor) used for frequency scanning.

data(lynx)
lsp(lynx)


per=lsp(lynx,ofac=5)
#ofac: The oversampling factor. Must be an integer>=1. Larger values of ofac lead to finer scanning of frequencies but may be time-consuming for large datasets and/or large frequency ranges (from...to).
getpeaks(per,6) # obtain the 6 largest peaks

?lsp()


data(ibex)
datetime <- as.POSIXlt(ibex$date)
plot(datetime,ibex$temp,pch=19,cex=0.3)

whT=ibex[,2:3]
data(ibex)
ibex.spec <- lsp(ibex[,2:3],type='period', from=12,to=36,ofac=10, plot=FALSE)
plot.lsp(ibex.spec, main="Tb in Capra ibex",xlabel="Period (h)",ylabel="Power",level=FALSE, xlim=c(0,0.1))

per=lsp(lynx,ofac=5)
pershow(per)

data(ibex)
ibex.spec <- lsp(ibex[,2:3],type='period', from=12,to=36,ofac=10, plot=FALSE)
plot.lsp(ibex.spec, main="Tb in Capra ibex",xlabel="Period (h)",ylabel="Power",level=FALSE)
plot(lsp(lynx), xlim=c(0,0.1))+theme_lsp(25)

#For a more robust - but potentially time-consuming estimation of p-values (when n is large) see randlsp. Significance levels in both lsp and randlsp increase with the number of frequencies inspected. Therefore, if the frequency-range of interest can be narrowed down a priori, use arguments “from” and “to” to do so.
data(ibex)
lsp(ibex[,2:3],ofac=5)
lsp(ibex$temp,times=ibex$hours,type='period',ofac=5)




chla_samples=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)

chla_samples$datetime <- as.POSIXlt(chla_samples$Date2)

chla_samples$decimal=decimal_date(chla_samples$datetime)
plot(chla_samples$datetime,chla_samples$chla.ug.per.L,pch=19,cex=0.3)

lsp(no_dup_metadata_KBytNERR$SW.Temperature.at.site,times=no_dup_metadata_KBytNERR$Date2,type='period',ofac=5)

pdf("Plotted_SWTemperature.pdf")
plot(chla_samples$datetime,chla_samples$SW.Temperature.at.site,pch=19,cex=0.3,  xlab="Time", ylab="Seawater Temperature (C)")
dev.off()

pdf("Output_Randlsp.pdf")
randlsp( x= chla_samples$SW.Temperature.at.site,
                times = chla_samples$decimal,
                type = 'period',
                plot = T)


summary.randlsp(c)


summary(lsp( x= chla_samples$SW.Temperature.at.site,
                times = chla_samples$decimal,
                type = 'period', to=1.1,
                plot = T))


randlsp( x= chla_samples$chla.ug.per.L,
                times = chla_samples$decimal,
                type = 'period',
                plot = T)


HP1=subset(chla_samples, Site=="HP1")
datetime <- as.POSIXlt(HP1$Date2)

HP1$decimal=decimal_date(datetime)

plot(datetime,HP1$SW.Temperature.at.site,pch=19,cex=0.3)
randlsp( x= HP1$SW.Temperature.at.site,
                times = decimal,
                type = 'period',
                plot = T, trace=TRUE)
randlsp( x= HP1$SW.Temperature.at.site,
                times = decimal,
                type = 'period',
                plot = T, trace=TRUE, to=1.1)


chla_samples=no_dup_metadata_KBytNERR%>% drop_na(PEUK.mL)

chla_samples$datetime <- as.POSIXlt(chla_samples$Date2)

chla_samples$decimal=decimal_date(chla_samples$datetime)
plot(chla_samples$datetime,chla_samples$PEUK.mL,pch=19,cex=0.3)


randlsp(x= chla_samples$PRO.mL,
                times = chla_samples$decimal,
                type = 'period',
                plot = T)
summary(randlsp( x= chla_samples$PRO.mL,
                times = chla_samples$decimal,
                type = 'period',
                plot = T))

summary(randlsp( x= HP1$SYN.mL,
                times = HP1$decimal,
                type = 'period',
                plot = T))


STO1=subset(chla_samples, Site=="STO1")
datetime <- as.POSIXlt(STO1$Date2)

STO1$decimal=decimal_date(datetime)
summary(randlsp( x= STO1$PRO.mL,
                times = STO1$decimal,
                type = 'period',
                plot = T))



plot.lsp()

chla_samples=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)

chla_samples$datetime <- as.POSIXlt(chla_samples$Date2)

chla_samples$decimal=decimal_date(chla_samples$datetime)
plot(chla_samples$datetime,chla_samples$Phosphate,pch=19,cex=0.3)

summary(randlsp( x= chla_samples$Phosphate,
                times = chla_samples$decimal,
                type = 'period',
                plot = T))



```




```{r}

chla=read.csv()
no_dup_metadata_KBytNERR$Date2
chla_samples=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)
chla_samples$Date2 
chla_samples$Date2 <- as.Date(chla_samples$Date2, format="%Y-%m-%d")
plot(chla_samples$Date2,chla_samples$Phosphate,pch=19,cex=0.3)

chla_samples$datetime <- as.POSIXlt(chla_samples$Date2)

chla_samples$decimal=decimal_date(chla_samples$datetime)


```






https://docs.astropy.org/en/stable/timeseries/lombscargle.html


https://cran.r-project.org/web/packages/wql/vignettes/wql-package.html?fbclid=IwAR2PqHgCm2j0RODFbJDdnyU1cr2BOqh939U2agpnNnujUgpK6J0HV3pnYGg#63_time_series_decomposition

```{r}

HP1_simple=HP1%>%select(decimal,chla.ug.per.L )

chl27 <- sfbayChla[, "s27"]
d1 <- decompTs(HP1_simple)
plot(d1, nc = 1, main = "Station 27 Chl-a decomposition")

```





```{r}
install.packages("devtools")
devtools::install_github("adw96/breakaway")
library(breakaway)
```



```{r}



chla <- no_dup_metadata_KBytNERR%>%group_by(Clustering, Season)%>%tally()


chla <- no_dup_metadata_KBytNERR%>%group_by(Site, Season)%>%tally()


test=melt(no_dup_metadata_KBytNERR)


no_dup_metadata_KBytNERR$Date1 <- as.Date(no_dup_metadata_KBytNERR$Date1,format = "%m/%d/%y")
no_dup_metadata_KBytNERR$Date1

off=subset(no_dup_metadata_KBytNERR, Environment=="Offshore")
p <- ggplot(data=off, aes(x=Date1, y=PRO.mL))
p + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2017-09-29"), xmax = as.Date("2017-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01) + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2018-09-29"), xmax = as.Date("2018-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01) + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2019-09-29"), xmax = as.Date("2019-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01)+ facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2020-09-29"), xmax = as.Date("2020-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01)+ geom_rect(aes(xmin = as.Date("2018-03-30"), xmax = as.Date("2018-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01) + geom_rect(aes(xmin = as.Date("2019-03-30"), xmax = as.Date("2019-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01) +geom_rect(aes(xmin = as.Date("2020-03-30"), xmax = as.Date("2020-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01)+geom_rect(aes(xmin = as.Date("2021-03-30"), xmax = as.Date("2021-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01)




```
```{r}

no_dup_metadata_FCM=read.csv("v2_nodups_woAprFP_metadata_2019_2021_11252021.csv", header=TRUE, fill=TRUE)



Supergroup4_means=no_dup_metadata_KBytNERR%>% drop_na(PRO.mL)%>% dplyr::group_by(Site, Month)%>%dplyr::summarise(Mean=round(mean(PRO.mL),2), sd=round(sd(PRO.mL),2))
Supergroup4_means


Supergroup4_means$Site=factor(Supergroup4_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

Supergroup4_means$Month=factor(Supergroup4_means$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

f3=Supergroup4_means %>%
  ggplot(aes(x=Month, y=Mean)) + facet_wrap(~Site, scales="free_x")+ geom_bar(stat="identity")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))

f3

no_dup_metadata_KBytNERR$Month=factor(no_dup_metadata_KBytNERR$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

no_dup_metadata_KBytNERR$Season=factor(no_dup_metadata_KBytNERR$Season, levels=c("Spring", "Summer", "Fall", "Winter"))
no_dup_metadata_KBytNERR$Year=factor(no_dup_metadata_KBytNERR$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
no_dup_metadata_KBytNERR$Year

ggplot(no_dup_metadata_KBytNERR, aes(Month, PRO.mL)) + facet_wrap(~Site, scales="free_x") +
  geom_jitter(width = 0.25, aes(colour = Month, shape=FCM.Machine))+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))



ggplot(no_dup_metadata_KBytNERR, aes(Month, PRO.mL)) + facet_wrap(~Site, scales="free_x") +
  geom_bar(aes(), stat="identity", position="stack")+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))


Supergroup4_sum$Supergroup4

Supergroup4_means$Supergroup4=factor(Supergroup4_means$Supergroup4, levels=c("D_0__Archaea", "Eukaryota", "Eukaryota_plas", "Cyanobacteria","D_0__Bacteria"))

p <- ggplot(data=Supergroup4_means, aes(x=Site, y=Mean, fill=Supergroup4))
p + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("All Taxa")

```

```{r}
library(reshape2)
total=subset_taxa(physeq_FCM, Analysis=="Total")
sample_data(total)
test=psmelt(total)

Phytototal=subset_taxa(physeq_FCM, Analysis=="PhytoplanktonTotal")
sample_data(Phytototal)
test=psmelt(Phytototal)

rel.pro.phyto.total=subset(test, OTU=="rel.pro.phyto.total")


###lets just look at prochlorococcus relative to all phytoplankton FCM data
rel.pro.phyto.total_means=rel.pro.phyto.total%>% drop_na(Abundance)%>% dplyr::group_by(Site, Month)%>%dplyr::summarise(Mean=round(mean(Abundance),2), sd=round(sd(Abundance),2))
rel.pro.phyto.total_means


rel.pro.phyto.total_means$Site=factor(rel.pro.phyto.total_means$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

rel.pro.phyto.total_means$Month=factor(rel.pro.phyto.total_means$Month, levels=c("April", "May", "June", "July", "August", "September", "October", "November", "December", "January", "February", "March"))

rel.pro.phyto.total_means$Season=factor(rel.pro.phyto.total_means$Season, levels=c("Spring", "Summer", "Fall", "Winter"))

rel.pro.phyto.total_means$Month

p <- ggplot(data=rel.pro.phyto.total_means, aes(x=Month, y=Mean))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Relative abundance of Pro across phytoplankton FCM") + ylab("Mean Relative Abundance (%)") #+geom_rect(aes(xmin = as.numeric(rel.pro.phyto.total_means$Month[[2]]),
                                        #xmax = as.numeric(rel.pro.phyto.total_means$Month[[4]]),
                                        # ymin = -Inf, ymax = Inf), fill = "yellow", alpha=0.01)+ scale_x_discrete(drop = FALSE)

ggsave("Rel_Pro_phytoplankton_FCM.pdf")




p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=Group))
p + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") 
ggsave("All_Phytoplankton_curated.pdf", width=13)

test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
test$Date1
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Group))
p + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2017-09-29"), xmax = as.Date("2017-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01) + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2018-09-29"), xmax = as.Date("2018-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01) + facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2019-09-29"), xmax = as.Date("2019-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01)+ facet_wrap(~Environment+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("FCM") + geom_rect(aes(xmin = as.Date("2020-09-29"), xmax = as.Date("2020-12-26"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01)+ geom_rect(aes(xmin = as.Date("2018-03-30"), xmax = as.Date("2018-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01) + geom_rect(aes(xmin = as.Date("2019-03-30"), xmax = as.Date("2019-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01) +geom_rect(aes(xmin = as.Date("2020-03-30"), xmax = as.Date("2020-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01)+geom_rect(aes(xmin = as.Date("2021-03-30"), xmax = as.Date("2021-06-27"),
             ymin = -Inf, ymax = Inf),  
             fill = "yellow", alpha=0.01)


```



```{r}


no_dup_metadata_KBytNERR=read.csv("v2_nodups_woAprFP_metadata_2019_2021_11252021.csv", header=TRUE, fill=TRUE)


chla=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)

chla$Site=factor(chla$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "SB", "CB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

chla$Season=factor(chla$Season, levels=c("Spring", "Summer", "Winter", "Fall"))

HP1=subset(chla, Site=="HP1")
AR=subset(chla, Site=="AR")
SR8=subset(chla, Site=="SR8")
SB=subset(chla, Site=="SB")
CB=subset(chla, Site=="CB")
NB=subset(chla, Site=="NB")
SR2=subset(chla, Site=="SR2")
NR2=subset(chla, Site=="NR2")
STO1=subset(chla, Site=="STO1")
NTO1=subset(chla, Site=="NTO1")

zscore_HP1=HP1 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_HP1<-subset(zscore_HP1, zscore<3)


zscore_SB=SB %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_SB<-subset(zscore_SB, zscore<3)


zscore_AR=AR %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_AR<-subset(zscore_AR, zscore<3)

zscore_SR8=SR8 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_SR8<-subset(zscore_SR8, zscore<3)

zscore_CB=CB %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_CB<-subset(zscore_CB, zscore<3)

zscore_NB=NB %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_NB<-subset(zscore_NB, zscore<3)

zscore_SR2=SR2 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_SR2<-subset(zscore_SR2, zscore<3)


zscore_NR2=NR2 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_NR2<-subset(zscore_NR2, zscore<3)


zscore_NTO1=NTO1 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_NTO1<-subset(zscore_NTO1, zscore<3)


zscore_STO1=STO1 %>% mutate(zscore = abs((chla.ug.per.L - mean(chla.ug.per.L))/sd(chla.ug.per.L)))
outlier_STO1<-subset(zscore_STO1, zscore<3)

outlier_HP1$SampleID
outlier_SB$SampleID
outlier_AR$SampleID
outlier_CB$SampleID
outlier_NB$SampleID
outlier_SR2$SampleID
outlier_NR2$SampleID
outlier_STO1$SampleID
outlier_NTO1$SampleID
outlier_SR8$SampleID


library("multcomp")
Chladata3 <- read.csv('metadata092921.csv', row.names = 1)
Chladata3$Enviro2 = as.factor(Chladata3$Enviro2)



shapiro.test(outlier_AR$chla.ug.per.L)
shapiro.test(log(outlier_AR$chla.ug.per.L))

 kruskal.test(chla.ug.per.L ~ Season, data = NB)




outlier_HP1$Season = as.factor(outlier_HP1$Season)
outlier_AR$Season = as.factor(outlier_AR$Season)
outlier_SR8$Season = as.factor(outlier_SR8$Season)
outlier_SB$Season = as.factor(outlier_SB$Season)
outlier_CB$Season = as.factor(outlier_CB$Season)
outlier_NB$Season = as.factor(outlier_NB$Season)
outlier_SR2$Season = as.factor(outlier_SR2$Season)
outlier_NR2$Season = as.factor(outlier_NR2$Season)
outlier_STO1$Season = as.factor(outlier_STO1$Season)
outlier_NTO1$Season = as.factor(outlier_NTO1$Season)




### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(chla.ug.per.L ~ Season, data = outlier_NTO1)
  amod
  summary(amod)
  
  kruskal.test(chla.ug.per.L ~ Season, data = NB)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  
  ggplot(CB, aes(x = chla.ug.per.L)) +
  geom_histogram(aes(color = Season, fill = Season), 
                position = "identity", bins = 30, alpha = 0.4) + facet_wrap(~Season, scales = "free_y") +
  scale_color_manual(values = c("#00AFBB", "#E7B800", "green", "red")) +
  scale_fill_manual(values = c("#00AFBB", "#E7B800", "green", "red")) +ggtitle("CB")

  hist(log(CB$chla.ug.per.L),breaks=20)
  hist(log(AR$chla.ug.per.L),breaks=20)
 hist(HP1$chla.ug.per.L,breaks=20)
 hist(sqrt(HP1$chla.ug.per.L),breaks=20)
 hist(log(SR8$chla.ug.per.L),breaks=20)
hist(log(SB$chla.ug.per.L),breaks=20)


shapiro.test(sqrt(HP1$chla.ug.per.L))
 
 
 lmMod <- lm(log(chla.ug.per.L) ~ Season, data = HP1)
 par(mfrow=c(2,2)) # init 4 charts in 1 panel
plot(lmMod)
car::ncvTest(lmMod)
#reject the null hypothesis that the variance of the residuals is constant and infer that heteroscedasticity is indeed present

```


```{r}

Feb_blooms=subset_samples(rel2All_phyto, Date1=="11/16/20" | Date1=="9/22/20" | Date1=="12/23/20" |Date1=="2/12/21")
Feb_blooms=subset_samples(Feb_blooms, Site=="CB" | Site=="HP1" | Site=="SB" | Site=="SR8")


April_blooms=subset_samples(rel2All_phyto,  Date1=="11/9/19" |Date1=="1/11/19"|Date1=="2/25/19" |Date1=="3/21/19" | Date1=="4/22/19")
April_blooms=subset_samples(April_blooms, Site=="NR2" | Site=="SR2" | Site=="STO1" | Site=="NTO1")


all_melt=psmelt(Feb_blooms)
all_melt$Date1 <- as.Date(all_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=all_melt, aes(x=Sampling.Order, y=Abundance, fill=Curated_v3))+ facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("Feb Bloom")  + scale_fill_manual(values=palette2) +theme(legend.position = "right")
p

ggsave("Feb_blooms_rel2.pdf", width=20, height=11)

p <- ggplot(data=all_melt, aes(x=Sampling.Order, y=SYN.mL))+ facet_wrap(~Site, scales="free_x") + geom_line(aes(), stat="identity") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("April Bloom")+theme(legend.position = "right")
p



all_melt=psmelt(April_blooms)
all_melt$Date1 <- as.Date(all_melt$Date1,format = "%m/%d/%y")
p <- ggplot(data=all_melt, aes(x=Sampling.Order, y=Abundance, fill=Curated_v3))+ facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("April Bloom")  + scale_fill_manual(values=palette2) +theme(legend.position = "right")
p

ggsave("April_blooms_rel2.pdf", width=20, height=11)

p <- ggplot(data=all_melt, aes(x=Sampling.Order, y=SYN.mL))+ facet_wrap(~Site, scales="free_x") + geom_line(aes(), stat="identity") +theme(axis.text.x = element_text(angle = 90)) +ggtitle("April Bloom")+theme(legend.position = "right")
p

ggsave("April_blooms_rel2.pdf", width=20, height=11)


```




```{r}
##SAR86

rel2_Plas_Bac_16S

###SAR86
SAR86 = subset_taxa(rel2_Plas_Bac_16S, Order == "D_3__SAR86_clade")  
ntaxa(SAR86)

plot_bar(SAR86, fill = "ASV2")

plot_bar(SAR86, x="timepoint", fill="Species") + facet_wrap(~site, scales="free_x")
pdf("check_ASV2.pdf")
plot_tree(SAR86, ladderize="left", label.tips="ASV2",  title="SAR86 Relative Abundances OTU Phylogram") 
dev.off()
###SAR86 table 

SAR11w5_NoChloroNoMito = subset_taxa(NoChloroNoMito, Order == "D_3__SAR11 clade")
ntaxa(SAR11w5_NoChloroNoMito)
tax_table(SAR11w5_NoChloroNoMito)
taxa_names(SAR86)

taxa_names(SAR86) <- paste("SAR86_ASV-", taxa_names(SAR86), sep="")
taxa_names(SAR86)
taxa_names(SAR86) <- 1:ntaxa(SAR86)
taxa_names(SAR86)
taxa_names(SAR86) <- paste("SAR86_ASV", taxa_names(SAR86), sep="")
taxa_names(SAR86)
plot_tree(SAR86, label.tips="taxa_names", ladderize="left")

tax_table(SAR86)
write.csv(tax_table(SAR86), "SAR86_tax.csv")
##note use this and change the main tax file



write.csv(taxa_names(SAR86), "SAR86.csv")


```

```{bash}
cd /Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq
grep -A 1 -f <(cut -d ',' -f 2 SAR86.csv | tr -d "\"") 2021_2019_merged-dna-sequences.fasta | sed '/^--$/d' >> SAR86_ASV.fasta

###get the seqeunces into tab form
awk 'BEGIN{RS=">"}{print "#"$1"\t"$2;}' SAR86_ASV.fasta | tail -n+2 > SAR86_ASV_reformat2.txt

```


```{r}


test=psmelt(SAR86)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

test$Sampling.Order=as.factor(test$Sampling.Order)
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=ASV2))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  +ggtitle("SAR96 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "Dark2")  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black", fill = NA,size = 1))
ggsave("SAR86_byASV.pdf", width =20)




```


```{r}
metadata_KBytNERR


df_Order_Class<- metadata_KBytNERR %>% drop_na(HBACT.mL) %>% group_by(Site, Season)%>% summarise(mean = round(mean(HBACT.mL), 1), sd=round(sd(HBACT.mL), 1)) 


df_Order_Class$Season=factor(df_Order_Class$Season, levels=c("Summer", "Fall", "Winter", "Spring"))

f3=df_Order_Class%>%
  ggplot(aes( x=as.factor(Season), y=mean)) + geom_bar(stat="identity")+ facet_wrap(~Site) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.title.y = element_text(size = 20), axis.text=element_text(size=20), legend.text=element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + scale_fill_manual(values=palette2) +ggtitle("Heterotrophic Bacteria Cell Counts") + theme(legend.title = element_blank()) + ylab("Average HBACT cell count")

f3





```


```{r}

write.csv(tax_table(rel2Bac_16S ), "Check_98.csv")

SAR86_bact=subset_taxa(rel2Bac_16S , Order=="D_3__SAR86_clade")


test=psmelt(SAR86_bact)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=ASV2))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("SAR86 Abundance Relative to Heterotrophic Community") + scale_color_brewer(palette = "palette2") +xlab("") + ylab("Relative Abundance") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("SAR86_Heterotrophic_Community.pdf", width=20)


```



```{r}

write.csv(tax_table(rel2_Het_Bac_16S), "Check_98.csv")

SAR86_het=subset_taxa(rel2_Het_Bac_16S, Order=="D_3__SAR86_clade")


test=psmelt(SAR86_het)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=ASV2))
p + facet_wrap(~Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("SAR86 Abundance Relative to Heterotrophic Community") + scale_color_brewer(palette = "palette2") +xlab("") + ylab("Relative Abundance") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("SAR86_Heterotrophic_Community.pdf", width=20)



df_Order_Class<- SAR86_het %>%psmelt() %>% group_by(ASV2,Order,Site, Season)%>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 


df_Order_Class$Season=factor(df_Order_Class$Season, levels=c("Summer", "Fall", "Winter", "Spring"))

f3=df_Order_Class%>%
  ggplot(aes(fill=as.factor(ASV2), x=as.factor(Season), y=mean)) + geom_bar(stat="identity")+ facet_wrap(~Site) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.title.y = element_text(size = 20), axis.text=element_text(size=20), legend.text=element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + scale_fill_manual(values=palette2) +ggtitle("SAR86 Heterotrophic Bacteria Only") + theme(legend.title = element_blank()) + ylab("Average Relative Abundance (%)")

f3

ggsave("SAR86_Season_HetBac_only.pdf", width=15, height=10)

```



```{r}

glom=tax_glom(Plas_Bac_16S, "Order")


ds = phyloseq_to_deseq2(glom, ~Season+Site)
ds_pol = DESeq(ds, fitType='mean')
alpha = 0.05
res = results(ds_pol, alpha=alpha)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(glom)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "physeq_DESEQ_Season_Class_glom.csv")


```


```{r}

##########
Abun_By_Class=tax_glom(SAR86, "Order")
otu_table(Abun_By_Class)
ntaxa(Abun_By_Class)
nsamples(Abun_By_Class)

df_Order_Class<- Abun_By_Class %>%psmelt() %>% group_by(OTU,Order,Site, Season)%>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 

library(reshape2)
df_Class_Site_mean=dcast(df_Order_Class, Site ~ Salmaso_Taxonomy, value.var ="mean")
df_Class_Site_sd=dcast(df_Order_Class, OTU+Phylum ~ Enviro, value.var = "sd")

write.csv(df_Class_Site_mean, "df_Class_Enviro_mean.csv")
write.csv(df_Class_Site_sd, "df_Class_Enviro_sd.csv")


library(dplyr)
df_Order_Class_top=subset(df_Order_Class, mean>5)

library(randomcoloR)
n <- 45
palette2 <- distinctColorPalette(n)







f3=df_Order_Class%>%
  ggplot(aes(fill=as.factor(Order), x=as.factor(Season), y=mean)) + geom_bar(stat="identity")+ facet_wrap(~Site) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ theme(axis.title.y = element_text(size = 20), axis.text=element_text(size=20), legend.text=element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + scale_fill_manual(values=palette2) +ggtitle("") + theme(legend.title = element_blank()) + ylab("Average Relative Abundance (%)")

f3

ggsave("SAR86_Season.pdf", width=15, height=10)

```








```{r}
###SAR11
library(randomcoloR)
n <- 18
palette2 <- distinctColorPalette(n)


SAR11_Sites= subset_samples(rel.abun, Site == "HP1" |Site == "STO1")

SAR11= subset_taxa(SAR11_Sites, Order == "D_3__SAR11_clade")
ntaxa(SAR11)

library(RColorBrewer)
display.brewer.all()


test=psmelt(SAR11)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")
p <- ggplot(data=test, aes(x=Date1, y=Abundance, fill=Genus))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack", width = 20) +theme(axis.text.x = element_text(angle = 90)) +ggtitle("SAR11 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "set") +  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
ggsave("SAR11_byGenus.pdf")



test=psmelt(SAR11)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

test$Sampling.Order=as.factor(test$Sampling.Order)
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=Genus))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  +ggtitle("SAR11 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "Dark2")  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black", fill = NA,size = 1))
ggsave("SAR11_byGenus.pdf", width =10)


test=psmelt(rel.abun)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")



library(randomcoloR)
n <- 183
palette2 <- distinctColorPalette(n)

test$Sampling.Order=as.factor(test$Sampling.Order)
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=Order))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  +ggtitle("SAR11 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "Dark2")  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black", fill = NA,size = 1))
ggsave("SAR11_byGenus.pdf", width=10)

```


```{r}

test=psmelt(SAR11)
test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

test$Sampling.Order=as.factor(test$Sampling.Order)
p <- ggplot(data=test, aes(x=Sampling.Order, y=Abundance, fill=Genus))
p + facet_wrap(~Enviro2+Site, scales="free_x") + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  +ggtitle("SAR11 Abundance Relative to Entire Community (w/ Chloroplasts)") + scale_color_brewer(palette = "Dark2")  +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black", fill = NA,size = 1))
ggsave("SAR11_byGenus.svg", width =10)

```




```{r}


test=psmelt(rel2All_phyto)
test$Curated_v3=factor(test$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))




test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

HP1_t=subset(test, Site=="HP1")
p_HP1_t <- ggplot(data=HP1_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance")
p_HP1_t



STO1_t=subset(test, Site=="STO1")
p_STO1_t <- ggplot(data=STO1_t, aes(x=Sampling.Order, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90))  + scale_fill_manual(values=coloresbarplot)+  theme(legend.position = "right")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("STO1") + ylab("Relative abundance")
p_STO1_t




library(rstatix)

library(scales)
library(cowplot)
chla <- no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L) 
summary(chla)


HP1_c=subset(chla, Site=="HP1")

p_HP1_c <- ggplot(data=HP1_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_HP1_c




STO1_c=subset(chla, Site=="STO1")
p_STO1_c <- ggplot(data=STO1_c, aes(x=Sampling.Order, y=chla.ug.per.L))+geom_line(color="black") +geom_point()+ theme(legend.title = element_blank()) + scale_y_continuous(trans = log10_trans())+ ylab("Chlorophyll-a (ug per L)") +xlab("") + theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
p_STO1_c


#################

tog_HP1=plot_grid(p_HP1_c, p_HP1_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2), rel_widths = c(1, -0.1, 1))
tog_HP1

tog_STO1=plot_grid(p_STO1_c, p_STO1_t, labels = c('', ''), align = 'v',label_size = 12, ncol = 1, rel_heights = c(0.5,2))
tog_STO1

plot_grid(tog_HP1, tog_STO1, align="hv", ncol=3, nrow=4)

#plot_grid(p_HP1_c, p_AR_c, p_HP1_t, p_AR_t, labels = c('', '', "", ""), align = 'vh',label_size = 12, ncol = 2, rel_heights = c(0.5,2))


ggsave("chla_barchart_HP1_STO1_legend.svg", width=20, height=24)



```



```{r}

no_dup_metadata_KBytNERR$Site=factor(no_dup_metadata_KBytNERR$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "CB", "SB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))



    geom_violin(color="black", fill="gray53") 

library(rstatix)

library(scales)
chla <- no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L) %>% group_by(Site) %>% summarise(mean = mean(chla.ug.per.L)) 

Phos <- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=Phosphate))+geom_boxplot(color="black", fill="gray53") 
Phos=Phos + theme(legend.title = element_blank()) + ylab("Phosphate (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Phos

Nitrate <- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=NitrateNitrite))+geom_boxplot(color="black", fill="gray53") 
Nitrate=Nitrate + theme(legend.title = element_blank()) + ylab("Nitrate+Nitrite (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Nitrate




Silicate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=Silicate))+geom_boxplot(color="black", fill="gray53") 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Silicate (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Silicate




Ammonia<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=Ammonia))+geom_boxplot(color="black", fill="gray53") 
Ammonia=Ammonia + theme(legend.title = element_blank()) + ylab("Ammonia (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Ammonia


MyPlots = list(Phos, Nitrate,Silicate, Ammonia)

pdf("Nutrient_boxplots.pdf")
MyPlots
dev.off()



no_dup_metadata_KBytNERR$Season=factor(no_dup_metadata_KBytNERR$Season, levels=c("Summer", "Fall", "Winter", "Spring"))


Ammonia<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Season), y=HBACT.mL))+geom_boxplot(color="black", fill="gray53") + facet_wrap(~Site, scales="free_y")
Ammonia=Ammonia + theme(legend.title = element_blank()) + ylab("HBACT per mL") +xlab("Season") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Ammonia


ggsave("HBACT_per_mL_perSeason.pdf")


ntaxa(All_phyto)

```



```{r}


rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))
ntaxa(rel.abun)
#3987
nsamples(rel.abun)
#366

write.csv(otu_table(physeq_Dav_no_dup_clean_no18S), "Count_data_physeq_Dav_no_dup_clean_no18S.csv")
write.csv(tax_table(physeq_Dav_no_dup_clean_no18S), "Taxonomy_Count_data_physeq_Dav_no_dup_clean_no18S.csv")

All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")

write.csv(otu_table(All_phyto), "Count_data_All_phyto.csv")
write.csv(tax_table(All_phyto), "Taxonomy_Count_data_All_phyto.csv")

#counts.filtered.phyto=read.csv("Count_data_All_phyto_taxonomy.csv", header=TRUE)

Count_tax_physeq_Dav_no_dup_clean_no18S.csv

counts.filtered.phyto=read.csv("Count_tax_physeq_Dav_no_dup_clean_no18S.csv", header=TRUE)
# (3) Create taxonomy key:
names(counts.filtered.phyto)
seq_counts<-counts.filtered.phyto[1:367]
tax_key<-counts.filtered.phyto[c(1,368:384)]; head(tax_key[1:2,])
# Modify seq_counts data frame so row.names = OTU.ID and all values are numeric
row.names(seq_counts)<-seq_counts$OTU.ID
seq_counts$OTU.ID<-NULL; head(seq_counts[1:2,])

```



```{r}
## Another approach to PCA ordination w/Compositional Data: Log-Ratio Transformations
# Log-ratio
log_rats<-data.frame(compositions::ilr(t(seq_counts)))
write.csv(log_rats, "log_rats_all.csv")
log_rats=read.csv("log_rats_all.csv", row.names=1)

??compositions
# These are complicated to interpret however because you move from D simplex to D-1 euclidean
# But using these we can see we are now in invertible covariance regime

## Checking the difference the log-ratio made on the data characteristics
new_covdet<-det(as.matrix(log_rats)%*%t(log_rats)) #Original Count Data
new_covdet # After doing a log ratio, you see that none of the axes are equal to 0
```



```{r}
euc_dmat<-dist(log_rats) 
plot(hclust(euc_dmat))

library(reshape2); 
library(vegan); 
library(dplyr)
library(ade4); 
library(plotly)
library(compositions); 
library(pracma); 
library(DESeq2); 
library(fpc); 
library(tidyverse)
library(purrr)
library(cluster)
library(RColorBrewer)
library(ape)
library(factoextra)

df=euc_dmat


library("metagMisc")
test=dist2list(df, tri = TRUE)
test2=dcast(test,col~row)

ugh=as.data.frame(df)

k2 <- kmeans(df, centers = 2, nstart = 25)
str(k2)
fviz_cluster(k2, data = df)


k2 <- kmeans(df, centers = 2, nstart = 25)
k3 <- kmeans(df, centers = 3, nstart = 25)
k4 <- kmeans(df, centers = 4, nstart = 25)
k5 <- kmeans(df, centers = 5, nstart = 25)

# plots to compare
p1 <- fviz_cluster(k2, geom = "point", data = df) + ggtitle("k = 2")
p1
p2 <- fviz_cluster(k3, geom = "point",  data = df) + ggtitle("k = 3")
p2
p3 <- fviz_cluster(k4, geom = "point",  data = df) + ggtitle("k = 4")
p4 <- fviz_cluster(k5, geom = "point",  data = df) + ggtitle("k = 5")


library(gridExtra)
pdf("test.pdf")
grid.arrange(p1, p2, p3, p4, nrow = 2)
dev.off()
ggsave("all_taxa_euc_lograt_gridarrange.pdf")
set.seed(123)


set.seed(123)

# function to compute total within-cluster sum of square 
wss <- function(k) {
  kmeans(df, k, nstart = 10 )$tot.withinss
}

# Compute and plot wss for k = 1 to k = 15
k.values <- 1:15

# extract wss for 2-15 clusters
wss_values <- map_dbl(k.values, wss)

plot(k.values, wss_values,
       type="b", pch = 19, frame = FALSE, 
       xlab="Number of clusters K",
       ylab="Total within-clusters sum of squares")


# function to compute average silhouette for k clusters
avg_sil <- function(k) {
  km.res <- kmeans(df, centers = k, nstart = 25)
  ss <- silhouette(km.res$cluster, dist(df))
  mean(ss[, 3])
}

# Compute and plot wss for k = 2 to k = 15
k.values <- 2:15

# extract avg silhouette for 2-15 clusters
avg_sil_values <- map_dbl(k.values, avg_sil)

plot(k.values, avg_sil_values,
       type = "b", pch = 19, frame = FALSE, 
       xlab = "Number of clusters K",
       ylab = "Average Silhouettes")





fviz_nbclust(k2, kmeans, method = "wss")

fviz_nbclust(test, kmeans, method = "silhouette")

set.seed(123)
gap_stat <- clusGap(test, FUN = kmeans, nstart = 25,
                    K.max = 10, B = 50)
# Print the result
print(gap_stat, method = "firstmax")

fviz_gap_stat(gap_stat)


set.seed(123)
final <- kmeans(df, 4, nstart = 25)
print(final)
fviz_cluster(final, data = df)


```


```{r}
library(colorspace)
library(dendextend)
library(DESeq2)


Clustering_type <- factor(sample_data(All_phyto)$Clustering, levels=c("Cluster_1","Cluster_2","Cluster_3"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_cluster <- c("coral1","burlywood1","lightblue")
col_Clustering_type <- colors_cluster[Clustering_type]





Enviro_type <- factor(sample_data(All_phyto)$Environment, levels=c("Estuary", "Coastal", "Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_c <- c("firebrick3","gold", "dodgerblue1")
col_Enviro_type <- colors_c[Enviro_type]



Season_type <- factor(sample_data(All_phyto)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkorange","darkturquoise", "darkorchid","darkolivegreen1" )
col_Season_type <- colors_Season[Season_type]


Year_type <- factor(sample_data(All_phyto)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]


Site_type <- factor(sample_data(All_phyto)$Site, levels=c("WAI2", "KAHO","HP1", "CB","AR" ,"SR8", "SB", "NB", "SR2", "NR2", "STO1", "NTO1"))
colors_Site <- c('#7F0000','#f94279',"#ffd000","#ffa812","#c78100", "#ff9966", "#c46210", "#ff7e00","#009dcc","#00bfcc", '#0085ff','#0045ff')
col_Site_type <- colors_Site[Site_type]


euc_dmat<-dist(log_rats) 
plot(hclust(euc_dmat))

euc_clust <- hclust(euc_dmat)
d1=hclust(euc_dmat)%>% as.dendrogram() 
plot(d1)

### for the colored bars 
k2 <- cutree(d1, k = 2)
cols2 <- c("#0082CE", "#CC476B")[k2]

### for the colored bars 
k234 <- cutree(d1, k = 2:4)


pdf("By_testing.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1 %>% 
    set("labels_cex",0.2) %>% 
    plot(main="Euclidean Complete", horiz = TRUE)
colored_bars(cbind(k234[,3:1], col_Site_type, col_Season_type, col_Enviro_type), d1, rowLabels = c(paste0("k = ", 4:2), "Sites","Seasons" ,"Enviroment"), horiz = TRUE, y_shift=6)
dev.off()



```





```{r}

# AND change command so its the same.
lograt_pca<-prcomp(log_rats)
# Look at this
lograt_variances<-lograt_pca$sdev^2/sum(lograt_pca$sdev^2)
barplot(lograt_variances,
        main='Log-Ratio PCA Screeplot',
        xlab='PC Axis',
        ylab='% Variance',
       col=c(rep('black',1),rep('grey',18)),
       cex.names=1.5,cex.axis=1.5,cex.lab=1.5,cex.main=1.5)
legend('topright',fill=c('black','grey'),c('Should Present','??? Judgment Call'))

## We conclude a more faithful representation is to plot this ordination in 2D because the 3rd and 4th axes appear
## very similar, and we can't construct a 4D plot
pca_lograt_frame<-data.frame(lograt_pca$x,
                          time_of_day=gsub('^.*_','',rownames(lograt_pca$x)))

pca_lograt_fram_meta=pca_lograt_frame %>%
    rownames_to_column("SampleID")

library(tidyverse)
library(reshape2)

pca_lograt_frame_meta=left_join(pca_lograt_fram_meta,no_dup_metadata_KBytNERR, by="SampleID")




#no_dup_metadata_KBytNERR=read.csv("v4_nodups_wo_AprFP_2019_2021_03142022.csv", header=TRUE, fill=TRUE)




ggplot(pca_lograt_frame_meta, groups=as.factor(pca_lograt_frame_meta$kmeans3))+
    geom_point(aes(x=PC1,y=PC2,colour=as.factor(pca_lograt_frame_meta$kmeans3)))+
    ylab(paste0('PC2 ',round(lograt_variances[2]*100,2),'%'))+
    xlab(paste0('PC1 ',round(lograt_variances[1]*100,2),'%'))+
    ggtitle('Log-Ratio PCA Ordination')+
    coord_fixed(ratio=lograt_variances[2]/lograt_variances[1])+
    theme_bw()+ scale_colour_manual(name="Clustering", values= c("#8b0000","#e6d300", "#00528b"))
ggsave("PCA_all_taxa_log_ratio.pdf")




ggplot(pca_lograt_frame_meta, groups=as.factor(pca_lograt_frame_meta$Season))+
    geom_point(aes(x=PC1,y=PC2,colour=as.factor(pca_lograt_frame_meta$Season)))+
    ylab(paste0('PC2 ',round(lograt_variances[2]*100,2),'%'))+
    xlab(paste0('PC1 ',round(lograt_variances[1]*100,2),'%'))+
    ggtitle('Log-Ratio PCA Ordination')+
    coord_fixed(ratio=lograt_variances[2]/lograt_variances[1])+
    theme_bw()+ scale_colour_manual(name="Clustering", values= c("#8b0000","#e6d300", "#00528b", "purple"))
ggsave("PCA_all_taxa_log_ratio_season.pdf")


ggplot(pca_lograt_frame_meta, groups=as.factor(pca_lograt_frame_meta$Site))+
    geom_point(aes(x=PC1,y=PC2,colour=as.factor(pca_lograt_frame_meta$Site), shape=as.factor(pca_lograt_frame_meta$Season)))+
    ylab(paste0('PC2 ',round(lograt_variances[2]*100,2),'%'))+
    xlab(paste0('PC1 ',round(lograt_variances[1]*100,2),'%'))+
    ggtitle('Log-Ratio PCA Ordination')+
    coord_fixed(ratio=lograt_variances[2]/lograt_variances[1])+
    theme_bw()+ scale_colour_manual(name="Clustering", values= c("#8b0000","#e6d300", "#00528b", "purple", "orange", "green", "black", "coral", "pink", "tan", "aquamarine", "grey"))
ggsave("PCA_all_taxa_log_ratio_site_season.pdf")

```
```{r}
jac_dmat<-vegdist(t(seq_counts),method="jaccard") # p/a example
pcoa_jac<-ape::pcoa(jac_dmat) #perform PCoA
# Now we need to inspect how the PCoA turned out. We look at a Screeplot for this.
# The screeplot shows the amount of variance explained by each axis
samp_no<-dim(seq_counts)[2]
jac_variances<-pcoa_jac$values$Relative_eig
par(mar=c(5,6,4,1)+.1)

barplot(jac_variances,
        xlab='Principal Coordinate Axis',
        ylab='% Variance',
       col=c(rep('black',2),'darkgrey',rep('lightgrey',16)))
legend('topright',fill=c('black','darkgrey','lightgrey'),c('Should Present','Judgment Call','Unnecessary to Present'),cex=2)

```
```{r}

euc_dmat<-dist(log_rats) 
plot(hclust(euc_dmat))

## Conduct ordination w/distance matrix
pcoa_euc<-ape::pcoa(euc_dmat)
#euc_variances<-pcoa_euc$sdev^2/sum(pcoa_euc$sdev^2)
euc_variances<-pcoa_euc$values$Relative_eig
par(mar=c(5,6,4,1)+.1)
barplot(euc_variances,
        xlab='Principal Coordinate Axis',
        ylab='% Variance',
       col=c(rep('black',2),rep('darkgrey',2),rep('lightgrey',17)))
legend('topright',fill=c('black','darkgrey','lightgrey'),c('Should Present','Judgment Call','Unnecessary'),cex=2)
#
```


```{r}
pcoa_jac_frame<-data.frame(pcoa_jac$vectors,time_of_day=gsub('^.*_','',rownames(pcoa_jac$vectors)))
eigenvalues<-round(jac_variances,4)*100
plot_ly(pcoa_jac_frame,type='scatter3d',mode='markers',
        x=~Axis.2,y=~Axis.3,z=~Axis.1,colors=~brewer.pal(6,'Set1'),color=~time_of_day)%>%
  layout(font=list(size=18),
         #title='PCoA Jaccard Distance',
         scene=list(xaxis=list(title=paste0('Co 2 ',eigenvalues[2],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
         yaxis=list(title=paste0('Co 3 ',eigenvalues[3],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
         zaxis=list(title=paste0('Co 1 ',eigenvalues[1],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
        aspectratio = list(x=3*eigenvalues[2]/eigenvalues[1])), y=3*eigenvalues[3]/eigenvalues[1], z=3)

## To compare with the euclidean distance ordination representing differences in relative composition
#pcoa_euc_frame<-data.frame(pcoa_euc$x,
#                          time_of_day=gsub('^.*_','',rownames(pcoa_euc$x)))
pcoa_euc_frame<-data.frame(pcoa_euc$vectors,time_of_day=gsub('^.*_','',rownames(pcoa_euc$vectors)))
euc_eigenvalues<-round(euc_variances,4)*100
plot_ly(pcoa_euc_frame,type='scatter3d',mode='markers',
        x=~Axis.3,y=~Axis.2,z=~Axis.1,colors=~brewer.pal(6,'Set1'),color=~time_of_day)%>%
  layout(font=list(size=18),
         #title='PCoA Euclidean Distance',
         scene=list(xaxis=list(title=paste0('Co 3 ',euc_eigenvalues[3],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
         yaxis=list(title=paste0('Co 2 ',euc_eigenvalues[2],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
         zaxis=list(title=paste0('Co 1 ',euc_eigenvalues[1],'%'),
                    showticklabels=FALSE,zerolinecolor='black'),
                   aspectratio = list(x=3*euc_eigenvalues[3]/euc_eigenvalues[1], 
                                      y=3*euc_eigenvalues[2]/euc_eigenvalues[1], 
                                      z=3)))
```

```{r}
library(vegan)

# To corroborate the 3-D plot, use a simple average 
## hierarchical clustering and plot the dendrogram.
## Some of the same pattern emerge, but it is not as 
## clear as the 3-D representation.
check=as.table(vegdist(t(seq_counts)))
write.csv(as)

cluster_ex<-hclust(vegdist(t(seq_counts),method='jaccard'),method="complete") #Using Jaccard distance so apples to apples
plot(cluster_ex,main='Jaccard Hierarchical Agglomerative Clustering',xlab='',sub='')

```

```{r}
set.seed(071510) # setting random seed to assure NMDS comes out the same every time
## So we can compare the relative composition based distance metric to the presence/absence based distance metric
euc_nmds<-metaMDS(euc_dmat,k=2,autotransform=FALSE)
jac_nmds<-metaMDS(jac_dmat,k=2,autotransform=FALSE)

euc_nmds$stress #Stress ~0.2094912
jac_nmds$stress #Stress ~0.1391334 So the Jaccard NMDS is a ***slightly*** more parsimonious ordination.
```

```{r}
# Additionally, the axes for NMDS are totally arbitrary, so axis scaling does not matter
# and data can be rotated/reflected about axes and the NMDS is still the same
euc_frame<-data.frame(euc_nmds$points,
                         time_of_day=gsub('^.*_','',rownames(log_rats)))
jac_frame<-data.frame(jac_nmds$points,
                     time_of_day=gsub('^.*_','',rownames(log_rats)))
## Plotting euclidean distance NMDS
ggplot(euc_frame,aes(x=MDS1,y=MDS2,col=time_of_day))+
    geom_point(size=2)+
    theme_bw()+ggtitle('Euclidean Distance NMDS')+ theme(legend.position="none")
## Plotting Jaccard distance NMDS
ggplot(jac_frame,aes(x=MDS1,y=MDS2,col=time_of_day))+
    geom_point(size=2)+
    theme_bw()+ggtitle('Jaccard Distance NMDS')+ theme(legend.position="none")
## For paper figures
euc_fig<-ggplot(euc_frame,aes(x=MDS1,y=MDS2,col=time_of_day))+
    geom_point(size=10)+
    scale_color_brewer(palette='Set1',name='Time of Day')+
    theme_bw()+theme(legend.position='none',text=element_text(size=24))+
xlab('NMDS 1')+ylab('NMDS 2')
jac_fig<-ggplot(jac_frame,aes(x=MDS1,y=MDS2,col=time_of_day))+
    geom_point(size=10)+
    scale_color_brewer(palette='Set1',name='Time of Day')+
    theme_bw()+theme(legend.position='none',text=element_text(size=24))+
xlab('NMDS1')+ylab('NMDS2')
euc_fig
jac_fig
```









######Now this gets into co-occurrence 



```{r}
## We stabilize the variances so that they are no longer correlated to the species abundances.
transformed_counts<-DESeq2::varianceStabilizingTransformation(as.matrix(seq_counts))
```


```{r}
## Check to see how the transformation did
within_trans_means<-apply(transformed_counts,1,mean)
within_trans_vars<-apply(transformed_counts,1,var)
# Plot:
plot(within_trans_means,within_trans_vars)

```
```{r}
# Since the goal is to identify repeating day/night patterns, we need to remove trends we are not considering,
# i.e., linear trends
#?pracma::detrend() for more information
transformed_detrended<-apply(transformed_counts,1,pracma::detrend)
# Now we rescale so numbers are again intercomparable and now overdispersion has been
# dealt with
trans_dt_scaled<-apply(transformed_detrended,2,scale)
## We lost rownames so tack those back on
rownames(trans_dt_scaled)<-colnames(transformed_counts)
#
## Comparing the distribution of the data along every step of the pipeline -- notice how by the time we get to the 
## z-scores we're looking more like a standard normal
hist(transformed_counts,
    main='VST Sequence Count Observations',
    xlab='# Total Observations',
    ylab='Frequency')
hist(transformed_detrended,
    main='VST+Detrended Data',
    xlab='Total Observations Accounting for Linear Trends',
    ylab='Frequency')
hist(trans_dt_scaled,
    main='VST+Detrended+Scaled Data (Z-scores)',
    xlab='Expression Level Relative to per-OTU Avg',
    ylab='Frequency')
```


```{r}
## Check to see how the transformation did
within_trans_means<-apply(trans_dt_scaled,1,mean)
within_trans_vars<-apply(trans_dt_scaled,1,var)
# Plot:
plot(within_trans_means,within_trans_vars)

```



```{r}
# Now we're ready to generate a distance matrix for clustering
# Recall what we previously learned about different distance metrics.
# We use euclidean distance below:
temporal_dmat<-dist(t(trans_dt_scaled))
n_clusts<-2:20 # This sets the range of # clusters we divide the data into
# First, the classic hierarchical agglomerative clustering
hc_full_cluster<-hclust(temporal_dmat)
plot(hc_full_cluster)
hc_clusts<-lapply(n_clusts,function(x) cutree(hc_full_cluster,x))
## Now we try k-medoids, also a very popular clustering method
## This particular implementation is not optimized for speed, so this may take a minute or so
kmed_clusts<-lapply(n_clusts, function(x) cluster::pam(temporal_dmat,k=x))



```



```{r}
## Comparing cluster outcomes
hc_stats<-lapply(hc_clusts,function(x) fpc::cluster.stats(temporal_dmat,
                                                          clustering=x))
kmed_stats<-lapply(kmed_clusts, function(x) fpc::cluster.stats(temporal_dmat,
                                                             clustering=x$clustering))

## We write a helper function to be less redundant
ripping_stats<-function(list,func){
  ## Essentially all this function does is implements a function (func) on a list
  ## and coerces the output to a column vector (this will be handy when we want to make a data frame)
  output<-do.call(rbind,lapply(list,func))
  return(output)
}
func_list<-rep(list(function(x) x$cluster.number,
                function(x) x$within.cluster.ss,
                function(x) x$avg.silwidth,
                function(x) x$ch),2)
stats_list<-rep(list(hc_stats,kmed_stats),each=4)
collected_stats<-purrr::map2(stats_list,func_list,ripping_stats)
nclusts<-rep(n_clusts,length(collected_stats))
method<-rep(c('hc_agg','kmed'),each=length(n_clusts)*length(collected_stats)/2)
ind_name<-rep(c('n','ss','sil','ch'),each=length(n_clusts)) %>%
  rep(2)
                    
## Collecting outputs for plotting                    
index_frame<-data.frame(index=do.call(rbind,collected_stats),
                        nc=nclusts,
                        Method=method,
                        ind=ind_name)
index_frame %>%
  filter(ind=='ss') %>%
  ggplot(aes(x=nc,y=index,col=Method)) +
  geom_point() +
  geom_line(aes(group=Method)) +
  ylab('Within Cluster Sum Square Error') +
  xlab('Number Clusters')+
  theme_bw()
## Interpretation of plots:                    
# As we add more clusters (yaxis), we want the sum square error to decrease
# As we allow for more clusters, we are able to describe more of the variability in the data
# so the sum square error decreases.
## trade-off - it will always decrease when you add clusters, but we want to 
## add as fewer clusters as possible, hence in this example kmed > hc
# for any given amount of clusters, kmed has less error. So we will continue using kmed approach.
```
```{r}
## Plotting out the calinski-harabasz indices for each clustering
## We want this number to be high
index_frame %>%
  filter(ind=='ch') %>%
  ggplot(aes(x=nc,y=index,col=Method)) +
  geom_point() +
  geom_line(aes(group=Method)) +
  ylab('C-H Stat') +
  xlab('Number Clusters')+
theme_bw()
```
```{r}
## Silhouette profile: Looking at species variability w/in our 8 clusters
## Extract silhouette profile for the clustering we select
silhouette_profile_kmed8<-cluster::silhouette(kmed_clusts[[13]])
## Reformat for easier plotting
silhouette_frame<-data.frame(cluster=silhouette_profile_kmed8[,1],
                             neighbor=silhouette_profile_kmed8[,2],
                             dist=silhouette_profile_kmed8[,3],
                             otu_id=rownames(silhouette_profile_kmed8))

## Sorting by cluster and ranking by silhouette width
new_silframe<-silhouette_frame %>%
  arrange(dist) %>%
  group_by(cluster) %>%
  mutate(idno=1:n(),
         tot_num=n())

## Plotting silhouette profile of OTUs for each cluster
ggplot(new_silframe,aes(x=idno,y=dist))+
  geom_bar(stat='identity') +
  coord_flip() +
  ggtitle('Silhouette Profiles for Individual Clusters') +
  facet_wrap(~cluster)

```
```{r}
# What is the taxonomic composition of each of the 8 clusters?
## To make ecological sense of the temporal clusters,
## we can pull out a representative species and use their signal over time
## to summarize the temporal trends across the whole cluster
medoid_otus<-kmed_clusts[[13]]$medoids
## Extracting those time series
medoid_dynamics<-trans_dt_scaled[,medoid_otus]

```

```{r}
# Using the output from:
head(silhouette_frame)
# We can re-build what the taxonomic composition of each 
## of these clusters is.
range(silhouette_frame$cluster) # n=8 clusters with test data

```
```{r}
# Join cluster information with taxonomic information
colnames(tax_key)[1]<-"otu_id"
tax_bycluster<-left_join(silhouette_frame, tax_key, by="otu_id") # Recall the taxonomy key we made from above?
head(tax_bycluster[1:4,])
```
```{r}
# Summarize taxa with cluster information
tax_bycluster_sum<-tax_bycluster %>%
    group_by(cluster, Curated_v3) %>%
    summarise(richness =  n()) %>% #Get richness information
    group_by(Curated_v3) %>%
    mutate(n_per_tax=sum(richness)) %>% #Figure out total number of OTUs assigned to each taxon
    as.data.frame

```


```{r}
# Plot cluster-to-cluster taxonomic differences
## Values are equal to the total number of OTUs 
## belonging to each taxonomic group
#
tax_color<-c("#67000d","#e31a1c","#dd3497","#fcbba1","#fed976","#fc8d59","#a63603","#addd8e","#7f2704","#238b45","#a1d99b","#081d58","#1f78b4","#a6cee3","#8c6bb1","#9e9ac8","#984ea3","#081d58","#662506","#ffffff","#969696","#525252","#000000")


#green
coloresbarplot=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="forestgreen",
"Archaeplastida:Bathycoccus"="darkolivegreen4", 
"Archaeplastida:Ostreococcus"="darkolivegreen3",
"Archaeplastida:Micromonas"="darkseagreen",
"Archaeplastida:Pedinophyceae"="limegreen",
"Archaeplastida:Other"="chartreuse4", 
"Hacrobia:Other"="thistle2",
"Hacrobia:Cryptomonadales"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiales"="red",
"Stramenopiles:Diatoms"="gold",
"Stramenopiles:Dictyochophyceae"="khaki1",
"Stramenopiles:Pelagophyceae"="darkgoldenrod3",
"Stramenopiles:Other"="yellow",
'Eukaryota:Other'="gray44",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")





tax_bycluster_sum$Curated_v3=factor(tax_bycluster_sum$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",
"Eukaryota:Other"))




#
ggplot(tax_bycluster_sum, aes(x=cluster, y=richness, fill=as.factor(Curated_v3)))+
    geom_bar(color="black", stat="identity")+
    coord_flip()+
    scale_fill_manual(values=coloresbarplot)+
    theme_minimal()+
    scale_x_continuous(breaks=1:13,labels=as.character(1:13),name='Cluster #')+
    scale_y_continuous(name='# Species')+
theme(legend.position='bottom',text=element_text(size=18))+ ## Make the cluster #s happen
guides(fill=guide_legend(nrow=7,byrow=TRUE))

ggsave("Clustering_of_taxonomic_Communities.pdf", width=15, height=10)










```


```{r}

physeq_Dav_no_dup_clean

Cluster1=subset_samples(noNA_All_phyto, kmeans3=="Cluster_1")
Cluster2=subset_samples(noNA_All_phyto, kmeans3=="Cluster_2")
Cluster3=subset_samples(noNA_All_phyto, kmeans3=="Cluster_3")
physeq_Dav_no_dup_clean


##3i think chosing this 
bl.phy.asinh <- transform_sample_counts(Cluster3, function(x) asinh(x))
physeq.list <- NULL

psmelt_dplyr = function(physeq) {
  #Implementation of the psmelt from phyloseq with dplyr
  # It is indeed faster, without further complications or differences. 
  # (well, in fact, its way more prone to give errors if the output is not well established, 
  # it doesn't check anything)
  
  sd = data.frame(sample_data(physeq)) %>% 
    rownames_to_column("Sample")
  TT = data.frame(as(tax_table(physeq),'matrix')) %>%
    rownames_to_column("OTU")
  if(taxa_are_rows(physeq)){
    otu.table = data.frame(as(otu_table(physeq),"matrix"),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  } else {
    otu.table = data.frame(t(as(otu_table(physeq),"matrix")),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  }
  
  all <- otu.table %>%
    left_join(TT, by = "OTU") %>%
    gather_("Sample", "Abundance", setdiff(colnames(otu.table), "OTU")) %>%
    left_join(sd, by = "Sample") %>% 
    select(Sample, Abundance, OTU, everything())
  
  return(all)
}


tsdf.0.1 <- bl.phy.asinh %>% 
  psmelt_dplyr() 


library(tidyverse)
library(dplyr)


df.ts <- tsdf.0.1 %>% 
  arrange(decimal_dat) %>% 
  select(decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)

library(lomb)

lomb.03  <-  df.ts %>% 
  map(~lsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5))

ibex.spec <- lsp(ibex[,2:3],type='period', from=12,to=36,ofac=10, plot=FALSE)
plot.lsp(ibex.spec, main="Tb in Capra ibex",xlabel="Period (h)",ylabel="Power",level=FALSE, xlim=c(0,0.1))


df.ts.X <- XX %>% 
  arrange(decimal_dat) %>% 
  select(decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)


XX=subset(tsdf.0.1, OTU=="ba8a34138faff25a32e8fca2df7342d1")


lomb.03.X  <-  df.ts.X %>% 
  map(~lsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5))



  
lomb.02  <-  df.ts %>% 
  map(~randlsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5))



summary.randlsp(lomb.02)





write.csv(lomb.sea.02, "lomb_cluster.csv")









library(fdrtool)
lomb.sea.03.ofac <- tibble( asv = names(lomb.03),
                       pval = map_dbl(lomb.03, ~.x$p.value),
                       peak = map_dbl(lomb.03, ~.x$peak),
                       interval  = map(lomb.03, ~.x$peak.at),
                       int.min = map_dbl(interval, ~.[[2]]),
                       int.max = map_dbl(interval, ~.[[1]])) %>% 
  drop_na(int.max) %>%
  mutate(qval = fdrtool::fdrtool(pval, statistic = 'pvalue')$qval) %>% 
  filter(qval <= 0.01)


results.lomb03 <- lomb.03[lomb.sea.03 %>% pull(asv)]

?lil.strip

# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))


periodoplots <- map(results.lomb03, ~tibble( scanned = .x$scanned,
                          power = .x$power)) %>% 
  bind_rows(.id = 'asv') %>% 
  split(.$asv) %>% 
  map(~ggplot(.x, aes(scanned, power)) + 
        geom_line(aes(group = asv)) + 
        facet_wrap(~asv) + 
        lil.strip)


periodoplots




```

```{r}


HP1=subset_samples(noNA_All_phyto, Site=="HP1")
Cluster2=subset_samples(noNA_All_phyto, kmeans3=="Cluster_2")
Cluster3=subset_samples(noNA_All_phyto, kmeans3=="Cluster_3")
physeq_Dav_no_dup_clean


##3i think chosing this 
bl.phy.asinh <- transform_sample_counts(HP1, function(x) asinh(x))
physeq.list <- NULL

psmelt_dplyr = function(physeq) {
  #Implementation of the psmelt from phyloseq with dplyr
  # It is indeed faster, without further complications or differences. 
  # (well, in fact, its way more prone to give errors if the output is not well established, 
  # it doesn't check anything)
  
  sd = data.frame(sample_data(physeq)) %>% 
    rownames_to_column("Sample")
  TT = data.frame(as(tax_table(physeq),'matrix')) %>%
    rownames_to_column("OTU")
  if(taxa_are_rows(physeq)){
    otu.table = data.frame(as(otu_table(physeq),"matrix"),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  } else {
    otu.table = data.frame(t(as(otu_table(physeq),"matrix")),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  }
  
  all <- otu.table %>%
    left_join(TT, by = "OTU") %>%
    gather_("Sample", "Abundance", setdiff(colnames(otu.table), "OTU")) %>%
    left_join(sd, by = "Sample") %>% 
    select(Sample, Abundance, OTU, everything())
  
  return(all)
}


tsdf.0.1 <- bl.phy.asinh %>% 
  psmelt_dplyr()

tsdf.0.2=subset(tsdf.0.2, OTU=="asv1"| OTU=="asv2"|OTU=="asv3"|OTU=="asv4"|OTU=="asv5")


chla_samples$datetime <- as.POSIXlt(chla_samples$Date2)

chla_samples$decimal=decimal_date(chla_samples$datetime)


df.ts <- tsdf.0.1 %>% 
  arrange(Date2) %>% 
  select(Date2, decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)



  
lomb.02.HP1  <-  df.ts %>% 
  map(~randlsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T))



summary.randlsp(lomb.02)


library(fdrtool)
lomb.sea.HP1 <- tibble( asv = names(lomb.02.HP1),
                       pval = map_dbl(lomb.02, ~.x$p.value),
                       peak = map_dbl(lomb.02, ~.x$peak),
                       interval  = map(lomb.02, ~.x$peak.at),
                       int.min = map_dbl(interval, ~.[[2]]),
                       int.max = map_dbl(interval, ~.[[1]])) %>% 
  mutate(qval = fdrtool::fdrtool(pval, statistic = 'pvalue')$qval) %>% 
  filter(qval <= 0.01, peak >= 10, int.max <= 2)



write.csv(lomb.sea.02, "lomb_cluster.csv")




```




```{r}
results.lomb02 <- lomb.02[lomb.sea.02 %>% pull(asv)]

?lil.strip

# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))


periodoplots <- map(results.lomb02, ~tibble( scanned = .x$scanned,
                          power = .x$power)) %>% 
  bind_rows(.id = 'asv') %>% 
  split(.$asv) %>% 
  map(~ggplot(.x, aes(scanned, power)) + 
        geom_line(aes(group = asv)) + 
        facet_wrap(~asv) + 
        lil.strip)


periodoplots
```





```{r}





####for nMDS

library(devtools)
library("metagMisc")
library(readr)
library(vegan)
library(fossil)
library(ggplot2)
library(mvabund)
library(dendextend)
library(colorspace)
library("factoextra")
library("cluster")




Enviro <- read.csv("/Users/sarahtucker/Documents/Dissertation/chapter 2/metadata/clustering_v5_only_inMiSeqAnalyses.csv", row.names = 1)
Enviro=Enviro%>%drop_na()
pop=Enviro[,11:18]


pop


#standardize Sepal.Width and Sepal.Length 
iris_new <- pop %>% mutate_each_(list(~scale(.) %>% as.vector),
                                  vars = c("PRO.mL","SYN.mL", "PEUK.mL", "HBACT.mL", "chla.ug.per.L", "SW.Temperature.at.site", "Salinity", "pH"))



Enviro <- read.csv("all_nut_data_KByT_HOT_v3.csv", row.names = 1)
MGonly=subset(Enviro, Metagenome_ID !="NA")


KByT=subset(Enviro, Site !="WAI2")
KByT=subset(KByT, Site !="KAHO")
wNutrients=subset(KByT, Phosphate !="NA")
wNutriendsNoNA=subset(wNutrients, chla.ug.per.L !="NA")
wNutriendsNoNA=subset(wNutriendsNoNA, PRO.mL !="NA")
pop=wNutriendsNoNA[,12:21]

test=pop%>% drop_na()

pop=Enviro[,12:21]
pop
d_iris <- dist(wNutriendsNoNA[,12:21], method="euclidean") 
hist(d_iris)

names(wNutriendsNoNA[,12:21])

normalize(wNutriendsNoNA[,12:21])
?normalize()

#load dplyr package
library(dplyr)

#load dplyr package
library(dplyr)

#standardize Sepal.Width and Sepal.Length 
iris_new <- wNutriendsNoNA %>% mutate_each_(list(~scale(.) %>% as.vector),
                                  vars = c("PRO.mL","SYN.mL", "PEUK.mL", "HBACT.mL", "chla.ug.per.L", "Phosphate", "Silicate", "NitrateNitrite", "SW.Temperature.at.site", "Salinity"))




#define Min-Max normalization function
min_max_norm <- function(x) {
    (x - min(x)) / (max(x) - min(x))
  }

#apply Min-Max normalization to first four columns in iris dataset
iris_norm <- as.data.frame(lapply(wNutriendsNoNA[,12:21], min_max_norm))
rownames(iris_norm) <- rownames(wNutriendsNoNA)

#view first six rows of normalized iris dataset
head(iris_norm)
hist(iris_new2)

#hc_euc_complete  = iris_norm %>% dist(method="euclidean")  %>% hclust(method = "complete") %>% as.dendrogram() 
#plot(hc_euc_complete)

n_clusts<-2:20 # This sets the range of # clusters we divide the data into
# First, the classic hierarchical agglomerative clustering
temporal_dmat  = iris_new %>% dist(method="euclidean")
hc_full_cluster<-hclust(temporal_dmat)
plot(hc_full_cluster)
hc_clusts<-lapply(n_clusts,function(x) cutree(hc_full_cluster,x))
## Now we try k-medoids, also a very popular clustering method
## This particular implementation is not optimized for speed, so this may take a minute or so
kmed_clusts<-lapply(n_clusts, function(x) cluster::pam(temporal_dmat,k=x))



hc_ward  = iris_new %>% dist(method="euclidean")  %>% hclust(method = "complete")
plot(hc_ward)
```





```{r}
## Comparing cluster outcomes
hc_stats<-lapply(hc_clusts,function(x) fpc::cluster.stats(temporal_dmat,
                                                          clustering=x))
kmed_stats<-lapply(kmed_clusts, function(x) fpc::cluster.stats(temporal_dmat,
                                                             clustering=x$clustering))

## We write a helper function to be less redundant
ripping_stats<-function(list,func){
  ## Essentially all this function does is implements a function (func) on a list
  ## and coerces the output to a column vector (this will be handy when we want to make a data frame)
  output<-do.call(rbind,lapply(list,func))
  return(output)
}
func_list<-rep(list(function(x) x$cluster.number,
                function(x) x$within.cluster.ss,
                function(x) x$avg.silwidth,
                function(x) x$ch),2)
stats_list<-rep(list(hc_stats,kmed_stats),each=4)
collected_stats<-purrr::map2(stats_list,func_list,ripping_stats)
nclusts<-rep(n_clusts,length(collected_stats))
method<-rep(c('hc_agg','kmed'),each=length(n_clusts)*length(collected_stats)/2)
ind_name<-rep(c('n','ss','sil','ch'),each=length(n_clusts)) %>%
  rep(2)
                    
## Collecting outputs for plotting                    
index_frame<-data.frame(index=do.call(rbind,collected_stats),
                        nc=nclusts,
                        Method=method,
                        ind=ind_name)
index_frame %>%
  filter(ind=='ss') %>%
  ggplot(aes(x=nc,y=index,col=Method)) +
  geom_point() +
  geom_line(aes(group=Method)) +
  ylab('Within Cluster Sum Square Error') +
  xlab('Number Clusters')+
  theme_bw()
## Interpretation of plots:                    
# As we add more clusters (yaxis), we want the sum square error to decrease
# As we allow for more clusters, we are able to describe more of the variability in the data
# so the sum square error decreases.
## trade-off - it will always decrease when you add clusters, but we want to 
## add as fewer clusters as possible, hence in this example kmed > hc
# for any given amount of clusters, kmed has less error. So we will continue using kmed approach.

km=cluster::pam(iris_new, 5)
plot(cluster::pam(temporal_dmat, 5))

library("factoextra")
library("cluster")

distance <- get_dist(iris_new)
fviz_dist(distance, gradient = list(low = "#00AFBB", mid = "white", high = "#FC4E07"))

fviz_cluster(km, data=iris_norm, ggtheme = theme_minimal())

??fviz_cluster()

fviz_cluster(km, data = iris_new)

set.seed(123)
gap_stat <- clusGap(iris_new, FUN = kmeans, nstart = 25,
                    K.max = 10, B = 50)

fviz_gap_stat(gap_stat)

```

```{r}
## Plotting out the calinski-harabasz indices for each clustering
## We want this number to be high
index_frame %>%
  filter(ind=='ch') %>%
  ggplot(aes(x=nc,y=index,col=Method)) +
  geom_point() +
  geom_line(aes(group=Method)) +
  ylab('C-H Stat') +
  xlab('Number Clusters')+
theme_bw()
```



```{r}


library(reshape2); 
library(vegan); 
library(dplyr)
library(ade4); 
library(plotly)
library(compositions); 
library(pracma); 
library(DESeq2); 
library(fpc); 
library(tidyverse)
library(purrr)
library(cluster)
library(RColorBrewer)
library(ape)
df=iris_new

k2 <- kmeans(df, centers = 2, nstart = 25)
str(k2)
fviz_cluster(k2, data = df)


k2 <- kmeans(df, centers = 2, nstart = 25)
k3 <- kmeans(df, centers = 3, nstart = 25)
k4 <- kmeans(df, centers = 4, nstart = 25)
k5 <- kmeans(df, centers = 5, nstart = 25)

# plots to compare
p1 <- fviz_cluster(k2, geom = "point", data = df) + ggtitle("k = 2")
p1
p2 <- fviz_cluster(k3, geom = "point",  data = df) + ggtitle("k = 3")
p2
p3 <- fviz_cluster(k4, geom = "point",  data = df) + ggtitle("k = 4")
p4 <- fviz_cluster(k5, geom = "point",  data = df) + ggtitle("k = 5")


library(gridExtra)
grid.arrange(p1, p2, p3, p4, nrow = 2)
ggsave("gridarrange.pdf")
set.seed(123)

fviz_nbclust(df, kmeans, method = "wss")

fviz_nbclust(df, kmeans, method = "silhouette")

set.seed(123)
gap_stat <- clusGap(df, FUN = kmeans, nstart = 25,
                    K.max = 10, B = 50)
# Print the result
print(gap_stat, method = "firstmax")

fviz_gap_stat(gap_stat)


set.seed(123)
final <- kmeans(df, 3, nstart = 25)
print(final)
fviz_cluster(final, data = df)

pp=iris_new %>%
  mutate(Cluster = final$cluster) %>%
  group_by(Cluster) %>%
  summarise_all("mean")

iris_new

#https://afit-r.github.io/kmeans_clustering
poo=as.data.frame(final$cluster)
poo$SampleID=row.names(poo)
names(poo)


names(poo)[names(poo) == 'final$cluster'] <- "kmeans5"
names(poo)




iris_new$SampleID=row.names(iris_new)

iris_new=left_join(iris_new, poo, by="SampleID")

names(iris_new)[names(iris_new) == 'final$cluster'] <- "kmeans3"
names(iris_new)




mtcars.pca <- prcomp(iris_new[,1:8], center = TRUE,scale. = FALSE)





summary(mtcars.pca)
str(mtcars.pca)
#install_github("vqv/ggbiplot")

library(ggbiplot)

ggbiplot(mtcars.pca)

Metagenome_type <- factor(iris_new$Metagenome, levels=c("None","KByT", "Aloha"))
iris_new$Metagenome=factor(iris_new$Metagenome, levels=c("None", "KByT", "Aloha"))

labels=rownames(prcomp_enviro)

ggbiplot(mtcars.pca, groups=as.factor(iris_new$kmeans3))+
  ggtitle("")+geom_point(aes(colour=as.factor(iris_new$kmeans3))) + scale_colour_manual(name="Clustering", values= c("#8b0000","#e6d300", "#00528b"))+
  theme(legend.position = "bottom")+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +theme(legend.position = "bottom") 
ggsave("PCA_Average_kmeans_scaled_KByT_4year.pdf", width=10, height=7)

iris_new$kmeans3

write.csv(iris_new, "clustering_kmeans.csv")





iris_new$SampleID=row.names(iris_new)

iris_new=left_join(iris_new, poo, by="SampleID")

names(iris_new)[names(iris_new) == 'final$cluster'] <- "kmeans3"
names(iris_new)

names(iris_new)[names(iris_new) == 'SampleID'] <- "ID_Clustering"


no_dup=read.csv("v2_nodups_woAprFP_metadata_2019_2021_11252021.csv", header=TRUE, fill=TRUE)

iris2=iris_new %>% select("ID_Clustering", "kmeans3")
no_dup_clustering=left_join(no_dup, iris2, by="ID_Clustering")

write.csv(no_dup_clustering, "v3_nodups_woAprFP_metadata_2019_2021_03112022.csv")



```



```{r}
######### EDA time series ############

## Some fast functions to visuazlie the data 

Cluster1=subset_samples(noNA_All_phyto, kmeans3=="Cluster_1")

p0.rel <- transform(Cluster1, "compositional")

p0.rel2=psmelt(p0.rel)

darkzone <- data.frame(x = c(2017:2020) + 0.75,
                       x2 = c(2018:2021) + 0.25)


thedark <- geom_rect(data=darkzone,
                     fill="grey",alpha=0.4,
                     aes(xmin=x,
                         xmax=x2),
                     ymin=-Inf,ymax=Inf, inherit.aes = FALSE)

# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))


theme_timeseries <- list(
  thedark ,
  scale_x_continuous(breaks = seq(2017, 2021, 1)) ,
  xlab("Year") ,
  ylab("Relative abundance"),
  lil.strip)

plot_timeseries <- function(ASV, psmelt = psmelt.relab,
                            title = NULL, color = "Curated_v3",
                            wrap = ~Site, percent = TRUE, scale = "fixed"){
  # A simple plotting of a timeseries. A vector of the asvs for plotting is given
  # and the results are plotted through the psmelt.
  
  require(lubridate)
  
  basic <-   psmelt %>% 
    filter(OTU %in% ASV) %>% 
    #mutate(decimaldat = decimal_date(Date) ) %>% 
    ggplot(aes(decimal_dat, Abundance)) + 
    geom_line(aes_string(group = "OTU", color = color)) +
    ggtitle(title) +
    theme_timeseries  
  
  if (!is.null(wrap)) {
    basic = basic +
      facet_wrap(wrap, scale = scale)
  }
  
  if (percent) {
    basic = basic + 
      ylab('Relative abundance') +
      scale_y_continuous(labels = scales::percent)
  }
  
  if (color == "phylogroup") {
    basic = basic +
      coloresbarplot
  }
  
  basic
}


plot_timeseries(p0.rel2)

p0.rel2


ggplot(p0.rel2, aes(factor(Month, levels=month.name), Abundance, colour = Phylum)) +
  geom_point() +
  facet_wrap(~Site)







```

