---
title: "Phyloseq_July12022_16SKByT_NERR"
author: "Sarah Tucker"
date: "7/1/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{bash}

###get the seqeunces into tab form
awk 'BEGIN{RS=">"}{print "#"$1"\t"$2;}' 2021_2019_merged-dna-sequences.fasta | tail -n+2 > 2021_2019_merged-dna-sequences.txt

```

```{r}
tax=read.csv("taxonomy_table_final_tucker_v9.csv")
seq=read.csv("v2_2021_2019_merged-dna-sequences.csv")

tax10=left_join(tax, seq, by="ASV")
write.csv(tax10,"taxonomy_table_final_tucker_v9_seq.csv")
```




```{r}
rm(list=ls())

# load libraries
library("ggplot2") 
library("phyloseq") 
library(ape)
library("knitr")
#library(dada2)
library(methods)
library(plotly)
library(tidyverse)
library(ggpubr)
library(compositions)
library(reshape2)
library(fpc)
library(pracma)
library(cluster)
library(ade4)
library(vegan)
library(DESeq2)
library(RColorBrewer)
library(randomcoloR)
library(purrr)
library(lomb)
library(lubridate)
library(tidyverse)
library(mgcv)
library(colorspace)
library(dendextend)
library(DESeq2)
library(DivNet)
library(breakaway)

library(microbiome)
library(viridis)
#devtools::install_github("microsud/microbiomeutilities")
#BiocManager::install("microbiome")
library(microbiomeutilities)

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)


###this is from the R tutorial document"https://github.com/joey711/phyloseq/files/1795523/QIIME2_to_Phyloseq.pdf"


setwd("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq") 

getwd()
```

```{r}
colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="darkseagreen",
"Coastal"="#E4C244")

colorEnviro=c("Offshore"="dodgerblue1",
"Coastal" ="gold",
"Nearshore"="firebrick3")

colorscluster=c("Cluster_1"="dodgerblue3",
"Cluster_2" ="darkseagreen",
"Cluster_3"="coral", 
"Cluster_4"="#b19cd9",
"Cluster_5"="gray44")


colorssite <- c("WAI2"='#7F0000', "KAHO"='#f94279',"HP1"="#ffd000", "CB"="#ffa812","AR"="#c78100" ,"SR8"="#ff9966", "SB"="#c46210", "NB"="#ff7e00", "SR2"="#00cc95", "NR2"="#00cc2f", "STO1"='#0085ff', "NTO1"='#0045ff')



```


```{bash}
cd /Users/sarahtucker/Documents/Davidson Fellowship/
R CMD INSTALL specificity_0.1.12.9000.tgz
```




```{r}

physeq_Dav_no_dup_clean_no18S




metadata=as.data.frame(sample_data(All_phyto_Nutrients_chla))
metadata$Salinity

taxonomy=as.data.frame(tax_table(All_phyto_Nutrients_chla))

library(tibble)
taxonomy <- tibble::rownames_to_column(taxonomy, "OTUID")

taxonomy=as.data.frame(taxonomy)
jj=endophyte$taxonomy


All_phyto_Nutrients=subset_samples(All_phyto, Silicate!="NA")
All_phyto_Nutrients_chla=subset_samples(All_phyto_Nutrients, chla.ug.per.L!="NA")
nsamples(All_phyto_Nutrients_chla)

otutable=as.matrix(t(otu_table(All_phyto_Nutrients_chla)))
otutable <- prop_abund(otutable)
otutable_ovr10 <- occ_threshold(otutable, threshold=30)


#`metadata`: table of data where each row corresponds to a sample, and each column is a different metadata category (e.g. PlantGenus, Elevation, Lon=longitude, Lat=latitude)

nrow(metadata)
names(metadata)

ncol(otutable)
ncol(otutable_ovr10)

all(rownames(otutable_ovr10) == rownames(metadata))

spec_ncores <- 4
# make empty list for specificity values
specs_list <- list()
# specificity for elevation:
specs_list$"Salinity" <- phy_or_env_spec(otutable_ovr10, env=metadata$Salinity,
  n_sim=500, n_cores=spec_ncores)

specs_list$"Temperature" <- phy_or_env_spec(otutable_ovr10, env=metadata$SW.Temperature.at.site,
  n_sim=500, n_cores=spec_ncores)

specs_list$"Silicate" <- phy_or_env_spec(otutable_ovr10, env=metadata$Silicate,
  n_sim=500, n_cores=spec_ncores)

specs_list$"Nitrate" <- phy_or_env_spec(otutable_ovr10, env=metadata$NitrateNitrite,
  n_sim=500, n_cores=spec_ncores)

specs_list$"Phosphate" <- phy_or_env_spec(otutable_ovr10, env=metadata$Phosphate,
  n_sim=500, n_cores=spec_ncores)


specs_list$"Chla" <- phy_or_env_spec(otutable_ovr10, env=metadata$chla.ug.per.L,
  n_sim=500, n_cores=spec_ncores)

specs_list$"Stream" <- phy_or_env_spec(otutable_ovr10, env=metadata$Kaneohe.stream,
  n_sim=500, n_cores=spec_ncores)

pdf("specificity2.pdf")
plot_specs_violin(specs_list, label_cex=0.6, cols=c("red", "blue", "green", "purple", "pink", "gold", "black"))
dev.off

pdf("Correl_spec.pdf")
plot_pairwise_spec(specs_list)
dev.off()
?plot_pairwise_spec()
specificity.shiny::plot_specs_shiny(sl=specs_list, fd=taxonomy)

specs_df_30 <- aggregate_specs_list(specs_list, byFeature=FALSE, fd=taxonomy, fd_id=1)

write.csv(specs_df_30, "spec30.csv")

# save specificity results
  save(list="specs_list", file="specs_results_30.rdata")
```

```{r}
occ_power <- function(x, reps_per_occ=10, min_occ=5, max_occ=NULL,  ...){
	calc_occ <- function(x){sum(x > 0)}

	if(is.null(max_occ)){ max_occ <- calc_occ(x) }
	occs <- rep(seq(from=min_occ, to=max_occ), reps_per_occ)

	xpositions <- seq_along(x)
	abunds_mat <- do.call("cbind", lapply(occs, function(occ){
		xi <- x
		while(calc_occ(xi) > occ){
			xi[sample(xpositions, 1)] <- 0
		}
		return(xi)
	}))
	out <- phy_or_env_spec(abunds_mat=abunds_mat, denom_type="sim_center", ...)
	out$occupancy <- colSums(abunds_mat > 0) # recalculate empirical occupancy
	return(out)
}



library(specificity)
data(endophyte)
	env <- endophyte$metadata$Elevation
	# # find an empirical feature vector with high occupancy and strong specificity.
	abunds_mat <- occ_threshold(prop_abund(endophyte$otutable), 100)
	empirical_specs <- phy_or_env_spec(abunds_mat, env, denom_type="sim_center", n_cores=4)
	 plot(empirical_specs$Spec ~ colSums(abunds_mat > 0))
	 x <- abunds_mat[, which.min(empirical_specs$Spec)]
	 # run this function. speed up by lowering reps_per_occ.
	sim_specs <- occ_power(x, env=env, n_cores=4, chunksize=250)
	# # plot
	plot(sim_specs$Pval ~ sim_specs$occupancy)
	# # calculate false negative rate (FNR) for each occupancy
	 fnrs <- aggregate(Pval~occupancy, sim_specs, FUN=function(x){ sum(x >= 0.05) / length(x) })
	 colnames(fnrs) <- c("occupancy", "FNR")
	# # add spline fit
	fnrs$FNR_spline <- smooth.spline(x=fnrs$occupancy,y=fnrs$FNR)$y
	# # make a plot
	 plot(FNR~occupancy, data=fnrs)
	#points(FNR_spline ~ occupancy, data=fnrs, type="l")






	library(specificity)

	 env <- metadata$SW.Temperature.at.site
	# # find an empirical feature vector with high occupancy and strong specificity.
	abunds_mat <- occ_threshold(prop_abund(otutable), 100)
	empirical_specs <- phy_or_env_spec(abunds_mat, env, denom_type="sim_center", n_cores=4)
	plot(empirical_specs$Spec ~ colSums(abunds_mat > 0))
	x <- abunds_mat[, which.min(empirical_specs$Spec)]
	# # run this function. speed up by lowering reps_per_occ.
	 sim_specs <- occ_power(x, env=env, n_cores=4, chunksize=250)
	# # plot
	plot(sim_specs$Pval ~ sim_specs$occupancy)
	# # calculate false negative rate (FNR) for each occupancy
	fnrs <- aggregate(Pval~occupancy, sim_specs, FUN=function(x){ sum(x >= 0.05) / length(x) })
	colnames(fnrs) <- c("occupancy", "FNR")
	# # add spline fit
	fnrs$FNR_spline <- smooth.spline(x=fnrs$occupancy,y=fnrs$FNR)$y
	# # make a plot
	pdf("Simulation_occupancy.pdf")
	plot(FNR~occupancy, data=fnrs)
	points(FNR_spline ~ occupancy, data=fnrs, type="l")+xline(30)
dev.off()
```





```{r}


phyto = read.csv("taxonomy_table_final_tucker_v7.csv", sep=",") 


group=specs_df %>%group_by(Curated_v3)%>%tally()
specs_df <- aggregate_specs_list(specs_list, byFeature=TRUE, fd=taxonomy, fd_id=1)
generalists=specs_df%>%filter(Salinity_Spec >0, Temperature_Spec >0, Silicate_Spec >0, Phosphate_Spec >0, Stream_Spec>0)


sig_specs=specs_df%>%filter(Pval <0.05)
generalists=specs_df%>%filter(Spec>0)

group=sig_specs%>%group_by(Class)%>%tally()

write.csv(sig_specs, "test_sig_specs.csv")


salinity_spec=sig_specs%>%subset(Variable=="Salinity")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Salinity_Pval = Pval  ,
    Salinity_Spec= Spec,
   ASV=FeatureID
    )
salinity_spec$Salinity_sig="sig"

silicate_spec=sig_specs%>%subset(Variable=="Silicate")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Silicate_Pval = Pval  ,
    Silicate_Spec= Spec,
   ASV=FeatureID
    )

silicate_spec$Silicate_sig="sig"

stream_spec=sig_specs%>%subset(Variable=="Stream")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Stream_Pval = Pval  ,
    Stream_Spec= Spec,
   ASV=FeatureID
    )
stream_spec$Stream_sig="sig"

temp_spec=sig_specs%>%subset(Variable=="Temperature")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Temp_Pval = Pval  ,
    Temp_Spec= Spec,
   ASV=FeatureID
    )
temp_spec$Temp_sig="sig"

chla_spec=sig_specs%>%subset(Variable=="Chla")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Chla_Pval = Pval  ,
    Chla_Spec= Spec,
   ASV=FeatureID
    )

chla_spec$chla_sig="sig"

phosphate_spec=sig_specs%>%subset(Variable=="Phosphate")%>%select(FeatureID,Pval,Spec)%>% 
  rename(
   Phosphate_Pval = Pval  ,
    Phosphate_Spec= Spec,
   ASV=FeatureID
    )

phosphate_spec$phosphate_sig="sig"

tog=full_join(phosphate_spec, chla_spec, by="ASV")

tog2=full_join(temp_spec, stream_spec, by="ASV")

tog3=full_join(salinity_spec, silicate_spec, by="ASV")

tog4=full_join(tog, tog2, by="ASV")

tog5=full_join(tog4, tog3, by="ASV")

tog6=left_join(phyto, tog5, by="ASV")

write.csv(tog6, "phyto_v8.csv")

sig_specs2=sig_specs%>%select(FeatureID,Variable,Pval,Spec )

sig_specs_melt=reshape2::melt(sig_specs2, ~Pval+Spec)

idk=reshape2::dcast(sig_specs, ~FeatureID,id = c("Variable","Pval","Spec"))

molten.data <- reshape2::melt(sig_specs2, id = c("Variable","Pval","Spec"))

sig_specs_count=sig_specs%>%group_by(Variable, Curated_v3)%>%tally()

sig_specs$distinct_variables=sig_specs$Variable

library(tidyverse)
sig_specs=sig_specs%>% 
  mutate(distinct_variables = str_replace(distinct_variables,"Salinity", "Grouped")) %>% 
  mutate(distinct_variables = str_replace(distinct_variables, "Silicate", "Grouped")) %>% 
  mutate(distinct_variables= str_replace(distinct_variables, "Phosphate", "Grouped"))%>% 
  mutate(distinct_variables= str_replace(distinct_variables, "Chla", "Grouped"))
sig_specs
  
sig_specs_count_grouped=sig_specs%>%group_by(distinct_variables,ASV)%>%distinct()

sig_specs_count_grouped=sig_specs%>%
  group_by(distinct_variables, Curated_v3) %>% distinct(ASV)

write.csv(sig_specs_count_grouped, "sig_variables_specific.csv")



temp_unique=subset_taxa(rel2All_phyto, Temp_sig=="sig")


temp_unique2=psmelt(temp_unique)

names(temp_unique2)

ggplot(temp_unique2, aes(x=Abundance, y=SW.Temperature.at.site, color=Curated_v3, shape=phytorelabun_max_class))+geom_point()+facet_wrap(~CuratedASV, scales="free_x")+scale_color_manual(values=coloresbarplot)

ggsave("unique_temp_spec30.pdf", width=15, height=15)


gg=as.data.frame(tax_table(temp_unique))
write.csv(gg, "temperature_ASV.csv")


temp_unique3=temp_unique2%>%group_by(phytorelabun_max_class,Curated_v3)%>%tally()
temp_unique3$ok=temp_unique3$n/366



count_specs=sig_specs%>%group_by(Variable,ASV, Curated_v3)%>%tally()

temp_unique2$Curated_v3=factor(temp_unique2$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))


ggplot(count_specs, aes(x=Variable, y=n, fill=Curated_v3))+geom_bar(stat='identity')+
  scale_x_discrete(guide = guide_axis(angle = 90))+scale_fill_manual(values=coloresbarplot)+theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+theme(legend.position="none")+xlab("")+ylab("Significant ASVs")
ggsave("Specificity_by_taxonomy_sig_30.pdf")



coloresbarplot=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="forestgreen",
"Archaeplastida:Bathycoccus"="darkolivegreen4", 
"Archaeplastida:Ostreococcus"="darkolivegreen3",
"Archaeplastida:Micromonas"="darkseagreen",
"Archaeplastida:Pedinophyceae"="limegreen",
"Archaeplastida:Other"="chartreuse4", 
"Hacrobia:Other"="thistle2",
"Hacrobia:Cryptomonadales"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiales"="red",
"Stramenopiles:Diatoms"="gold",
"Stramenopiles:Dictyochophyceae"="khaki1",
"Stramenopiles:Pelagophyceae"="darkgoldenrod3",
"Stramenopiles:Other"="yellow",
'Eukaryota:Other'="gray44",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")




sig_specs_count$Curated_v3=factor(sig_specs_count$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))






```




```{r}
specific=subset_taxa(All_phyto, Stream_sig=="sig" | Temp_sig =="sig" | Silicate_sig=="sig" | phosphate_sig=="sig"| chla_sig=="sig"| Salinity_sig=="sig")
ntaxa(specific)
write.csv(as.data.frame(tax_table(specific)), "speific_ASV.csv")

specific=subset_taxa(All_phyto, Temp_sig=="sig" | Silicate_sig=="sig" | phosphate_sig=="sig"| Stream_sig=="sig" | chla_sig=="sig" | Salinity_sig=="sig")
ntaxa(specific)

ff=as.data.frame(tax_table(specific))
ff=ff%>%dplyr::select(Stream_sig, Temp_sig,Silicate_sig,phosphate_sig,  chla_sig, Salinity_sig, Curated_v3, Genus)

ff[is.na(ff)] <- 0
write.csv(ff, "idk.csv")

df <- ff %>% 
  mutate_all(funs(str_replace(., "sig", "1")))
df


df[genres] = df[genres] == 1

test=as.data.frame(tax_table(specific))
specific_temp=subset_taxa(All_phyto,Temp_sig=="sig" | Silicate_sig=="NA" | phosphate_sig=="NA" | Stream_sig=="NA" | chla_sig=="NA" | Salinity_sig=="NA")
ntaxa(specific_temp)

test$Silicate_sig
test$Temp_sig
specific_temp=test%>%filter(Temp_sig=="sig" &Silicate_sig!="sig")


library(ggplot2)
library(ComplexUpset)

load(movies)
head(movies, 3)


genres = colnames(df)

pdf("upsetplot_specificity.pdf")
upset(df, genres, name='genre', width_ratio=0.1)
dev.off()
```


```{r}

taxonomy_Phyto_binary <- transform_sample_counts(All_phyto, function(abund) 1*(abund>0))

mami_spec=subset_taxa(taxonomy_Phyto_binary , CuratedASV=="ASV198" | CuratedASV=="ASV201")
kk=psmelt(mami_spec)
ggplot(kk, aes(x=Abundance, y=chla.ug.per.L, color=CuratedASV))+geom_point() + scale_y_continuous(trans = log10_trans())


ggplot(kk, aes(x=SW.Temperature.at.site, y=Abundance,color=CuratedASV)) +
  geom_density(stat="identity") +
  theme(legend.position=c(.2,0.8))

ggplot(kk, aes(x=SW.Temperature.at.site,y=Abundance, color=CuratedASV)) +
  geom_point(stat="identity") +
  theme(legend.position=c(.2,0.8))

ggplot(kk, aes(x=SW.Temperature.at.site,y=Abundance, color=CuratedASV)) +
  geom_point(stat="identity") +
  theme(legend.position=c(.2,0.8))



mami_spec=subset_taxa(rel2All_phyto, ASV=="a908b00f4edf992ef1db8de70b4d7615" | ASV=="ea325221e74caca2941b3bc9db57c62d")
kk=psmelt(mami_spec)
ggplot(kk, aes(x=Abundance, y=SW.Temperature.at.site, color=CuratedASV))+geom_point()


```




```{r}
library(vegan)
mantel(dist(metadata$SW.Temperature.at.site), dist(metadata$chla.ug.per.L))
mantel(dist(metadata$SW.Temperature.at.site), dist(metadata$Salinity))
mantel(dist(metadata$SW.Temperature.at.site), dist(metadata$Silicate))

mantel(dist(metadata$Salinity), dist(metadata$Silicate))
mantel(dist(metadata$N), dist_algae)

mantel(dist(metadata$p), dist_geo)
mantel(dist(metadata$p), dist_algae)

mantel(dist_algae, dist_geo)


# table of number of significant species
  nsig <- sapply(X=specs_list, FUN=function(df){sum(df$Pval <= 0.05)})
  nsig <- sort(nsig)
  
  
# table of how many features are within each category
  nfeatures <- table(metadata$SW.Temperature.at.site)

# organize nfeatures and nsig, compare
  nfeatures2 <- sapply(X=names(nsig), FUN=function(nm){nfeatures[names(nfeatures) == nm]})
  pdf("diagnostic_nmetabolites_vs_nsig.pdf")
  plot(nsig ~ nfeatures2, xlab="log Number of metabolomic features in matrix",
    ylab="number of significant bacterial species")
  dev.off()


# draw plots
  pdf("barplot_sorted_all.pdf")
  par(mar=c(12,4,1,1))
  par(mfrow=c(2,1))
  centers2 <- barplot(nsig, ylab="# Sig. Species", 
    las=2, cex.names=0.4)
  dev.off()

  pdf("barplot_sorted_nonzero.pdf")
  nsig_nonzero <- nsig[nsig>0]
  par(mar=c(12,4,1,1))
  par(mfrow=c(2,1))
  centers2 <- barplot(nsig_nonzero, ylab="# Sig. Species", 
    las=2, cex.names=1)
  dev.off()

  pdf("barplot_sorted_nonzero_angle.pdf")
  barplot_rotatex <- function(x, labels_vec, labels_cex, labels_angle, ...) {
      plt <- barplot(x, xaxt="n", ...)
      text(plt, par("usr")[3], labels = labels_vec, srt = labels_angle, adj = c(1,1), xpd = TRUE, cex=labels_cex) 
  }
  par(mar=c(12,4,1,1))
  barplot_rotatex(nsig_nonzero, names(nsig_nonzero), 0.9, 45, ylab="Number of significant species")
  dev.off()


```


```{r}
library(remotes)
library(specificity)
specificity::
packageVersion("remotes")
library(ape)
library(Rcpp)
library(fields)
library(tools)
library(devtools)


suppressMessages(library(specificity))
# the version of specificity with which this vignette was built:
packageVersion("specificity")
# check what objects are within endophyte
names(endophyte)
# attach "endophyte" data set from specificity
attach(endophyte)

gg=endophyte$otutable

 otutable <- prop_abund(otutable)
  set.seed(12345)
  keep <- sample(c(rep(TRUE, 300), rep(FALSE, nrow(otutable) - 300)))
  otutable <- otutable[keep,]
  metadata <- metadata[keep,]
 
  otutable_ovr10 <- occ_threshold(otutable, threshold=10)
# how many species are we left with?
ncol(otutable)
ncol(otutable_ovr10)

all(rownames(otutable_ovr10) == rownames(metadata))

spec_ncores <- 4
# make empty list for specificity values
specs_list <- list()
# specificity for elevation:
specs_list$"Elevation" <- phy_or_env_spec(otutable_ovr10, env=metadata$Elevation,
  n_sim=500, n_cores=spec_ncores)

# specificity for rainfall
specs_list$"Rainfall" <- phy_or_env_spec(otutable_ovr10, env=metadata$Rainfall, 
  n_sim=500, n_cores=spec_ncores)
# specificity for evapotranspiration
specs_list$"Evapotranspiration" <- phy_or_env_spec(otutable_ovr10,
  env=metadata$Evapotranspiration, n_sim=500, n_cores=spec_ncores )
# specificity for host phylogeny - converting tree to dist will take a while!
specs_list$"Host Phylogeny" <- phy_or_env_spec(otutable_ovr10,
  hosts=metadata$PlantGenus, hosts_phylo=supertree, n_sim=500, 
  n_cores=spec_ncores)


geo_distmat <- distcalc(lat=metadata$Lat, lng=metadata$Lon, sampIDs=metadata$SampleID)
# specificity for geographic distance
specs_list$"Geographic Distance" <- phy_or_env_spec(otutable_ovr10, env=geo_distmat,
  n_sim=500, n_cores=spec_ncores)

plot_specs_violin(specs_list, label_cex=0.6, cols=c("red", "blue", "gold", "green", "black"))

plot_pairwise_spec(specs_list)

specs_df <- aggregate_specs_list(specs_list, byFeature=FALSE, fd=taxonomy, fd_id=1)

library(ggplot2)
ggplot(specs_df, aes(x=Variable, y=Spec, fill=Variable)) + 
  geom_violin() + 
  geom_jitter(width=0.2, size=0.2)


  remotes::install_github("darcyj/specificity.shiny")
specificity.shiny::plot_specs_shiny(sl=specs_list, fd=endophyte$taxonomy)

```
 










```{r}
library(vegan)

taxonomy_Phyto_binary <- transform_sample_counts(All_phyto, function(abund) 1*(abund>0))
NERRSamples=subset_samples(taxonomy_Phyto_binary, Sampling.Order=="30"|Sampling.Order=="32"|Sampling.Order=="33"|Sampling.Order=="35")
dat2=sample_data(NERRSamples)

HP1=subset_samples(NERRSamples, Site=="HP1")
HP1_binary_table =as.data.frame((otu_table(HP1)))
HP1_binary_table$sum=rowSums(HP1_binary_table)
HP1_binary_table$HP1[HP1_binary_table$sum > 0] = 1
HP1_binary_table$HP1[is.na(HP1_binary_table$HP1)] <- 0
HP1_binary_table$HP1
HP1_binary_table2=HP1_binary_table%>%select(HP1)

AR=subset_samples(NERRSamples, Site=="AR")
AR_binary_table =as.data.frame((otu_table(AR)))
AR_binary_table$sum=rowSums(AR_binary_table)
AR_binary_table$AR[AR_binary_table$sum > 0] = 1
AR_binary_table$AR[is.na(AR_binary_table$AR)] <- 0
AR_binary_table$AR
AR_binary_table2=AR_binary_table%>%select(AR)

CB=subset_samples(NERRSamples, Site=="CB")
CB_binary_table =as.data.frame((otu_table(CB)))
CB_binary_table$sum=rowSums(CB_binary_table)
CB_binary_table$CB[CB_binary_table$sum > 0] = 1
CB_binary_table$CB[is.na(CB_binary_table$CB)] <- 0
CB_binary_table$CB
CB_binary_table2=CB_binary_table%>%select(CB)


SR8=subset_samples(NERRSamples, Site=="SR8")
SR8_binary_table =as.data.frame((otu_table(SR8)))
SR8_binary_table$sum=rowSums(SR8_binary_table)
SR8_binary_table$SR8[SR8_binary_table$sum > 0] = 1
SR8_binary_table$SR8[is.na(SR8_binary_table$SR8)] <- 0
SR8_binary_table$SR8
SR8_binary_table2=SR8_binary_table%>%select(SR8)


WAI2=subset_samples(NERRSamples, Site=="WAI2")
WAI2_binary_table =as.data.frame((otu_table(WAI2)))
WAI2_binary_table$sum=rowSums(WAI2_binary_table)
WAI2_binary_table$WAI2[WAI2_binary_table$sum > 0] = 1
WAI2_binary_table$WAI2[is.na(WAI2_binary_table$WAI2)] <- 0
WAI2_binary_table$WAI2
WAI2_binary_table2=WAI2_binary_table%>%select(WAI2)

KAHO=subset_samples(NERRSamples, Site=="KAHO")
dat=sample_data(KAHO)
KAHO_binary_table =as.data.frame((otu_table(KAHO)))
KAHO_binary_table$sum=rowSums(KAHO_binary_table)
KAHO_binary_table$KAHO[KAHO_binary_table$sum > 0] = 1
KAHO_binary_table$KAHO[is.na(KAHO_binary_table$KAHO)] <- 0
KAHO_binary_table$KAHO
KAHO_binary_table2=KAHO_binary_table%>%select(KAHO)


SB=subset_samples(NERRSamples, Site=="SB")
SB_binary_table =as.data.frame((otu_table(SB)))
SB_binary_table$sum=rowSums(SB_binary_table)
SB_binary_table$SB[SB_binary_table$sum > 0] = 1
SB_binary_table$SB[is.na(SB_binary_table$SB)] <- 0
SB_binary_table$SB
SB_binary_table2=SB_binary_table%>%select(SB)

NB=subset_samples(NERRSamples, Site=="NB")
NB_binary_table =as.data.frame((otu_table(NB)))
NB_binary_table$sum=rowSums(NB_binary_table)
NB_binary_table$NB[NB_binary_table$sum > 0] = 1
NB_binary_table$NB[is.na(NB_binary_table$NB)] <- 0
NB_binary_table$NB
NB_binary_table2=NB_binary_table%>%select(NB)


NR2=subset_samples(NERRSamples, Site=="NR2")
NR2_binary_table =as.data.frame((otu_table(NR2)))
NR2_binary_table$sum=rowSums(NR2_binary_table)
NR2_binary_table$NR2[NR2_binary_table$sum > 0] = 1
NR2_binary_table$NR2[is.na(NR2_binary_table$NR2)] <- 0
NR2_binary_table$NR2
NR2_binary_table2=NR2_binary_table%>%select(NR2)

SR2=subset_samples(NERRSamples, Site=="SR2")
SR2_binary_table =as.data.frame((otu_table(SR2)))
SR2_binary_table$sum=rowSums(SR2_binary_table)
SR2_binary_table$SR2[SR2_binary_table$sum > 0] = 1
SR2_binary_table$SR2[is.na(SR2_binary_table$SR2)] <- 0
SR2_binary_table$SR2
SR2_binary_table2=SR2_binary_table%>%select(SR2)

STO1=subset_samples(NERRSamples, Site=="STO1")
STO1_binary_table =as.data.frame((otu_table(STO1)))
STO1_binary_table$sum=rowSums(STO1_binary_table)
STO1_binary_table$STO1[STO1_binary_table$sum > 0] = 1
STO1_binary_table$STO1[is.na(STO1_binary_table$STO1)] <- 0
STO1_binary_table$STO1
STO1_binary_table2=STO1_binary_table%>%select(STO1)

NTO1=subset_samples(NERRSamples, Site=="NTO1")
NTO1_binary_table =as.data.frame((otu_table(NTO1)))
NTO1_binary_table$sum=rowSums(NTO1_binary_table)
NTO1_binary_table$NTO1[NTO1_binary_table$sum > 0] = 1
NTO1_binary_table$NTO1[is.na(NTO1_binary_table$NTO1)] <- 0
NTO1_binary_table$NTO1
NTO1_binary_table2=NTO1_binary_table%>%select(NTO1)


binary_sites4=binary_sites2
binary_sites4$sum=rowSums(binary_sites4)
binary_sites5=binary_sites4%>%select(sum)
NB_binary_table2,SB_binary_table2,AR_binary_table2,CB_binary_table2,SR8_binary_table2,NR2_binary_table2, SR2_binary_table2,STO1_binary_table2, NTO1_binary_table2

binary_sites=do.call("cbind", list(KAHO_binary_table2, WAI2_binary_table2,HP1_binary_table2))
binary_sites2=as.data.frame(t(binary_sites))

binary_sites2

taxonomy_Phyto_binary_table =as.data.frame(t(otu_table(taxonomy_Phyto_binary)))




out <- nestedtemp(binary_sites2)
out
plot(out, names = TRUE)
pdf("nestedness_Sites_Coastal.pdf", height=14,width=9)
plot(out, kind="incid", names = TRUE, order=FALSE)
dev.off()


nestedchecker(binary_sites2)
oecosimu(binary_sites2, nestedchecker, "quasiswap")
## Another Null model and standardized checkerboard score
oecosimu(binary_sites2, nestedchecker, "r00", statistic = "C.score")


nestednodf
library(vegan)
nestedbetasor(binary_sites2)
nestedbetajac(binary_sites2)
nestednodf(binary_sites2)
oecosimu(binary_sites2, nestedtemp, "quasiswap")
oecosimu(binary_sites2, nestedbetajac, "quasiswap")
oecosimu(binary_sites2, nestednodf, "quasiswap")


out <- nestednodf(binary_sites2)
out
pdf("nestedness_nodf_NERR_Samples_Pond.pdf", height=6,width=5)
plot(out, names = TRUE, col = "black")
 dev.off()
```


```{r}
no_dup_metadata_KBytNERR$decimal_dat

names(no_dup_metadata_KBytNERR)
stuff<- no_dup_metadata_KBytNERR%>%subset(Site=="HP1")%>%select(Sampling.Order,decimal_dat)
write.csv(stuff, "sampling_distance.csv")


```


###All sites





```{r}
library(vegan)

data(sipoo)
## Matrix temperature
out <- nestedtemp(sipoo)
out
plot(out)
plot(out, kind="incid")
## Use oecosimu to assess the non-randomness of checker board units
nestedchecker(sipoo)
oecosimu(sipoo, nestedchecker, "quasiswap")
## Another Null model and standardized checkerboard score
oecosimu(sipoo, nestedchecker, "r00", statistic = "C.score")



taxonomy_Phyto_binary <- transform_sample_counts(All_phyto, function(abund) 1*(abund>0))

HP1=subset_samples(taxonomy_Phyto_binary, Site=="HP1")
HP1_binary_table =as.data.frame((otu_table(HP1)))
HP1_binary_table$sum=rowSums(HP1_binary_table)
HP1_binary_table$HP1[HP1_binary_table$sum > 0] = 1
HP1_binary_table$HP1[is.na(HP1_binary_table$HP1)] <- 0
HP1_binary_table$HP1
HP1_binary_table2=HP1_binary_table%>%select(HP1)

AR=subset_samples(taxonomy_Phyto_binary, Site=="AR")
AR_binary_table =as.data.frame((otu_table(AR)))
AR_binary_table$sum=rowSums(AR_binary_table)
AR_binary_table$AR[AR_binary_table$sum > 0] = 1
AR_binary_table$AR[is.na(AR_binary_table$AR)] <- 0
AR_binary_table$AR
AR_binary_table2=AR_binary_table%>%select(AR)

CB=subset_samples(taxonomy_Phyto_binary, Site=="CB")
CB_binary_table =as.data.frame((otu_table(CB)))
CB_binary_table$sum=rowSums(CB_binary_table)
CB_binary_table$CB[CB_binary_table$sum > 0] = 1
CB_binary_table$CB[is.na(CB_binary_table$CB)] <- 0
CB_binary_table$CB
CB_binary_table2=CB_binary_table%>%select(CB)


SR8=subset_samples(taxonomy_Phyto_binary, Site=="SR8")
SR8_binary_table =as.data.frame((otu_table(SR8)))
SR8_binary_table$sum=rowSums(SR8_binary_table)
SR8_binary_table$SR8[SR8_binary_table$sum > 0] = 1
SR8_binary_table$SR8[is.na(SR8_binary_table$SR8)] <- 0
SR8_binary_table$SR8
SR8_binary_table2=SR8_binary_table%>%select(SR8)


WAI2=subset_samples(taxonomy_Phyto_binary, Site=="WAI2")
WAI2_binary_table =as.data.frame((otu_table(WAI2)))
WAI2_binary_table$sum=rowSums(WAI2_binary_table)
WAI2_binary_table$WAI2[WAI2_binary_table$sum > 0] = 1
WAI2_binary_table$WAI2[is.na(WAI2_binary_table$WAI2)] <- 0
WAI2_binary_table$WAI2
WAI2_binary_table2=WAI2_binary_table%>%select(WAI2)

KAHO=subset_samples(taxonomy_Phyto_binary, Site=="KAHO")
KAHO_binary_table =as.data.frame((otu_table(KAHO)))
KAHO_binary_table$sum=rowSums(KAHO_binary_table)
KAHO_binary_table$KAHO[KAHO_binary_table$sum > 0] = 1
KAHO_binary_table$KAHO[is.na(KAHO_binary_table$KAHO)] <- 0
KAHO_binary_table$KAHO
KAHO_binary_table2=KAHO_binary_table%>%select(KAHO)


SB=subset_samples(taxonomy_Phyto_binary, Site=="SB")
SB_binary_table =as.data.frame((otu_table(SB)))
SB_binary_table$sum=rowSums(SB_binary_table)
SB_binary_table$SB[SB_binary_table$sum > 0] = 1
SB_binary_table$SB[is.na(SB_binary_table$SB)] <- 0
SB_binary_table$SB
SB_binary_table2=SB_binary_table%>%select(SB)

NB=subset_samples(taxonomy_Phyto_binary, Site=="NB")
NB_binary_table =as.data.frame((otu_table(NB)))
NB_binary_table$sum=rowSums(NB_binary_table)
NB_binary_table$NB[NB_binary_table$sum > 0] = 1
NB_binary_table$NB[is.na(NB_binary_table$NB)] <- 0
NB_binary_table$NB
NB_binary_table2=NB_binary_table%>%select(NB)


NR2=subset_samples(taxonomy_Phyto_binary, Site=="NR2")
NR2_binary_table =as.data.frame((otu_table(NR2)))
NR2_binary_table$sum=rowSums(NR2_binary_table)
NR2_binary_table$NR2[NR2_binary_table$sum > 0] = 1
NR2_binary_table$NR2[is.na(NR2_binary_table$NR2)] <- 0
NR2_binary_table$NR2
NR2_binary_table2=NR2_binary_table%>%select(NR2)

SR2=subset_samples(taxonomy_Phyto_binary, Site=="SR2")
SR2_binary_table =as.data.frame((otu_table(SR2)))
SR2_binary_table$sum=rowSums(SR2_binary_table)
SR2_binary_table$SR2[SR2_binary_table$sum > 0] = 1
SR2_binary_table$SR2[is.na(SR2_binary_table$SR2)] <- 0
SR2_binary_table$SR2
SR2_binary_table2=SR2_binary_table%>%select(SR2)

STO1=subset_samples(taxonomy_Phyto_binary, Site=="STO1")
STO1_binary_table =as.data.frame((otu_table(STO1)))
STO1_binary_table$sum=rowSums(STO1_binary_table)
STO1_binary_table$STO1[STO1_binary_table$sum > 0] = 1
STO1_binary_table$STO1[is.na(STO1_binary_table$STO1)] <- 0
STO1_binary_table$STO1
STO1_binary_table2=STO1_binary_table%>%select(STO1)

NTO1=subset_samples(taxonomy_Phyto_binary, Site=="NTO1")
NTO1_binary_table =as.data.frame((otu_table(NTO1)))
NTO1_binary_table$sum=rowSums(NTO1_binary_table)
NTO1_binary_table$NTO1[NTO1_binary_table$sum > 0] = 1
NTO1_binary_table$NTO1[is.na(NTO1_binary_table$NTO1)] <- 0
NTO1_binary_table$NTO1
NTO1_binary_table2=NTO1_binary_table%>%select(NTO1)


binary_sites4=binary_sites2
binary_sites4$sum=rowSums(binary_sites4)
binary_sites5=binary_sites4%>%select(sum)
#KAHO_binary_table2, WAI2_binary_table2,NB_binary_table2, SB_binary_table2, AR_binary_table2,CB_binary_table2, HP1_binary_table2,SR8_binary_table2
#NR2_binary_table2, SR2_binary_table2


binary_sites=do.call("cbind", list(KAHO_binary_table2, WAI2_binary_table2,NB_binary_table2, SB_binary_table2, AR_binary_table2,CB_binary_table2, HP1_binary_table2,SR8_binary_table2,NR2_binary_table2, SR2_binary_table2,STO1_binary_table2, NTO1_binary_table2))
binary_sites2=as.data.frame(t(binary_sites))

binary_sites2

taxonomy_Phyto_binary_table =as.data.frame(t(otu_table(taxonomy_Phyto_binary)))




out <- nestedtemp(binary_sites2)
out
plot(out, names = TRUE)
pdf("nestedness_Sites_Coastal.pdf", height=14,width=9)
plot(out, kind="incid", names = TRUE, order=FALSE)
dev.off()


nestedchecker(binary_sites2)
oecosimu(binary_sites2, nestedchecker, "quasiswap")
## Another Null model and standardized checkerboard score
oecosimu(binary_sites2, nestedchecker, "r00", statistic = "C.score")


nestednodf
library(vegan)
nestedbetasor(binary_sites2)
nestedbetajac(binary_sites2)
nestednodf(binary_sites2)
oecosimu(binary_sites2, nestedtemp, "quasiswap")
oecosimu(binary_sites2, nestedbetajac, "quasiswap")
oecosimu(binary_sites2, nestednodf, "quasiswap")


out <- nestednodf(binary_sites2)
out
pdf("nestedness_nodf_Sites.pdf", height=6,width=5)
plot(out, names = TRUE, col = "black", kind="incid")
dev.off()
```


```{r}


Coastal=subset_samples(taxonomy_Phyto_binary, Cluster_DeSeq_Spatial=="Coastal")
Coastal_binary_table =as.data.frame((otu_table(Coastal)))
Coastal_binary_table$sum=rowSums(Coastal_binary_table)
Coastal_binary_table$Coastal[Coastal_binary_table$sum > 0] = 1
Coastal_binary_table$Coastal[is.na(Coastal_binary_table$Coastal)] <- 0
Coastal_binary_table$Coastal
Coastal_binary_table2=Coastal_binary_table%>%select(Coastal)


Transition=subset_samples(taxonomy_Phyto_binary, Cluster_DeSeq_Spatial=="Transition")
Transition_binary_table =as.data.frame((otu_table(Transition)))
Transition_binary_table$sum=rowSums(Transition_binary_table)
Transition_binary_table$Transition[Transition_binary_table$sum > 0] = 1
Transition_binary_table$Transition[is.na(Transition_binary_table$Transition)] <- 0
Transition_binary_table$Transition
Transition_binary_table2=Transition_binary_table%>%select(Transition)


Offshore=subset_samples(taxonomy_Phyto_binary, Cluster_DeSeq_Spatial=="Offshore")
Offshore_binary_table =as.data.frame((otu_table(Offshore)))
Offshore_binary_table$sum=rowSums(Offshore_binary_table)
Offshore_binary_table$Offshore[Offshore_binary_table$sum > 0] = 1
Offshore_binary_table$Offshore[is.na(Offshore_binary_table$Offshore)] <- 0
Offshore_binary_table$Offshore
Offshore_binary_table2=Offshore_binary_table%>%select(Offshore)


Coastal_binary_table3=filter(Coastal_binary_table2,Coastal==1)
write.csv(Coastal_binary_table3, "Coastal_binary_table3.csv")

Transition_binary_table3=filter(Transition_binary_table2,Transition==1)
write.csv(Transition_binary_table3, "Transition_binary_table3.csv")


Offshore_binary_table3=filter(Offshore_binary_table2,Offshore==1)
write.csv(Offshore_binary_table3, "Offshore_binary_table3.csv")


Offshore_binary_table3=filter(Offshore_binary_table2,Offshore==1)
write.csv(Offshore_binary_table3, "Offshore_binary_table3.csv")

binary_Spatial=do.call("cbind", list(Offshore_binary_table2, Coastal_binary_table2, Transition_binary_table2))
binary_Spatial2=as.data.frame(t(binary_Spatial))

oecosimu(binary_Spatial2, nestedtemp, "quasiswap")
oecosimu(binary_Spatial2, nestedbetajac, "quasiswap")
oecosimu(binary_Spatial2, nestednodf, "quasiswap")


distr=read.csv("distribution_Spatail_Venn.csv")
tax2= read.csv("taxonomy_table_final_tucker.csv", sep=",") 

diatoms=subset(tax4, Class=="Bacillariophyta")
diatoms2=diatoms%>%group_by(Binary_Venn)%>%tally()

tax4=left_join(tax2,distr, by="ASV")
write.csv(tax4, "taxonomy_table_final_tucker_v2.csv")

tax5=subset(tax4, Binary_Venn=="Offshore")
tax5=tax5%>%group_by(Class,Genus)%>%tally()
tax5


out <- nestednodf(binary_Spatial2)
out
pdf("nestedness_nodf_Spatial.pdf", height=6,width=5)
plot(out, names = TRUE, col = "black")
dev.off()

```




```{r}
library(indicspecies)
install.packages("indicspecies")
data(wetland)
groups = c(rep(1, 17), rep(2, 14), rep(3,10))
groups

indval = multipatt(wetland, groups, 
                   control = how(nperm=999)) 
summary(indval)




```





```{r}
no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", header=TRUE, fill=TRUE)
names(no_dup_metadata_KBytNERR)

Temp=no_dup_metadata_KBytNERR %>% drop_na(Photosynthetic.radiaion)%>%dplyr::group_by(Season_Aka)%>%dplyr::summarise(Mean=round(mean(Photosynthetic.radiaion),1), sd=round(sd(Photosynthetic.radiaion),1), n=n())



Temp=no_dup_metadata_KBytNERR %>% drop_na(wind.max)%>%dplyr::group_by(Season_Aka)%>%dplyr::summarise(Temp.Mean=round(mean(wind.max),1), sd=round(sd(wind.max),1), n=n())

wind.max
```



```{r}
no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", header=TRUE, fill=TRUE)

avi=no_dup_metadata_KBytNERR%>%group_by(Site,Enviro_Clustering,Cluster_DeSeq_Spatial)%>%tally()

avi2=avi
avi2$Enviro_rename=avi2$Enviro_Clustering


library("dplyr")
# Replace on selected column
avi2  <- avi2 %>% 
  mutate(Enviro_rename = str_replace(Enviro_rename, "Nearshore", "Coastline")) %>% 
  mutate(Enviro_rename = str_replace(Enviro_rename, "Coastal", "InnerBay")) %>% 
  mutate(Enviro_rename = str_replace(Enviro_rename, "Offshore", "OuterBay"))
avi2 

names(no_dup_metadata_KBytNERR)

install.packages("ggalluvial")
library(ggalluvial)
avi$Cluster_DeSeq_Spatial

is_alluvia_form(as.data.frame(avi), axes = 1:3, silent = TRUE)
avi$Cluster_DeSeq_Spatial=factor(avi$Cluster_DeSeq_Spatial, levels=c("Coastal", "Transition", "Offshore"))

avi$Site=factor(avi$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "CB", "SB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))


avi$Cluster_DeSeq_Spatial

avi$Enviro_Clustering




ggplot(as.data.frame(avi),
       aes(y = n, axis1 = as.factor(Cluster_DeSeq_Spatial), axis2 = as.factor(Enviro_Clustering))) +
  geom_alluvium(aes(fill = Cluster_DeSeq_Spatial), width = 1/12) +
  geom_stratum(width = 1/12, fill = "black", color = "grey") +
  geom_label(stat = "stratum", aes(label = after_stat(stratum))) +
  scale_x_discrete(limits = c("Community \n Composition", "Environmental \n Clustering"), expand = c(.05, .05)) +
  scale_fill_brewer(type = "qual", palette = "Set1") +
  ggtitle("Clustering by Community Composition compared to environment")

ggsave("alluvial_plot.pdf")




ggplot(as.data.frame(avi),
       aes(y = n, axis1 = as.factor(Cluster_DeSeq_Spatial), axis2 = as.factor(Site)), axis3=Enviro_Clustering) +
  geom_alluvium(aes(fill = Cluster_DeSeq_Spatial), width = 1/8) +
  geom_stratum(width = 1/8, fill = "black", color = "grey") +
  geom_label(stat = "stratum", aes(label = after_stat(stratum))) +
  scale_x_discrete(limits = c("Community \n Composition", "Site", "Environmental \n Clustering"), expand = c(.05, .05)) +
  scale_fill_brewer(type = "qual", palette = "Set1") +
  ggtitle("Clustering by Community Composition compared to environment")

ggsave("alluvial_plot.pdf")


 geom_alluvium(aes(fill = Class),
                width = 0, knot.pos = 0, reverse = FALSE) +
  guides(fill = FALSE) +
  geom_stratum(width = 1/8, reverse = FALSE) +
  geom_text(stat = "stratum", aes(label = after_stat(stratum)),
            reverse = FALSE) +
  scale_x_continuous(breaks = 1:3, labels = c("Survived", "Sex", "Class"))


 


ggplot(as.data.frame(avi),
       aes(y = n, axis1 = as.factor(Cluster_DeSeq_Spatial), axis2 = as.factor(Site)), axis3=Enviro_Clustering) +
  geom_alluvium(aes(fill = Cluster_DeSeq_Spatial), width = 1/8) +
  geom_stratum(width = 1/8, fill = "black", color = "grey") +
  geom_label(stat = "stratum", aes(label = after_stat(stratum))) +
   scale_x_continuous(breaks = 1:3, labels = c("Survived", "Sex", "Class")) +
  scale_fill_brewer(type = "qual", palette = "Set1") +
  ggtitle("Clustering by Community Composition compared to environment")

avi$Enviro_Clustering

names(avi)

poop=factor(avi$Enviro_Clustering, levels=c("Nearshore", "Coastal", "Offshore"))


avi2$Cluster_DeSeq_Spatial=factor(avi2$Cluster_DeSeq_Spatial, levels=c("Coastal", "Transition", "Offshore"))

avi2$Site=factor(avi2$Site, levels=c("WAI2", "KAHO","SB", "AR","CB","HP1","NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))

svg("alluvium.svg", height=4, width=4)

ggplot(as.data.frame(avi2),
       aes(y = n,
           axis1 = as.factor(Cluster_DeSeq_Spatial), axis2 =as.factor(Site), axis3 = as.factor(Enviro_rename))) +
  geom_alluvium(aes(fill = Cluster_DeSeq_Spatial),
                width = 0) +
  guides(fill = FALSE) +
  geom_stratum(width = 1/8) +
  geom_text(stat = "stratum", aes(label = after_stat(stratum))) +
  scale_x_continuous(breaks = 1:3, labels = c("Community \n Composition", "Site", "Environmental \n Clustering")) +
  ggtitle("")+
  scale_fill_manual(values = c(Coastal = "#E4C244", Transition= "darkseagreen",
                              Offshore = "dodgerblue3"))
ggsave("alluvium.pdf")
dev.off()


colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="darkseagreen",
"Coastal"="#E4C244")


  

```




```{r}

#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# read in otu table
no_dup_otu_table_KBytNERR = read.csv("otu_final_tucker.csv", sep=",", row.names=1) 
no_dup_otu_table_KBytNERR = as.matrix(no_dup_otu_table_KBytNERR)

# import as phyloseq objects
no_dup_OTU_KBytNERR = otu_table(no_dup_otu_table_KBytNERR, taxa_are_rows = TRUE) 


taxonomy_Phyto = read.csv("taxonomy_table_final_tucker_v10.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)
ntaxa(TAX_Phyto)

#v5_nodups_wo_AprFP_2019_2021_04212022.csv
#no_dup_metadata_KBytNERRv3.csv
#v6_nodups_wo_AprFP_2019_2021_07062022.csv


no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", header=TRUE, fill=TRUE)

rownames(no_dup_metadata_KBytNERR) <- no_dup_metadata_KBytNERR$SampleID
no_dup_metadata_KBytNERR$SampleID
META_no_dup_KBytNERR = sample_data(no_dup_metadata_KBytNERR)
nsamples(META_no_dup_KBytNERR)



physeq_Dav_no_dup = phyloseq(no_dup_OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav_no_dup=merge_phyloseq(physeq_Dav_no_dup, META_no_dup_KBytNERR)
ntaxa(physeq_Dav_no_dup)


#4462
#4196
#4117
```




```{r}

##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean <- subset_taxa(physeq_Dav_no_dup, !is.na(Domain) & !Domain %in% c("", "unclassified_Root"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean)
ntaxa(physeq_Dav_no_dup_clean)
#4343
table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)


#check to see all of the uncharacterized phylum
table(tax_table(physeq_Dav_no_dup_clean)[, "Phylum"], exclude = NULL)


##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean_no18S <- subset_taxa(physeq_Dav_no_dup_clean, !is.na(Domain) & !Domain %in% c("", "Eukaryota"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean_no18S)
ntaxa(physeq_Dav_no_dup_clean_no18S)
#3987
table(tax_table(physeq_Dav_no_dup_clean_no18S)[, "Domain"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))
ntaxa(rel.abun)
#3987
#3680
nsamples(rel.abun)
#366

```





```{r}

#Bac_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Bac_16S= subset_taxa(physeq_Dav_no_dup_clean_no18S,Domain == "D_0__Bacteria")
ntaxa(Bac_16S)

Arc_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Arc_16S= subset_taxa(physeq_Dav_no_dup_clean_no18S,Domain == "D_0__Bacteria")
ntaxa(Bac_16S)


rel2Bac_16S <- transform_sample_counts(Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


Just_Het_Bac_16S= subset_taxa(Bac_16S,Phylum != "D_1__Cyanobacteria")
ntaxa(Just_Het_Bac_16S)

rel2_Het_Bac_16S <- transform_sample_counts(Just_Het_Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 
ntaxa(rel2_Het_Bac_16S)


Cyano = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria")
ntaxa(Cyano)


All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto)

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


All_phyto_Rel = subset_taxa(rel.abun,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto_Rel)
table(tax_table(All_phyto)[, "Phylum"], exclude = NULL)





rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)
#3987
nsamples(rel.abun)
#366


```


```{r}



svg("Salinity_Box_Site.svg", height=2, width=6)
Sailinity<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=blueberry, y=Salinity, color=Site))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter(aes(shape = Enviro_Clustering))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2) 
Sailinity
dev.off()
ggsave("Salinity_sites.pdf", height=3, width=6)
```





```{r}

names(no_dup_metadata_KBytNERR)

no_dup_metadata_KBytNERR$Cluster_DeSeq_Spatial
ff=no_dup_metadata_KBytNERR%>%drop_na(Dva)%>%group_by(Cluster_DeSeq, Season_Aka)%>%tally()
fff=no_dup_metadata_KBytNERR%>%drop_na(Dva)%>%subset(Cluster_DeSeq_Spatial=="Coastal")


names(no_dup_metadata_KBytNERR)
svg("Salinity_Box_Site.svg", height=2, width=6)
Sailinity<- ggplot(data=fff, aes(x=Cluster_DeSeq, y=Dva, color=Cluster_DeSeq))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Sailinity
dev.off()
ggsave("Salinity_sites.pdf", height=3, width=6)


Sailinity<- ggplot(data=fff, aes(x=Season_Aka, y=Dva))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Sailinity

PRO_coastal=no_dup_metadata_KBytNERR%>%drop_na(Dva)%>%subset(Cluster_DeSeq_Spatial=="Coastal")


PRO_coastal$Cluster_DeSeq=factor(PRO_coastal$Cluster_DeSeq, levels=c("Cluster_3","Cluster_4", "Cluster_5"))

  amod <- aov(Dva ~ Cluster_DeSeq, data =PRO_coastal)
  amod
  summary(amod)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  
library("multcomp")
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))


```
```{r}

Syn_mean=no_dup_metadata_KBytNERR%>%drop_na(SYN.mL)%>%subset(Cluster_DeSeq_Spatial=="Coastal")%>%group_by(Cluster_DeSeq)%>%summarise(mean=mean(SYN.mL))

Syn_coastal=no_dup_metadata_KBytNERR%>%drop_na(SYN.mL)%>%subset(Cluster_DeSeq_Spatial=="Coastal")

Sailinity<- ggplot(data=Syn_coastal, aes(x=Cluster_DeSeq, y=SYN.mL, color=Cluster_DeSeq))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Sailinity
  

Syn_coastal$Cluster_DeSeq=factor(Syn_coastal$Cluster_DeSeq, levels=c("Cluster_3","Cluster_4", "Cluster_5"))

  amod <- aov(SYN.mL ~ Cluster_DeSeq, data =Syn_coastal)
  amod
  summary(amod)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  
library("multcomp")
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))

```


```{r}



PRO_mean=no_dup_metadata_KBytNERR%>%drop_na(PRO.mL)%>%subset(Cluster_DeSeq_Spatial=="Coastal")%>%group_by(Cluster_DeSeq)%>%summarise(mean=mean(PRO.mL))

PRO_coastal=no_dup_metadata_KBytNERR%>%drop_na(PRO.mL)%>%subset(Cluster_DeSeq_Spatial=="Coastal")

Sailinity<- ggplot(data=Syn_coastal, aes(x=Cluster_DeSeq, y=PRO.mL, color=Cluster_DeSeq))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Sailinity
  

PRO_coastal$Cluster_DeSeq=factor(Syn_coastal$Cluster_DeSeq, levels=c("Cluster_3","Cluster_4", "Cluster_5"))

  amod <- aov(PRO.mL ~ Cluster_DeSeq, data =PRO_coastal)
  amod
  summary(amod)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  
library("multcomp")
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))

```


```{r}
tax= read.csv("taxonomy_table_final_tucker_v9.csv", sep=",", row.names=1) 

names(tax)
count_tax=tax%>%group_by(DeSeq_Spatial_Cluster,phytorelabun_max_class)%>%tally()

count_tax2=tax%>%group_by(phytorelabun_max_class)%>%tally()

```




```{r}
names(no_dup_metadata_KBytNERR)

no_dup_metadata_KBytNERR$Cluster_DeSeq_Spatial
ff=no_dup_metadata_KBytNERR%>%drop_na(Fuco)%>%group_by(Cluster_DeSeq, Season_Aka)%>%tally()
fff=no_dup_metadata_KBytNERR%>%drop_na(Fuco)%>%subset(Cluster_DeSeq_Spatial=="Coastal")


names(no_dup_metadata_KBytNERR)

Sailinity<- ggplot(data=fff, aes(x=Cluster_DeSeq, y=Fuco, color=Cluster_DeSeq))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Sailinity
fff$Cluster_DeSeq
fff$Cluster_DeSeq=factor(fff$Cluster_DeSeq, levels=c("Cluster_3","Cluster_4", "Cluster_5"))

  amod <- aov(Fuco ~ Cluster_DeSeq, data =fff)
  amod
  summary(amod)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  
library("multcomp")
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))

fuco_mean=fff%>%drop_na(Fuco)%>%group_by(Cluster_DeSeq)%>%summarise(mean=mean(Fuco))
  
```






```{r}
svg("Salinity_Box_Site.svg", height=2, width=6)
Sailinity<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=blueberry, y=pH, color=Site))+geom_boxplot() 
Sailinity=Sailinity + theme(legend.title = element_blank()) + ylab("Sailinity") +xlab("") +geom_jitter(aes(shape = Enviro_Clustering))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2) 
Sailinity
dev.off()
ggsave("Salinity_sites.pdf", height=3, width=6)
```



```{r}


colorssite2 <- c("WAI2"='#7F0000', "KAHO"='#f94279',"SB"="#c46210","AR"="#c78100" , "CB"="#ffa812","HP1"="#ffd000","SR8"="#ff9966",  "NB"="#ff7e00", "SR2"="#00cc95", "NR2"="#00cc2f", "STO1"='#0085ff', "NTO1"='#0045ff')


blueberry=factor(no_dup_metadata_KBytNERR$Site, levels=c("WAI2","KAHO","SB","AR","CB","HP1","SR8", "NB", "SR2", "NR2", "STO1", "NTO1"))

svg("Silicate_Box_Site.svg", height=2, width=6)
Silicate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=blueberry, y=Silicate, color=Site))+geom_boxplot() 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Silicate \n (uM)") +xlab("") +geom_jitter(aes(shape = Enviro_Clustering))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())
Silicate
dev.off()
ggsave("Silicate_clusters.svg", height=3, width=6)
```


```{r}
svg("chlorophylla_Site.svg",height=2, width=6 )
Silicate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=blueberry, y=chla.ug.per.L, color=Site))+geom_boxplot() 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Chlorophyll a(ug L)") +xlab("") +geom_jitter(aes(shape = Enviro_Clustering))+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())
Silicate
dev.off()
colors_Enviro2 <- c("Nearshore"="firebrick3","Offshore"= "dodgerblue1","Coastal"="gold")
colors_Enviro2 <- c("Nearshore"="firebrick3","Offshore"= "dodgerblue1","Coastal"="gold")

library(scales)
test=no_dup_metadata_KBytNERR%>%drop_na(Enviro_Clustering)%>%drop_na(chla.ug.per.L)

svg("chlorophylla_Season.svg",height=2, width=6 )


chla<-ggplot(data=test, aes(x=factor(Enviro_Clustering), y=chla.ug.per.L, fill=Season_Aka))+geom_boxplot(outlier.shape = NA) +geom_jitter(aes(shape =factor(Enviro_Clustering), color=factor(Enviro_Clustering)))+ scale_color_manual(values=colors_Enviro2)
chla=chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a(ug L)") +xlab("")+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+scale_fill_manual(values=c("#D3D3D3", "#565656"))+ geom_hline(aes(yintercept = 2.5), linetype="dashed")+ geom_hline(aes(yintercept = 8), linetype="dashed")
chla

dev.off()

test$Enviro_Clustering
shapes <- c(17,16,15)

test$Enviro_Clustering=factor(test$Enviro_Clustering, levels=c("Nearshore", "Coastal", "Offshore"))


svg("chlorophylla_Season.svg",height=2, width=6 )
chla<-ggplot(data=test, aes(x=factor(Enviro_Clustering), y=chla.ug.per.L, fill=Season_Aka))+geom_boxplot(outlier.shape = NA) +geom_jitter(aes(shape =factor(Enviro_Clustering), color=factor(Site)))+ scale_color_manual(values=colorssite2 )+scale_shape_manual(values=shapes)
chla=chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a (ug L)") +xlab("")+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+scale_fill_manual(values=c("#D3D3D3", "#565656"))+ geom_hline(aes(yintercept = 2.5), linetype="dashed")+ geom_hline(aes(yintercept = 8), linetype="dashed")
chla
dev.off()

```

```{r}
?month.abb
no_dup_metadata_KBytNERR$Month
no_dup_metadata_KBytNERR$Date2
no_dup_metadata_KBytNERR$Month = factor(no_dup_metadata_KBytNERR$Month, levels==month.name)
your_data$month = factor(your_data$month, levels = month.abb)
list(month.abb)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="darkseagreen",
"Coastal"="#E4C244")


library(ggplot2)


no_dup_metadata_KBytNERR=no_dup_metadata_KBytNERR%>%drop_na(Enviro_Clustering)
no_dup_metadata_KBytNERR$Month_abb=factor(no_dup_metadata_KBytNERR$Month_abb, levels=c("Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar"))




scale_x_discrete(limits = c(-1, 0))



coastal=subset(no_dup_metadata_KBytNERR, Enviro_Clustering=="Coastal")
Nearshore=subset(no_dup_metadata_KBytNERR, Enviro_Clustering=="Nearshore")
Offshore=subset(no_dup_metadata_KBytNERR, Enviro_Clustering=="Offshore")
coastal$dev

a=ggplot(data = Coastal, aes(x=decimal_dat, y=chla.ug.per.L, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Temperature (C)")+xlab("") + scale_y_continuous(trans = log10_trans())
a


a=ggplot(data = no_dup_metadata_KBytNERR, aes(x=decimal_dat, y=chla.ug.per.L, group=Enviro_Clustering, color=Enviro_Clustering, shape=Enviro_Clustering))+
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Temperature (C)")+xlab("") + scale_y_continuous(trans = log10_trans())+ geom_hline(aes(yintercept = 1.75, color="firebrick3"), linetype="dashed")+ geom_hline(aes(yintercept = 1.54, color="gold"), linetype="dashed")+ geom_hline(aes(yintercept = 0.77, color="dodgerblue3"), linetype="dashed")
a


Coastal %>% mutate(diff= 1.54+abs(1.54-chla.ug.per.L)) %>%ggplot(aes(x=decimal_dat, y=diff))+geom_point()+scale_y_continuous(limits = c(-1, 30))

Nearshore %>% mutate(diff= 1.75+abs(1.75-chla.ug.per.L)) %>%ggplot(aes(x=decimal_dat, y=diff))+geom_point()+scale_y_continuous(limits = c(-1, 30))

Offshore %>% mutate(diff= 1.57+abs(1.57-chla.ug.per.L)) %>%ggplot(aes(x=decimal_dat, y=diff))+geom_point()+scale_y_continuous(limits = c(-1, 30))

```





######
EVALUATING SEQUENCING DEPTH
######



```{r}

#### note here we are still including 18S 
##########
############
#############


##not using cleaned, cleaned removes sequences based on our poor taxonomic assignments not based on well it was sequenced

sdt = data.frame(as(sample_data(physeq_Dav_no_dup_clean_no18S), "data.frame"),
                 TotalReads = sample_sums(physeq_Dav_no_dup_clean_no18S), keep.rownames = TRUE)
sdt

write.csv(sdt, "sdt.csv")

pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth +facet_wrap(~Season)

pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth +facet_wrap(~Environment)


pSeqDepth = ggplot(sdt, aes(TotalReads)) + geom_histogram() + ggtitle("Sequencing Depth")
pSeqDepth 


summary(sdt$TotalReads)

sd(sdt$TotalReads)

sdt2 <-sdt%>% summarise(average=mean(TotalReads), SD=sd(TotalReads))
sdt2
sdt3 <-sdt  %>% group_by(Site)%>% summarise(average=mean(TotalReads), SD=sd(TotalReads))



write.csv(sdt3, "test.csv")

ggqqplot(sdt$TotalReads)


##sdt3=subset(sdt, SampleID != "NBJa0121" & SampleID != "CBJn0218")


library("multcomp")

sdt$Environment=as.factor(sdt$Environment)
sdt$Season=as.factor(sdt$Season)

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Environment, data = sdt)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Environment= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = sdt)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
coastal=subset(sdt, Environment=="Coastal")
transition=subset(sdt,Environment=="Estuary")
offshore=subset(sdt, Environment=="Offshore")

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = coastal)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = transition)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
    ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Season, data = offshore)

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  ??multicomp
  
  
  
  ### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(TotalReads ~ Site, data = sdt)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Site= "Tukey"))
  summary(g, test = adjusted("holm"))
  
```
```{r}
###make sure the right otu table is in here rel abun without chloro, cell # 
Abun_By_Order=tax_glom(rel.abun, "Order")
gg=as.data.frame(sort(taxa_sums(Abun_By_Order), decreasing=TRUE))

top15<- names(sort(taxa_sums(Abun_By_Order), decreasing=TRUE))[1:15]

top15
#ps.top20 <- transform_sample_counts(Cells_physeq_KByt2, function(OTU) OTU/sum(OTU))
ps.top15<- prune_taxa(top15, Abun_By_Order)
gg=as.data.frame(tax_table(ps.top15))
plot_bar(ps.top15, fill="Order") + facet_wrap(~Site, scales="free_x")
ggsave("Top15byOrder_allTaxa_Barplots.pdf")

maybe=get_top_taxa(physeq_Dav_no_dup_clean_no18S, 15, relative = TRUE)



rel2All_phyto_sum= rel2All_phyto %>% psmelt()%>% dplyr::group_by(Curated_v3,SampleID,Cluster_DeSeq_Spatial)%>%dplyr::summarise(sum=sum(Abundance)*100)
rel2All_phyto_sum


rel2All_phyto_sum2=rel2All_phyto_sum%>% dplyr::group_by(Curated_v3,SampleID,Cluster_DeSeq_Spatial)%>%filter(sum>20)
               
               
####perhaps one strategy would be to get the average of Syn in each cluster and then subset for samples when Syn goes below its average to see if other taxa become more abundant and if so, which ones and how many times for each
#for instsance there are samples where diatoms reach >35 and malliephycea >20 but this definition i think splits up this group tooo much 
## need to make a new curated with syn, pro, cyanobium and then maybe class level for others 



rel2All_phyto_means=rel2All_phyto_sum %>% dplyr::group_by(Curated_v3,Cluster_DeSeq_Spatial)%>%dplyr::summarise(Mean=round(mean(sum),2), sd=round(sd(sum),2))
rel2All_phyto_means





Abun_By_Order=tax_glom(rel.abun, "Order")
df_Order_total <- Abun_By_Order %>% psmelt() %>% group_by(OTU,Order) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 




df_Order_total$percent_total_mean=df_Order_total$mean*100
df_Order_total$percent_total_SD=df_Order_total$sd*100


test=df_Order_total[df_Order_total$percent_total_mean %in% tail(sort(df_Order_total$percent_total_mean),15),]
write.csv(test, "Mean_allsamples_Order.csv")

filter(Sample)

Abun_By_Order=tax_glom(rel2All_phyto, "Class")
ntaxa(Abun_By_Order)

###not tax glom by Curated does not work

df_Order_total <- rel2All_phyto %>% psmelt() %>% group_by(OTU,Genus)%>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 
df_Order_total$percent_total_mean=df_Order_total$mean*100
df_Order_total$percent_total_SD=df_Order_total$sd*100



df_Order_total2=df_Order_total%>%select(OTU,Sample, Abundance, Cluster_DeSeq, Cluster_DeSeq_Spatial, Curated_v3)

AllPhyto_abun_ASV_enviro=reshape2::dcast(df_Order_total2, OTU+Curated_v3 ~ Sample, value.var ="Abundance")


beaver

df_Order_total3 <- df_Order_total2 %>% group_by(OTU,Curated_v3,Cluster_DeSeq_Spatial)%>% summarise(mean = mean(Abundance), sd=sd(Abundance), max=max(Abundance), min=min(Abundance)) 





Abun_By_Order=tax_glom(rel2All_phyto, "Class")
df_Order_total <- Abun_By_Order %>% psmelt() %>% group_by(OTU,Class) %>% summarise(mean = mean(Abundance), sd=sd(Abundance)) 

df_Order_total$percent_total_mean=df_Order_total$mean*100
df_Order_total$percent_total_SD=df_Order_total$sd*100


test=df_Order_total[df_Order_total$percent_total_mean %in% tail(sort(df_Order_total$percent_total_mean),15),]
write.csv(test, "rel2allPhytoMean_allsamples_Order.csv")



```


```{r}

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

rel.abun.df=as.data.frame(otu_table(rel.abun))
dim(rel.abun.df)


rel.abun.df$totalrelabunMax <- apply(rel.abun.df, 1, max)
rel.abun.df$totalrelabunMax=rel.abun.df$totalrelabunMax*100
rel.abun.df$totalrelabunMax

rel.abun.df$totalrelabun_max_class="NA"

  
rel.abun.df2=rel.abun.df %>% 
  mutate(totalrelabun_max_class = case_when(totalrelabunMax >= 25 ~ 'dominant'
                                ,totalrelabunMax >= 0.5 ~ 'abundant'
                                ,TRUE ~ 'minor'
                                )
  )                    
ll=rel.abun.df2%>%group_by(totalrelabun_max_class)%>%tally()

library(tibble)
gg2 <- tibble::rownames_to_column(rel.abun.df2, "ASV")
gg2=gg2%>%select(ASV, totalrelabunMax, totalrelabun_max_class)


taxonomy_occurrence = read.csv("taxonomy_table_final_tucker_v4.csv", sep=",", row.names=1)
ubiq2=left_join(taxonomy_occurrence, gg2, "ASV")

write.csv(ubiq2, "taxonomy_table_final_tucker_v5.csv")
view(ubiq2)






```

```{r}
rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


rel.abun.df=as.data.frame(otu_table(rel2All_phyto))
dim(rel.abun.df)




rel.abun.df$phytorelabunMax <- apply(rel.abun.df, 1, max)
rel.abun.df$phytorelabunMax=rel.abun.df$phytorelabunMax*100
rel.abun.df$phytorelabunMax

rel.abun.df$phytorelabun_max_class="NA"

  
rel.abun.df2=rel.abun.df %>% 
  mutate(phytorelabun_max_class = case_when(phytorelabunMax >= 25 ~ 'dominant'
                                ,phytorelabunMax >= 0.5 ~ 'abundant'
                                ,TRUE ~ 'minor'
                                )
  )                    
ll=rel.abun.df2%>%group_by(phytorelabun_max_class)%>%tally()



library(matrixStats)

rel.abun.df2$phytorelabunMedian = rowMedians(as.matrix(rel.abun.df2[,c(1:366)]))
rel.abun.df2

rel.abun.df2$phytorelabunMedian=rel.abun.df2$phytorelabunMedian*100

library(tibble)
gg2 <- tibble::rownames_to_column(rel.abun.df2, "ASV")
gg2=gg2%>%select(ASV, phytorelabunMax, phytorelabun_max_class, phytorelabunMedian)




taxonomy_occurrence = read.csv("taxonomy_table_final_tucker_v5.csv", sep=",", row.names=1)
ubiq2=left_join(taxonomy_occurrence, gg2, "ASV")

write.csv(ubiq2, "taxonomy_table_final_tucker_v6.csv")

ubiq2
library(scales)

svg("ra_occurrences4.svg",width=13, height=4)
ggplot(ubiq2, aes(x=occur, y=phytorelabunMax, color=Curated_v3, size=phytorelabunMedian))+geom_point()+scale_color_manual(values=coloresbarplot)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) +geom_vline(aes(xintercept = 10, color="gray"),linetype="dotted")+geom_vline(aes(xintercept = 75, color="gray"),linetype="dotted")+geom_vline(aes(xintercept = 100, color="gray"),linetype="dotted")+geom_hline(aes(yintercept = 0.5, color="gray"),linetype="dotted")+geom_hline(aes(yintercept = 25, color="gray"),linetype="dotted")+scale_y_continuous(trans = log10_trans())+ylab("Maxumim abundance (%)")+xlab("Frequency across samples (%)")+theme(legend.position="bottom")
dev.off()

ggsave("max_abun_freq.pdf", width=4, height=4)
size()
#+theme(legend.position="bottom")


test=psmelt(rel2All_phyto)


coloresbarplot=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="forestgreen",
"Archaeplastida:Bathycoccus"="darkolivegreen4", 
"Archaeplastida:Ostreococcus"="darkolivegreen3",
"Archaeplastida:Micromonas"="darkseagreen",
"Archaeplastida:Pedinophyceae"="limegreen",
"Archaeplastida:Other"="chartreuse4", 
"Hacrobia:Other"="thistle2",
"Hacrobia:Cryptomonadales"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiales"="red",
"Stramenopiles:Diatoms"="gold",
"Stramenopiles:Dictyochophyceae"="khaki1",
"Stramenopiles:Pelagophyceae"="darkgoldenrod3",
"Stramenopiles:Other"="yellow",
'Eukaryota:Other'="gray44",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")




ubiq2$Curated_v3=factor(ubiq2$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))





kk$Curated_v3=factor(kk$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))

other=ubiq2%>%subset(phytorelabun_max_class=="abundant" & occurrence=="frequent")%>%group_by(ASV, Curated_v3, phytorelabun_max_class,occurrence)

unique(ubiq2$phytorelabun_max_class)


kk=ubiq2%>%group_by(Curated_v3)%>%tally

kk=kk%>%drop_na()
ggplot(kk,aes(x=Curated_v3, y=n))+geom_bar(stat='identity')+
  scale_x_discrete(guide = guide_axis(angle = 90))


kk=ubiq2%>%group_by(Class, Curated_v3)%>%tally


kk=ubiq2%>%group_by(Class)%>%tally

kk=kk%>%drop_na(Curated_v3)

ggplot(kk,aes(x=Curated_v3, y=n, fill=Curated_v3))+geom_bar(stat='identity')+
  scale_x_discrete(guide = guide_axis(angle = 90))+scale_fill_manual(values=coloresbarplot)+theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+theme(legend.position="none")+xlab("")+ylab("Count of ASVs")
dev.off()

kk=kk%>%drop_na()
svg("Phytoplankton_ASVs_Count.svg", width=2.5,height=2)
ggplot(kk,aes(x=Curated_v3, y=n, fill=Curated_v3))+geom_bar(stat='identity')+
  scale_x_discrete(guide = guide_axis(angle = 90))+scale_fill_manual(values=coloresbarplot)+theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), axis.ticks.x=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+theme(legend.position="none")+xlab("")+ylab("Count of ASVs")
dev.off()


ggplot(kk,aes(x=Curated_v3, y=n, fill=Curated_v3))+geom_bar(stat='identity')+
  scale_x_discrete(guide = guide_axis(angle = 90))+scale_fill_manual(values=coloresbarplot)+theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+theme(legend.position="none")+xlab("")+ylab("Count of ASVs")
ggsave("labeled_barplot_ASVs_count.pdf")
dev.off()

```





```{r}

taxonomy_Phyto_binary <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(abund) 1*(abund>0))
taxonomy_Phyto_binary_table =as.data.frame(otu_table(taxonomy_Phyto_binary))


taxonomy_Phyto_binary_table2=taxonomy_Phyto_binary_table
taxonomy_Phyto_binary_table2$SUM=rowSums(taxonomy_Phyto_binary_table)
taxonomy_Phyto_binary_table2$occur=(taxonomy_Phyto_binary_table2$SUM/366)*100

#write.csv(rel2All_phyto_binary_table2,"rel2All_phyto_binary_table2.csv")

universal=taxonomy_Phyto_binary_table2%>%filter(occur>99)
head(universal)
taxonomy_Phyto_binary_table2%>%filter(occur>=75&occur<100)%>%tally()



noPond=subset_samples(physeq_Dav_no_dup_clean_no18S,Site=="SR8"|Site=="KAHO")

#check to see all of the uncharacterized phylum
rr=table(tax_table(physeq_Dav_no_dup_clean_no18S)[, "Phylum"], exclude = NULL)
gg=as.data.frame(rr)

ntaxa(All_phyto)

rr=table(tax_table(rel2All_phyto)[, "Class"], exclude = NULL)
mm=as.data.frame(rr)





#Bac_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Coxiellales= subset_taxa(rel2Bac_16S,Order == "D_3__Coxiellales")
ntaxa(Coxiellales)

pdf("Coxiellales.pdf", width=44,height=5)
plot_bar(Coxiellales, fill="ASV")
dev.off()

D_3__Coxiellales


	
D_3__Rickettsiales


taxonomy_Phyto_binary <- transform_sample_counts(All_phyto, function(abund) 1*(abund>0))
taxonomy_Phyto_binary_table =as.data.frame(otu_table(taxonomy_Phyto_binary))


taxonomy_Phyto_binary_table2=taxonomy_Phyto_binary_table
taxonomy_Phyto_binary_table2$SUM=rowSums(taxonomy_Phyto_binary_table)
taxonomy_Phyto_binary_table2$occur=(taxonomy_Phyto_binary_table2$SUM/366)*100
#write.csv(rel2All_phyto_binary_table2,"rel2All_phyto_binary_table2.csv")
taxonomy_Phyto_binary_table2$occurrence="NA"

  
taxonomy_Phyto_binary_table3=taxonomy_Phyto_binary_table2 %>% 
  mutate(occurrence = case_when(occur >= 100 ~ 'universal'
                                ,occur >= 75 ~ 'frequent'
                                ,occur >= 10 ~ 'common'
                                ,TRUE ~ 'rare'
                                )
  )                    


           
kk=taxonomy_Phyto_binary_table3%>%group_by(occurrence)%>%tally()
          
          
names(taxonomy_Phyto_binary_table3)
library(tibble)
gg2 <- tibble::rownames_to_column(taxonomy_Phyto_binary_table3, "ASV")
gg2=gg2%>%select(ASV, occurrence, occur)


taxonomy_occurrence = read.csv("taxonomy_table_final_tucker_v2.csv", sep=",", row.names=1)
ubiq2=left_join(taxonomy_occurrence, gg2, "ASV")

write.csv(ubiq2, "taxonomy_table_final_tucker_v4.csv")


halp = read.csv("halp.csv", sep=",") 

newotu=left_join(halp, gg2, by="ASV")

write.csv(newotu,"newotu_again.csv")

  
library(tibble)
df=df[,1:2]
df[,2]="ubiq"
colnames(df) <- c("ASV","ubiq")

taxonomy_occurrence = read.csv("taxonomy_table_final_tucker_v3.csv", sep=",", row.names=1)
ubiq2=left_join(taxonomy_occurrence, df, "ASV")



taxonomy_Phyto_binary_table2%>%filter(75>occur<10)%>%tally()


gg2=gg
library(tibble)
gg2 <- tibble::rownames_to_column(taxonomy_Phyto_binary_table3, "ASV")
gg2=gg2%>%select(ASV,SUM)
gg2$remove="remove"
halp = read.csv("halp.csv", sep=",") 

newotu=left_join(halp, gg2, by="ASV")

write.csv(newotu,"newotu_again.csv")

summary(taxonomy_Phyto_binary_table2$SUM)
taxonomy_Phyto_binary_table2%>%filter(SUM<2)%>%tally()

GP = filter_taxa(taxonomy_Phyto_binary , function(x) sum(x > 3) > (length(x)), TRUE)
```



```{r}


rel_otu_table_phyto=as.data.frame(otu_table(rel2All_phyto))


rel2All_phyto_binary <- transform_sample_counts(rel2All_phyto, function(abund) 1*(abund>0))
rel2All_phyto_binary_table =as.data.frame(otu_table(rel2All_phyto_binary))
rel2All_phyto_binary_table2=rel2All_phyto_binary_table
rel2All_phyto_binary_table2$SUM=rowSums(rel2All_phyto_binary_table)
rel2All_phyto_binary_table2$occur=(rel2All_phyto_binary_table2$SUM/366)*100
write.csv(rel2All_phyto_binary_table2,"rel2All_phyto_binary_table2.csv")

summary(rel2All_phyto_binary_table2$occur)
rel2All_phyto_binary_table2%>%filter(occur>=75&occur<=99)%>%tally()

Mydata %>% filter(x>=3&x<=7)

write.csv(e, "AllPhyto_binary.csv")

AllPhyto_abun_ASV <- All_phyto %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 
AllPhyto_abun_ASV_enviro=reshape2::dcast(AllPhyto_abun_ASV, OTU ~ Enviro2, value.var ="mean")


AllPhyto_binary <- AllPhyto_binary %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(sum = sum(Abundance)) 
AllPhyto_binary_enviro=dcast(AllPhyto_binary, OTU ~ Enviro2, value.var ="sum")


```


```{r}
AllPhyto_abun_ASV <- All_phyto %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(mean = round(mean(Abundance)*100, 1), sd=round(sd(Abundance)*100, 1)) 
AllPhyto_abun_ASV_enviro=reshape2::dcast(AllPhyto_abun_ASV, OTU ~ Enviro2, value.var ="mean")


AllPhyto_binary <- AllPhyto_binary %>% psmelt() %>% group_by(OTU,Enviro2) %>% summarise(sum = sum(Abundance)) 
AllPhyto_binary_enviro=dcast(AllPhyto_binary, OTU ~ Enviro2, value.var ="sum")
```








```{r}


diatom=subset_taxa(All_phyto_Rel, Class=="Bacillariophyta")
ntaxa(diatom)
plot_tree(diatom, color="Cluster_DeSeq_Spatial", label.tips="taxa_names", ladderize="left")


pro=subset_taxa(All_phyto_Rel, Genus=="D_5__Cyanobium_PCC-6307")
ntaxa(pro)
plot_tree(pro, color="Cluster_DeSeq", label.tips="taxa_names", ladderize="left", size="abundance")



pro=subset_taxa(All_phyto_Rel, Genus=="D_5__Synechococcus_CC9902")
ntaxa(pro)
plot_tree(pro, color="Cluster_DeSeq", label.tips="taxa_names", ladderize="left", size="abundance")



pro=subset_taxa(All_phyto_Rel, Genus=="D_5__Synechococcus_CC9902")
ntaxa(pro)
plot_tree(pro, color="Cluster_DeSeq", label.tips="taxa_names", ladderize="left", size="abundance")



```


```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", fill=TRUE)

PRO.mL_coastal=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")


ggplot(PRO.mL_coastal, aes(x=SW.Temperature.at.site, y=PRO.mL, color=Season_Aka, shape=Cluster_DeSeq))+geom_point()
ggsave("temp_PROml_coastal.pdf")




no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", fill=TRUE)

Fuco_coastal=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")

ggplot(Fuco_coastal, aes(x=SW.Temperature.at.site, y=Fuco, color=Season_Aka, shape=Cluster_DeSeq))+geom_point()

ggplot(Fuco_coastal, aes(x=Season_Aka, y=Fuco, fill=Season_Aka))+geom_boxplot()


ggplot(Fuco_coastal, aes(x=Season_Aka, y=chla.ug.per.L, fill=Season_Aka))+geom_boxplot()+ scale_y_continuous(trans = log10_trans())



chla <- Fuco_coastal%>%drop_na(chla.ug.per.L) 
summary(chla)

chla$Season_Aka=as.factor(chla$Season_Aka)
chla$Cluster_DeSeq=as.factor(chla$Cluster_DeSeq)
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(chla.ug.per.L ~ Cluster_DeSeq, data = chla)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))




```





```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", fill=TRUE)

SYN.mL_cluster3=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq=="Cluster_3")

ggplot(SYN.mL_cluster3, aes(x=SW.Temperature.at.site, y=SYN.mL))+geom_point()


SYN.mL_cluster3=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")

robot2

ggplot(SYN.mL_cluster3, aes(x=SW.Temperature.at.site, y=SYN.mL, color=Season_Aka, shape=Cluster_DeSeq))+geom_point()
ggsave("temp_Synml_coastal.pdf")

fit=lm(SYN.mL~SW.Temperature.at.site,data=SYN.mL_cluster3)
summary(fit)

ggplot(SYN.mL_cluster3, aes(x=SW.Temperature.at.site, y=SYN.mL))+geom_point()+
    geom_smooth(method=lm)


ggplot(SYN.mL_cluster3, aes(x=Site, y=SW.Temperature.at.site, fill=Season_Aka))+geom_boxplot() 



```



```{r}
link2data <- "https://raw.githubusercontent.com/cmdlinetips/data/master/palmer_penguins.csv"
penguins <- read_csv(link2data)
penguins <- penguins %>% drop_na()

names(penguins)
library(tidyverse)
X <- penguins %>% dplyr::select("bill_depth_mm", "bill_length_mm")%>%scale()

Y <- penguins %>%
  dplyr::select(flipper_length_mm,body_mass_g) %>%
  scale()
head(Y)

install.packages("CCA")
library(CCA)
cc_results <- cancor(X,Y)
```
```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", fill=TRUE)

coastal_tax_enviro=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq=="Cluster_4"| Cluster_DeSeq=="Cluster_5")%>%dplyr::select(c("SampleID","PRO.mL", "Salinity","NitrateNitrite", "chla.ug.per.L", "pH",
                                                       "SYN.mL", 
                                                       "PEUK.mL", 
                                                       "HBACT.mL", 
                                                       "Silicate",
                                                         "Phosphate",
                                                    "SW.Temperature.at.site", "Cluster_DeSeq"))%>%drop_na()


coastal_tax_noNut=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")%>%dplyr::select(c("SampleID","PRO.mL", "Salinity","chla.ug.per.L", "pH",
                                                       "SYN.mL", 
                                                       "PEUK.mL", 
                                                       "HBACT.mL","SW.Temperature.at.site", "Cluster_DeSeq"))%>%drop_na()



pop=coastal_tax_enviro[,2:12]
prcomp_enviro=coastal_tax_enviro
mtcars.pca <- prcomp(prcomp_enviro[,2:12], center = TRUE,scale. = TRUE)


summary(mtcars.pca)
str(mtcars.pca)
#install_github("vqv/ggbiplot")

library(ggbiplot)

ggbiplot(mtcars.pca)


labels=rownames(prcomp_enviro)

ggbiplot(mtcars.pca, groups=prcomp_enviro$Cluster_DeSeq)+
  scale_colour_manual(name="Clustering", values= c("#8b0000","#00528b","#e6d300"))+
  ggtitle("")+geom_point(aes(colour=prcomp_enviro$Cluster_DeSeq), size = 3)+
  theme(legend.position = "bottom")+theme(axis.text=element_text(size=24), legend.text=element_text(size=24), axis.title=element_text(size=24,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +theme(legend.position = "bottom") 

ggsave("enviro_grouping_taxonomy_v2.pdf", width=10, height=10)


```






```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", fill=TRUE)

coastal_tax_enviro=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")%>%dplyr::select(c("SampleID","PRO.mL", "Salinity","NitrateNitrite", "chla.ug.per.L", "pH",
                                                       "SYN.mL", 
                                                       "PEUK.mL", 
                                                       "HBACT.mL", 
                                                       "Silicate",
                                                         "Phosphate",
                                                    "SW.Temperature.at.site", "Cluster_DeSeq"))%>%drop_na()


coastal_tax_noNut=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")%>%dplyr::select(c("SampleID","PRO.mL", "Salinity","chla.ug.per.L", "pH",
                                                       "SYN.mL", 
                                                       "PEUK.mL", 
                                                       "HBACT.mL","SW.Temperature.at.site", "Cluster_DeSeq"))%>%drop_na()



pop=coastal_tax_enviro[,2:12]
prcomp_enviro=coastal_tax_enviro
mtcars.pca <- prcomp(prcomp_enviro[,2:12], center = TRUE,scale. = TRUE)


summary(mtcars.pca)
str(mtcars.pca)
#install_github("vqv/ggbiplot")

library(ggbiplot)

ggbiplot(mtcars.pca)


labels=rownames(prcomp_enviro)

ggbiplot(mtcars.pca, groups=prcomp_enviro$Cluster_DeSeq)+
  scale_colour_manual(name="Clustering", values= c("#8b0000","#00528b","#e6d300"))+
  ggtitle("")+geom_point(aes(colour=prcomp_enviro$Cluster_DeSeq), size = 3)+
  theme(legend.position = "bottom")+theme(axis.text=element_text(size=24), legend.text=element_text(size=24), axis.title=element_text(size=24,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +theme(legend.position = "bottom") 

ggsave("enviro_grouping_taxonomy_v2.pdf", width=10, height=10)





library(vegan)

dist <- vegdist(Enviro[,11:18],  method = "bray")

# PCoA is not included in vegan. 
# We will use the ape package instead
library(ape)
PCOA <- pcoa(dist)

# plot the eigenvalues and interpret
barplot(PCOA$values$Relative_eig[1:10])
# Can you also calculate the cumulative explained variance of the first 3 axes?

# Some distance measures may result in negative eigenvalues. In that case, add a correction:
PCOA <- pcoa(dist)

# Plot your results
biplot.pcoa(PCOA)

# You see what`s missing? 
# Indeed, there are no species plotted on this biplot. 
# That's because we used a dissimilarity matrix (sites x sites) 
# as input for the PCOA function. 
# Hence, no species scores could be calculated. 
#However, we could work around this problem like this:
biplot.pcoa(PCOA, Enviro[,11:18])








Enviro <- read.csv("metadata092921_noAprilWAI2KAHO_nMDS_Enviro.csv", row.names = 1)



Enviro_only2=Enviro[5:12]


head(Enviro)

clustering=Enviro[,5:12]

Enviro_only <- read.csv("metadata092921_noAprilKAHOWAI2_nMDS_only.csv", row.names = 1)
Enviro_only=Enviro_only %>% drop_na()



set.seed(123)
######try nMDS
bray=vegdist(Enviro_only, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, autotransform = FALSE)









#perform NMDS
Meta_NMDS = metaMDS(bray, k = 2)

#build a data frame with NMDS coordinates and metadata
 MDS1 = Meta_NMDS$points[,1]
 MDS2 = Meta_NMDS$points[,2]
 NMDS = data.frame(MDS1 = MDS1, MDS2 = MDS2, Enviro4=Enviro$Enviro4)
 #NMDS
 #pdf("metadata_nMDS_NERR.pdf")
 ggplot(NMDS, aes(x=MDS1, y=MDS2, col=Enviro$Enviro2, shape=Enviro$Season)) +
   geom_point() + geom_text(aes(label=Enviro$Site)) +
labs(col="type") +
 theme_bw() +
 labs(title = "NMDS Plot of YSI,chla,FCM data w Bray Distance")
 
 
 
 

 
 
 p= ggplot(NMDS, aes(x=MDS1, y=MDS2, col=Enviro$Enviro4)) +
   geom_point()  +
labs(col="Environment Types") +
 labs(title = "NMDS Plot of YSI,chla,FCM data w Bray Distance") + theme(axis.text=element_text(size=14), legend.text=element_text(size=14),panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 1))

 p + scale_color_manual(values = c("#e9f12d", "#b4145e","#1d78d4"))
 ggsave("nMDS_YSI_FCM_CHLA_v3.svg")
 
 
 

Meta_NMDS$stress
#stress values below 0.05 represent a good fit 

#In a formal analysis, MDS plots are usually accompanied by some multivariate statistical test of dissimilarity between treatment/observational groups, e.g., via the adonis function in vegan.

#anosim_time = anosim(bray, Enviro$Enviro2)
 #anosim_time # take a look at results
 #summary(anosim_time)
# plot(anosim_time)

 
#adonis(bray~Enviro2,Enviro)

library(pairwiseAdonis)
pairwise.adonis2(coastal_tax_enviro[,2:12]~Cluster_DeSeq,Cluster_DeSeq)


library(pairwiseAdonis)
pairwise.adonis2(bray~Season*Enviro2,Enviro)

 
library(pairwiseAdonis)
pairwise.adonis2(bray~Site,Enviro)




#adonis_location = adonis(dis ~ site, pd)
#adonis_location # take a look at results; there are no summary() or plot() methods included
 


 ####bdisp
#bdisp=betadisper(bray,Enviro$Enviro2, bias.adjust = TRUE)
#anova(bdisp)
#TukeyHSD(bdisp)
#boxplot(bdisp)






#ANOSIM is a method that tests whether two or more groups of samples are significantly different (similar to adonis, above). You can specify a category in the metadata to separate samples into groups and then test whether there are significant differences between those groups.  Since ANOSIM is nonparametric, statistical significance is determined through permutations.Note: ANOSIM only works with a categorical variable that is used to do the grouping. Mantel is recommended for continuous variables. Running ANOSIM on groups with very different dispersions can lead to unreliable results. Groups with very different dispersions may produce high R values, even if there's no real difference in their centroids. If differences in group dispersion are as meaningful to your analysis as differences in group centre, this may not be an issue.One- and Two-way ANOSIM are available

##Adonis essentially says that our communities are statistically different from each other, contrary to what our plot seemed to show.  Let’s take a look at PermANOVA, using adonis(), which is generally considered to be more robust than anosim().

```






```{r}

#rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
#ntaxa(rel.abun)

#head(tax_table(rel.abun))
#ntaxa(rel.abun)
#3987
#nsamples(rel.abun)
#366

#write.csv(otu_table(physeq_Dav_no_dup_clean_no18S), "Count_data_physeq_Dav_no_dup_clean_no18S.csv")
#write.csv(tax_table(physeq_Dav_no_dup_clean_no18S), "Taxonomy_Count_data_physeq_Dav_no_dup_clean_no18S.csv")

#All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
#ntaxa(All_phyto)
#write.csv(otu_table(All_phyto), "Count_data_All_phyto.csv")
#write.csv(tax_table(All_phyto), "Taxonomy_Count_data_All_phyto.csv")
#fred=as.data.frame(tax_table(All_phyto))

#counts.filtered.phyto=read.csv("Count_tax_physeq_Dav_no_dup_clean_no18S.csv", header=TRUE)

#counts.filtered.phyto=read.csv("Count_data_All_phyto_taxonomy.csv", header=TRUE)


#counts.filtered.phyto=read.csv("Count_tax_physeq_Dav_no_dup_clean_no18S.csv", header=TRUE)
# (3) Create taxonomy key:
#names(counts.filtered.phyto)
#seq_counts<-counts.filtered.phyto[1:367]
#tax_key<-counts.filtered.phyto[c(1,368:384)]; head(tax_key[1:2,])
# Modify seq_counts data frame so row.names = OTU.ID and all values are numeric
#row.names(seq_counts)<-seq_counts$OTU.ID
#seq_counts$OTU.ID<-NULL; head(seq_counts[1:2,])


#transformed_counts<-DESeq2::varianceStabilizingTransformation(as.matrix(seq_counts))
#head(transformed_counts)
#class(transformed_counts)
#transformed_counts2=as.data.frame(transformed_counts)
#write.csv(transformed_counts2, "transformed_counts_physeq_Dav_no_dup_clean_no18S")
#just_phyto_subset_transformed_counts.csv
#transformed_counts2$ASV=row.names(transformed_counts2)
#transformed_counts2$ASV
#tosubset=left_join(fred, transformed_counts2, by="ASV")

#write.csv(tosubset, "tosubset.csv")
phyto_try2.csv

phyto_subset_transformed=read.csv("phyto_try2.csv", row.names=1)
euc_dist <- dist(t(phyto_subset_transformed))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 
plot(d1)
?hclust()

### for the colored bars 
k234 <- cutree(d1, k = 2:5)

pdf("wtf.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance just phytoplankton", horiz = TRUE)
colored_bars(cbind(k234[,4], col_Year_type, col_Season_type, col_Site_type), d1, rowLabels = c(paste0("k = ", 5),"Year", "Seasons" , "Site"), horiz = TRUE, y_shift=20)
dev.off()



pdf("wtf.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance just phytoplankton", horiz = TRUE)
colored_bars(cbind(k234[,4], col_Year_type, col_Season_type, col_Site_type, col_Season_Aka_type), d1, rowLabels = c(paste0("k = ", 5),"Year", "Seasons" , "Site"), horiz = TRUE, y_shift=20)
dev.off()





svg("VST_Dendrogram_VST_counts_just_phyto_4.svg", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance just phytoplankton", horiz = TRUE)
colored_bars(cbind(k234[,4]), d1, rowLabels = c(paste0("k = ", 5)), horiz = TRUE, y_shift=20)
dev.off()



pdf("VST_Dendrogram_VST_counts_just_phyto_4_season.pdf", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="VST Euclidean distance just phytoplankton", horiz = TRUE)
colored_bars(cbind(k234[,4], col_Clustering_type, col_Season_Aka_type), d1, rowLabels = c(paste0("k = ", 5),"Cluster", Season_aka), horiz = TRUE, y_shift=20)
dev.off()


col_Season_Aka_type

#if (!requireNamespace("BiocManager", quietly = TRUE))
  #  install.packages("BiocManager")

#BiocManager::install("DESeq2")
library(colorspace)
library(dendextend)
library(DESeq2)

All_16S_plastid=subset_taxa(physeq_Dav_no_dup_clean,Domain == "Eukaryota_plas" |Domain == "D_0__Bacteria" |Domain == "D_0__Archaea")
table(tax_table(All_16S_plastid)[, "Phylum"], exclude = NULL)

Clustering_type <- factor(sample_data(All_16S_plastid)$Cluster_DeSeq, levels=c("Cluster_1","Cluster_2","Cluster_3", "Cluster_4", "Cluster_5"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_cluster <- c("dodgerblue3","darkseagreen","coral","#b19cd9", "gray44")
col_Clustering_type <- colors_cluster[Clustering_type]


Site_type <- factor(sample_data(All_16S_plastid)$Site, levels=c("WAI2", "KAHO","HP1", "CB","AR" ,"SR8", "SB", "NB", "SR2", "NR2", "STO1", "NTO1"))
colors_Site <- c('#7F0000','#f94279',"#ffd000","#ffa812","#c78100", "#ff9966", "#c46210", "#ff7e00","#009dcc","#00bfcc", '#0085ff','#0045ff')
col_Site_type <- colors_Site[Site_type]


Enviro_type <- factor(sample_data(All_16S_plastid)$Environment, levels=c("Estuary", "Coastal", "Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_c <- c("firebrick3","gold", "dodgerblue1")
col_Enviro_type <- colors_c[Enviro_type]



Season_type <- factor(sample_data(All_16S_plastid)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkorange","darkturquoise", "darkorchid","darkolivegreen1" )
col_Season_type <- colors_Season[Season_type]

sample_data(All_16S_plastid)$Season_Aka

Season_Aka_type <- factor(sample_data(All_16S_plastid)$Season_Aka, levels=c("Dry", "Wet"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season_Aka <- c("red", "blue")
col_Season_Aka_type <- colors_Season_Aka[Season_Aka_type]


Year_type <- factor(sample_data(All_16S_plastid)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]



```

```{r}



Season_type <- factor(sample_data(aa)$Season, levels=c("Summer", "Fall", "Winter", "Spring"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Season <- c("darkturquoise", "darkolivegreen1","darkorange","darkorchid" )

Year_type <- factor(sample_data(aa)$Year, levels=c("2017", "2018", "2019", "2020", "2021"))
#n_Season_types <- length(unique(Season_type))
#Season_cols <- colorspace::rainbow_hcl(n_Season_types, c = 70, l  = 50)
colors_Year <- c("oldlace","lightpink","hotpink1","deeppink", "red1", "maroon")
col_Year_type <- colors_Year[Year_type]


aa <- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq,Date1, Site)%>%tally() 
aa$Date1 <- as.Date(aa$Date1,format = "%m/%d/%y")
bb=subset(aa, Cluster_DeSeq=="Cluster_4" | Cluster_DeSeq=="Cluster_5")
ggplot(data=bb, aes(x=as.Date(Date1), y=n, fill=Cluster_DeSeq))+ geom_bar(aes(), stat="identity", position="stack")+xlab("Date")
ggsave("Cluster_over_time.pdf")


aa <- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq,Date1, Site)%>%tally() 
aa$Date1 <- as.Date(aa$Date1,format = "%m/%d/%y")
bb=subset(aa, Cluster_DeSeq=="Cluster_4" | Cluster_DeSeq=="Cluster_5"| Cluster_DeSeq=="Cluster_3")
ggplot(data=bb, aes(x=as.Date(Date1), y=n, fill=Cluster_DeSeq))+ geom_bar(aes(), stat="identity", position="stack")+xlab("Date")
ggsave("Cluster_over_time_coastal.pdf")


aa <- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq,Date1, Site, Season)%>%tally() 
aa$Date1 <- as.Date(aa$Date1,format = "%m/%d/%y")
bb=subset(aa, Cluster_DeSeq=="Cluster_4" | Cluster_DeSeq=="Cluster_5"| Cluster_DeSeq=="Cluster_3")
cc=subset(bb, Season=="Winter")
ggplot(data=cc, aes(x=as.Date(Date1), y=n, fill=Cluster_DeSeq))+ geom_bar(aes(), stat="identity", position="stack")+xlab("Date")
ggsave("Cluster_over_time_coastal.pdf")


aa=no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq, Site)%>%tally()
aa$Site <- factor(aa$Site, levels=c("SR8", "KAHO","HP1", "CB","AR" ,"SR8", "SB", "NB", "SR2", "NR2", "STO1", "NTO1"))
colors_Site <- c('#7F0000','#f94279',"#ffd000","#ffa812","#c78100", "#ff9966", "#c46210", "#ff7e00","#00cc95","#00cc2f", '#0085ff','#0045ff')

ggplot(data=aa, aes(x=Cluster_DeSeq, y=n, fill=as.factor(Site)))+ geom_bar(aes(), stat="identity", position="stack")+scale_fill_manual(values=colorssite)+ylab("n Samples")+xlab("Clustering by Phytoplankton Community Composition")
ggsave("Clustering_explain_Site.pdf")

aa=no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq, Season_Aka)%>%tally()

ggplot(data=aa, aes(x=Cluster_DeSeq, y=n, fill=Season_Aka))+ geom_bar(aes(), stat="identity", position="stack") +ylab("n Samples")+xlab("Clustering by Phytoplankton Community Composition")
ggsave("Clustering_explain_Season_AKA.pdf")

+scale_fill_manual(values=colors_Season)

aa=no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq, Year)%>%tally()


ggplot(data=aa, aes(x=Cluster_DeSeq, y=n, fill=as.factor(Year)))+ geom_bar(aes(), stat="identity", position="stack") +scale_fill_manual(values=colors_Year)+ylab("n Samples")+xlab("Clustering by Phytoplankton Community Composition")
ggsave("Clustering_explain_Year.pdf")




aa=no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq, Site)%>%tally()

```




```{r}
Ammonia<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Cluster_DeSeq), y=SYN.mL))+geom_boxplot(color="black", fill="gray53") 
Ammonia=Ammonia + theme(legend.title = element_blank()) + ylab("SYN per mL") +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Ammonia

ggsave("SYN_per_mL_perCluster.pdf")

library(multcomp)




chla <- no_dup_metadata_KBytNERR%>%drop_na(PRO.mL) 
summary(chla)

chla$Cluster_DeSeq=as.factor(chla$Cluster_DeSeq)

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(PRO.mL ~ Cluster_DeSeq, data = chla)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))

  
 Ammonia<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Cluster_DeSeq), y=PEUK.mL))+geom_boxplot(color="black", fill="gray53") 
Ammonia=Ammonia + theme(legend.title = element_blank()) + ylab("PEUK per mL") +xlab("") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Ammonia

ggsave("PEUK_per_mL_perCluster.pdf")

 
library(scales)
chla<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=chla.ug.per.L, color=Cluster_DeSeq))+geom_boxplot() 
chla=chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a \n (ug per L)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) +scale_y_continuous(trans = log10_trans())
chla

ggsave("chla_clusters.pdf", height=2)

Nitrate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=NitrateNitrite, color=Cluster_DeSeq))+geom_boxplot() 
Nitrate=Nitrate + theme(legend.title = element_blank()) + ylab("Nitrate+Nitrite \n (uM)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
Nitrate

ggsave("Nitrate_clusters.pdf", height=2)

Phosphate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=Phosphate, color=Cluster_DeSeq))+geom_boxplot() 
Phosphate=Phosphate + theme(legend.title = element_blank()) + ylab("Phosphate \n (uM)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
Phosphate

ggsave("Phosphate_clusters.pdf", height=2)

Silicate<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=Silicate, color=Cluster_DeSeq))+geom_boxplot() 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Silicate \n (uM)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster)  + scale_y_continuous(trans = log10_trans())
Silicate

ggsave("Silicate_clusters.pdf", height=2)
library(cowplot)
prow=plot_grid(chla, Silicate, Nitrate,Phosphate, align="hv", ncol=1, nrow=4)

prow=plot_grid(chla+theme(legend.position="none"), Nitrate+theme(legend.position="none"),Phosphate+theme(legend.position="none"), Silicate+theme(legend.position="none"), align="hv", ncol=1, nrow=4, rel_widths = c(1, -0.1, 1))

prow

legend_b <- get_legend(
  chla + 
    guides(color = guide_legend(nrow = 1)) +
    theme(legend.position = "bottom")
)

# add the legend underneath the row we made earlier. Give it 10%
# of the height of one plot (via rel_heights).
plot_grid(prow, legend_b, ncol = 1, rel_heights = c(1,.1))
ggsave("all_nutrients_clusters.pdf", height=10, width=8)



SYN<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=SYN.mL, color=Cluster_DeSeq))+geom_boxplot() 
SYN=SYN + theme(legend.title = element_blank()) + ylab("Synechococcus \n cells per mL") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
SYN

ggsave("SYN_clusters.pdf", height=2)


HBACT<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=HBACT.mL, color=Cluster_DeSeq))+geom_boxplot() 
HBACT=HBACT + theme(legend.title = element_blank()) + ylab("Heterotrophic \n prokaryotic \n cells per mL") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
HBACT

ggsave("HBACT_clusters.pdf", height=2)


PEUK<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=PEUK.mL, color=Cluster_DeSeq))+geom_boxplot() 
PEUK=PEUK + theme(legend.title = element_blank()) + ylab("Photosynthetic \n picoeukaryotic \n cells per mL") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
PEUK
ggsave("PEUK_clusters.pdf", height=2)



PRO<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=PRO.mL, color=Cluster_DeSeq))+geom_boxplot() 
PRO=PRO + theme(legend.title = element_blank()) + ylab("Prochlorococcus \n cells per mL") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
PRO
ggsave("PRO_clusters.pdf", height=2)



Salinity<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=Salinity, color=Cluster_DeSeq))+geom_boxplot() 
Salinity=Salinity+ theme(legend.title = element_blank()) + ylab("Salinity") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
Salinity

ggsave("SalinityCluster.pdf",height=2)



pH<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=pH, color=Cluster_DeSeq))+geom_boxplot() 
pH=pH+ theme(legend.title = element_blank()) + ylab("pH") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
pH

ggsave("pHCluster.pdf",height=2)
names(no_dup_metadata_KBytNERR)

Temp<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=Cluster_DeSeq, y=SW.Temperature.at.site, color=Cluster_DeSeq))+geom_boxplot() 
Temp=Temp+ theme(legend.title = element_blank()) + ylab("Seawater \n Temperature (C)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster) 
Temp

ggsave("TempCluster.pdf",height=2)





 + scale_y_continuous(trans = log10_trans())

  geom_jitter(width = 0.1, aes(colour = Month))
  
  
```






```{r}
ggplot(no_dup_metadata_KBytNERR, aes(x=chla.ug.per.L)) + geom_vline(xintercept = 2.5)+
  geom_histogram(position="identity")+facet_wrap(~Cluster_DeSeq,scales="free" )

aa<- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq)%>%tally(chla.ug.per.L>2.5) 
names(aa)[names(aa) == "n"] <- "nmeso"
bb <- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq)%>%tally()
cc<- no_dup_metadata_KBytNERR%>%group_by(Cluster_DeSeq)%>%tally(chla.ug.per.L>8) 
names(cc)[names(aa) == "n"] <- "neutrophic"

dd=cbind(aa,bb)
ee=cbind(dd,cc)
write.csv(ee, "Chlorophyllastatus.csv")


stuff<- no_dup_metadata_KBytNERR%>%select(SW.Temperature.at.site, Site, Month, SampleID)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Month)%>%dplyr::summarise(mean=round(mean(SW.Temperature.at.site),1), sd=round(sd(SW.Temperature.at.site),1))

stuff<- no_dup_metadata_KBytNERR%>%drop_na(SW.Temperature.at.site)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Month)%>%dplyr::summarise(mean=round(mean(SW.Temperature.at.site),1), sd=round(sd(SW.Temperature.at.site),1))
write.csv(stuff, "temp_by_month.csv")

stuff<- no_dup_metadata_KBytNERR%>%drop_na(HBACT.mL)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Month)%>%dplyr::summarise(mean=round(mean(HBACT.mL),1), sd=round(sd(HBACT.mL),1))
write.csv(stuff, "hbatmonthcluster.csv")

stuff<- no_dup_metadata_KBytNERR%>%drop_na(SYN.mL)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Month)%>%dplyr::summarise(mean=round(mean(SYN.mL),1), sd=round(sd(SYN.mL),1))
write.csv(stuff, "SYNmonthcluster.csv")




stuff<- no_dup_metadata_KBytNERR%>%drop_na(SW.Temperature.at.site)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Month)%>%dplyr::summarise(mean=round(mean(SW.Temperature.at.site),1), sd=round(sd(SW.Temperature.at.site),1))
write.csv(stuff, "temp_by_month.csv")

stuff<- no_dup_metadata_KBytNERR%>%drop_na(SW.Temperature.at.site)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Season)%>%dplyr::summarise(mean=round(mean(SW.Temperature.at.site),1), sd=round(sd(SW.Temperature.at.site),1))
write.csv(stuff, "temp_Season.csv")

stuff<- no_dup_metadata_KBytNERR%>%drop_na(HBACT.mL)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Season)%>%dplyr::summarise(mean=round(mean(HBACT.mL),1), sd=round(sd(HBACT.mL),1))
write.csv(stuff, "hbatseasoncluster.csv")

stuff<- no_dup_metadata_KBytNERR%>%drop_na(SYN.mL)%>%dplyr::group_by(Cluster_DeSeq_Spatial, Season)%>%dplyr::summarise(mean=round(mean(SYN.mL),1), sd=round(sd(SYN.mL),1))
write.csv(stuff, "SYNseasoncluster.csv")




stuff<- no_dup_metadata_KBytNERR%>%dplyr::group_by(Cluster_DeSeq)%>%dplyr::summarise(mean=round(mean(chla.ug.per.L),1), sd=round(sd(chla.ug.per.L),1)) %>% tally()

stuff<- no_dup_metadata_KBytNERR%>%drop_na(PRO.mL)%>%dplyr::group_by(Cluster_DeSeq)%>%dplyr::summarise(mean=round(mean(PRO.mL),1), sd=round(sd(PRO.mL),1)) 

stuff<- no_dup_metadata_KBytNERR%>%drop_na(PRO.mL)%>% filter(PRO.mL > 0) %>% dplyr::group_by(Cluster_DeSeq)%>%tally()

notoligo=subset(no_dup_metadata_KBytNERR, chla.ug.per.L>2.5)%>%select(Month, SampleID, Site,Season, chla.ug.per.L)


summarise(PRO.mL, count_gt_0)

count_gt_0 = function(x) {sum(x > 0)}
count_gt_0(no_dup_metadata_KBytNERR$PRO.mL)

count(PRO.mL>0)

```

#####to check for outliers
```{r}
library(dplyr)

vars <- c("SW.Temperature.at.site","pH", "Salinity", "chla.ug.per.L", "PEUK.mL", "HBACT.mL", "SYN.mL", "PRO.mL", "Phosphate", "Silicate", "NitrateNitrite", "Ammonia")

Outlier_check <- vector("list", length(vars))

for (i in vars) {

  Outlier_check[[i]] <- no_dup_metadata_KBytNERR %>%
    group_by(Cluster_DeSeq) %>% 
    summarise(min = min(get(i), na.rm = TRUE),
              max = max(get(i), na.rm = TRUE),
              median = median(get(i), na.rm = TRUE),
              MAD = mad(get(i), na.rm = TRUE),
              MAD_lowlim = median - (3 * MAD),
              MAD_highlim = median + (3 * MAD),
              Outliers_low = any(get(i) < MAD_lowlim, na.rm = TRUE),
              Outliers_high = any(get(i) > MAD_highlim, na.rm = TRUE)
    )
  assign(paste(i, "Outlier_check", sep = "_"), i)
}

Outlier_check

Outlier_check
Outline_Temp=as.data.frame(Outlier_check$SW.Temperature.at.site)
Outline_pH=as.data.frame(Outlier_check$pH)
Outline_Salinity=as.data.frame(Outlier_check$Salinity)
Outline_chla=as.data.frame(Outlier_check$chla.ug.per.L)
Outline_PEUK=as.data.frame(Outlier_check$PEUK.mL)
Outline_HBACT=as.data.frame(Outlier_check$HBACT.mL)
Outline_SYN=as.data.frame(Outlier_check$SYN.mL)
Outline_PRO=as.data.frame(Outlier_check$PRO.mL)
Outline_Phosphate=as.data.frame(Outlier_check$Phosphate)
Outline_Silicate=as.data.frame(Outlier_check$Silicate)
Outline_NitrateNitrite=as.data.frame(Outlier_check$NitrateNitrite)
Outline_Ammonia=as.data.frame(Outlier_check$Ammonia)



Outline_enviro_mean=do.call("cbind", list(Outline_Temp, Outline_pH, Outline_Salinity, Outline_chla, Outline_PEUK,Outline_HBACT, Outline_SYN, Outline_PRO,Outline_Phosphate, Outline_Silicate, Outline_NitrateNitrite,Outline_Ammonia))

write.csv(Outline_enviro_mean,"Outline_enviro_mean.csv")




```





```{r}

vars <- c("SW.Temperature.at.site","pH", "Salinity", "chla.ug.per.L", "PEUK.mL", "HBACT.mL", "SYN.mL", "PRO.mL", "Phosphate", "Silicate", "NitrateNitrite", "Ammonia")

Outlier_check <- vector("list", length(vars))

for (i in vars) {

  Outlier_check[[i]] <- no_dup_metadata_KBytNERR %>%
    group_by(Cluster_DeSeq) %>% 
    summarise(min = min(get(i), na.rm = TRUE),
              max = max(get(i), na.rm = TRUE),
              median = median(get(i), na.rm = TRUE),
              MAD = mad(get(i), na.rm = TRUE),
              MAD_lowlim = median - (3 * MAD),
              MAD_highlim = median + (3 * MAD),
              Outliers_low = any(get(i) < MAD_lowlim, na.rm = TRUE),
              Outliers_high = any(get(i) > MAD_highlim, na.rm = TRUE)
    )
  assign(paste(i, "Outlier_check", sep = "_"), i)
}

Outlier_check
```


```{r}

vars <- c("SW.Temperature.at.site","pH", "Salinity", "chla.ug.per.L", "PEUK.mL", "HBACT.mL", "SYN.mL", "PRO.mL", "Phosphate", "Silicate", "NitrateNitrite", "Ammonia")

Outlier_check <- vector("list", length(vars))

for (i in vars) {

  Outlier_check[[i]] <- no_dup_metadata_KBytNERR %>%
    group_by(Cluster_DeSeq_Spatial, Season) %>% 
    summarise(mean=round(mean(get(i),na.rm = TRUE),1),
              sd=round(sd(get(i),na.rm = TRUE),1),
              min=min(get(i),na.rm = TRUE),
              max=max(get(i),na.rm = TRUE)
    )


}

Outlier_check
Season_Temp=as.data.frame(Outlier_check$SW.Temperature.at.site)
Season_pH=as.data.frame(Outlier_check$pH)
Season_Salinity=as.data.frame(Outlier_check$Salinity)
Season_chla=as.data.frame(Outlier_check$chla.ug.per.L)
Season_PEUK=as.data.frame(Outlier_check$PEUK.mL)
Season_HBACT=as.data.frame(Outlier_check$HBACT.mL)
Season_SYN=as.data.frame(Outlier_check$SYN.mL)
Season_PRO=as.data.frame(Outlier_check$PRO.mL)
Season_Phosphate=as.data.frame(Outlier_check$Phosphate)
Season_Silicate=as.data.frame(Outlier_check$Silicate)
Season_NitrateNitrite=as.data.frame(Outlier_check$NitrateNitrite)
Season_Ammonia=as.data.frame(Outlier_check$Ammonia)



Season_enviro_mean=do.call("cbind", list(Season_Temp, Season_pH, Season_Salinity, Season_chla, Season_PEUK,Season_HBACT, Season_SYN, Season_PRO,Season_Phosphate, Season_Silicate, Season_NitrateNitrite,Season_Ammonia))

write.csv(Season_enviro_mean,"Season_enviro_mean.csv")



```




```{r}
library(dplyr)


vars <- c("SW.Temperature.at.site","pH", "Salinity", "chla.ug.per.L", "PEUK.mL", "HBACT.mL", "SYN.mL", "PRO.mL", "Phosphate", "Silicate", "NitrateNitrite", "Ammonia")

Outlier_check <- vector("list", length(vars))

for (i in vars) {

  Outlier_check[[i]] <- no_dup_metadata_KBytNERR %>%
    group_by(Cluster_DeSeq_Spatial, Season) %>% 
    summarise(mean=round(mean(get(i),na.rm = TRUE),1),
              sd=round(sd(get(i),na.rm = TRUE),1),
              min=min(get(i),na.rm = TRUE),
              max=max(get(i),na.rm = TRUE)
    )
  
}

Outlier_check

rename(!!paste0('NO_', 2) := mpg, 'SI' = carb)

 assign(paste(get(i), "booop", sep = "_"), get(i))
row.names(SumsIa3V)[hh[1]] <- gsub("mean",vars) 

Season_Temp=as.data.frame(Outlier_check$SW.Temperature.at.site)
Season_pH=as.data.frame(Outlier_check$pH)
Season_Salinity=as.data.frame(Outlier_check$Salinity)
Season_chla=as.data.frame(Outlier_check$chla.ug.per.L)
Season_PEUK=as.data.frame(Outlier_check$PEUK.mL)
Season_HBACT=as.data.frame(Outlier_check$HBACT.mL)
Season_SYN=as.data.frame(Outlier_check$SYN.mL)
Season_PRO=as.data.frame(Outlier_check$PRO.mL)
Season_Phosphate=as.data.frame(Outlier_check$Phosphate)
Season_Silicate=as.data.frame(Outlier_check$Silicate)
Season_NitrateNitrite=as.data.frame(Outlier_check$NitrateNitrite)
Season_Ammonia=as.data.frame(Outlier_check$Ammonia)



```




```{r}
# Plot cluster-to-cluster taxonomic differences
## Values are equal to the total number of OTUs 
## belonging to each taxonomic group
#
tax_color<-c("#67000d","#e31a1c","#dd3497","#fcbba1","#fed976","#fc8d59","#a63603","#addd8e","#7f2704","#238b45","#a1d99b","#081d58","#1f78b4","#a6cee3","#8c6bb1","#9e9ac8","#984ea3","#081d58","#662506","#ffffff","#969696","#525252","#000000")


test=psmelt(rel2All_phyto)


coloresbarplot=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="forestgreen",
"Archaeplastida:Bathycoccus"="darkolivegreen4", 
"Archaeplastida:Ostreococcus"="darkolivegreen3",
"Archaeplastida:Micromonas"="darkseagreen",
"Archaeplastida:Pedinophyceae"="limegreen",
"Archaeplastida:Other"="chartreuse4", 
"Hacrobia:Other"="thistle2",
"Hacrobia:Cryptomonadales"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiales"="red",
"Stramenopiles:Diatoms"="gold",
"Stramenopiles:Dictyochophyceae"="khaki1",
"Stramenopiles:Pelagophyceae"="darkgoldenrod3",
"Stramenopiles:Other"="yellow",
'Eukaryota:Other'="gray44",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")




test$Curated_v3=factor(test$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))




test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")



############################3

Three_t=subset(test, Cluster_DeSeq=="Cluster_3")
p_3<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +facet_wrap(~Cluster_DeSeq)
p_3

ggsave("Cluster_3.pdf")


Three_t=subset(test, Cluster_DeSeq=="Cluster_4")
p_4<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +facet_wrap(~Cluster_DeSeq)
p_4

ggsave("Cluster_4.pdf")


Three_t=subset(test, Cluster_DeSeq=="Cluster_5")
p_5<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +facet_wrap(~Cluster_DeSeq)
p_5

ggsave("Cluster_5.pdf")


Three_t=subset(test, Cluster_DeSeq=="Cluster_2")
p_2 <- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +facet_wrap(~Cluster_DeSeq)
p_2

ggsave("Cluster_2.pdf")


Three_t=subset(test, Cluster_DeSeq=="Cluster_1")
p_1<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +facet_wrap(~Cluster_DeSeq)
p_1

ggsave("Cluster_1.pdf")


library(cowplot)

plot_grid(p_3, p_4, p_5,p_1, p_2, align="hv", ncol=3, nrow=2)

ggsave("Clusters_color_Ordered_Site_Sample_v2.pdf", width=24, height=10)
```


```{r}


rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID, Cluster_DeSeq, Curated_v3) %>% dplyr::summarise(sum = sum(Abundance)) 


names(rel2Phyto_ASV_total)

rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Cluster_DeSeq, Curated_v3) %>% dplyr::summarise(mean = mean(sum), sd=sd(sum)) 

rel2Phyto_ASV_Cluster $percent_total_mean=rel2Phyto_ASV_Cluster  $mean*100
rel2Phyto_ASV_Cluster $percent_total_SD=rel2Phyto_ASV_Cluster $sd*100



coastal_rel=rel2Phyto_ASV_Cluster%>%subset(Cluster_DeSeq=="Cluster_3" | Cluster_DeSeq=="Cluster_4" | Cluster_DeSeq=="Cluster_5")



coastal_rel$Curated_v3=factor(coastal_rel$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))




test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")



############################3
svg("Coastal_RA_Clusters_Mean.svg", width=2, height=3)

p_3<- ggplot(data=coastal_rel, aes(x=Cluster_DeSeq, y=percent_total_mean, fill=as.factor(Curated_v3))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.title=element_text(size=12,face="bold"), axis.text.x = element_blank()) + scale_fill_manual(values=coloresbarplot)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") 
p_3
dev.off()
ggsave("Coastal_Clusters.pdf", width=2, height=3)



colorscluster_coastal=c(
"Cluster_3"="coral", 
"Cluster_4"="#b19cd9",
"Cluster_5"="gray44")



colorscluster=c("Cluster_1"="dodgerblue3",
"Cluster_2" ="darkseagreen",
"Cluster_3"="coral", 
"Cluster_4"="#b19cd9",
"Cluster_5"="gray44")

coastal_envi=subset(no_dup_metadata_KBytNERR, Cluster_DeSeq=="Cluster_3" | Cluster_DeSeq=="Cluster_4" | Cluster_DeSeq=="Cluster_5")

svg("Temp_coastal_Box_Site.svg", width=2, height=2)
Temp<- ggplot(data=coastal_envi, aes(x=Cluster_DeSeq, y=SW.Temperature.at.site, color=Cluster_DeSeq))+geom_boxplot() 

Temp2=Temp + theme(legend.title = element_blank()) + ylab("Temperature C") +xlab("") +geom_jitter(aes())+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster_coastal) + theme(legend.position = "none")
Temp2
dev.off()
ggsave("Temperature_Coastal_Clusters.pdf", width=2, height=2)


svg("Temp_Syn_coastal.svg", width=3, height=2)
SYN<- ggplot(data=coastal_envi, aes(x=SW.Temperature.at.site, y=SYN.mL, color=Cluster_DeSeq))+geom_point() + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorscluster_coastal) + theme(legend.position = "none")+ylab("Synechococcus \n cells per mL")+xlab("Temperature C")
SYN
dev.off()
ggsave("Temperature_SY_Coastal_Clusters.pdf", width=3, height=2)

```



```{r}
coastal_envi$Cluster_DeSeq=factor(coastal_envi$Cluster_DeSeq, levels=c("Cluster_3", "Cluster_4", "Cluster_5"))

ggplot(data=coastal_envi, aes(x=Cluster_DeSeq, y=PEUK.mL, color=Cluster_DeSeq))+geom_boxplot() 
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(PEUK.mL~ Cluster_DeSeq, data = coastal_envi)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))
  
library(multcomp)


coastal_envi_noNA$Cluster_DeSeq=factor(coastal_envi_noNA$Cluster_DeSeq, levels=c("Cluster_3", "Cluster_4", "Cluster_5"))

coastal_envi_noNA=coastal_envi%>%drop_na(Silicate)

library(scales)
?log10_trans()

ggplot(data=coastal_envi_noNA, aes(x=Cluster_DeSeq, y=chla.ug.per.L, color=Cluster_DeSeq))+geom_boxplot() +scale_y_continuous(trans=log10_trans())
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(chla.ug.per.L~ Cluster_DeSeq, data = coastal_envi_noNA)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Cluster_DeSeq= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
  
```




```{r}
library(tidyverse) 
#install.packages("speedyseq")
#library(speedyseq)
library(mgcv)
library(broom)
library(microbiome)
library(microbiomeutilities)
?transform()


test=subset_samples(Coastal_All_phyto, chla.ug.per.L!="NA")

clr=transform(Coastal_All_phyto, transform="clr")

psmelt.clr.intragenus=psmelt(clr)


psmelt.clr.intragenus$SampleID

models.par <- psmelt.clr.intragenus  %>% 
  filter(OTU %in% taxa_names(bl.phy.gen.filt)) %>% 
  filter(Sample_name %in% sample_names(bl.phy.gen.filt)) %>% 
  select(OTU, Abundance,  Temperature, Chla_total,
         PO4, NH4, NO2, NO3, Si,
         PNF_Micro, HNF_Micro) 


models.par <- psmelt.clr.intragenus  %>% 
  filter(OTU %in% taxa_names(Coastal_All_phyto)) %>% 
  filter(SampleID %in% sample_names(Coastal_All_phyto)) %>% 
  dplyr::select(OTU, Abundance, Date1, SW.Temperature.at.site, chla.ug.per.L)  %>% 
  pivot_longer(names_to = 'parameter',
               values_to = 'value',
               cols = -c(OTU, Abundance, Date1)) %>% 
  group_by(OTU,parameter) %>% 
  nest() %>% 
  plyr::mutate(model = map(data, ~gam(Abundance ~ s(value, k = 2),
                                data = .x, method = 'REML')),
         results = map(model, glance),
         pval.term = map_dbl(model, ~ summary(.)$s.pv) ,
         R.square = map_dbl(model, ~ summary(.)$r.sq)) 


# how many are close to linearlity 
edfs <- models.par %>% 
  pull(model) %>% 
  map(~summary(.x)) %>% 
  map_dbl(~.x$edf)
table(edfs >= 1.5)

results <- models.par %>% 
  dplyr::select(OTU, parameter, results, pval.term, R.square) %>% 
  unnest(results)

results_pval=results%>%filter(pval.term <= 0.01)

plot(results_pval)

```









```{r}
# Adding column based on other column:
dd2=no_dup_metadata_KBytNERR
dd2$Site_Code=dd2$Site

no_dup_metadata_KBytNERR$Site_Code=no_dup_metadata_KBytNERR$Site
no_dup_metadata_KBytNERRv2=no_dup_metadata_KBytNERR %>%
  mutate(Site_Code = case_when(
    Site == "NTO1"~ "1",
   Site == "STO1" ~ "2",
   Site == "NR2"~ "3",
   Site == "SR2"~ "4",
   Site == "NB"~ "5",
   Site == "SR8"~ "6",
   Site == "CB"~ "7",
   Site == "SB"~ "8",
   Site == "AR"~ "9",
    Site == "HP1" ~ "10",
   Site == "KAHO" ~ "11",
   Site == "SR8"~ "12"
    ))



no_dup_metadata_KBytNERR$Site_Code=no_dup_metadata_KBytNERR$Site
no_dup_metadata_KBytNERRv2=no_dup_metadata_KBytNERRv2 %>%
  mutate(Site_Code = case_when(
    Site == "NTO1"~ "A",
   Site == "STO1" ~ "B",
   Site == "NR2"~ "C",
   Site == "SR2"~ "D",
   Site == "NB"~ "E",
   Site == "SR8"~ "F",
   Site == "CB"~ "G",
   Site == "SB"~ "H",
   Site == "AR"~ "I",
    Site == "HP1" ~ "J",
   Site == "KAHO" ~ "K",
   Site == "WAI2"~ "L"
    ))


no_dup_metadata_KBytNERRv3 = unite(no_dup_metadata_KBytNERRv2,col='Ordered_Site_Sample'
, c('Sampling.Order2', 'Site_Code', "SampleID") , sep = "_", remove = FALSE)

write.csv(no_dup_metadata_KBytNERRv3,"no_dup_metadata_KBytNERRv3.csv")

no_dup_metadata_KBytNERRv3$Ordered_Site_Sample
names(no_dup_metadata_KBytNERR$Site_Code)

colorssite <- c("WAI2"='#7F0000', "KAHO"='#f94279',"HP1"="#ffd000", "CB"="#ffa812","AR"="#c78100" ,"SR8"="#ff9966", "SB"="#c46210", "NB"="#ff7e00", "SR2"="#00cc95", "NR2"="#00cc2f", "STO1"='#0085ff', "NTO1"='#0045ff')



no_dup_metadata_KBytNERR$Sampling.Order


no_dup_metadata_KBytNERR$Sampling.Order2=no_dup_metadata_KBytNERR$Sampling.Order

no_dup_metadata_KBytNERRv2=no_dup_metadata_KBytNERR %>%
  mutate(Sampling.Order2= case_when(
    Sampling.Order == "1"~ "AA",
 Sampling.Order == "2" ~ "AB",
Sampling.Order == "3"~ "AC",
Sampling.Order == "4"~ "AD",
Sampling.Order == "5"~ "AE",
Sampling.Order == "6"~ "AF",
Sampling.Order == "7"~ "AG",
Sampling.Order == "8"~ "AH",
Sampling.Order == "9"~ "AI",
Sampling.Order == "10" ~ "AJ",
Sampling.Order == "11" ~ "AK",
Sampling.Order == "12"~ "AL",
    Sampling.Order == "13"~ "AM",
 Sampling.Order == "14" ~ "AN",
Sampling.Order == "15"~ "AO",
Sampling.Order == "16"~ "AP",
Sampling.Order == "17"~ "AQ",
Sampling.Order == "18"~ "AR",
Sampling.Order == "19"~ "AS",
Sampling.Order == "20"~ "AT",
Sampling.Order == "21"~ "AU",
Sampling.Order == "22" ~ "AV",
Sampling.Order == "23" ~ "AW",
Sampling.Order == "24"~ "AX",
    Sampling.Order == "25"~ "AY",
 Sampling.Order == "26" ~ "AZ",
Sampling.Order == "27"~ "BA",
Sampling.Order == "28"~ "BB",
Sampling.Order == "29"~ "BC",
Sampling.Order == "30"~ "BD",
Sampling.Order == "31"~ "BE",
Sampling.Order == "32"~ "BF",
Sampling.Order == "33"~ "BG",
Sampling.Order == "34" ~ "BH",
Sampling.Order == "35" ~ "BI",
Sampling.Order == "36"~ "BJ"
    ))


```

https://bioinformatics.psb.ugent.be/webtools/Venn/

```{r}

Cluster_1=subset_samples(All_phyto, Cluster_DeSeq=="Cluster_1")
nsamples(Cluster_1)
Cluster_1_melt=psmelt(Cluster_1)
Cluster_1_melt2=subset(Cluster_1_melt, Abundance != "0")
unique_Cluster_1=distinct(Cluster_1_melt2, OTU, .keep_all = TRUE)
write.csv(unique_Cluster_1, "unique_Cluster_1.csv")

Cluster_2=subset_samples(All_phyto, Cluster_DeSeq=="Cluster_2")
nsamples(Cluster_2)
Cluster_2_melt=psmelt(Cluster_2)
Cluster_2_melt2=subset(Cluster_2_melt, Abundance != "0")
unique_Cluster_2=distinct(Cluster_2_melt2, OTU, .keep_all = TRUE)
write.csv(unique_Cluster_2, "unique_Cluster_2.csv")


Cluster_3=subset_samples(All_phyto, Cluster_DeSeq=="Cluster_3")
nsamples(Cluster_3)
Cluster_3_melt=psmelt(Cluster_3)
Cluster_3_melt2=subset(Cluster_3_melt, Abundance != "0")
unique_Cluster_3=distinct(Cluster_3_melt2, OTU, .keep_all = TRUE)
write.csv(unique_Cluster_3, "unique_Cluster_3.csv")


Cluster_4=subset_samples(All_phyto, Cluster_DeSeq=="Cluster_4")
nsamples(Cluster_4)
Cluster_4_melt=psmelt(Cluster_4)
Cluster_4_melt2=subset(Cluster_4_melt, Abundance != "0")
unique_Cluster_4=distinct(Cluster_4_melt2, OTU, .keep_all = TRUE)
write.csv(unique_Cluster_4, "unique_Cluster_4.csv")

Cluster_5=subset_samples(All_phyto, Cluster_DeSeq=="Cluster_5")
nsamples(Cluster_5)
Cluster_5_melt=psmelt(Cluster_5)
Cluster_5_melt2=subset(Cluster_5_melt, Abundance != "0")
unique_Cluster_5=distinct(Cluster_5_melt2, OTU, .keep_all = TRUE)
write.csv(unique_Cluster_5, "unique_Cluster_5.csv")

ntaxa(All_phyto)

Cluster_1=subset_samples(All_phyto)
nsamples(Cluster_1)
Cluster_1_melt=psmelt(Cluster_1)
melt2=subset(Cluster_1_melt, Abundance != "0")
unique_melt2=distinct(melt2, OTU, .keep_all = TRUE)

write.csv(unique_melt2, "test.csv")

write.csv(as.data.frame(tax_table(All_phyto)), "okauy.csv")

write.csv(as.data.frame(otu_table(All_phyto)), "better.csv")

4040f83030d99604c9292f47ded6ab33

```

MHMHHMMMH so we recognize that there are only 514 with phytoplankton reads. 

```{r}
Cluster_1_2_unique=read.csv("Cluster_1_2_unique.csv")
all_phyto_taxa=as.data.frame(tax_table(All_phyto))
names(all_phyto_taxa)
                             
Cluster_1_2_unique_tax=left_join(Cluster_1_2_unique,all_phyto_taxa, by="ASV" )

write.csv(Cluster_1_2_unique_tax, "Cluster_1_2_unique_tax.csv")


Cluster2345
Cluster_1_2_unique=read.csv("Cluster2345.csv")
all_phyto_taxa=as.data.frame(tax_table(All_phyto))
names(all_phyto_taxa)
                             
Cluster_1_2_unique_tax=left_join(Cluster_1_2_unique,all_phyto_taxa, by="ASV" )

write.csv(Cluster_1_2_unique_tax, "Cluster_2345_unique_tax.csv")



```




```{r}

rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID, Cluster_DeSeq, Class) %>% dplyr::summarise(sum = sum(Abundance)) 



rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Cluster_DeSeq, Class) %>% dplyr::summarise(mean = mean(sum), sd=sd(sum)) 



rel2Phyto_ASV_Cluster $percent_total_mean=rel2Phyto_ASV_Cluster  $mean*100
rel2Phyto_ASV_Cluster $percent_total_SD=rel2Phyto_ASV_Cluster $sd*100


top5=rel2Phyto_ASV_Cluster%>%subset(Class=="D_2__Cyanobacteriia"| Class=="Mamiellophyceae"| Class=="Bacillariophyta"| Class=="Prymnesiophyceae"| Class=="Pelagophyceae")


bp=ggplot(top5, aes(x=as.factor(Cluster_DeSeq), y=as.factor(Class), fill = percent_total_mean)) +
    geom_tile(aes(fill = percent_total_mean)) + 
    geom_text(aes(label = round(percent_total_mean, 1))) +
  scale_fill_viridis(option="viridis",trans = 'log10')  + labs(fill = "Mean Relative Abundance \n as Percent of Phytoplankton Abundance")+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+xlab("")+ylab("")
bp 



```

```{r}
phyto_subset_transformed=read.csv("phyto_try2.csv")

diatom1=phyto_subset_transformed%>%subset(X=="f3ee74de7eead5fdc014bfd4d3c48295")
diatom2=phyto_subset_transformed%>%subset(X=="e5adae9e96a5065fb7a94a7db136332e")
diatom3=phyto_subset_transformed%>%subset(X=="85125408ea648cf8e6317b860cd2f1ac")




gom=tax_glom(All_phyto, "Class")


install.packages("corncob")
library(corncob)




```



```{r}

rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID, Cluster_DeSeq_Spatial, Curated_v3) %>% dplyr::summarise(sum = sum(Abundance)) 


names(rel2Phyto_ASV_total)

rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Cluster_DeSeq_Spatial, Curated_v3) %>% dplyr::summarise(mean = mean(sum), sd=sd(sum)) 

gg=rel2Phyto_ASV_Cluster%>%subset(Curated_v3=="Cyanobacteria:Synechococcus"|Curated_v3=="Stramenopiles:Diatoms")

ugh=dcast(rel2Phyto_ASV_Cluster, Curated_v3~Cluster_DeSeq, value.var ="mean")
ugh$sum=rowSums(ugh[2:6])
ugh$mean=rowMeans(ugh[2:6])

rel2Phyto_Abund_enviro_mean=dcast(rel2Phyto_Abund_enviro, OTU+ASV ~ Enviro2, value.var ="mean")



rel2Phyto_ASV_Cluster $percent_total_mean=rel2Phyto_ASV_Cluster  $mean*100
rel2Phyto_ASV_Cluster $percent_total_SD=rel2Phyto_ASV_Cluster $sd*100

top10=subset(rel2Phyto_ASV_Cluster, Curated_v3=="Stramenopiles:Pelagophyceae"| Curated_v3=="Archaeplastida:Bathycoccus"| Curated_v3=="Archaeplastida:Ostreococcus"| Curated_v3=="Archaeplastida:Micromonas"| Curated_v3=="Archaeplastida:Other"| Curated_v3=="Hacrobia:Prymnesiales"| Curated_v3=="Archaeplastida:Mamiellophyceae"| Curated_v3=="Stramenopiles:Diatoms"| Curated_v3=="Cyanobacteria:Prochlorococcus"| Curated_v3=="Cyanobacteria:Synechococcus"| Curated_v3=="Cyanobacteria:Cyanobium")




test=psmelt(rel2All_phyto)
top10$Curated_v3=factor(top10$Curated_v3, levels=c("Hacrobia:Prymnesiales","Archaeplastida:Other",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Mamiellophyceae","Stramenopiles:Pelagophyceae",
"Stramenopiles:Diatoms","Cyanobacteria:Prochlorococcus","Cyanobacteria:Cyanobium","Cyanobacteria:Synechococcus"
))


top10$Cluster_DeSeq_Spatial=factor(top10$Cluster_DeSeq_Spatial, levels=c("Coastal", "Transition", "Offshore"))


svg("heatmap_Spatial_Clusters_top10.svg", width=8, height=6)
bp=ggplot(top10, aes(x=as.factor(Cluster_DeSeq_Spatial), y=as.factor(Curated_v3), fill = percent_total_mean)) +
    geom_tile(aes(fill = percent_total_mean)) + 
    geom_text(aes(label = round(percent_total_mean, 1))) +
  scale_fill_viridis(option="viridis",trans = 'log10')  + labs(fill = "Mean Relative Abundance \n as Percent of \n Phytoplankton Abundance")+theme(axis.text=element_text(size=11), legend.text=element_text(size=11), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+xlab("")+ylab("")
bp 

dev.off()

ggsave("heatmap_Clusters_spatial_mean.pdf", width=8, height=6)



test=psmelt(rel2All_phyto)
rel2Phyto_ASV_Cluster$Curated_v3=factor(rel2Phyto_ASV_Cluster$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))

library(rstatix)
library(scales)
?log10_trans()

names(rel2Phyto_ASV_Cluster)

bp=ggplot(rel2Phyto_ASV_Cluster, aes(x=as.factor(Cluster_DeSeq), y=as.factor(Curated_v3), fill = percent_total_mean)) +
    geom_tile(aes(fill = percent_total_mean)) + 
    geom_text(aes(label = round(percent_total_mean, 1))) +
  scale_fill_viridis(option="viridis",trans = 'log10')  + labs(fill = "Mean Relative Abundance \n as Percent of Phytoplankton Abundance")+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
bp 

ggsave("heatmap_Clusters_mean.pdf", width=14)


#############3

p0.rel <- transform(ralph, "clr")


rel2Phyto_ASV_total <-p0.rel %>% psmelt() %>% group_by(SampleID, Cluster_DeSeq, Curated_v3) %>% summarise(sum = sum(Abundance)) 


rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% group_by(Cluster_DeSeq, Curated_v3) %>% summarise(mean = mean(sum), sd=sd(sum)) 

rel2Phyto_ASV_Cluster $percent_total_mean=rel2Phyto_ASV_Cluster  $mean*100
rel2Phyto_ASV_Cluster $percent_total_SD=rel2Phyto_ASV_Cluster $sd*100


rel2Phyto_ASV_Cluster$Curated_v3=factor(rel2Phyto_ASV_Cluster$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))

library(rstatix)
library(scales)
?log10_trans()

names(rel2Phyto_ASV_Cluster)

bp=ggplot(rel2Phyto_ASV_Cluster, aes(x=as.factor(Cluster_DeSeq), y=as.factor(Curated_v3), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) +
  scale_fill_viridis(option="viridis")  + labs(fill = "Mean CLR Phytoplankton Abundance")+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
bp 

ggsave("heatmap_Clusters_mean_CLR.pdf", width=14)


```


```{r}
library(DESeq2)

All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")

diagdds_pol = phyloseq_to_deseq2(All_phyto, ~ Cluster_DeSeq)
diagdds_pol = DESeq(diagdds_pol, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)



taxonomy_Phyto_we = read.csv("taxonomy_table_final_tucker_v8.csv", sep=",") 

sigtab2=sigtab [,c(1:6,15)]
sigtab2$DeSeq_Cluster5_Variable=sigtab$Domain
sigtab2$DeSeq_Cluster5_Variable="DeSeq_Cluster5_Variable"
sigtab2=as.data.frame(sigtab2)
paper=left_join(taxonomy_Phyto_we,sigtab2,by="ASV")

write.csv(paper,"another_tax9.csv")
left_join()


res <- results(diagdds_pol, contrast=c("Cluster_DeSeq", "Cluster_4","Cluster_5"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_Clustering_deSeq_Transition_Offshore.csv")



```









```{r}



All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")

diagdds_pol = phyloseq_to_deseq2(All_phyto, ~ Cluster_DeSeq_Spatial)
diagdds_pol = DESeq(diagdds_pol, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_Clustering_deSeq_Spaital.csv")

taxonomy_Phyto_we = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_v6.csv", sep=",") 

sigtab2=sigtab [,c(1:6,15)]
sigtab2$DeSeq_Spatial_Cluster=sigtab$Domain
sigtab2$DeSeq_Spatial_Cluster="DeSeq_Spatial_Variable"
sigtab2=as.data.frame(sigtab2)
paper=left_join(taxonomy_Phyto_we,sigtab2,by="ASV")

write.csv(paper,"taxonmy_all_Deseqdata.csv")
left_join()


res <- results(diagdds_pol, contrast=c("Cluster_DeSeq_Spatial", "Transition","Offshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "DESeq_All_Phyto_Clustering_deSeq_Transition_Offshore.csv")

Transition_Offshore=sigtab

Transition_Offshore$Transition_Offshore="Transition_Offshore"

names(Transition_Offshore)[names(Transition_Offshore) == 'log2FoldChange'] <- 'TO_log2FoldChange'
names(Transition_Offshore)[names(Transition_Offshore) == 'padj'] <- 'TO_padj'
names(Transition_Offshore)[names(Transition_Offshore) == 'baseMean'] <- 'TO_baseMean'

Transition_Offshore_v2=Transition_Offshore%>%select(ASV, TO_log2FoldChange, TO_padj, TO_baseMean,Transition_Offshore)

names(Transition_Offshore)


Coastal_Transition$Coastal_Transition="Coastal_Transition"

names(Coastal_Transition)[names(Coastal_Transition) == 'log2FoldChange'] <- 'CT_log2FoldChange'
names(Coastal_Transition)[names(Coastal_Transition) == 'padj'] <- 'CT_padj'
names(Coastal_Transition)[names(Coastal_Transition) == 'baseMean'] <- 'CT_baseMean'

Coastal_Transition_v2=Coastal_Transition%>%select(ASV, CT_log2FoldChange, CT_padj, CT_baseMean,Coastal_Transition)



Coastal_Offshore$Coastal_Offshore="Coastal_Offshore"

names(Coastal_Offshore)[names(Coastal_Offshore) == 'log2FoldChange'] <- 'CO_log2FoldChange'
names(Coastal_Offshore)[names(Coastal_Offshore) == 'padj'] <- 'CO_padj'
names(Coastal_Offshore)[names(Coastal_Offshore) == 'baseMean'] <- 'CO_baseMean'

Coastal_Offshore_v2=Coastal_Offshore%>%select(ASV, CO_log2FoldChange, CO_padj, CO_baseMean,Coastal_Offshore)



COT=full_join(Coastal_Offshore_v2,Coastal_Transition_v2, by="ASV")
COOT=full_join(COT,Transition_Offshore_v2, by="ASV")

maybe=full_join(COOT,sigtab,by="ASV")

write.csv(maybe, "all_comparisons_DESEQ_Spatial.csv")

```








```{r}



library(microbiome)
library(viridis)
#devtools::install_github("microsud/microbiomeutilities")
#BiocManager::install("microbiome")
library(microbiomeutilities)

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)

p0.rel <- transform(ralph, "compositional")
deseq_diff=subset_taxa(p0.rel, ASV=="47e0463d97f67bf6288a979f3ff70bfb" | ASV=="936685ee710e6e0ad190df2c0e863f75" |ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20" |ASV=="49db984d738a8889309dfffd963f5a75" | ASV=="300296cb89902c1721ff71f7cbc316b7" |ASV=="406bb14741ded383380219611b8564d9" |ASV=="5e7894007198ccd65ab8b06f349d8ee3" |ASV=="bbc1f42b124d03523f023f7f393454ed" |
ASV=="b313fb940989d495b780f97175df05f5" |ASV=="024dcc1ab6ba58bbb754448afd9c1127" |ASV=="558d9745881b8379f4553fc42b83cccd" |ASV=="44095b9d0c379f3ca649b693174bbdfc" |
ASV=="13f195313f60ce69ff1d3f24e072cef5" |ASV=="0a474686e6615f230a75dc3b07db3cea" |
ASV=="4801416d0b718170ba5bb7218e243280" |ASV=="26c4bf94339bbcabcf2b9e16f24427c7" |
ASV=="e8445e0b9c8d35e2a59579a2fef6dca6" |ASV=="e5a35d824858b0a8c85ae863af9caed0" |ASV=="79dcf9764c6c03f92472e5dfeb29da75" |ASV=="84db962c29105bd1d38d0ff8779ea923" |ASV=="42cd59d539b4edfa4ca8128418cca49f" |ASV=="8ce7d5b4b52e422fbbd2d0c84e245273" |
ASV=="4038caa74cef581938e69f3acc20304d" |ASV=="6bfe56565a740b481b5328f950267bc6" |
ASV=="33417ecc6e666f82ad205b38b0bccc25" |ASV=="efd9be6e5ad49f28be30f42932f882e8" |ASV=="6ce8db6d8ee1fdaa3e32fbf15415a65b" |ASV=="b0dba0147194c2bf5465c38eaa04e559" |
ASV=="a7618398706dd809d14fffaee9833339" |
ASV=="e6510d6615987f47739507a9d5d83d95" |
ASV=="957c121f10a6486973716afdaed844f3" |
ASV=="ecfbb028e04b03f207bba70c7554b75e" |
ASV=="fb938011d56121722a6326ff4640809d" |
ASV=="c944890292c1813482d05b6f6fcb9562" |
ASV=="1c950ec439291f32b15aa87ae3417e4c" |
ASV=="d0d06719369a3471cd8c6a6335e3386f" |
ASV=="fb6279754cb58f9c830b8d2d3c748752" |
ASV=="211f2e171afc061e6745fed9d0d33583" |
ASV=="e6719aefdda6941be9f597a88fc3384e" |
ASV=="23880d5ecb51a362e9282e5cc82b6c8e" |
ASV=="7a6c38e7a3af6fa54bd7ea028cc73118" |
ASV=="10768a5c1a314cd9f73baee744c7df10" |
ASV=="1bff589486f68e77b4476d6837244cd6" |
ASV=="537fce5b9b68f9ea0e912bb4a01b4102" |
ASV=="a908b00f4edf992ef1db8de70b4d7615" |
ASV=="2b5da926477e5820dfdc9bac8d2ce40b" |
ASV=="ea325221e74caca2941b3bc9db57c62d" |
ASV=="c0d135a5439f71cc414c405cb36ed08b" |
ASV=="83bd0cb9b9e313ae5a863dcb031a23ac" |
ASV=="d7676849144b7e62e70d06262fecca34" |
ASV=="0032fa9e7bb07ae8dd0a4958274ddae9" |
ASV=="6393eee32af2c373d3177b743ca57fe1" |
ASV=="622994d1a3b21bd9df22236d03434b84" |
ASV=="86215bfc26ab6967d26b6adcbc011490" |
ASV=="8c759af3c4151ead0c56e2e97d0c510e" |
ASV=="bd6e2afd311cd170d79defd15fd77ad8" |
ASV=="ee7d879868f6e2a04692c147ed497c88" |
ASV=="1a01dca998f93882d0290802a579c464" |
ASV=="a4c9be481e6bba68cfcc839c1b1019b2" |
ASV=="87b66d169861236181e60e3ff3edf5d0" |
ASV=="edf69e9168c8dd37c91686aeeab2d0a4" |
ASV=="30346abbc7e6c49a65bfda9e13db3a6d" |
ASV=="0b45d16cc99dcdc9b5bc12b637c5d971" |
ASV=="f1749331353145c63288216e243cfbae" |
ASV=="9107af9e4ab0766c43fd15901c926f3d" |
ASV=="c6ebc08d6495e327d01cbd509e32f0c3" |
ASV=="01e3d73fa64a2f6b2a0329089680ef4f" |
ASV=="85125408ea648cf8e6317b860cd2f1ac" |
ASV=="c357a7d4eade7ed830cc23b54139c249" |
ASV=="a30f7c557c17f43e77119cdea6e75559" |
ASV=="e5adae9e96a5065fb7a94a7db136332e" |
ASV=="b0432b08b95cbc2cb7e69d4ae87019f6" |
ASV=="f3ee74de7eead5fdc014bfd4d3c48295" |
ASV=="b7b54b056986d6f0aec5a03a7a837dae" |
ASV=="3334ddadd2f68f4f767c229665a55956" |
ASV=="fb2c4366cf3b10ea36e892dc28f4a97a" |
ASV=="7ea549b78d0a14ade45725fb23cfb8aa" |
ASV=="03296caf3fa6eb808ac51270bcfb0b28" |
ASV=="2fef69a9baf521fdc9b24730a466d4ff" |
ASV=="9ff9314b8621bd217f32b05328815dd8" |
ASV=="e06f50344798d12e83c6e163975ad4cd" |
ASV=="185e4e0d8237652eb8a42c05db02d95e" |
ASV=="e0bcd67e37d43165add5c7c89efb52fb")



ntaxa(deseq_diff)

grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
simple_heatmap(deseq_diff,
               group.facet = "Cluster_DeSeq",
               group.order = NULL,
               abund.thres = 0.005,
               prev.thres = 0.01,
               level = "ASV",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange=TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "Log-ratio transformed Abundances of ASVs with Significantly Different Abundances Across Clusters")


?transform()

ggsave("CLR_Clusters_ASV_DeSeq.pdf", width=10, height=10)


```


```{r}
p0.rel <- transform(All_phyto, "compositional")
deseq_diff=subset_taxa(p0.rel, DeSeq_Spatial_Cluster=="DeSeq_Spatial_Variable")
ntaxa(deseq_diff)

deseq_diff=subset_taxa(deseq_diff, phytorelabun_max_class=="abundant" | phytorelabun_max_class=="dominant")



grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
simple_heatmap(deseq_diff,
               group.facet = "Cluster_DeSeq_Spatial",
               group.order = NULL,
               abund.thres = 0.005,
               prev.thres = 0.001,
               level = "Genus",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange=TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "rel Abundances of ASVs with Significantly Different Abundances Across Clusters")

ggsave("deseq_heatmap.pdf", width=10, height=20)

```


```{r}
p0.rel <- transform(All_phyto, "compositional")
deseq_diff=subset_taxa(p0.rel, DeSeq_Spatial_Cluster=="DeSeq_Spatial_Variable")
ntaxa(deseq_diff)


asv_switch=subset_taxa(deseq_diff, Genus=="Phaeocystis" | Genus=="Bathycoccus"| Genus=="Ostreococcus"| Genus=="Pyramimonas"| Genus=="Teleaulax"| Genus=="Dictyochophyceae_XXX" | Genus=="unclassified_Mamiellaceae" | Genus=="unclassified_Cryptomonadales_X")

?simple_heatmap()


grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "red"))

heat.cols <- grad_ab(10)
simple_heatmap(p0.rel,
               group.facet = "Cluster_DeSeq_Spatial",
               group.order = c("Coastal", "Transition", "Offshore"),
               abund.thres = 0.0005,
               prev.thres = 0.0001,
               level = "Genus",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange= TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "")

ggsave("Deseq_ASVs_collasped.pdf", height=12, width=8)

```
```{r}

asv_switch=subset_taxa(deseq_diff, Genus=="Phaeocystis" | Genus=="Bathycoccus"| Genus=="Ostreococcus"| Genus=="Pyramimonas"| Genus=="Teleaulax")



?simple_heatmap()


grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "red"))

heat.cols <- grad_ab(10)
simple_heatmap(asv_switch,
               group.facet = "Cluster_DeSeq_Spatial",
               group.order = c("Coastal", "Transition", "Offshore"),
               abund.thres = 0.0005,
               prev.thres = 0.0001,
               level = "Genus",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange= TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "")
ggsave("ASV_switching.pdf")
```


```{r}
p0.rel <- transform(All_phyto, "compositional")

mami=subset_taxa(p0.rel, Class=="Mamiellophyceae")
ntaxa(mami)

asv_switch=subset_taxa(deseq_diff, Class=="Mamiellophyceae")
ntaxa(mami)

heat.cols <- grad_ab(10)
simple_heatmap(mami,
               group.facet = "Cluster_DeSeq_Spatial",
               group.order = c("Coastal", "Transition", "Offshore"),
               abund.thres = 0.0005,
               prev.thres = 0.0001,
               level = "Genus_ASV",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange= TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "")

ggsave("Mamimello.pdf")
```



```{r}
CO=read.csv("C_O.csv")

library(ggplot2)
theme_set(theme_bw())
data(mtcars)
# Plot

mtcars$`car name` <- rownames(mtcars)  # create new column for car names
mtcars$mpg_z <- round((mtcars$mpg - mean(mtcars$mpg))/sd(mtcars$mpg), 2)  # compute normalized mpg
mtcars$mpg_type <- ifelse(mtcars$mpg_z < 0, "below", "above")  # above / below avg flag
mtcars <- mtcars[order(mtcars$mpg_z), ]  # sort
mtcars$`car name` <- factor(mtcars$`car name`, levels = mtcars$`car name`)  # convert to factor to retain sorted order in plot.


ggplot(mtcars, aes(x=`car name`, y=mpg_z, label=mpg_z)) + 
  geom_point(stat='identity', aes(col=mpg_type), size=6)  +
  scale_color_manual(name="Mileage", 
                     labels = c("Above Average", "Below Average"), 
                     values = c("above"="#00ba38", "below"="#f8766d")) + 
  geom_text(color="white", size=2) +
  labs(title="Diverging Dot Plot", 
       subtitle="Normalized mileage from 'mtcars': Dotplot") + 
  ylim(-2.5, 2.5) +
  coord_flip()


CO$Curated_v3=factor(CO$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))


CO$ASV <- factor(CO$ASV, levels = CO$ASV)

ggplot(CO, aes(x=as.factor(ASV), y=CO_log2FoldChange, label=CO_log2FoldChange)) + 
  geom_point(stat='identity', aes(col=Curated_v3), size=6)  +
  scale_color_manual(name="Mileage", 
                     labels = c("Coastal", "Offshore"), 
                     values = c("Coastal"="#00ba38", "Offshore"="#f8766d")) + 
  geom_text(color="white", size=2) +
  labs(title="Diverging Dot Plot", 
       subtitle="Normalized mileage from 'mtcars': Dotplot") + 
  coord_flip()



ggplot(CO, aes(x=ASV, y=CO_log2FoldChange, label=CO_log2FoldChange)) + 
  geom_point(stat='identity', aes(col=Curated_v3, shape=direction), size=6)  +
  scale_color_manual(values = coloresbarplot) + 
  labs() + 
  coord_flip()+
    scale_x_discrete(labels = CO$Genus)


```



```{r}


CO=read.csv("CO_genus.csv")

CO$Curated_v3=factor(CO$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))


CO$ASV <- factor(CO$ASV, levels = CO$ASV)

svg("logFold_deseq_dotplot.svg", width=9, height=8)
ggplot(CO, aes(x=ASV, y=CO_log2FoldChange, label=CO_log2FoldChange)) + 
  geom_point(stat='identity', aes(col=Curated_v3, shape=direction), size=6)  +
  scale_color_manual(values = coloresbarplot) + 
  labs() + 
  coord_flip()+
    scale_x_discrete(labels = CO$Genus)+ geom_hline(yintercept=0, linetype='dotted', col = 'red', size=1)+ylab("Log-fold change")
dev.off()

ggsave("logFold_deseq_dotplot.pdf", width=9, height=8)
```
















```{r}
dv_water_order <- divnet(All_phyto, ncores = 4)
```


```{r}

divnet_Season<- All_phyto %>%
  divnet(formula = ~ Season_Aka, ncores = 6)

p=plot(divnet_Season$shannon, 
     All_phyto, 
     col = "Season_Aka")
p

save(list="divnet_Season", file="divnet_Season.rdata")


dv=divnet_Season

# my DivNet object - replace yours here :) 
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(All_phyto))

df2=left_join(df, meta, by="SampleID")

gg=df2%>%  arrange(desc(Site_Code), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=df2%>%group_by(Season_Aka)%>%summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

svg("Shannons_season.svg", width=2, height=2)
gg_aveerage%>% ggplot(aes(x = Season_Aka, xend = Season_Aka, color=Season_Aka))+
  geom_point(aes(x = Season_Aka, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text())+xlab("Season")+  theme(legend.position="none")
dev.off()
ggsave("Season_differences.pdf", width=2, height=2)





```



```{r}
library(breakaway)
library(DivNet)

divnet_Season_Site<- All_phyto %>%
  divnet(formula = ~ Site*Season_Aka, ncores = 6)

pdf("site_season_alpha.pdf")
p=plot(divnet_Season_Site$shannon, 
     All_phyto, 
     col = "Season_Aka")
p    
dev.off()
save(list="divnet_Season_Site", file="divnet_Season_Site.rdata")

  

dv=divnet_Season_Site

# my DivNet object - replace yours here :) 
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(All_phyto))


df2=left_join(df, meta, by="SampleID")

gg=df2%>%  arrange(desc(Site_Code), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=gg%>%group_by(Site, Season_Aka)%>%summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

gg_aveerage$Site_Season <- paste(gg_aveerage$Site, gg_aveerage$Season_Aka, sep="_")

gg_aveerage$Site=factor(gg_aveerage$Site, levels=c("WAI2", "KAHO", "SB", "AR", "CB", "HP1", "SR8", "NB", "SR2", "NR2", "STO1", "NTO1" ))
gg_aveerage%>% ggplot(aes(x = Site_Season, xend = Site_Season, color=Site))+
  geom_point(aes(x = Site_Season, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)
ggsave("alpha_diversity_Site_order_averages.pdf")





divnet_Site <- All_phyto %>%
  divnet(X = "Site", ncores = 2)
divnet_Site$shannon



#testDiversity(divnet_Site, "shannon")
```


```{r}

Coastal_All_phyto=subset_samples(All_phyto, Cluster_DeSeq_Spatial=="Coastal")
library(DivNet)
library(tidyverse)

divnet_Coastal_Month <- Coastal_All_phyto %>%
  divnet(X = "Month", ncores = 4)
divnet_Site$shannon


p=plot(divnet_Coastal_Month$shannon, 
     Coastal_All_phyto, 
     col = "Month")
p  

save(list="divnet_Coastal_Month", file="divnet_Coastal_Month.rdata")


dv=divnet_Coastal_Month
# my DivNet object - replace yours here :) 
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(Coastal_All_phyto))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Month)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

gg=df2%>%  arrange(desc(Site_Code), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=gg%>%group_by(Month_abb)%>%summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

gg%>% ggplot(aes(x = SampleID, xend = SampleID, color=Month))+
  geom_point(aes(x = SampleID, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


gg_aveerage$Month

gg_aveerage$Month_abb=factor(gg_aveerage$Month_abb, levels=c("Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar"))

svg("month_alpha_shannons.svg", width=4, height=2)
gg_aveerage%>% ggplot(aes(x = Month_abb, xend = Month_abb, color=Month_abb))+
  geom_point(aes(x = Month_abb, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+  theme(legend.position="none")+xlab("Month")
dev.off()
ggsave("alpha_diversity_shannons_month.pdf", width=4, height=2)


Coastal_All_phyto

?testDiversity()
testDiversity(divnet_Coastal_Month, "shannon")
Season_Month <- divnet_Coastal_Month$`bray-curtis`



ba <- breakaway(Coastal_All_phyto)
ba

bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(Coastal_All_phyto, "Month_abb"))
bet=as.data.frame(bt$table)
write.csv(bet, "richness_month.csv")


ests <- sapply(ba[["richness"]], function(x) x$estimate)

lci <- sapply(ba[["richness"]], function(x) x$interval[1])
uci <- sapply(ba[["richness"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 ba$X) # the X variables I used in my model

df

plot(ba, Coastal_All_phyto, color = "Season_Aka")
ggsave("estimate_richness.pdf")



richness=read.csv("richness_month.csv")
richness$Month=factor(richness$Month, levels=c("Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar" ))

svg("richness_Month.svg", width=4, height=2)
richness%>%ggplot(aes(x = Month, xend=Month, color=Month))+
  geom_point(aes(x = Month, y = est)) +
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+  theme(legend.position="none")
dev.off()
ggsave("Month_ASV_richness.pdf",width=4, height=2)

```


```{r}


ba <- breakaway(Coastal_All_phyto)
ba

bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(Coastal_All_phyto, "Season_Aka"))
bet=as.data.frame(bt$table)
write.csv(bet, "richness_month.csv")


ests <- sapply(ba[["richness"]], function(x) x$estimate)

lci <- sapply(ba[["richness"]], function(x) x$interval[1])
uci <- sapply(ba[["richness"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 ba$X) # the X variables I used in my model

df

plot(ba, Coastal_All_phyto, color = "Season_Aka")
ggsave("estimate_richness.pdf")



richness=read.csv("richness_month.csv")
richness$Month=factor(richness$Month, levels=c("Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar" ))

svg("richness_Month.svg", width=4, height=2)
richness%>%ggplot(aes(x = Month, xend=Month, color=Month))+
  geom_point(aes(x = Month, y = est)) +
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+  theme(legend.position="none")
dev.off()
ggsave("Month_ASV_richness.pdf",width=4, height=2)

```



```{r}
NERRSamples=subset_samples(All_phyto ,Sampling.Order=="30"|Sampling.Order=="32"|Sampling.Order=="33"|Sampling.Order=="35")

divnet_Site <- All_phyto %>%
  divnet(X = "Site", ncores = 2)
divnet_Site$shannon
#testDiversity(divnet_Site, "shannon")

save(list="divnet_Site", file="divnet_Site.rdata")

library(phyloseq)
names(sample_data(All_phyto))

f=divnet_Site$shannon%>%
  summary %>%
  add_column("SampleNames" = All_phyto %>% otu_table %>% sample_names)%>%
  add_column("SampleType" = All_phyto %>% sample_data %>% (function(x) x$Site))%>%
  add_column("Enviro" = All_phyto %>% sample_data %>% (function(x) x$Enviro_Clustering))



ggplot(f, aes(x = SampleNames, xend = estimate)) +
  geom_point(aes(x = SampleNames, y = estimate)) +
  geom_segment(aes(y = lower, yend = upper)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


ggplot(f, aes(x=SampleNames, y=estimate))+geom_point( aes(col = Enviro))+geom_segment(aes(y = lower, yend = upper))

?alpha_estimates()
p=plot(divnet_Site$shannon, 
     All_phyto, 
     col = "Site")
p    
ggsave("alpha_diversity_site_justphyto_Simpson.pdf")


dv=divnet_Site
# my DivNet object - replace yours here :) 
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(All_phyto))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Site)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)


gg=df2%>%  arrange(desc(Site_Code), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=gg%>%group_by(Site)%>%summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

gg%>% ggplot(aes(x = SampleID, xend = SampleID, color=Site))+
  geom_point(aes(x = SampleID, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)
ggsave("alpha_diversity_Site_order.pdf")


#####this one
svg("alpha_diversity_Site_order_averages.svg", width=4, height=2)
gg_aveerage$Site=factor(gg_aveerage$Site, levels=c("WAI2", "KAHO", "SB", "AR", "CB", "HP1", "SR8", "NB", "SR2", "NR2", "STO1", "NTO1" ))
gg_aveerage%>% ggplot(aes(x = Site, xend = Site, color=Site))+
  geom_point(aes(x = Site, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)+xlab("Site")+  theme(legend.position="none")
dev.off()
ggsave("alpha_diversity_Site_order_averages.pdf", width=4, height=2)



df2%>%ggplot(aes(x = SampleID, xend = SampleID, color=Site))+facet_wrap(~Site, scales="free_y")+
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)


kk=simplifyBeta(divnet_Season, All_phyto, "euclidean", "Site")

# You can plot this easily
simplifyBeta(divnet_Site, All_phyto, "bray-curtis", "Site") %>%
  ggplot(aes(x = interaction(Covar1, Covar2), 
             y = beta_est)) +
  geom_point() +
  geom_linerange(aes(ymin = lower, ymax = upper)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  xlab("") + ylab("Estimates of Bray-Curtis distance")
ggsave("Beta_diversity_Site.pdf")

```







```{r}
dv=simplifyBeta(divnet_Site, All_phyto, "bray-curtis", "Site")

ests <- sapply(dv[["bray-curtis"]], function(x) x$bray-curtis)
lci <- sapply(dv[["bray-curtis"]], function(x) x$interval[1])
uci <- sapply(dv[["bray-curtis"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(All_phyto))




dv
dv=divnet_Site
test=interaction(Covar1, Covar2)# my DivNet object - replace yours here :) 
ests <- sapply(dv[["bray-curtis"]], function(x) x$bray-curtis)
lci <- sapply(dv[["bray-curtis"]], function(x) x$interval[1])
uci <- sapply(dv[["bray-curtis"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(All_phyto))


df2=left_join(df, meta, by="SampleID")


beta=simplifyBeta(divnet_Site, All_phyto, "bray-curtis", "Site")

strawberry

library(corrplot)
M = cor(mtcars)


library(reshape2)

beta2=dcast(beta, beta_est)

library(ggplot2)
ggplot(data = beta, aes(x=Covar1, y=Covar2, fill=beta_est)) + 
  geom_tile()

beta2=read.csv("beta_cast_v3.csv", row.names=1)
fff=as.matrix(beta2)

f1 = colorRamp2(seq(min(0), max(1), length = 3), c("blue", "#EEEEEE", "red"))
col_fun = colorRamp2(c(0, 0.5,1), c("blue", "white", "red"))

pdf("HEATMAP_BRAY_CURTIS.pdf")
Heatmap(fff , name = "mat",
    column_title = "", col=col_fun, column_order = colnames(fff), row_order = rownames(fff))
dev.off()

pdf("HEATMAP_BRAY_CURTIS_clustered.pdf")
Heatmap(fff , name = "mat",
    column_title = "Estimated Bray Curtis differences", col=col_fun)
dev.off()

library(ComplexHeatmap)
library(circlize)
?colorRamp2()


mat_with_na = fff
na_index = sample(c(TRUE, FALSE), nrow(fff)*ncol(fff), replace = TRUE, prob = c(1, 9))
mat_with_na[na_index] = NA



ggplot(data = beta3, aes(x=Covar1, y=Covar2, fill=beta_est)) + 
  geom_tile()

beta_cast=reshape2::dcast(beta, Covar1 ~ Covar2, value.var ="beta_est")

view(mtcars)

write.csv(beta_cast, "beta_cast.csv")

# You can plot this easily
simplifyBeta(divnet_Site, All_phyto, "bray-curtis", "Site") %>%
  ggplot(aes(x = interaction(Covar1, Covar2), 
             y = beta_est)) +
  geom_point() +
  geom_linerange(aes(ymin = lower, ymax = upper)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  xlab("") + ylab("Estimates of Bray-Curtis distance")


ggsave("Beta_diversity_Site.pdf")


```



```{r}
install.packages("ggheatmap")
library(ggheatmap)
library(tidyr)
library(aplot)
set.seed(123)
df <- matrix(runif(600,0,10),ncol = 12)
colnames(df) <- paste("sample",1:12,sep = "")
rownames(df) <- sapply(1:50, function(x)paste(sample(LETTERS,3,replace = FALSE),collapse = ""))
head(df)
#example 1
text_rows <- sample(rownames(df),3)
p <- ggheatmap(df,scale="none", cluster_rows  = TRUE,cluster_cols = TRUE,
          text_show_rows = text_rows)%>%
          ggheatmap_theme(1,theme =list(
           theme(axis.text.x = element_text(angle = 90,face = "bold"),
           axis.text.y = element_text(colour = "red",face = "bold"))
  ))

library(heatmaply)
x <- heatmapr(iris[,-5], scale = "column", colors = "Blues")
ggheatmap(x)

```




```{r}
ba <- breakaway(All_phyto)
ba

bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(All_phyto, "Site"))
bet=as.data.frame(bt$table)



ests <- sapply(ba[["richness"]], function(x) x$estimate)

lci <- sapply(dv[["richness"]], function(x) x$interval[1])
uci <- sapply(dv[["richness"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df

plot(ba, All_phyto, color = "Season_Aka", shape="Season_Aka")+scale_color_manual(values=colorssite2)
ggsave("estimate_richness.pdf")



```



```{r}


bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(All_phyto, "Season_Aka"))
bet=as.data.frame(bt$table)

ests <- sapply(ba[["estimates"]], function(x) x$estimates)
error <- sapply(ba[["error"]], function(x) x$error)

lci <- sapply(dv[["bray-curtis"]], function(x) x$interval[1])
uci <- sapply(dv[["bray-curtis"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, error, 
                 ba$X) # the X variables I used in my model

df

plot(ba, All_phyto, color = "Season_Aka", shape="Season_Aka")
ggsave("estimate_richness.pdf")

```



```{r}

richness=read.csv("estimate_ASV_richness.csv")
richness$Site=factor(richness$Site, levels=c("WAI2", "KAHO", "SB", "AR", "CB", "HP1", "SR8", "NB", "SR2", "NR2", "STO1", "NTO1" ))

svg("richness_site.svg", width=4, height=2)
richness%>%ggplot(aes(x = Site, xend=Site, color=Site))+
  geom_point(aes(x = Site, y = est)) +
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+ scale_color_manual(values=colorssite2)+  theme(legend.position="none")
dev.off()
ggsave("ASV_richness.pdf")



```



```{r}


```



```{r}

Coastal_All_phyto=subset_samples(All_phyto, Cluster_DeSeq_Spatial=="Coastal")
nsamples(Coastal_All_phyto)
pp=as.data.frame(divnet_Site[shannon])

simplifyBeta(divnet_Site, All_phyto, "shannon", "Site")

divnet_Month_Coastal <- Coastal_All_phyto %>%
  divnet(X = "Month", ncores = 2)

plot(divnet_Month$shannon, 
     All_phyto, 
     col = "Month")
ggsave("alpha_diversity_month.pdf", width=14, height=8)


testDiversity(divnet_Month, "shannon")

# This is just a wrapper for betta(), btw
testDiversity(divnet_Season, "shannon")

Season_bc <- divnet_Season$`bray-curtis`

# Uniquely, DivNet also has variance estimates:
gg=simplifyBeta(divnet_Site, All_phyto, "bray-curtis", "Site")







###############site
sampledata(All_phyto)

divnet_cluster <- All_phyto %>%
  divnet(X = "Cluster_DeSeq", ncores = 1)


divnet_temperature <- All_phyto %>%
  divnet(X = "Season", ncores = 2)

dv_water_st <- water %>%
  divnet(X = "SampleType", ncores = 2)

plot(divnet_temperature$shannon, 
     All_phyto, 
     col = "Site")
ggsave("alpha_diversity_season.pdf")
# This is just a wrapper for betta(), btw
testDiversity(divnet_temperature, "shannon")

?testDiversity
# To test hypotheses about beta diversity
# using DivNet, let's pull out our
# estimated distance matrix:
bc <- divnet_temperature$`bray-curtis`

# You'll notice that all samples with the same
# SampleType at the same estimate. DivNet uses
# covariate information to share strength
# across samples and obtain an estimate
# about the beta diversity of the *ecosystems*
#   not the samples

# We can consider the  unique rows using
bc %>% unique

# Uniquely, DivNet also has variance estimates:
ll=simplifyBeta(divnet_temperature, All_phyto, "bray-curtis", "Site")
simplifyBeta(divnet_temperature, All_phyto, "euclidean", "Site")

# You can plot this easily
simplifyBeta(divnet_temperature, All_phyto, "bray-curtis", "Site") %>%
  ggplot(aes(x = interaction(Covar1, Covar2), 
             y = beta_est)) +
  geom_point() +
  geom_linerange(aes(ymin = lower, ymax = upper)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  xlab("") + ylab("Estimates of Bray-Curtis distance")

ggsave("Beta_diversity.pdf")



set.seed(3423)

X <- diag(nrow(sample_data(All_phyto)))

divnet_phylum <-  divnet(W = All_phyto,
                         X = X,
                         variance = "none")


ait_test <- testBetaDiversity(dv = divnet_phylum, 
                              sample_specimen_matrix = divnet_phylum$X,
                              h0 = "aitchison",
                              n_boot = 1000,
                    groups = sample_data(All_phyto)$char)
ait_test$p_value



#################################
#################################

HP1=subset_samples(All_phyto, Site=="HP1")

divnet_temperature <- HP1 %>%
  divnet(X = "Sampling.Order", ncores = 2)


gg=simplifyBeta(divnet_temperature, HP1, "bray-curtis", "decimal_dat")

gg %>% mutate(diff= abs(Covar1-Covar2)) %>% group_by(diff) %>%summarize(median=median(beta_est))%>%ungroup()%>%ggplot(aes(x=diff, y=median))+geom_line()+geom_smooth()+
  geom_linerange(aes(ymin = lower, ymax = upper))



gg %>% mutate(diff= abs(Covar1-Covar2)) %>% group_by(diff)%>%ggplot(aes(x=diff, y=beta_est))+geom_line()+geom_smooth()+theme_classic()+xlab("Time lag (years)")+ylab("Estimates of Bray-Curtis distance")+ggtitle("HP1")
ggsave("HP1_beta_diversity.pdf")

write.csv(gg, "test_diversity.csv")

gg %>% mutate(diff= abs(Covar1-Covar2)) %>% subset(Covar1=="2017.641096" | Covar2=="2017.641096")%>%ggplot(aes(x=diff, y=beta_est))+geom_point()+geom_smooth()+theme_classic()+xlab("Time lag (years)")+ylab("Estimates of Bray-Curtis distance")+ggtitle("HP1")


library(vegan)
HP1=subset_samples(All_phyto, Site=="HP1")

Hp1_otu=as.data.frame(otu_table(HP1))
summary(colSums(Hp1_otu))

Hp1_otu2=t(Hp1_otu)

yy=as.data.frame(sample_data(HP1))

yy$sample=yy$SampleID

dream=yy%>%as_tibble() %>%select(sample, decimal_dat)

dream$name=dream$sample

mice_dist=avgdist(Hp1_otu2, sample=4265)

p=mice_dist%>%
  as.matrix() %>%
  as_tibble(rownames="sample") %>%
  pivot_longer(-sample) %>%
  filter(sample < name)

sleep=left_join(p, dream, by="sample")

sleep2=left_join(sleep, dream, by="name")

names(sleep2)

sleep2$decimal_dat.x

sleep2 %>% mutate(diff=abs(decimal_dat.x-decimal_dat.y)) %>%group_by(diff)%>% summarize(median=median(value))%>% ungroup() %>% ggplot(aes(x=diff, y=median))+geom_line()
ggsave("Hp1_community_similarity.pdf")



library(vegan)
AR=subset_samples(All_phyto, Site=="AR")

AR_otu=as.data.frame(otu_table(AR))
summary(colSums(AR_otu))

AR_otu2=t(AR_otu)

yy=as.data.frame(sample_data(AR))

yy$sample=yy$SampleID

dream=yy%>%as_tibble() %>%select(sample, decimal_dat)



mice_dist=avgdist(AR_otu2, sample=5195)

p=mice_dist%>%
  as.matrix() %>%
  as_tibble(rownames="sample") %>%
  pivot_longer(-sample) %>%
  filter(sample < name)

sleep=left_join(p, dream, by="sample")

dream$name=dream$sample
sleep2=left_join(sleep, dream, by="name")

names(sleep2)

sleep2$decimal_dat.x

sleep2 %>% mutate(diff=abs(decimal_dat.x-decimal_dat.y)) %>%dplyr::group_by(diff)%>% dplyr::summarize(median=median(value))%>% ungroup() %>% ggplot(aes(x=diff, y=median))+geom_line()+xlab("time lag in years")+ylab("Bray Curtis Similarity")
ggsave("AR_community_similarity.pdf")


library(dplyr)

library(vegan)
NTO1=subset_samples(All_phyto, Site=="NTO1")

NTO1_otu=as.data.frame(otu_table(NTO1))
summary(colSums(NTO1_otu))

NTO1_otu2=t(NTO1_otu)

yy=as.data.frame(sample_data(NTO1))

yy$sample=yy$SampleID

dream=yy%>%as_tibble() %>%select(sample, decimal_dat)



mice_dist=avgdist(NTO1_otu2, sample=3255)

p=mice_dist%>%
  as.matrix() %>%
  as_tibble(rownames="sample") %>%
  pivot_longer(-sample) %>%
  filter(sample < name)

sleep=left_join(p, dream, by="sample")

dream$name=dream$sample
sleep2=left_join(sleep, dream, by="name")

names(sleep2)

sleep2$decimal_dat.x

sleep2 %>% mutate(diff=abs(decimal_dat.x-decimal_dat.y)) %>%group_by(diff)%>% summarize(median=median(value))%>% ungroup() %>% ggplot(aes(x=diff, y=median))+geom_line()
ggsave("NTO1_community_similarity.pdf")





#################################
#################################

AR=subset_samples(All_phyto, Site=="AR")

divnet_temperature <- AR %>%
  divnet(X = "decimal_dat", ncores = 2)


hh=simplifyBeta(divnet_temperature, AR, "bray-curtis", "decimal_dat")

hh %>% mutate(diff= abs(Covar1-Covar2)) %>% group_by(diff) %>%summarize(median=median(beta_est))%>%ungroup()%>%ggplot(aes(x=diff, y=median))+geom_line()+geom_smooth()+
  geom_linerange(aes(ymin = lower, ymax = upper))


hh %>% mutate(diff= abs(Covar1-Covar2)) %>% group_by(diff)%>%ggplot(aes(x=diff, y=beta_est))+geom_line()+geom_smooth()+theme_classic()+xlab("Time lag (years)")+ylab("Estimates of Bray-Curtis distance")+ggtitle("AR")
ggsave("AR_beta_diversity.pdf")

write.csv(gg, "test_diversity.csv")



```



```{r}

library(remotes)
# install.packages("remotes")   ## run this line if you do not already have remotes installed
#remotes::install_github("adw96/breakaway")
#remotes::install_github("adw96/DivNet")
library(DivNet)
library(magrittr)
library(ggplot2)

## The importance of verifying assumptions

#As with all statistical models, the assumptions of `DivNet` should be verified before interpreting the results. In this vignette, I will show you how to investigate a key assumption of the `DivNet` model: that the log-ratio relative abundances are linear (or at least not strongly curved).

#You do not need to do this if you are fitting `DivNet` with categorical covariates, only if you are fitting `DivNet` with continuous covariates.

## Testing the changes in Shannon diversity with temperature

#We are going to use the Lee et al dataset and investigate the effect of temperature on phylum-level Shannon diversity. Recall that this dataset contains samples of microbial communities living on seafloor rocks near a hydrothermal vent.

#We first load the data and necessary packages, aggregate counts to the phylum level, and subset to samples with temperature data.



data(Lee)
lee_phylum <- Lee %>%
  tax_glom(taxrank = "Phylum") %>%
  subset_samples(temp != "<NA>")
#Annoyingly, temperature is a categorical variable in this dataset, so we convert it to a continuous variable and run `DivNet`, modelling the log-ratios as a linear function of temperature:

#lee_phylum %>% sample_data %>% temp
temp_continuous <- lee_phylum %>% sample_data %$% temp %>% as.character %>% as.numeric
divnet_temperature <- lee_phylum %>%
  divnet(X = temp_continuous, base=1)
#I'm a big fan of plotting the estimated diversity against the covariate before doing any hypothesis testing to make sure that everything looks reasonable.

shannon_table <- divnet_temperature %$%
  shannon %>%
  summary %>%
  cbind(temp_continuous)
shannon_table %>%
  ggplot(aes(x = temp_continuous, y = estimate)) + geom_point() +
  geom_linerange(aes(x = temp_continuous, ymin = lower, ymax = upper))

#Looks good to me! Let's go ahead and see if that upward trend is statistically significant given those wide error bars:

betta(shannon_table$estimate, shannon_table$error,
      cbind(1, divnet_temperature$X))$table

?divnet()

#We reject the null hypothesis of no significant linear increase in Shannon diversity with temperature with $p < 10^{-3}$.

# Assessing model fit

#Before taking that result too seriously, let's check for model misspecification. The `DivNet` model performs well when there is not significant curvature in the log-ratios. To investigate this, let's pull out the observed log-ratios (`observed_yy`) as well as the fitted log-ratios (`fitted_yy`).


fitted_yy <- DivNet::to_log_ratios(divnet_temperature$fitted_z,
                                   base = 1)
observed_yy <- lee_phylum %>% otu_table %>%
  as.matrix %>% t %>%
  DivNet::to_log_ratios(base = 1)

#Now let's put both the observed log-ratios and the fitted log-ratios against temperature. `DivNet` assumes a linear model for all log-ratios, and since there are 20 taxa in this dataset, there are 19 log-ratios. Therefore I'm going to write a function that plots the log-ratios.


draw_log_ratios <- function(taxon) {
  data.frame(temperature = temp_continuous,
             log_ratio = fitted_yy[,taxon],
             log_ratio_observed = observed_yy[,taxon]) %>%
    ggplot(aes(x = temperature, y = log_ratio_observed)) +
    geom_point() +
    xlab("Temperature") +
    ylab("log-ratios") +
    geom_point(aes(y = log_ratio), col = "blue") +
    ggtitle("Observed (black) and fitted (blue) log-ratios against temperature")
}
draw_log_ratios(taxon=1)
draw_log_ratios(taxon=2)
draw_log_ratios(taxon=3)

#I don't see any significant curvature in any of these plots. Note that it's not important that these scatterplots display a strong linear trend -- just that they are not strongly nonlinear. Note that we found that model is robust to mild nonlinearity, but its performance decreases as the nonlinear trend gets stronger. 

#In theory I should run the function and inspect those plots, but given that there may be hundreds or thousands of taxa model, I would recommend checking the most abundant taxa, since these are the ones that are most likely to throw off diversity estimation if misspecified.

?DivNet::divnet

divnet_phylum_char <- lee_phylum %>%
  divnet(X = "char", ncores = 2)

#Now that we have `divnet_phylum_char`, let's now compare the plug-in Shannon index with the divnet estimates

library(ggplot2)
divnet_phylum_char$shannon %>% 
  plot(lee_phylum, color = "char") +
  xlab("Basalt characteristic") +
  ylab("Shannon diversity estimate\n(phylum level)") +
  coord_cartesian(ylim = c(0,2))

#You will notice that the plug-in  estimates of Shannon diversity are different for each sample, but there is only a single DivNet estimate for each  characteristic (along with error bars). For characteristics for which many samples were observed, there are smaller error bars than for samples for which there was only one sample (seems reasonable -- we had less data).

## Where do the variance estimates come from?

#When taxa cluster together (spatially), you can imagine that your samples are more likely to be different from each other. For example, you may happen upon a patch of Microbe A in your first sample, but get a patch of Microbe B in your second sample. The statistics word for this is variance, and it is well studied that dependence structures (like spatial organisation) lead to greater variance. Plug-in estimates of diversity (the ones you're familiar with: where you just take the sample data and calculate the diversity index on it) ignore any dependence structure. This leads to understatements of variance (samples are more varying than the model that underpins the estimate allows), and exaggerated risk of concluding significant differences when none exist (p values are smaller than they should be, leading to an increased Type 1 error rate).
#DivNet addresses this by explicitly estimating those dependence structures, and then using them to come up with variance estimates (useful for the error bars). Let's see that in action.


## Hypothesis testing with DivNet

#I think the most common way ecologists currently test hypotheses about diversity is with a t-test. For example, if biological replicates are available, they would take the estimated index for one group and compared to the estimated index of the second group (Don't stop reading! You shouldn't do this!):

plugin <- tax_glom(Lee, taxrank="Phylum") %>%
            estimate_richness(measures = "Shannon") %$% Shannon
char <- Lee %>% sample_data %$% char
t.test(plugin[char == "altered"],
       plugin[char == "glassy"])

#Underpinning this approach is a major problem: the plug-in estimates of diversity are not very good because they don't account for missing taxa or undersampling or oversampling of taxa. For the Shannon index, estimated diversity is too low, with a known negative bias. They are also based on the multinomial model, which ignores taxon-taxon interactions and the additional variance attributable to them.

#DivNet addresses these problems by accounting for oversampling and undersampling, and modelling these interactions. This gives us better variance estimates with which to do hypothesis testing.

#Let's say we want to compare the Shannon index across the different characteristics. The package `breakaway` provides an implementation for statistical inference for alpha diversity called `betta`. We are going to use this function here for inference on the Shannon index.

#To set this up, we need a vector of our alpha diversity estimates, a vector of the standard errors in these estimates, and a design matrix

estimates <- divnet_phylum_char$shannon %>% summary %$% estimate
ses <- sqrt(divnet_phylum_char$`shannon-variance`)
X <- breakaway::make_design_matrix(lee_phylum, "char")
betta(estimates, ses, X)$table

#The intercept term is our altered basalts (they are the only ones that aren't listed), and so all comparisons are made to this baseline. We see that using DivNet we conclude that there is a significant differences between altered and glassy basalts, unlike what we saw with the t-test (DivNet p value is `r betta(estimates, ses, X)$table["predictorsglassy",3]` not `r round(t.test(plugin[char == "altered"], plugin[char == "glassy"])$p.value, 2)`). Note that the estimated differences between altered and glassy are similar (DivNet estimates a change in Shannon diversity of `r round(betta(estimates, ses, X)$table["predictorsglassy", 1], 3)` while the t-test estimated `r round(mean(plugin[char == "glassy"])-mean(plugin[char == "altered"]), 3)`), but the difference is the lower standard error (greater precision in estimating the difference) from DivNet. 

#We also see some differences in alpha diversity across the other different groups: biofilm basalts have significantly different diversity (at the phylum level) than altered basalts. If we want to do the global test of whether or not there are differences across categories, we can get the p value as follows:
betta(estimates, ses, X)$global[2]

data("GlobalPatterns")
# Recall that the best way to adjust for this is by estimating the 
# number of missing species using a species richness estimate.
# This will take a while, but it will speed things up later
water <- GlobalPatterns %>%
  subset_samples(SampleType %in% c("Freshwater", 
                                   "Freshwater (creek)", 
                                   "Ocean", 
                                   "Sediment (estuary)")) %>%
  tax_glom("Order")

water
ba <- breakaway(water)
ba 
plot(ba, water, color = "SampleType")

bt <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(water, "SampleType"))
bt$table


# to test if DivNet loaded correctly, try
dv_water_testing <- divnet(water, ncores = 2, tuning = "test")

dv_water <- divnet(water, ncores = 2)
dv_water %>% names

dv_water$shannon %>%
  summary %>%
  add_column("SampleNames" = water %>% otu_table %>% sample_names)

# or plot them:
plot(dv_water$shannon, 
     water, 
     col = "SampleType")
plot(water %>% sample_shannon, 
     water, 
     col = "SampleType") + ylim(0, 3.5)

dv_water_st <- water %>%
  divnet(X = "SampleType", ncores = 2)

plot(dv_water_st$shannon, 
     water, 
     col = "SampleType")
# This is just a wrapper for betta(), btw
testDiversity(dv_water_st, "shannon")




## To test hypotheses about beta diversity
# using DivNet, let's pull out our
# estimated distance matrix:
bc <- dv_water_st$`bray-curtis`

# You'll notice that all samples with the same
# SampleType at the same estimate. DivNet uses
# covariate information to share strength
# across samples and obtain an estimate
# about the beta diversity of the *ecosystems*
#   not the samples

# We can consider the  unique rows using
bc %>% unique

# Uniquely, DivNet also has variance estimates:
simplifyBeta(dv_water_st, water, "bray-curtis", "SampleType")
simplifyBeta(dv_water_st, water, "euclidean", "SampleType")

# You can plot this easily
simplifyBeta(dv_water_st, water, "bray-curtis", "SampleType") %>%
  ggplot(aes(x = interaction(Covar1, Covar2), 
             y = beta_est)) +
  geom_point() +
  geom_linerange(aes(ymin = lower, ymax = upper)) + 
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  xlab("") + ylab("Estimates of Bray-Curtis distance")


library(doSNOW)
library(doParallel)
library(foreach)
#

```














```{r}

######### EDA time series ############

## Some fast functions to visuazlie the data 

Cluster1=subset_samples(rel2All_phyto, Cluster_DeSeq=="Cluster_1")

otu_table(rel2All_phyto)

#p0.rel <- transform(Cluster1, "compositional")

p0.rel2=psmelt(rel2All_phyto)

darkzone <- data.frame(x = c(2017:2020) + 0.75,
                       x2 = c(2018:2021) + 0.25)


thedark <- geom_rect(data=darkzone,
                     fill="grey",alpha=0.4,
                     aes(xmin=x,
                         xmax=x2),
                     ymin=-Inf,ymax=Inf, inherit.aes = FALSE)

# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))


theme_timeseries <- list(
  thedark ,
  scale_x_continuous(breaks = seq(2017, 2021, 1)) ,
  xlab("Year") ,
  ylab("Relative abundance"),
  lil.strip)

plot_timeseries <- function(ASV, psmelt = p0.rel2,
                            title = NULL, color = "Curated_v3",
                            wrap = ~Site, percent = TRUE, scale = "fixed"){
  # A simple plotting of a timeseries. A vector of the asvs for plotting is given
  # and the results are plotted through the psmelt.
  
  require(lubridate)
  
  basic <-   psmelt %>% 
    filter(OTU %in% ASV) %>% 
    #mutate(decimaldat = decimal_date(Date) ) %>% 
    ggplot(aes(decimal_dat, Abundance)) + 
    geom_line(aes_string(group = "OTU", color = color)) +
    ggtitle(title) +
    theme_timeseries  
  
  if (!is.null(wrap)) {
    basic = basic +
      facet_wrap(wrap, scale = scale)
  }
  
  if (percent) {
    basic = basic + 
      ylab('Relative abundance') +
      scale_y_continuous(labels = scales::percent)
  }
  
  if (color == "phylogroup") {
    basic = basic +
      coloresbarplot
  }
  
  basic
}


plot_timeseries(p0.rel2)+scale_color_manual(values=coloresbarplot)
ggsave("timeseries_plot_Site.pdf", width=20, height=20)

p0.rel2









```





```{r}
library(tidyverse)
library(dplyr)


##3i think chosing this 
bl.phy.asinh <- transform_sample_counts(All_phyto, function(x) asinh(x))
physeq.list <- NULL

psmelt_dplyr = function(physeq) {
  #Implementation of the psmelt from phyloseq with dplyr
  # It is indeed faster, without further complications or differences. 
  # (well, in fact, its way more prone to give errors if the output is not well established, 
  # it doesn't check anything)
  
  sd = data.frame(sample_data(physeq)) %>% 
    rownames_to_column("Sample")
  TT = data.frame(as(tax_table(physeq),'matrix')) %>%
    rownames_to_column("OTU")
  if(taxa_are_rows(physeq)){
    otu.table = data.frame(as(otu_table(physeq),"matrix"),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  } else {
    otu.table = data.frame(t(as(otu_table(physeq),"matrix")),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  }
  
  all <- otu.table %>%
    left_join(TT, by = "OTU") %>%
    gather_("Sample", "Abundance", setdiff(colnames(otu.table), "OTU")) %>%
    left_join(sd, by = "Sample") %>% 
    select(Sample, Abundance, OTU, everything())
  
  return(all)
}


tsdf.0.1 <- bl.phy.asinh %>% 
  psmelt_dplyr() 


df.ts <- tsdf.0.1 %>% 
  arrange(decimal_dat) %>% 
  select(decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)

library(lomb)

lomb.03  <-  df.ts %>% 
  map(~lsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5))



library(fdrtool)
lomb.sea.03.ofac <- tibble( asv = names(lomb.03),
                       pval = map_dbl(lomb.03, ~.x$p.value),
                       peak = map_dbl(lomb.03, ~.x$peak),
                       interval  = map(lomb.03, ~.x$peak.at),
                       int.min = map_dbl(interval, ~.[[2]]),
                       int.max = map_dbl(interval, ~.[[1]])) %>% 
  drop_na(int.max) %>%
  mutate(qval = fdrtool::fdrtool(pval, statistic = 'pvalue')$qval) %>% 
  filter(qval <= 0.01)


results.lomb03 <- lomb.03[lomb.sea.03.ofac  %>% pull(asv)]



# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))

plotsPath = "periodoplots_significant.pdf"
pdf(file=plotsPath)  

periodoplots <- map(results.lomb03, ~tibble( scanned = .x$scanned,
                          power = .x$power)) %>% 
  bind_rows(.id = 'asv') %>% 
  split(.$asv) %>% 
  map(~ggplot(.x, aes(scanned, power)) + 
        geom_line(aes(group = asv)) + 
        facet_wrap(~asv) + 
        lil.strip)

periodoplots
dev.off()


#tosubset=lomb.sea.03.ofac%>%select(asv, pval)
#tosubset$pval="periodic"
#names(tosubset)[1] <- "ASV"
#names(tosubset)[2] <- "Periodicity"
#taxonomy_Phyto2 = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_v5.csv", sep=",") 


#taxonomy_Phyto3=left_join(taxonomy_Phyto2,tosubset, by="ASV")

#write.csv(taxonomy_Phyto3, "all_taxonomy_pr2_euk_chloro_unassigned_correct_v6.csv")

```




```{r}

library(microbiome)
library(viridis)
#devtools::install_github("microsud/microbiomeutilities")
#BiocManager::install("microbiome")
library(microbiomeutilities)

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)
ralph
p0.rel <- transform(All_phyto, "compositional")
deseq_diff=subset_taxa(p0.rel, Periodicity=="periodic")



ntaxa(deseq_diff)

grad_ab <- colorRampPalette(c("#96d4ca","#d3f3f1", "#7c65a9"))
heat.cols <- grad_ab(10)
simple_heatmap(deseq_diff,
               group.facet = "Cluster_DeSeq",
               group.order = NULL,
               abund.thres = 0.005,
               prev.thres = 0.01,
               level = "ASV",
               scale.color = "log10",
               na.fill = "white",
               color.fill = heat.cols,
               taxa.arrange=TRUE,
               remove.other=TRUE,
               panel.arrange="grid",
               ncol=NULL,
               nrow=NULL) + 
  labs(title = "Log-ratio transformed Abundances of ASVs with Significantly Different Abundances Across Clusters")


?transform()

ggsave("CLR_Clusters_ASV_DeSeq.pdf", width=10, height=10)


```



```{r}

periodic_ASVs=as.data.frame(tax_table(deseq_diff))

count_periodic_ASVs=periodic_ASVs%>%group_by(Curated_v3)%>%tally()
count_periodic_ASVs_genus=periodic_ASVs%>%group_by(Genus)%>%tally()
count_periodic_ASVs=periodic_ASVs%>%group_by(Genus,Curated_v3)%>%tally()

tog=left_join(count_periodic_ASVs_genus, curate, by="Genus")

phyto_ASVs=as.data.frame(tax_table(All_phyto))
curate=phyto_ASVs%>%select(Genus, Curated_v3)%>%distinct(Genus,.keep_all = TRUE )

count_phyto_ASVs=phyto_ASVs%>%group_by(Curated_v3)%>%tally()




p0.rel <- transform(All_phyto, "c")
deseq_diff=subset_taxa(rel2All_phyto, Periodicity=="periodic")

periodic_ASVs_otu=as.data.frame(otu_table(deseq_diff))
periodic_ASVs_otu$max<- apply(periodic_ASVs_otu, 1, max)
periodic_ASVs_otu$permax=periodic_ASVs_otu$max*100

test=periodic_ASVs_otu%>%filter(permax>5)
```


```{r}

p0.rel <- transform(All_phyto, "log")
deseq_diff=subset_taxa(p0.rel, Periodicity=="periodic")
write.csv(tax_table(deseq_diff), "periodic_ASVs.csv")

deseq_diff=subset_taxa(p0.rel, phytorelabun_max_class=="dominant" | phytorelabun_max_class=="abundant")
deseq_diff=subset_taxa(p0.rel, Periodicity=="periodic")

test=psmelt(deseq_diff)



or <- order(
  taxa_sums(deseq_diff), 
  decreasing=TRUE
  )[1:118]


ass <- otu_table(deseq_diff) %>% 
  data.frame()

write.csv(ass, "heatmap_to_order.csv")

colnames(ass) <- sample_names(deseq_diff)

tx <- tax_table(deseq_diff) %>% 
  data.frame()

tx <- dplyr::mutate(tx, 
                    usename = paste0(
                      "ASV_", 
                      seq(1, ntaxa(deseq_diff)), 
                      "_", 
                      tax_table(deseq_diff)[, "Curated_v3"]
                      ), 
                    rnames = taxa_names(deseq_diff)
                    )


tx <- dplyr::mutate(tx, 
                    usename = paste0(
                      tax_table(deseq_diff)[, "CuratedASV"], 
                      "_", 
                      tax_table(deseq_diff)[, "Curated_v3"]
                      ), 
                    rnames = taxa_names(deseq_diff)
                    )

tx
tx$usename

write.csv(tx, "tx.csv")

ass <- ass[or, ]
rownames(ass) <- tx$usename[or]

write.csv(ass, "ass.csv")

df <- sample_data(deseq_diff) %>% 
  data.frame()

df <- df[ , c(
  "Cluster_DeSeq_Spatial",
  "Season_Aka"
    )] 


df <- with(
  df, 
  df[order(Season_Aka, Cluster_DeSeq_Spatial),]
  )



Cluster_DeSeq_Spatial<- c("royalblue3", "seagreen1", "pink")
names(Cluster_DeSeq_Spatial) <- levels(df$Cluster_DeSeq_Spatial)


Season_Aka <- c("seagreen1", "pink")
names(Season_Aka) <- levels(df$Season_Aka)

ann_colors <-  list(
  Cluster_DeSeq_Spatial = Cluster_DeSeq_Spatial,
  Season = Season
  )


poo=t(ass)
quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

#OG playing with 70
mat_breaks <- quantile_breaks(poo, n =100)

mat_breaks

mat_breaks2 <- seq(min(0.01), max(0.4), length.out = 100)
mat_breaks <- quantile_breaks(mat_breaks2, n =100)

#ass=read.csv("practice_t_reordered_ass.csv", row.names = 1)

??cubicl()
library(pals)
library(RColorBrewer)


pdf("heatmap_log.pdf", height=20, width=15)
pheatmap(
  ass, 
   color = colorRampPalette(rev(brewer.pal(n = 7, name =
"RdYlBu")))(100), 
  cluster_rows = TRUE,
  cluster_cols = FALSE,
  fontsize_row = 14, 
  fontsize_col = 10,
  annotation_col=df
  )
dev.off()
annotation_colors = ann_colors
  #annotation_col = df, 

```





```{r}
library(microbiome)
library(viridis)
#devtools::install_github("microsud/microbiomeutilities")
#BiocManager::install("microbiome")
library(microbiomeutilities)

####start here as an example 3102020
##Figure to use with Mid 
library("pheatmap")
library(viridis)
library(pals)
write.csv(t(otu_table((SAR11w5.rel_NoChloroNoMito))),"SAR11w5_rel_nochloro.csv") 
rex=as.data.frame(sample_data(SAR11w5.rel_NoChloroNoMito))
write.csv(rex, file="rex2_noChloro.csv")



SAR11Edited=read.csv(file = "ORDERED_otu_heat_New_Seasons_nochloro.csv", header= TRUE, row.names=1)

reltable=otu_table(SAR11Edited, taxa_are_rows = T)

reltable=as.data.frame(reltable)

poo=t(reltable)

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}

#OG playing with 70
mat_breaks <- quantile_breaks(poo, n =10000)

mat_breaks

mat_breaks2 <- seq(min(poo), max(poo), length.out = 100)



colorBlindGrey8   <- c("#0072B2", "#56B4E9", "#E69F00","#999999", "#009E73","#F0E442", "#D55E00", "#CC79A7")

Tol=c("#332288", "#88CCEE", "#117733", "#999933", "#DDCC77", "#CC6677", "#882255", "#AA4499")


##TO TRY OUR LINEAR 


#write.csv(flip_sar11, "test.csv")

nameSAR11 <- read.csv(file = "taxa_name_ordered3.csv", header = TRUE, row.names=1) 
annotation_row=nameSAR11
annotation_row=as.data.frame(annotation_row)

labelsrow <- read.csv(file = "labels_row.csv", header = TRUE, row.names=1) 
labelsrow=as.data.frame(labelsrow)
labelsrow


#Note:this is with more of the metadata variables
#meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4.csv", header = TRUE, row.names = 1) 

meta_site <- read.csv(file = "meta_ORDERED_NewSeasons_otu_heat4A.csv", header = TRUE, row.names = 1) 

annotation_col=as.data.frame(meta_site)
annotation_col

col.pal <- RColorBrewer::brewer.pal(9, "Reds")

library(viridis)
aka4 = list(
  Season = c(Spring="#f7943e",Summer = "#E7665E", Fall= "#2ab0b9", Winter="#9D77FC"))

aka5 = list(
  Season = c(Winter="#ababff", Spring="#ffffab",Summer = "#ffabab", Fall= "#abffab"),
Subgroups=c(Ia="#0072B2", Ib="#56B4E9", IIa="#E69F00",IIb= "#999999",IIIa= "#009E73",IV= "#F0E442", Va= "#D55E00",Vb= "#CC79A7") )


library(pals)
#https://cran.r-project.org/web/packages/pals/vignettes/pals_examples.html

##cubicl works best with lots of breaks 10,000 but not 100,000
##tol.rainbow also works but only with 10,000 breaks



labels_row = c("E   S", "E   S", "E", "E", "E", "E", "E", "E", "     S", "     S", "", "", "", "", "", 
    "", "", "", "", "", "", "", "E", "E", "E", "E", "E   S", "E", "E", "E", "E", "", "", "", "", 
    "", "", "", "", "", "", "E   S", "E", "E", "E", "E", "E", "E", "E   S", "E   S", "E", "E", "", "", "", 
    "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", "", 
    "", "", "", "", "", "", "", "", "", "E   S", "E   S", "E", "E", "", "", "", "E", "", "", "", 
    "E   S", "E", "", "E", "E   S", "E", 
    "E   S", "", "", "", "E   S")

pdf(file = "SAR11_all_8222020_nochloro_newseasons_cubicl.pdf")



pheatmap(reltable, 
          color = cubicl(length(mat_breaks)-1),
        breaks=mat_breaks,
                   cluster_row = F,
                   cluster_cols = F,
        gaps_row = c(22,41,82,84,91,95,98),
         main = "",
                   annotation_col = annotation_col,
                   annotation_row = annotation_row,
        labels_row = labels_row,
                   fontsize = 12,
                   fontsize_row=4, 
                   fontsize_col = 2,
                border_color=NA,
annotation_colors = aka5,
        gaps_col=c(120,160))


```


```{r}

plot(no_dup_metadata_KBytNERR$decimal,no_dup_metadata_KBytNERR$Photosynthetic.radiaion,pch=19,cex=0.3)

```



```{r}
#install.packages("wql")
library(wql) 

#install.packages("lomb")
library(lomb)
library(lubridate)
library(tidyverse)


no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv5.csv", header=TRUE, fill=TRUE)
#no_dup_metadata_KBytNERR$Date2<- as.POSIXlt(no_dup_metadata_KBytNERR$Date2)
#no_dup_metadata_KBytNERR$Date2

#no_dup_metadata_KBytNERR$Date2 <- as.Date(no_dup_metadata_KBytNERR$Date2, format="%Y-%m-%d")

no_dup_metadata_KBytNERR=no_dup_metadata_KBytNERR%>%drop_na(Enviro_Clustering)
cluster_coastal=subset(no_dup_metadata_KBytNERR, Enviro_Clustering=="Coastal")
cluster_nearshore=subset(no_dup_metadata_KBytNERR,Enviro_Clustering=="Nearshore")
cluster_offshore=subset(no_dup_metadata_KBytNERR, Enviro_Clustering=="Offshore")

plot(cluster_coastal$decimal,cluster_coastal$SYN.mL,pch=19,cex=0.3)
plot(cluster_nearshore$decimal,cluster_nearshore$SYN.mL,pch=19,cex=0.3)
plot(cluster_offshore$decimal,cluster_offshore$SYN.mL,pch=19,cex=0.3)



####
plotsPath = "cluster_offshore_Temp.pdf"
pdf(file=plotsPath)  
    {
cluster_offshore_Temp=summary(randlsp(x= cluster_offshore$SW.Temperature.at.site,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_Temp
    } 
    dev.off() 
names(cluster_offshore_Temp)[1] <- 'cluster_offshore_Temp'


plotsPath = "cluster_nearshore_Temp.pdf"
pdf(file=plotsPath)  
    {
cluster_nearshore_Temp=summary(randlsp(x= cluster_nearshore$SW.Temperature.at.site,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_nearshore_Temp
  } 
    dev.off() 
names(cluster_nearshore_Temp)[1] <- 'cluster_nearshore_Temp'

plotsPath = "cluster_coastal_Temp.pdf"
pdf(file=plotsPath)  
    {
cluster_coastal_Temp=summary(randlsp(x= cluster_coastal$SW.Temperature.at.site,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_coastal_Temp
} 
    dev.off() 
names(cluster_coastal_Temp)[1] <- 'cluster_coastal_Temp'

####

#pHSalinity_cluster_offshore=subset(cluster_offshore, Site!="WAI2" & Site!="KAHO")
#pHSalinity_cluster_nearshore=subset(cluster_nearshore, Site!="WAI2" & Site!="KAHO")
#pHSalinity_cluster_coastal=subset(cluster_coastal, Site!="WAI2" & Site!="KAHO")



plotsPath = "cluster_offshore_pH.pdf"
pdf(file=plotsPath)  
    {   
cluster_offshore_pH=summary(randlsp(x= cluster_offshore$pH,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_pH
 
    } 
    dev.off() 

names(cluster_offshore_pH)[1] <- 'cluster_offshore_pH'


plotsPath = "cluster_nearshore_pH.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_pH=summary(randlsp(x= cluster_nearshore$pH,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_nearshore_pH
} 
    dev.off() 
names(cluster_nearshore_pH)[1] <- 'cluster_nearshore_pH'


plotsPath = "cluster_coastal_pH.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_pH=summary(randlsp(x= cluster_coastal$pH,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_coastal_pH
} 
    dev.off() 
names(cluster_coastal_pH)[1] <- 'cluster_coastal_pH'

#############Salinity



plotsPath = "cluster_offshore_Salinity.pdf"
pdf(file=plotsPath)  
    {   
cluster_offshore_Salinity=summary(randlsp(x= cluster_offshore$Salinity,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_Salinity
 
    } 
    dev.off() 

names(cluster_offshore_Salinity)[1] <- 'cluster_offshore_Salinity'


plotsPath = "cluster_nearshore_Salinity.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_Salinity=summary(randlsp(x= cluster_nearshore$Salinity,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_nearshore_Salinity
} 
    dev.off() 
names(cluster_nearshore_Salinity)[1] <- 'cluster_nearshore_Salinity'


plotsPath = "cluster_coastal_Salinity.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_Salinity=summary(randlsp(x= cluster_coastal$Salinity,
                times = luster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_coastal_Salinity
} 
    dev.off() 
names(cluster_coastal_Salinity)[1] <- 'cluster_coastal_Salinity'


#############3

plotsPath = "cluster_offshore_chla.pdf"
pdf(file=plotsPath)  
    {   
cluster_offshore_chla=summary(randlsp(x= cluster_offshore$chla.ug.per.L,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_chla
} 
    dev.off() 

names(cluster_offshore_chla)[1] <- 'cluster_offshore_chla'


plotsPath = "cluster_nearshore_chla.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_chla=summary(randlsp(x= cluster_nearshore$chla.ug.per.L,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_nearshore_chla
} 
    dev.off() 

names(cluster_nearshore_chla)[1] <- 'cluster_nearshore_chla'


####cluster_coastal 
cluster_coastal_nobloom=subset(cluster_coastal, chla.ug.per.L<8)
plotsPath = "cluster_coastal_chla.pdf"
pdf(file=plotsPath)  
    {   

cluster_coastal_chla=summary(randlsp(x= cluster_coastal$chla.ug.per.L,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_coastal_chla
} 
    dev.off() 
plot(cluster_coastal$decimal,cluster_coastal$chla.ug.per.L,pch=19,cex=0.3)
names(cluster_coastal_chla)[1] <- 'cluster_coastal_chla'

#############PEUK

plotsPath = "cluster_offshore_PEUK.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_PEUK=summary(randlsp(x= cluster_offshore$PEUK.mL,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_PEUK
} 
    dev.off() 
names(cluster_offshore_PEUK)[1] <- 'cluster_offshore_PEUK'


plot(cluster_offshore$decimal,cluster_offshore$PEUK.mL,pch=19,cex=0.3)


plotsPath = "cluster_nearshore_PEUK.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_PEUK=summary(randlsp(x= cluster_nearshore$PEUK.mL,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_PEUK
names(cluster_nearshore_PEUK)[1] <- 'cluster_nearshore_PEUK'


plot(cluster_nearshore$decimal,cluster_nearshore$PEUK.mL,pch=19,cex=0.3)


plotsPath = "cluster_coastal_PEUK.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_PEUK=summary(randlsp(x= cluster_coastal$PEUK.mL,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_PEUK)[1] <- 'cluster_coastal_PEUK'

cluster_coastal_PEUK
plot(cluster_coastal$decimal,cluster_coastal$PEUK.mL,pch=19,cex=0.3)


######HBACT

plotsPath = "cluster_offshore_HBACT.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_HBACT=summary(randlsp(x= cluster_offshore$HBACT.mL,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_HBACT
} 
    dev.off() 
names(cluster_offshore_HBACT)[1] <- 'cluster_offshore_HBACT'


plotsPath = "cluster_nearshore_HBACT.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_HBACT=summary(randlsp(x= cluster_nearshore$HBACT.mL,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_HBACT
names(cluster_nearshore_HBACT)[1] <- 'cluster_nearshore_HBACT'


plotsPath = "cluster_coastal_HBACT.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_HBACT=summary(randlsp(x= cluster_coastal$HBACT.mL,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_HBACT)[1] <- 'cluster_coastal_HBACT'

cluster_coastal_HBACT



######SYN

plotsPath = "cluster_offshore_Syn.pdf"
pdf(file=plotsPath)  
    {   
cluster_offshore_Syn=summary(randlsp(x= cluster_offshore$SYN.mL,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
names(cluster_offshore_Syn)[1] <- 'cluster_offshore_Syn'


plotsPath = "cluster_nearshore_Syn.pdf"
pdf(file=plotsPath)  
    {   

cluster_nearshore_Syn=summary(randlsp(x= cluster_nearshore$SYN.mL,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 

names(cluster_nearshore_Syn)[1] <- 'cluster_nearshore_Syn'

plotsPath = "cluster_coastal_Syn.pdf"
pdf(file=plotsPath)  
    {   

cluster_coastal_Syn=summary(randlsp(x= cluster_coastal$SYN.mL,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 

names(cluster_coastal_Syn)[1] <- 'cluster_coastal_Syn'



###okay I cant use PRO 
plotsPath = "cluster_offshore_PRO.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_PRO=summary(randlsp(x= cluster_offshore$PRO.mL,
                times = cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 

names(cluster_offshore_PRO)[1] <- 'cluster_offshore_PRO'

cluster_offshore_PRO


plot(cluster_offshore$decimal,cluster_offshore$PRO.mL,pch=19,cex=0.3)

plotsPath = "cluster_nearshore_PRO.pdf"
pdf(file=plotsPath)  
    {   

cluster_nearshore_PRO=summary(randlsp(x= cluster_nearshore$PRO.mL,
                times = cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 

names(cluster_nearshore_PRO)[1] <- 'cluster_nearshore_PRO'

cluster_nearshore_PRO
plot(cluster_nearshore$decimal,cluster_nearshore$PRO.mL,pch=19,cex=0.3)


plotsPath = "cluster_coastal_PRO.pdf"
pdf(file=plotsPath)  
    {   

cluster_coastal_PRO=summary(randlsp(x= cluster_coastal$PRO.mL,
                times = cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 

names(cluster_coastal_PRO)[1] <- 'cluster_coastal_PRO'

cluster_coastal_PRO
plot(cluster_coastal$decimal,cluster_coastal$PRO.mL,pch=19,cex=0.3)



######Phosphate
Nutrients_cluster_offshore=subset(cluster_offshore, Year!="2020" & Year!="2021")
Nutrients_cluster_nearshore=subset(cluster_nearshore, Year!="2020" & Year!="2021")
Nutrients_cluster_coastal=subset(cluster_coastal, Year!="2020" & Year!="2021")



plotsPath = "cluster_offshore_Phosphate.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_Phosphate=summary(randlsp(x= Nutrients_cluster_offshore$Phosphate,
                times = Nutrients_cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_Phosphate
} 
    dev.off() 
names(cluster_offshore_Phosphate)[1] <- 'cluster_offshore_Phosphate'


plotsPath = "cluster_nearshore_Phosphate.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_Phosphate=summary(randlsp(x= Nutrients_cluster_nearshore$Phosphate,
                times = Nutrients_cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_Phosphate
names(cluster_nearshore_Phosphate)[1] <- 'cluster_nearshore_Phosphate'


plotsPath = "cluster_coastal_Phosphate.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_Phosphate=summary(randlsp(x= Nutrients_cluster_coastal$Phosphate,
                times =Nutrients_cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_Phosphate)[1] <- 'cluster_coastal_Phosphate'

cluster_coastal_Phosphate


######Silicate


plotsPath = "cluster_offshore_Silicate.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_Silicate=summary(randlsp(x= Nutrients_cluster_offshore$Silicate,
                times = Nutrients_cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_Silicate
} 
    dev.off() 
names(cluster_offshore_Silicate)[1] <- 'cluster_offshore_Silicate'


plotsPath = "cluster_nearshore_Silicate.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_Silicate=summary(randlsp(x= Nutrients_cluster_nearshore$Silicate,
                times = Nutrients_cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_Silicate
names(cluster_nearshore_Silicate)[1] <- 'cluster_nearshore_Silicate'


plotsPath = "cluster_coastal_Silicate.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_Silicate=summary(randlsp(x= Nutrients_cluster_coastal$Silicate,
                times =Nutrients_cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_Silicate)[1] <- 'cluster_coastal_Silicate'

cluster_coastal_Silicate



#####Nitrate+Nitrite
Nutrients_cluster_offshore$NitrateNitrite
Nutrients_cluster_offshore$NitrateNitrite
Nutrients_cluster_offshore2=Nutrients_cluster_offshore%>%drop_na(NitrateNitrite)
plotsPath = "cluster_offshore_NitrateNitrite.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_NitrateNitrite=summary(randlsp(x= Nutrients_cluster_offshore$NitrateNitrite,
                times = Nutrients_cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_NitrateNitrite
} 
    dev.off() 
names(cluster_offshore_NitrateNitrite)[1] <- 'cluster_offshore_NitrateNitrite'


plotsPath = "cluster_nearshore_NitrateNitrite.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_NitrateNitrite=summary(randlsp(x= Nutrients_cluster_nearshore$NitrateNitrite,
                times = Nutrients_cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_NitrateNitrite
names(cluster_nearshore_NitrateNitrite)[1] <- 'cluster_nearshore_NitrateNitrite'


plotsPath = "cluster_coastal_NitrateNitrite.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_NitrateNitrite=summary(randlsp(x= Nutrients_cluster_coastal$NitrateNitrite,
                times =Nutrients_cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_NitrateNitrite)[1] <- 'cluster_coastal_NitrateNitrite'

cluster_coastal_NitrateNitrite




######Ammonia cant really do?

plotsPath = "cluster_offshore_Ammonia.pdf"
pdf(file=plotsPath)  
    {   

cluster_offshore_Ammonia=summary(randlsp(x= Nutrients_cluster_offshore$Ammonia,
                times = Nutrients_cluster_offshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
cluster_offshore_Ammonia
} 
    dev.off() 
names(cluster_offshore_Ammonia)[1] <- 'cluster_offshore_Ammonia'


plotsPath = "cluster_nearshore_Ammonia.pdf"
pdf(file=plotsPath)  
    {   
cluster_nearshore_Ammonia=summary(randlsp(x= Nutrients_cluster_nearshore$Ammonia,
                times = Nutrients_cluster_nearshore$decimal,
                type = 'period',
                plot = T,  ofac=5))
} 
    dev.off() 
cluster_nearshore_Ammonia
names(cluster_nearshore_Ammonia)[1] <- 'cluster_nearshore_Ammonia'


plotsPath = "cluster_coastal_Ammonia.pdf"
pdf(file=plotsPath)  
    {   
cluster_coastal_Ammonia=summary(randlsp(x= Nutrients_cluster_coastal$Ammonia,
                times =Nutrients_cluster_coastal$decimal,
                type = 'period',
                plot = T,  ofac=5, trace = TRUE))
} 
    dev.off() 
names(cluster_coastal_Ammonia)[1] <- 'cluster_coastal_Ammonia'

cluster_coastal_Ammonia










##############
#############
###############

SYN_lomb=do.call("cbind", list(cluster_offshore_Temp, cluster_nearshore_Temp, cluster_coastal_Temp, cluster_offshore_pH, cluster_nearshore_pH, cluster_coastal_pH,cluster_offshore_Salinity, cluster_nearshore_Salinity, cluster_coastal_Salinity,cluster_offshore_chla, cluster_nearshore_chla, cluster_coastal_chla,cluster_offshore_PEUK, cluster_nearshore_PEUK, cluster_coastal_PEUK,cluster_offshore_HBACT, cluster_nearshore_HBACT, cluster_coastal_HBACT,cluster_offshore_Syn, cluster_nearshore_Syn,cluster_coastal_Syn,cluster_offshore_PRO, cluster_nearshore_PRO, cluster_coastal_PRO,cluster_offshore_Phosphate, cluster_nearshore_Phosphate, cluster_coastal_Phosphate,cluster_offshore_Silicate, cluster_nearshore_Silicate, cluster_coastal_Silicate, cluster_offshore_NitrateNitrite, cluster_nearshore_NitrateNitrite, cluster_coastal_NitrateNitrite, cluster_offshore_Ammonia, cluster_nearshore_Ammonia, cluster_coastal_Ammonia))

write.csv(SYN_lomb,"environmental_lomb_v2.csv")

```





```{r}



c=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=DIN_DIP, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("DIN_DIP")+xlab("")
c



+scale_shape_manual(values=colorssite)
values = c(0, 5, 6, 15)
library(svglite)

svglite("DIN_P_3.svg", height=3, width=3)
p=ggplot(no_dup_metadata_KBytNERR, aes(x = Phosphate, y = DIN, color=Site, shape=Enviro_Clustering)) + geom_point(size=3) + geom_abline(slope = 16, yintercept=0)+ xlim(0,0.5)+ scale_color_manual(values=colorssite)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + geom_hline(aes(yintercept = 0.01, color="gray"), linetype="dotted")+geom_vline(aes(xintercept = 0.01, color="gray"),linetype="dotted")+ylab("DIN (uM)")+xlab("Phosphate (uM)")+theme(legend.position="none")
p
dev.off()




p=ggplot(no_dup_metadata_KBytNERR, aes(x = NitrateNitrite, y = Silicate, color=Site, shape=Enviro_Clustering)) + geom_point() + geom_abline(slope = 1, yintercept=0)+ scale_color_manual(values=colorssite)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + geom_hline(aes(yintercept = 0.01, color="gray"), linetype="dotted")+geom_vline(aes(xintercept = 0.01, color="gray"),linetype="dotted")
p

ggsave("DIN_P.pdf")


colorssite 

summary(no_dup_metadata_KBytNERR$DIN_DIP)
ggplot(no_dup_metadata_KBytNERR, x=DIN_DIP, color=DeSeq_Spatial_Cluster)+geom_box()

zebra

write.csv(stuff, "DIN_DIP_KByT.csv")

Coastal=subset(no_dup_metadata_KBytNERR, Cluster_DeSeq_Spatial=="Coastal")

stuff<- Coastal%>%drop_na(DIN_DIP)%>%dplyr::group_by(Cluster_DeSeq_Spatial,Season)%>%dplyr::summarise(mean=round(mean(DIN_DIP),2), sd=round(sd(DIN_DIP),2), n=n())




library("multcomp")

Coastal$Season=as.factor(Coastal$Season)

### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(DIN_Silicate~ Season, data = Coastal)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  



no_dup_metadata_KBytNERR$Site=factor(no_dup_metadata_KBytNERR$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "CB", "SB", "NB", "SR8", "SR2", "NR2", "STO1", "NTO1"))


Ammonia<- ggplot(data=Coastal, aes(x=Season, y=DIN_DIP, fill=Season))+geom_boxplot() 
Ammonia=Ammonia + theme(legend.title = element_blank()) + ylab("DIN DIP ratio") +xlab("Season") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Ammonia
ggsave("Coastal_only_Seasonal_DIN_DIP.pdf")


Phos <- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=Phosphate))+geom_boxplot(color="black", fill="gray53") 
Phos=Phos + theme(legend.title = element_blank()) + ylab("Phosphate (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Phos

Nitrate <- ggplot(data=no_dup_metadata_KBytNERR, aes(x=as.factor(Site), y=NitrateNitrite))+geom_boxplot(color="black", fill="gray53") 
Nitrate=Nitrate + theme(legend.title = element_blank()) + ylab("Nitrate+Nitrite (uM)") +xlab("Site") + theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) 
Nitrate




```




```{r}
?month.abb
no_dup_metadata_KBytNERR$Month
no_dup_metadata_KBytNERR$Date2
no_dup_metadata_KBytNERR$Month = factor(no_dup_metadata_KBytNERR$Month, levels==month.name)
your_data$month = factor(your_data$month, levels = month.abb)
list(month.abb)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="darkseagreen",
"Coastal"="#E4C244")


library(ggplot2)


no_dup_metadata_KBytNERR=no_dup_metadata_KBytNERR%>%drop_na(Enviro_Clustering)
no_dup_metadata_KBytNERR$Month_abb=factor(no_dup_metadata_KBytNERR$Month_abb, levels=c("Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar"))









a=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SW.Temperature.at.site, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Temperature (C)")+xlab("")
a

b=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=pH, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("pH")+xlab("")


c=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=Salinity, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Salinity (ppt)")+xlab("")




d=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=chla.ug.per.L,group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + scale_y_continuous( trans = log10_trans()) + xlab("")+ylab(bquote("Chlorophyll a ug L"^-1))




e=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=PEUK.mL, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Photosynthetic picoeukaryotic cells mL"^-1))


f=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=HBACT.mL, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Heterotrophic prokaryotic cells mL"^-1))

g=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SYN.mL, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Synechococus cells mL"^-1))

h=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=PRO.mL, group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Prochlorococcus cells mL"^-1))





i=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=Phosphate,group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ylab("Phosphate uM")+xlab("Month")


j=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=Silicate,group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + scale_y_continuous( trans = log10_trans()) +xlab("Month")+ylab("Silicate uM")

k=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=NitrateNitrite,group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +xlab("Month")+ylab("Nitrate+Nitrite uM")

l=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=Ammonia,group=Enviro_Clustering, color=Enviro_Clustering))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorEnviro)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +xlab("Month")+ylab("Ammonia uM")



library(cowplot)

plot_grid(a, b,c,d,e,f,g,h,i,j,k,l,labels = c('a', 'b','c', 'd','e', 'f','g', 'h','i', 'j','k', 'l'), align = 'hv',label_size = 12, ncol = 4)
ggsave("Enviro_sup_figure_spatial_month_based_enviro.pdf", width=19, height=12)



#############3
##############
#############
Coastal_Site=subset(no_dup_metadata_KBytNERR, Site=="HP1" | Site=="AR"| Site=="CB"| Site=="SB"|Site=="SR8" | Site=="NB")


p=ggplot(data = Coastal_Site, aes(x=as.Date(Date2), y=SYN.mL,group=Site, color=Site))+
  geom_line() +
  geom_point()+ xlab("")+ylab(bquote("Synechococus cells mL"^-1))+
  theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + geom_rect(aes(xmin = as.Date("2017-12-27"), xmax = as.Date("2018-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2018-12-27"), xmax = as.Date("2019-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2019-12-27"), xmax = as.Date("2020-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2020-12-27"), xmax = as.Date("2021-03-29"),
             ymin = -Inf, ymax = Inf),fill = "gray", alpha=0.01,color="gray")


p+geom_point(aes(x=as.Date(Date2), y = Coastal_Site$SW.Temperature.at.site*20000))+geom_line(aes(x=as.Date(Date2), y = SW.Temperature.at.site*20000)) +
  scale_y_continuous(sec.axis = ggplot2::sec_axis(~./20000,
                  name = "Temperature (C)"))
ggsave("Synechoccus_Timing_coastal_with_Temp.pdf")


#############3
##############
#############
Offshore_site=subset(no_dup_metadata_KBytNERR, Site=="NTO1" | Site=="STO1"| Site=="NR2"| Site=="SR2")


p=ggplot(data = Offshore_site, aes(x=as.Date(Date2), y=SYN.mL,group=Site, color=Site))+
  geom_line() +
  geom_point()+ xlab("")+ylab(bquote("Synechococus cells mL"^-1))+
  theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12,face="bold"), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + geom_rect(aes(xmin = as.Date("2017-12-27"), xmax = as.Date("2018-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2018-12-27"), xmax = as.Date("2019-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2019-12-27"), xmax = as.Date("2020-03-29"),
             ymin = -Inf, ymax = Inf),  
             fill = "gray", alpha=0.01,color="gray")+ geom_rect(aes(xmin = as.Date("2020-12-27"), xmax = as.Date("2021-03-29"),
             ymin = -Inf, ymax = Inf),fill = "gray", alpha=0.01,color="gray")
p

p+geom_point(aes(x=as.Date(Date2), y = Coastal_Site$SW.Temperature.at.site*20000))+geom_line(aes(x=as.Date(Date2), y = SW.Temperature.at.site*20000)) +
  scale_y_continuous(sec.axis = ggplot2::sec_axis(~./20000,
                  name = "Temperature (C)"))
ggsave("Synechoccus_Timing_coastal_with_Temp.pdf")



names(no_dup_metadata_KBytNERR)

ggplot(data = no_dup_metadata_KBytNERR, aes(x=decimal_dat, y=SYN.mL, group=Cluster_DeSeq_Spatial, color=Site))+
  geom_smooth(method=lm) +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("SYN.mL")+xlab("Date")



```


```{r}

ggplot(data = no_dup_metadata_KBytNERR, aes(x=SW.Temperature.at.site, y=SYN.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth(method=lm) +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("SYN.mL")+xlab("Temperature (C)")

library(ggpmisc)
install.packages('ggpmisc')
library(ggpmisc)
dettach("scales")
install.packages("scales")
nowai2nokaho=subset(no_dup_metadata_KBytNERR, Site!="WAI2"&Site!="KAHO")
ggplot(data = no_dup_metadata_KBytNERR, aes(x=SW.Temperature.at.site, y=SYN.mL, group=Site, color=Site)) +
  stat_poly_line() +
  stat_poly_eq(aes(label = paste(after_stat(eq.label),
                                 after_stat(rr.label), sep = "*\", \"*"))) +
  geom_point()

names(no_dup_metadata_KBytNERR)
old_Fcm=subset(no_dup_metadata_KBytNERR, FCM.Machine=="Old")
new_Fcm=subset(no_dup_metadata_KBytNERR, FCM.Machine=="New")


ggplot(data = old_Fcm, aes(x=SW.Temperature.at.site, y=PRO.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth(method=lm) +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("PRO.mL")+xlab("Temperature (C)")


ggplot(data = new_Fcm, aes(x=SW.Temperature.at.site, y=PRO.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth(method=lm) +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("PRO.mL")+xlab("Temperature (C)")




a=ggplot(data = new_Fcm, aes(x=Month_abb, y=PRO.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("New Prochlorococcus cells mL"^-1))

a


b=ggplot(data = old_Fcm, aes(x=Month_abb, y=PRO.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote(" Old Prochlorococcus cells mL"^-1))
b

c=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=PRO.mL, group=Cluster_DeSeq_Spatial, color=Cluster_DeSeq_Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote(" Together Prochlorococcus cells mL"^-1))
c





plot_grid(a, b,c,labels = c('a', 'b', "c"), align = 'hv',label_size = 12, ncol = 1)
ggsave("Old_new_FCM.pdf", width=6, height=12)




```




```{r}

mean_coverage_Q2Q3_cyano

library(data.table)

read.recruit=read.csv("mean_coverage_Q2Q3_cyano_v2.csv", header=TRUE, row.names = 1)
library(tidyverse)
library(reshape2)
library(dplyr)
??aggregate()
#woof=aggregate(read.recruit~ clade, read.recruit, sum)


samp.with.rownames <- data.frame(woof[,-1], row.names=woof[,1])

#samp.with.rownames2=samp.with.rownames[,-1]

freqs2 <- apply(read.recruit, 2, function(i) i/sum(i))

freqs3=freqs2*100
freqs3=as.data.frame(freqs3)

write.csv(freqs3,"test.csv")

freqs4<- freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(freqs4)




freqs3=read.csv("SCG_normalized_all_depths.csv", row.name=1)


sample_order <- factor(freqs4$rowname,level=c("MED4",
"MIT9515",
"AS9601",
"MIT9301",
"SB",
"MIT9314",
"GP2",
"MIT9107",
"MIT9201",
"MIT9202",
"MIT9215",
"MIT9302",
"MIT9311",
"MIT9321",
"PAC1",
"NATL1A",
"NATL2A",
"CCMP1375",
"MIT9211",
"MIT9303",
"MIT9313",
"GEYO",
"MIT9508",
"MITS9509",
"UW179A",
"CC9311",
"UW179B",
"WH8016",
"WH8020",
"CC9605",
"N19",
"N26",
"N32",
"N5",
"UW86",
"cyano3B",
"WH8109",
"WH8102",
"BL107",
"CC9902",
"RS9916",
"CC9616",
"KORDI100",
"WH7803",
"WH7805",
"RS9917",
"KORDI49",
"KORDI52",
"UW106",
"UW69",
"UW105",
"UW140",
"CB0101",
"PCC6307",
"PCC7001",
"RCC307"))
freqs4$colname


####this one is right???
level_order <- factor(freqs4$colname, level=c("KSe054HP",
"KSe055AR",
"KSe062NR",
"KSe063NB",
"KMa157AR",
"KMa159R8",
"KMa165NB",
"KOc293R8",
"KOc299NB",
"KFe387AR",
"KSe058SB",
"KMa156HP",
"KMa158CB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe386HP",
"KFe395NB",
"KFe389R8",
"KSe059R2",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KFe393NT",
"KFe416ST",
"KMa162ST",
"KMa163NT",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))


geom.text.size = 3
recruit=ggplot(freqs4, aes(x = level_order, y = sample_order, fill = value)) +
    geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 0)), size=geom.text.size) +
    scale_fill_gradient(low="#FBFEF9",high="#A63446") + scale_x_discrete(name="Metagenomes") + scale_y_discrete(name="Clade", position = "left") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, vjust = 1, 
hjust = -0.05, size=5)) + coord_flip()

recruit

ggsave('Metagenomes_Pro_Syn_Cyano.pdf', width=9.0, height=8)


```


```{r}



library(data.table)

readrecruit=read.csv("mean_coverage_Q2Q3_cyano.csv", header=TRUE, row.names = 1)
library(tidyverse)
library(reshape2)
library(dplyr)
library(stats)


woof=aggregate(.~ clade, data = readrecruit, FUN = "sum")
help(aggregate)

samp.with.rownames <- data.frame(woof[,-1], row.names=woof[,1])

#samp.with.rownames2=samp.with.rownames[,-1]

freqs2 <- apply(samp.with.rownames , 2, function(i) i/sum(i))

freqs3=freqs2*100
freqs3=as.data.frame(freqs3)

write.csv(freqs3,"test.csv")

freqs4<- freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(freqs4)




#freqs3=read.csv("SCG_normalized_all_depths.csv", row.name=1)


sample_order <- factor(freqs4$rowname,level=c("MED4",
"MIT9515",
"AS9601",
"MIT9301",
"SB",
"MIT9314",
"GP2",
"MIT9107",
"MIT9201",
"MIT9202",
"MIT9215",
"MIT9302",
"MIT9311",
"MIT9321",
"PAC1",
"NATL1A",
"NATL2A",
"CCMP1375",
"MIT9211",
"MIT9303",
"MIT9313",
"GEYO",
"MIT9508",
"MITS9509",
"UW179A",
"CC9311",
"UW179B",
"WH8016",
"WH8020",
"CC9605",
"N19",
"N26",
"N32",
"N5",
"UW86",
"cyano3B",
"WH8109",
"WH8102",
"BL107",
"CC9902",
"RS9916",
"CC9616",
"KORDI100",
"WH7803",
"WH7805",
"RS9917",
"KORDI49",
"KORDI52",
"UW106",
"UW69",
"UW105",
"UW140",
"CB0101",
"PCC6307",
"PCC7001",
"RCC307"))
freqs4$colname


####this one is right???
level_order <- factor(freqs4$colname, level=c("KSe054HP",
"KSe055AR",
"KSe062NR",
"KSe063NB",
"KOc293R8",
"KOc299NB",
"KSe058SB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KMa158CB",
"KMa156HP",
"KMa157AR",
"KMa159R8",
"KMa165NB",
"KFe387AR",
"KFe386HP",
"KFe389R8",
"KFe395NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KFe393NT",
"KFe416ST",
"KMa162ST",
"KMa163NT",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))


geom.text.size = 3
recruit=ggplot(freqs4, aes(x = level_order, y = rowname, fill = value)) +
    geom_tile(aes(fill = value)) + 
    geom_text(aes(label = round(value, 0)), size=geom.text.size) +
    scale_fill_gradient(low="#FBFEF9",high="#A63446") + scale_x_discrete(name="Metagenomes") + scale_y_discrete(name="Clade", position = "left") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, vjust = 1, 
hjust = -0.05, size=5)) + coord_flip()

recruit

ggsave('Metagenomes_Pro_Syn_Cyano_subclade.pdf', width=9.0, height=8)


```



```{r}


abun_diatoms=subset_taxa(rel2All_phyto, ASV=="85125408ea648cf8e6317b860cd2f1ac"| ASV=="c6ebc08d6495e327d01cbd509e32f0c3" | ASV=="2533d90c79f13829fa5d0dde39a7931e" | ASV=="f1749331353145c63288216e243cfbae" | ASV=="e5adae9e96a5065fb7a94a7db136332e" | ASV=="e3bad245be46508bf90676fdcc0ca26b" | ASV=="0b45d16cc99dcdc9b5bc12b637c5d971" | ASV== "9107af9e4ab0766c43fd15901c926f3d" | ASV=="0d2fef790750acb309a2cc43311050e1")
abun_diatoms=subset_taxa(abun_diatoms, phytorelabun_max_class=="abundant"| phytorelabun_max_class=="dominant")
ntaxa(abun_diatoms)
test=psmelt(abun_diatoms)
test=test%>%filter(phytorelabunMax > 0.5)



test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")



############################3

library(randomcoloR)
n <- 20
palette <- distinctColorPalette(n)


Three_t=subset(test, Cluster_DeSeq=="Cluster_3")
p_3<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_3

ggsave("abund_diatoms_cluster3.pdf", height=5, width=7)



Three_t=subset(test, Cluster_DeSeq=="Cluster_4")
p_4<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_4

ggsave("abund_diatoms_cluster4.pdf", height=5, width=7)

Three_t=subset(test, Cluster_DeSeq=="Cluster_5")
p_5<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_5

ggsave("abund_diatoms_cluster5.pdf", height=5, width=7)
```

```{r}



#fit 6 linear models 
pH_lin=lm(log(chla.ug.per.L)~pH+Site, data=Chladata)

```


```{r}

abun_mami=subset_taxa(rel2All_phyto, ASV=="300296cb89902c1721ff71f7cbc316b7"| ASV=="024dcc1ab6ba58bbb754448afd9c1127" | ASV=="b313fb940989d495b780f97175df05f5" | ASV=="c289fa66402d05b139a37184d8e4180e" | ASV=="bf0c8fa75326dd2c6ed5575c14c9cb20" | ASV=="406bb14741ded383380219611b8564d9" | ASV=="936685ee710e6e0ad190df2c0e863f75" | ASV== "47e0463d97f67bf6288a979f3ff70bfb" | ASV=="ba8a34138faff25a32e8fca2df7342d1")


ntaxa(abun_mami)
test=psmelt(abun_mami)




############################3

library(randomcoloR)
n <- 10
palette <- distinctColorPalette(n)


Three_t=subset(test, Cluster_DeSeq=="Cluster_3")
p_3<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_3

ggsave("abund_Mamiellophyceae_cluster3.pdf", height=5, width=7)



Three_t=subset(test, Cluster_DeSeq=="Cluster_4")
p_4<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_4

ggsave("abund_Mamiellophyceae_cluster4.pdf", height=5, width=7)

Three_t=subset(test, Cluster_DeSeq=="Cluster_5")
p_5<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_5

ggsave("abund_Mamiellophyceae_cluster5.pdf", height=5, width=7)
```


```{r}



abun_syn=subset_taxa(rel2All_phyto, Genus=="D_5__Synechococcus_CC9902")


ntaxa(abun_syn)
test=psmelt(abun_syn)




############################3

library(randomcoloR)
n <- 14
palette <- distinctColorPalette(n)


Three_t=subset(test, Cluster_DeSeq=="Cluster_3")
p_3<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_3

ggsave("abund_Syn_cluster3.pdf", height=5, width=7)



Three_t=subset(test, Cluster_DeSeq=="Cluster_4")
p_4<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_4

ggsave("abund_Syn_cluster4.pdf", height=5, width=7)

Three_t=subset(test, Cluster_DeSeq=="Cluster_5")
p_5<- ggplot(data=Three_t, aes(x=SampleID, y=Abundance, fill=ASV)) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x = element_text(angle = 90)) +  theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("HP1") + ylab("Relative abundance") +scale_fill_manual(values=palette)
p_5

ggsave("abund_Syn_cluster5.pdf", height=5, width=7)
```









```{r}
# library
library(ggridges)
library(ggplot2)
library(viridis)
install.packages("hrbrthemes")
library(hrbrthemes)
hrbrthemes::import_roboto_condensed()



view(lincoln_weather)
# Plot
ggplot(temp_unique2, aes(x = `SW.Temperature.at.site`, y = `CuratedASV`, stat = "Abundance")) +
  geom_density_ridges_gradient(scale = 3, rel_min_height = 0.01) +
  scale_fill_viridis(name = "Temp. [F]", option = "C") +
  labs(title = 'Temperatures in Lincoln NE in 2016') +
  theme_ipsum() +
    theme(
      legend.position="none",
      panel.spacing = unit(0.1, "lines"),
      strip.text.x = element_text(size = 8)
    )



temp_unique=subset_taxa(All_phyto, Temp_sig=="sig")

p0.rel <- transform(temp_unique, "compositional")

temp_unique2=psmelt(p0.rel)


names(temp_unique2)

ggplot(temp_unique2, aes(x=Abundance, y=SW.Temperature.at.site, color=Curated_v3, shape=phytorelabun_max_class))+geom_point()+facet_wrap(~CuratedASV, scales="free_x")

ggsave("unique_temp_spec30.pdf", width=15, height=15)








ggplot(temp_unique2, aes(y=ASV, x=SW.Temperature.at.site, height = density, group = ASV)) + geom_ridgeline()



g=subset(temp_unique2, CuratedASV=="ASV159")




ggplot(f, aes(x=SW.Temperature.at.site)) + 
  geom_density()

ggplot(g, aes(x=SW.Temperature.at.site)) + 
  geom_density()


ggplot(g, aes(x=SW.Temperature.at.site,y=Abundance)) +
  geom_density(stat="identity", fill=NA) +
  theme(legend.position=c(.2,0.8))


library(splines)

ggplot(g, aes(x=SW.Temperature.at.site,y=Abundance)) +
    geom_bar(stat="identity",position=position_dodge()) +
    geom_smooth(aes(y=Abundance), se=F,
                method="glm",
                formula=y~ns(x,8),
                family=gaussian(link="log"),
                show_guide = FALSE,lwd=0.7) +
    theme(legend.position=c(.2,0.8))

ggplot(no_dup_metadata_KBytNERR, aes(x=decimal_dat, y=wind.max))+geom_point()

ggplot(no_dup_metadata_KBytNERR, aes(x=Season_Aka, y=Photosynthetic.radiaion))+geom_boxplot()

```




```{r}
# Value used to transform the data
coeff <- 2

# A few constants
temperatureColor <- "#69b3a2"
priceColor <- rgb(0.2, 0.6, 0.9, 1)

ggplot(g, aes(x=SW.Temperature.at.site)) + 
  geom_density()+ 
  geom_point( aes(y=Abundance / coeff), size=2, color=priceColor) +
  
  scale_y_continuous(
    
    # Features of the first axis
    name = "Density Temperature (Celsius °)",
    
    # Add a second axis and specify its features
    sec.axis = sec_axis(~.*coeff, name="Relative Abundance")
  ) + 
  
  theme_ipsum() +

  theme(
    axis.title.y = element_text(color = temperatureColor, size=13),
    axis.title.y.right = element_text(color = priceColor, size=13)
  ) +

  ggtitle("Temperature down, price up")
```


```{r}
library(ggplot2)
install.packages("ggExtra")
library(ggExtra)
data(mpg, package="ggplot2")
# mpg <- read.csv("http://goo.gl/uEeRGu")





CLR_Phyto <- transform(All_phyto, "clr")

temp_unique_clr=subset_taxa(CLR_Phyto, Salinity_sig=="sig")

temp_unique2_clr=psmelt(temp_unique_clr)


temp_unique2_clr$Curated_v3=factor(temp_unique2_clr$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))


k=ggplot(temp_unique2_clr, aes(x=Salinity, y=Abundance, color=Curated_v3, shape=phytorelabun_max_class))+geom_point()+facet_wrap(~CuratedASV, scales="free_y")+scale_color_manual(values=coloresbarplot)+ylab("CLR Abunandance")
k
#+scale_x_continuous(trans = log10_trans())

ggsave("CLR_salinity_spec.pdf", width=10,height=10)





```

```{r}

g=subset(temp_unique2, CuratedASV=="ASV201")

# Scatterplot
h <- ggplot(g, aes(SW.Temperature.at.site, Abundance)) + 
  geom_point() + 
  geom_smooth(method="lm", se=F)+ylab("Log-ratio Abundnace")+xlab("Seawater Temperature (Celsius °)")
h
ggMarginal(h, type = "histogram", fill="transparent")


hot=no_dup_metadata_KBytNERR%>%filter(SW.Temperature.at.site >28)
hist(hot$SW.Temperature.at.site)

```





```{r}

library(corncob)

set.seed(1)

sam_data(All_phyto)

Coastal_All_phyto=subset_samples(All_phyto, Cluster_DeSeq_Spatial=="Coastal")

Coastal_All_phyto_noNA=subset_samples(Coastal_All_phyto, Silicate!="NA")
nsamples(Coastal_All_phyto_noNA)



da_analysis=differentialTest(formula=~ SW.Temperature.at.site,
                             phi.formula = ~1,
                             formula_null = ~1,
                             phi.formula_null = ~1,
                             test="Wald", boot=FALSE, data=Coastal_All_phyto, fdr_cutoff= 0.05)

table(da_analysis$significant_models)


pdf("Temp_analysis.pdf", height=15)
plot(da_analysis, level=c("Genus", "CuratedASV"))
dev.off()

?corncob::differentialTest()

install.packages("limma")
library(limma)
# phyloseq example
data(soil_phylum_small)
da_analysis <- contrastsTest(formula = ~ DayAmdmt,
                             phi.formula = ~ DayAmdmt,
                             contrasts_DA = list("DayAmdmt21 - DayAmdmt11",
                                                 "DayAmdmt22 - DayAmdmt21"),
                             data = soil_phylum_small,
                             fdr_cutoff = 0.05)

da_analysis
da_analysis$all_models





obtain_coeffs <- function(obj){
  data.frame(plot(obj, data_only = T) ,
             ASV = obj$significant_taxa)
}

  # Only strong coefficients
  # filter(abs(estimate) > 0.01)  %>% 
  # Only estimates with the confidence interval without taking the zero value
  # filter(sign(upper) == sign(lower))


coefs.all <- obtain_coeffs(da_analysis) %>% 
  left_join(tax, by = 'ASV')  %>% 
  rename( 'estimate' = x, 
          'upper' = xmax,
          'lower' = xmin,
          'par' = variable) %>% 
  mutate(par = str_replace(par,
                           '\nDifferential Abundance', ''))  %>% 
  # Only strong coefficients
  filter(abs(estimate) > 0.01)  %>% 
  # Only estimates with the confidence interval without taking the zero value
  filter(sign(upper) == sign(lower)) %>% 
  as_tibble()

write.csv(coefs.all, "coef_corncob_temp.csv")

coefs.all.sort$CuratedASV <- factor(coefs.all.sort$CuratedASV, levels = coefs.all.sort$CuratedASV)

coefs.all$Curated_v3=factor(coefs.all$Curated_v3, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Bathycoccus",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Pedinophyceae",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptomonadales",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiales",
"Stramenopiles:Diatoms",
"Stramenopiles:Dictyochophyceae",
"Stramenopiles:Pelagophyceae",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))
coefs.all$Curated_v3


coefs.all.sort=coefs.all %>% 
  arrange(Curated_v3)


ggplot(coefs.all.sort, aes(x=CuratedASV, xend=CuratedASV, color=Curated_v3))+
  geom_point(aes(x = CuratedASV, y = estimate)) +
  geom_segment(aes(y = lower, yend = upper)) +
  ylab(paste("")) +
  theme_bw() +
  theme(axis.text.x = element_text())+xlab("")+  theme(legend.position="none") +coord_flip()+scale_color_manual(values=coloresbarplot)
ggsave("temp_ASVs.pdf")


#coefficients for first significant model
diftest_coefs[1] 

#can also subset using a character vector because it's a named list


diftest_coefs$`44095b9d0c379f3ca649b693174bbdfc`


extractCoefficients <- function(diftest_res, taxa){
  
#add names to all_models list
  names(diftest_res$all_models) <- taxa_names(diftest_res$data)
  
  # default is all taxa
  if (missing(taxa)) {
    taxa <- taxa_names(diftest_res$data)
  } else {
    taxa <- taxa
  }
  
  purrr::map(
    diftest_res$all_models[taxa],
    ~ stats::coef(.)
  )
  
}


diftest_coefs <- extractCoefficients(da_analysis, da_analysis$significant_taxa)



```

```{r}
kk=lm(HBACT.mL~SW.Temperature.at.site, coastal_envi)
summary(kk)
plot(kk)

?glm()

kk=lm(Fuco~SW.Temperature.at.site, coastal_envi)
summary(kk)
plot(kk)

plot(HBACT.mL~SW.Temperature.at.site, coastal_envi)


```



