---
title: "Miseq Sep 2021 Qiime 2.4"
output: html_notebook
---



```{bash}

###idk why but the data came back to us not named like the typical casssava files 

cd /tank/tucker_data/2021_MiSeqFinal/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC-updated/BaseCalls/


for f in *R2_001.fastq.gz; do mv $f ${f%*R2_001.fastq.gz}L001_R2_001.fastq.gz; done

for f in *R1_001.fastq.gz; do mv $f ${f%*R1_001.fastq.gz}L001_R1_001.fastq.gz; done


```




```{python}

screen -S Miseq 

source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4

qiime tools import --show-importable-types

cd /tank/tucker_data/MiSeq/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC/BaseCalls



qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path /tank/tucker_data/MiSeq/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC/BaseCalls \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end.qza 
  
  

```


**(2) Generate a summary to evaluate the data, it will help determine how to truncate data in a future step.**

```{bash, eval=FALSE}
  
qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end.qza  \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/demux_AllSeqs.qzv

```



**(3) Use Cutadapt to remove primers**

https://forum.qiime2.org/t/trimming-primers-from-both-ends-of-paired-end-reads/9984
https://docs.qiime2.org/2017.12/plugins/available/cutadapt/demux-paired/


```{bash}

qiime cutadapt trim-paired \
    --i-demultiplexed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end-trimmed.qza \
    --p-cores 5\
    --verbose

###i dont think anything was trimmed really...

qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end-trimmed.qza  \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end-trimmed.qzv


```




###Processing data: DADA2 

####Sequence quality control and feature table construction with 

*Note: we did not have to do the demultiplexing step as OSU did that prior to sending us the files, while this is outlined in the MP tutorial, I have left it out and moved to the next section. I have also outlined using DADA2 (option 1 in the MP tutorial)*

* After viewing the demux_AllSeqs.qzv file, I decided not trim very much. Mike and I had a conversation about this because i am looking for phylogenetic signal having the longet sequences possible 

```{bash, eval=FALSE}


  qiime dada2 denoise-paired \
  --i-demultiplexed-seqs /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end-trimmed.qza \
  --p-trunc-len-f 250 \
  --p-trunc-len-r 250 \
  --p-trim-left-f 0 \
  --p-trim-left-r 0 \
  --o-representative-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/rep-seqs-dada2_AllSeqs.qza \
  --o-table /tank/tucker_data/MiSeq_Analysis_Sep082021/table-dada2_AllSeqs.qza \
  --o-denoising-stats /tank/tucker_data/MiSeq_Analysis_Sep082021/stats-dada2_AllSeqs.qza \
  --p-n-threads 15
  


qiime metadata tabulate \
  --m-input-file stats-dada2_AllSeqs.qza \
  --o-visualization stats-dada2_AllSeqs.qzv



```




```{bash}

###uggghh lets go for single reads 

qiime dada2 denoise-single \
  --i-demultiplexed-seqs demux-paired-end-trimmed.qza \
  --p-trunc-len 250 \
  --o-representative-sequences single_rep-seqs-dada2_AllSeqs.qza \
  --o-table single_table-dada2_AllSeqs.qza \
  --o-denoising-stats single_stats-dada2_AllSeqs.qza \
  --p-n-threads 20
  
  

qiime metadata tabulate \
  --m-input-file single_stats-dada2_AllSeqs.qza \
  --o-visualization single_stats-dada2_AllSeqs.qzv
  
  
qiime feature-table summarize \
  --i-table single_table-dada2_AllSeqs.qza \
  --o-visualization single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file sample_metadata.tsv
  
  
qiime feature-table tabulate-seqs \
  --i-data single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization single_rep-seqs-dada2_AllSeqs.qzv
  
```



```{bash}

##insert training bit here##

###in the future lets not truncate the reads! 



qiime feature-classifier extract-reads \
  --i-sequences silva-138-99-seqs.qza \
  --p-f-primer GTGCCAGCMGCCGCGGTAA \
  --p-r-primer CCGYCAATTYMTTTRAGTTT \
  --p-trunc-len 250 \
  --p-min-length 100 \
  --p-max-length 500 \
  --o-reads ref-seqs.qza

####The --p-trunc-len parameter should only be used to trim reference sequences if query sequences are trimmed to this same length or shorter. Paired-end sequences that successfully join will typically be variable in length. Single-end reads that are not truncated at a specific length may also be variable in length. For classification of paired-end reads and untrimmed single-end reads, we recommend training a classifier on sequences that have been extracted at the appropriate primer sites, but are not trimmed.

qiime feature-classifier fit-classifier-naive-bayes \
  --i-reference-reads ref-seqs.qza \
  --i-reference-taxonomy silva-138-99-tax.qza \
  --o-classifier classifier.qza



```




```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier classifier.qza \
  --i-reads single_rep-seqs-dada2_AllSeqs.qza \
  --o-classification single_read_taxonomy.qza

qiime metadata tabulate \
  --m-input-file single_read_taxonomy.qza \
  --o-visualization single_read_taxonomy.qzv
  

qiime taxa barplot --i-table single_table-dada2_AllSeqs.qza --i-taxonomy single_read_taxonomy.qza --o-visualization single_taxa-bar-plots.qzv
  
```


```{bash}

###renaming sample IDS

qiime feature-table group \
  --i-table single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file sample_metadata.tsv \
  --m-metadata-column blah \
  --p-mode sum \
  --o-grouped-table New_table_MiSeq2021.qza



qiime feature-table summarize \
  --i-table New_table_MiSeq2021.qza \
  --o-visualization New_table_Kbyt.qzv \
  --m-sample-metadata-file New_sample_metadata.tsv

qiime taxa barplot --i-table New_table_MiSeq2021.qza --i-taxonomy single_read_taxonomy.qza --o-visualization single_taxa_updated_metadata-bar-plots.qzv


```



```{bash}
##get rid of samples that were poorly sequenced I got rid of all samples under 1,000 quality control reads. This included all the negatives, but also a number of JdFR samples, 3 KByT samples (NBJu0836, R2De0742, R2Ap0782), and a mangrove sample (Mg020bot). It looks like people are not recommending repeating the denoising step and instead recommending just a filtering base on sample: https://forum.qiime2.org/t/remove-blank-samples/9119



qiime feature-table filter-samples \
  --i-table New_table_MiSeq2021.qza \
  --m-metadata-file New_stats_metadata.tsv \
  --p-where "[seq.quality]='keep'" \
  --o-filtered-table good-quality-filtered-table.qza

qiime taxa barplot --i-table good-quality-filtered-table.qza --i-taxonomy single_read_taxonomy.qza --o-visualization single_taxa_updated_metadata-bar-plots_filtered.qzv


#https://docs.qiime2.org/2021.4/tutorials/filtering/




qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences single_rep-seqs-dada2_AllSeqs.qza \
  --o-alignment single_aligned-rep-seqs.qza \
  --o-masked-alignment masked-single_aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza


#Saved FeatureData[AlignedSequence] to: single_aligned-rep-seqs.qza
#Saved FeatureData[AlignedSequence] to: masked-single_aligned-rep-seqs.qza
#Saved Phylogeny[Unrooted] to: unrooted-tree.qza
#Saved Phylogeny[Rooted] to: rooted-tree.qza


##more prep
qiime tools export \
  --input-path unrooted-tree.qza \
  --output-path exported-tree

qiime tools export \
  --input-path good-quality-filtered-table.qza \
  --output-path exported-good-quality-filtered-feature-table

cd exported-good-quality-filtered-feature-table
biom convert -i feature-table.biom -o feature-table.tsv --to-tsv


qiime tools export \
  --input-path unrooted-tree.qza \
  --output-path exported-tree

qiime tools export \
  --input-path single_read_taxonomy.qza \
  --output-path exported-taxonomy
 
 qiime tools export \
  --input-path single_aligned-rep-seqs.qza \
  --output-path exported-rep-seqs
 
  
 qiime tools export \
  --input-path masked-single_aligned-rep-seqs.qza \
  --output-path exported-masked-rep-seqs


single_read_taxonomy.qza

```


```{bash}



qiime feature-table filter-samples \
  --i-table good-quality-filtered-table.qza\
  --m-metadata-file just_KByT_metadata.tsv \
  --p-where "[Sample.Type]='KByT'" \
  --o-filtered-table good-quality-KByT.qza

qiime taxa barplot --i-table good-quality-KByT.qza --i-taxonomy single_read_taxonomy.qza --o-visualization good-quality-KByT-bar-plots_filtered.qzv


qiime tools export \
  --input-path good-quality-KByT.qza \
  --output-path exported-good-quality-KByT-feature-table

cd exported-good-quality-KByT-feature-table
biom convert -i feature-table.biom -o KByT-MiSeq-table.tsv --to-tsv


```



###combining runs 
“collapseNoMismatch”, that “ collapse together variants with no mismatches or internal indels but that could differ by terminal gaps, i.e. variants that differ by their lengths and nothing else”. 
https://forum.qiime2.org/t/dada2-collapsenomismatch/12479/7
https://forum.qiime2.org/t/formatting-a-qiime2-feature-table-for-the-r-version-of-dada2/13387
#https://forum.qiime2.org/t/merging-data-from-different-runs-plates-in-manifest-step-or-after/18985






```{bash}


qiime feature-table filter-samples \
  --i-table good-quality-filtered-table.qza\
  --m-metadata-file Mocks_only_metadata_v2.tsv\
  --p-where "[Sample.Type]='Mock'" \
  --o-filtered-table good-quality-Mock.qza

qiime taxa barplot --i-table good-quality-Mock.qza --i-taxonomy single_read_taxonomy.qza --o-visualization good-quality-Mock-bar-plots_filtered.qzv


qiime tools export \
  --input-path good-quality-Mock.qza \
  --output-path exported-good-quality-Mock-feature-table

cd exported-good-quality-Mock-feature-table
biom convert -i feature-table.biom -o MiSeq2021-Mock-table.tsv --to-tsv

```


Filtering!!

```{bash}

## do after joining 

#This filter can be applied to the feature axis to remove low abundance features from a table. For example, you can remove all features with a total abundance (summed across all samples) of less than 10 as follows.


qiime feature-table filter-features \
  --i-table table.qza \
  --p-min-frequency 10 \
  --o-filtered-table feature-frequency-filtered-table.qza
  
  
#This filtering is commonly used for filtering features that show up in only one or a few samples, based on the suspicion that these may not represent real biological diversity but rather PCR or sequencing errors (such as PCR chimeras). Features that are present in only a single sample could be filtered from a feature table as follows.

  
qiime feature-table filter-features \
  --i-table table.qza \
  --p-min-samples 2 \
  --o-filtered-table sample-contingency-filtered-table.qza

```





```{r}

info=read.csv("making_shiz.csv",header=TRUE)
seq=read.csv("sent_to_oregon.csv",header=TRUE)

library(dplyr)
?left_join()
tog=left_join(seq, info, by="barcode")

write.csv(tog, "Metadata_MiSeq2021.csv")

write.table(tog, file='test.tsv', quote=FALSE, sep='\t', col.names = NA)


stats=read.csv("stats_single_dada2_all.csv",header=TRUE)

meta=read.csv("sample_metadata.csv",header=TRUE)

tog=left_join(meta, stats, by="sampleID")


write.table(tog, file='stats_metadata.tsv', quote=FALSE, sep='\t', col.names = NA)




stats=read.csv("stats_denoising_251-2021.csv",header=TRUE)

meta=read.csv("/Users/sarahtucker/Documents/KByt/Miseq_2021_Prep/KByT_MiSeq_Analysis_092021/Trim251-redo2021/New_sample_metadata.csv", header=TRUE)

tog=left_join(meta, stats, by="other")


write.table(tog, file='stats_denoising_Redo_251-2021.tsv', quote=FALSE, sep='\t', col.names = NA)




```



```{r}


# load libraries
library("ggplot2") 
library("phyloseq") 
library(ape)
library("knitr")
library(dada2)
library(methods)
library(plotly)
library(tidyverse)
library(ggpubr)

###this is from the R tutorial document"https://github.com/joey711/phyloseq/files/1795523/QIIME2_to_Phyloseq.pdf"


#setwd("/Users/sarahtucker/Documents/Dissertation/chapter 1/miseq_run_82019") 

#getwd()


###now that you are in the right format you can bring things in as phyloseq objects


# read in otu table
otu_table_Mock = read.csv("MiSeq2021-Mock-table.csv", sep=",", row.names=1) 
otu_table_Mock = as.matrix(otu_table_Mock)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

taxonomy_2021 = read.csv("taxonomy-table-Miseq2021.csv", sep=",", row.names=1) 
taxonomy_2021 = as.matrix(taxonomy_2021)



#read in metadata
#make sure when you are saving from excel that you use a tab delineated format
####also see my R file making_metadata.Rmd in my Chapter_2 folder for a quick and easy way to integrate genetic metadata with environmental!
##except I did need to remember add metadata for R2 samples (R857R2, SB58R2, NT61R2)
##i put in an average of the month before and month after for FCM data of SBoC0294 
metadata_Mock= read.csv("Mocks_only_metadata.csv", header=TRUE, fill=TRUE)

rownames(metadata_Mock) <- metadata_Mock$SampleID
metadata_Mock$SampleID

#read in tree
phy_tree = read_tree("MiSeq_2021_tree.nwk")

# import as phyloseq objects
OTU_MOCK = otu_table(otu_table_Mock, taxa_are_rows = TRUE) 
TAX_2021= tax_table(taxonomy_2021)
#check that the names are correct 
#rank_names(TAX3)
#ncol(taxonomy_phylo3)
#ncol(TAX3)

#bring in your metadata 

META_MOCK = sample_data(metadata_Mock)
sample_names(META_MOCK)
#colnames(META)

#first make sure your sample names match 
#sample_names(META)
#sample_names(OTU)

#now create your phyloseq object
#not sure why, but if you make 2 objects and add the metadata on the merge step it keeps the column names 

physeq_Mock = phyloseq(OTU_MOCK, TAX_2021, phy_tree)

physeq_Mock2=merge_phyloseq(physeq_Mock, META_MOCK)
ntaxa(physeq_Mock)


```


#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-01-Answers.html

```{r}
write.csv(otu_table(physeq_Mock), "physeq_Mock_otu.csv")
write.csv(tax_table(physeq_Mock), "physeq_Mock_tax.csv")


physeq_Mock2 = prune_taxa(taxa_sums(physeq_Mock)>10, physeq_Mock)
ntaxa(physeq_Mock2)

physeq_Mock2_rel  = transform_sample_counts(physeq_Mock2, function(x) x / sum(x) )

write.csv(otu_table(physeq_Mock2_rel), "physeq_Mock_rel_otu2.csv")
write.csv(tax_table(physeq_Mock2), "physeq_Mock_tax2.csv")


```


```{r}

Mock_2021 = read.csv("mock_analysis_2021.csv", sep=",", row.names=1) 
library(dplyr)
library(plotly)
library(scales)
library(ggplot2)

getwd()

Mockdata <- read.csv('mock_analysis_2021.csv')


Everything = subset(Mockdata, select= c("Sample", "MGII", "Thaumarchaeota","OCS115","Flavo", "Prochlorococcus", "MGA", "Plancto", "SAR11", "SAR86", "AEGEAN169", "SAR116", "SAR202","Porticoccaceae", "Oceanospirillales", "Puniceicoccaceae"))




xform <- list(font = list(
    family = "sans-serif",
    size = 16),
  categoryorder = "array",
              categoryarray = c("Staggered_Expected","MoSTOrg2","MoSTOrg1","MoST30c1","MoST25c1","STOrig_SE_2019","Even_Expected","MoEvOrg2","MoEV20c1", "EVOrig_SE_2019"
))







data.table::melt(Everything, id.vars='Sample') %>%
plot_ly(x = ~Sample, y = ~value, type = 'bar', 
                name = ~variable, color = ~variable) %>%
      layout(yaxis = list(title = 'Count'), xaxis=xform, title= "Mock Communities Comparison Expected to 2021 MiSeq Run ", barmode = 'stack')





```



##############redoing all of 2019
#This is technically already how DADA2 works. It makes the error model using all the samples from the same run, then uses that model to denoise each read separately and independently, this is why it is nicely parallelizable.


```{python}

screen -S Miseq 

source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4

qiime tools import --show-importable-types

cd /tank/tucker_data/Base_Calls_manuscript



qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path /tank/tucker_data/Base_Calls_manuscript \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza 
  
  

```


**(2) Generate a summary to evaluate the data, it will help determine how to truncate data in a future step.**

```{bash, eval=FALSE}
  
qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qzv

```



**(3) Use Cutadapt to remove primers**

https://forum.qiime2.org/t/trimming-primers-from-both-ends-of-paired-end-reads/9984
https://docs.qiime2.org/2017.12/plugins/available/cutadapt/demux-paired/


```{bash}

qiime cutadapt trim-paired \
    --i-demultiplexed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza \
    --p-cores 5\
    --verbose

###i dont think anything was trimmed really...

qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza  \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qzv


```




```{bash}

###uggghh lets go for single reads 

qiime dada2 denoise-single \
  --i-demultiplexed-seqs /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 20
  
  

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qzv
  
  
qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv
  
  
qiime feature-table tabulate-seqs \
  --i-data single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization single_rep-seqs-dada2_AllSeqs.qzv
  
```




```{bash}

##insert training bit here##

###in the future lets not truncate the reads! 



qiime feature-classifier extract-reads \
  --i-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/silva-138-99-seqs.qza \
  --p-f-primer GTGCCAGCMGCCGCGGTAA \
  --p-r-primer CCGYCAATTYMTTTRAGTTT \
  --p-trunc-len 251 \
  --p-min-length 100 \
  --p-max-length 500 \
  --o-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-ref-seqs.qza

####The --p-trunc-len parameter should only be used to trim reference sequences if query sequences are trimmed to this same length or shorter. Paired-end sequences that successfully join will typically be variable in length. Single-end reads that are not truncated at a specific length may also be variable in length. For classification of paired-end reads and untrimmed single-end reads, we recommend training a classifier on sequences that have been extracted at the appropriate primer sites, but are not trimmed.

qiime feature-classifier fit-classifier-naive-bayes \
  --i-reference-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-ref-seqs.qza \
  --i-reference-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/silva-138-99-tax.qza \
  --o-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza



```



```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza \
  --i-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-classification /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qzv
  

qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_taxa-bar-plots.qzv
  
```


```{bash}

###renaming sample IDS

qiime feature-table group \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv \
  --m-metadata-column Seq \
  --p-mode sum \
  --o-grouped-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza

###start here for both

qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/New_Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv

qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_taxa-bar-plots.qzv


```


###Redo 2021 with 251




```{bash}

###uggghh lets go for single reads 

qiime dada2 denoise-single \
  --i-demultiplexed-seqs /tank/tucker_data/MiSeq_Analysis_Sep082021/demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 20
  
  

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_stats-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_stats-dada2_AllSeqs.qzv
  
  
qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/sample_metadata.tsv
  
  
qiime feature-table tabulate-seqs \
  --i-data single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization single_rep-seqs-dada2_AllSeqs.qzv
  
```




```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza \
  --i-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-classification /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qza

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qzv
  

qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_taxa-bar-plots.qzv
  
```





```{bash}

###renaming sample IDS

qiime feature-table group \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/sample_metadata.tsv \
  --m-metadata-column blah \
  --p-mode sum \
  --o-grouped-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_table-dada2_AllSeqs.qza



qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization  /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/New_sample_metadata.tsv

qiime taxa barplot --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_taxa-bar-plots.qzv


```



```{bash}
##get rid of samples that were poorly sequenced I got rid of all samples under 1,000 quality control reads. This included all the negatives, but also a number of JdFR samples, 3 KByT samples (NBJu0836, R2De0742, R2Ap0782), and a mangrove sample (Mg020bot). It looks like people are not recommending repeating the denoising step and instead recommending just a filtering base on sample: https://forum.qiime2.org/t/remove-blank-samples/9119



qiime feature-table filter-samples \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Redo_251_2021_wstats_metadata.tsv \
  --p-where "[keep.status]='keep'" \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Working-Samples-2021-251-filtered-table.qza
  



######have just KByT samples 

qiime feature-table filter-samples \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Working-Samples-2021-251-filtered-table.qza \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/Redo_251_2021_wstats_metadata.tsv \
  --p-where "[Sample.Type]='KByT'" \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/KByT-2021-251-filtered-table.qza
  


qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/KByT-2021-251-filtered-table.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_read_taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/KByT-Renamed-251-2021-single_taxa-bar-plots.qzv




#######Time to merge!! 
qiime feature-table merge \
  --i-tables /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/KByT-2021-251-filtered-table.qza \
  --i-tables /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-merged-table /tank/tucker_data/MiSeq_Analysis_Sep082021/merged_2021_2019_table.qza


qiime feature-table merge-seqs \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-merged-data /tank/tucker_data/MiSeq_Analysis_Sep082021/merged_2021_2019-rep-seqs.qza


qiime feature-table summarize \
  --i-table table.qza \
  --o-visualization table.qzv \
  --m-sample-metadata-file sample-metadata.tsv


qiime feature-table tabulate-seqs \
  --i-data rep-seqs.qza \
  --o-visualization rep-seqs.qzv






#This filter can be applied to the feature axis to remove low abundance features from a table. For example, you can remove all features with a total abundance (summed across all samples) of less than 10 as follows.


qiime feature-table filter-features \
  --i-table table.qza \
  --p-min-frequency 10 \
  --o-filtered-table feature-frequency-filtered-table.qza
  
  
#This filtering is commonly used for filtering features that show up in only one or a few samples, based on the suspicion that these may not represent real biological diversity but rather PCR or sequencing errors (such as PCR chimeras). Features that are present in only a single sample could be filtered from a feature table as follows.

  
qiime feature-table filter-features \
  --i-table table.qza \
  --p-min-samples 2 \
  --o-filtered-table sample-contingency-filtered-table.qza




#https://docs.qiime2.org/2021.4/tutorials/filtering/




qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences single_rep-seqs-dada2_AllSeqs.qza \
  --o-alignment single_aligned-rep-seqs.qza \
  --o-masked-alignment masked-single_aligned-rep-seqs.qza \
  --o-tree unrooted-tree.qza \
  --o-rooted-tree rooted-tree.qza


#Saved FeatureData[AlignedSequence] to: single_aligned-rep-seqs.qza
#Saved FeatureData[AlignedSequence] to: masked-single_aligned-rep-seqs.qza
#Saved Phylogeny[Unrooted] to: unrooted-tree.qza
#Saved Phylogeny[Rooted] to: rooted-tree.qza


##more prep
qiime tools export \
  --input-path unrooted-tree.qza \
  --output-path exported-tree

qiime tools export \
  --input-path good-quality-filtered-table.qza \
  --output-path exported-good-quality-filtered-feature-table

cd exported-good-quality-filtered-feature-table
biom convert -i feature-table.biom -o feature-table.tsv --to-tsv


qiime tools export \
  --input-path unrooted-tree.qza \
  --output-path exported-tree

qiime tools export \
  --input-path single_read_taxonomy.qza \
  --output-path exported-taxonomy
 
 qiime tools export \
  --input-path single_aligned-rep-seqs.qza \
  --output-path exported-rep-seqs
 
  
 qiime tools export \
  --input-path masked-single_aligned-rep-seqs.qza \
  --output-path exported-masked-rep-seqs


single_read_taxonomy.qza

```


```{bash}



qiime tools export \
  --input-path good-quality-KByT.qza \
  --output-path exported-good-quality-KByT-feature-table

cd exported-good-quality-KByT-feature-table
biom convert -i feature-table.biom -o KByT-MiSeq-table.tsv --to-tsv


```




```{r}
ugh <- read.csv('together_metadata_2019_2021.csv')

info <- read.csv('/Users/sarahtucker/Documents/Dissertation/chapter 2/metadata/metadata092321.csv')


maybe=left_join(ugh, info, by="Universal_Sample_ID")

write.csv(maybe, "new_joined_metadata_2019_2021.csv")



ugh <- read.csv('barcode_2021.csv')
ugh2 <- read.csv('universal_2021.csv')

maybe=left_join(ugh2, ugh, by="sampleID")

write.csv(maybe, "better_2021.csv")



okay=read.csv("better_2021_2019.csv")
info <- read.csv('/Users/sarahtucker/Documents/Dissertation/chapter 2/metadata/metadata092321.csv')
maybe=left_join(okay,info, by="Universal_Sample_ID")

write.csv(maybe, "together_metadata_2019_2021.csv")




okay=read.csv("together_metadata_2019_2021.csv")
info <- read.csv('read_num.csv')
maybe=left_join(okay,info, by="Seq2")

write.csv(maybe, "together_metadata_2019_2021_v2.csv")


```







