---
title: "Code_LO_Submission"
author: "Sarah Tucker"
date: "4/29/2024"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(echo = TRUE)
```


These analyses were conducted in various packages, but most predominately the Phyloseq package (McMurdie & Holmes, 2013). The code within represents analyses to create summary statistics and all figures included within the manuscript. 




```{r}

rm(list=ls())
# load libraries
library("ggplot2") 
library("phyloseq") 
library(tidyverse)
library(ggpubr)
library(reshape2)
library(DESeq2)
library(lomb)
library(DivNet)
library(breakaway)
library(seecolor)
library(microbiome)
library(dendextend)
library(scales)
library(svglite)
library(cowplot)
library("pheatmap")
library(viridis)
library(geojsonio)
require(sf)
library(MBA)
library(lubridate)
library(mgcv)
library(corrplot)
library(fdrtool)
library(multcomp)

```

#########
Current phyloseq objects
##########

```{r}
#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# read in otu table
no_dup_otu_table_KBytNERR = read.csv("otu_final_tucker.csv", sep=",", row.names=1) 
no_dup_otu_table_KBytNERR = as.matrix(no_dup_otu_table_KBytNERR)

# import as phyloseq objects
no_dup_OTU_KBytNERR = otu_table(no_dup_otu_table_KBytNERR, taxa_are_rows = TRUE) 

# read in tax table
taxonomy_Phyto = read.csv("taxonomy_table_clean_May2024.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)



no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)


rownames(no_dup_metadata_KBytNERR) <- no_dup_metadata_KBytNERR$SampleID
no_dup_metadata_KBytNERR$SampleID
META_no_dup_KBytNERR = sample_data(no_dup_metadata_KBytNERR)
nsamples(META_no_dup_KBytNERR)



physeq_Dav_no_dup = phyloseq(no_dup_OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav_no_dup=merge_phyloseq(physeq_Dav_no_dup, META_no_dup_KBytNERR)
ntaxa(physeq_Dav_no_dup)
#4117
```





```{r}

##Removing unwanted sequences
table(tax_table(physeq_Dav_no_dup)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean <- subset_taxa(physeq_Dav_no_dup, !is.na(Domain) & !Domain %in% c("", "unclassified_Root"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean)
ntaxa(physeq_Dav_no_dup_clean)
table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#check to see all of the uncharacterized phylum
table(tax_table(physeq_Dav_no_dup_clean)[, "Phylum"], exclude = NULL)
##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean_no18S <- subset_taxa(physeq_Dav_no_dup_clean, !is.na(Domain) & !Domain %in% c("", "Eukaryota"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean_no18S)
#366
ntaxa(physeq_Dav_no_dup_clean_no18S)
#3680
table(tax_table(physeq_Dav_no_dup_clean_no18S)[, "Domain"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))
ntaxa(rel.abun)
#3680
nsamples(rel.abun)
#366


All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto)
nsamples(All_phyto)

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))

All_phyto_Rel = subset_taxa(rel.abun,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto_Rel)
table(tax_table(All_phyto)[, "Phylum"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)
#3680
nsamples(rel.abun)
#366


```



######
Figure 1
######


```{r}

colorssite2 <- c("WAI2"='#000000', "KAHO"='#f94279',"AR"="#967969" ,"HP1"="#ffd000", "NB"="#ff8c00","SB"="#C47F2B","CB"="#ad0201ff","SR8"="#D7B4F3", "SR2"="#AAFF00", "NR2"="#049861", "STO1"='#00FFFF', "NTO1"='#0045ff', "ALOHA"="#7d7d7d")
print_color(colorssite2)
library(seecolor)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Nearshore"="#B70D6A")
print_color(colorspatial)
library(ggplot2)


KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")
aloha=subset(KBaloha, Site=="ALOHA")
summary(aloha$chla.ug.per.L)


blueberry=factor(KBaloha$Site,levels=c("WAI2", "KAHO", "HP1","SB","SR8","CB", "AR", "NB", "SR2", "NR2", "STO1", "NTO1", "ALOHA"))

svg("Silicate_Box_Site_v2.svg", height=2, width=6)
Silicate<- ggplot(data=KBaloha, aes(x=blueberry, y=Silicate, color=Site))+geom_boxplot(outlier.shape = NA) 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Silicate (uM)") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
Silicate
dev.off()
ggsave("Silicate_clusters_v2.pdf", height=3, width=6)




svg("chla_Box_Site.svg", height=2, width=6)
Chla<- ggplot(data=KBaloha, aes(x=blueberry, y=chla.ug.per.L, color=Site))+geom_boxplot(outlier.shape = NA) 
Chla=Chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a (ug/L)") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
Chla
dev.off()

ggsave("chla_sites_v2.pdf", height=3, width=6)


library(svglite)
library(cowplot)
svg("Chla_Silicate_Aloha_v2.svg", width=4, height=4)
plot_grid(Chla,Silicate,labels = c('B)', 'C)'), align = 'hv',label_size = 12, ncol = 1)
dev.off()
ggsave("chla_silicate_legend_v2.svg", width=7, height=11)



svg("DIN_P_4.svg", height=3, width=2)
p=ggplot(no_dup_metadata_KBytNERR, aes(x = Phosphate, y = DIN, color=Site)) + geom_point(size=1.5) + geom_abline(slope = 16, yintercept=0)+ xlim(0,0.5)+ scale_color_manual(values=colorssite2)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) +ylab("DIN (uM)")+xlab("Phosphate (uM)")+theme(legend.position="none")
p
ggsave("DIN_v2.pdf")
dev.off()

```


###Oʻahu Map Figure 1b
```{r}

library(geojsonio)
require(sf)

spdf <- geojson_read("Coastline.geojson",  what = "sp")

map <- read_sf("Coastline.geojson")

sites <- sf::st_as_sf(data.frame(longitude = c(-158.0), latitude = c(22.45)), coords = c("longitude", "latitude"), crs = 4326, 
agr = "constant")


svg("Map_Aloha_Oahu_blue.svg", width=2.5, height=2.5 )
hawaii  <- ggplot(data = map) +
     geom_sf(fill = "#f3eddd") +
     geom_sf(data = sites, size = 4, shape = 24, fill = "#7d7d7d")+
     coord_sf(xlim = c(-157.25, -158.73), ylim = c(21.20, 22.6), expand = FALSE) +
     theme(panel.grid.major = element_blank(), panel.background = element_rect(fill = "#47a4ff"), 
         panel.border = element_rect(fill = NA))+scale_x_discrete(position = "top")
hawaii
dev.off()
##Note labels were converted to degree minute seconds to match other plot in Figure 1 which was made in GIS 

#theme(panel.grid.major = element_line(colour = gray(0.8), linetype = "dashed", size = 0.2)
#+theme(axis.text.x = element_text(angle = 90, vjust = 1, hjust = 1))
```


######
Figure 2
######




```{r}

library(tidyverse)
library(lubridate)
library(reshape2)
library(MBA)
library(mgcv)
library(ggplot2)

ODV_colours <- c("#feb483", "#d31f2a", "#ffc000", "#27ab19", "#0db5e6", "#7139fe", "#d16cfa")

###the following can be repeated with all HPLC pigments

KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")


south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
names(south)
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,Fuco.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na()
str(south2)
south3=south2%>%reshape2::dcast(Site~decimal_dat)
names(south2)

south2$Site=as.numeric(south2$Site)

?mba.surf()

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Fuco.Rel') 


# Finally we create our gridded result
Fuco=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Fuco.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Fuco.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Fuco.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2,alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Diatoms \n Fucoxanthin:Tchla") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))
Fuco
ggsave("Fuco_ODV_KByT_SB.pdf")
```





######
Note Allo was below detection at Station Aloha 
######
```{r}
##############################Allo, Peri,X19But,X19Hex, Zea, Dva,
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")
SitelabsAllo <- c( "HP1", "SB","SR8", "SR2", "STO1")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
south2=south%>%dplyr::select( Site,decimal_dat,Allo.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%drop_na(Allo.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Allo.Rel') 


# Finally we create our gridded result
Allo=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Allo.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Allo.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Allo.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Crytophytes \n Allo:Tchla") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=SitelabsAllo)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))

Allo
ggsave("Allo_ODV_KByT_SB.pdf")

#########


```





######
```{r}
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")

##############################Peri
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,Peri.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(Peri.Rel)
str(south2)


south2$Site=as.numeric(south2$Site)


# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Peri.Rel') 
 # filter(depth < 0) %>% 
 # mutate(temp = round(temp, 1))

# Finally we create our gridded result
Peri=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Peri.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Peri.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Peri.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Dinoflagellates \n Peri:Tchla") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))


Peri
ggsave("Peri_ODV_KByT_SB.pdf")




```


```{r}

##############################X19But
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,X19But.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(X19But.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)


# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'X19But.Rel') 


# Finally we create our gridded result
X19But=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = X19But.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = X19But.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = X19But.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Pelagophytes \n But:Tchla") +
  coord_cartesian(expand = 0)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))+ scale_y_continuous(labels=Sitelabs)


X19But
ggsave("X19But_ODV_KByT_SB.pdf")



```


```{r}

##############################X19Hex
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,X19Hex.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(X19Hex.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'X19Hex.Rel') 

# Finally we create our gridded result
X19Hex=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = X19Hex.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = X19Hex.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = X19Hex.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Prymnesiophytes \n Hex:Tchla") +
  coord_cartesian(expand = 0)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))+ scale_y_continuous(labels=Sitelabs)


X19Hex
ggsave("X19Hex_ODV_KByT_SB.pdf")


```


```{r}


##############################Zea
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,Zea.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(Zea.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Zea.Rel') 


# Finally we create our gridded result
Zea=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Zea.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Zea.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Zea.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Cyanobacteria \n Zea:Tchla") +
  coord_cartesian(expand = 0)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))+ scale_y_continuous(labels=Sitelabs)


Zea
ggsave("Zea_ODV_KByT_SB.pdf")


```


```{r}
##############################Dva
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,Dva.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(Dva.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)


# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Dva.Rel') 

# Finally we create our gridded result
Dva=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Dva.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Dva.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Dva.Rel),  breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Prochlorococcus \n Dva:Tchla") +
  coord_cartesian(expand = 0)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))+ scale_y_continuous(labels=Sitelabs)


Dva
ggsave("Dva_ODV_KByT_SB.pdf")
```


```{r}

##############################Tchla
south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%dplyr::filter(decimal_dat<2019.31)%>%dplyr::filter(decimal_dat>2017.7)
Sitelabs <- c( "HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select( Site,decimal_dat,Tchla)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(Tchla)
str(south2)

south2$Site=as.numeric(south2$Site)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Tchla') 


# Finally we create our gridded result
Tchla=ggplot(data = ctd_mba, aes(y = Site, x =decimal_dat)) +
  geom_raster(aes(fill = Tchla)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Tchla), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Tchla), breaks=NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Total \n Chlorophyll a \n (ng/L)") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2018.0, 2018.5,  2019.0), labels=c( "Jan-18",  "Jul-18", "Jan-19"))

Tchla
ggsave("Tchla_ODV_KByT_SB.pdf")

```


```{r}
##################FCM- the following can be repeated with all FCM

KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")

south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
#south=south%>%filter(decimal_dat<2020.0)
Sitelabs <- c( "HP1","SB", "SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select(Site,decimal_dat,PRO.mL)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(PRO.mL)
str(south2)

is.na(south2$PRO.mL)

south2$Site=as.numeric(south2$Site)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'PRO.mL') 

# Finally we create our gridded result
svg("PRO.mL_ALOHA_Contour.svg", width=4, height=4)
PRO=ggplot(data = ctd_mba, aes(x =decimal_dat, y = Site)) +
  geom_raster(aes(fill = PRO.mL)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = PRO.mL), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = PRO.mL), breaks = NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y=Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Prochlorococcus \n cells mL") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2017.5, 2018.0,2018.5, 2019.0, 2019.5, 2020.0, 2020.5, 2021.0, 2021.5), labels=c( "", "2018", "","2019", "", "2020", "", "2021", ""))

PRO
dev.off()
ggsave("PRO_ODV_KByT_ALOHA.pdf")


```


```{r}
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")

south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")

Sitelabs <- c("HP1", "SB","SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select(Site,decimal_dat,SYN.mL)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(SYN.mL)
str(south2)

south2$Site=as.numeric(south2$Site)
summary(south2$SYN.mL)

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'SYN.mL') 



# Finally we create our gridded result
svg("SYN.mL_ALOHA_Contour.svg", width=4, height=4)
SYN=ggplot(data = ctd_mba, aes(x =decimal_dat, y = Site)) +
  geom_raster(aes(fill = SYN.mL)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = SYN.mL), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = SYN.mL), breaks = NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Synechococcus \n cells mL") +
  coord_cartesian(expand = 0)+ scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2017.5, 2018.0,2018.5, 2019.0, 2019.5, 2020.0, 2020.5, 2021.0, 2021.5), labels=c( "", "2018", "","2019", "", "2020", "", "2021", ""))


SYN
dev.off()
ggsave("SYN_ODV_KByT_ALOHA.pdf")
```

```{r}
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")

south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")

Sitelabs <- c("HP1","SB", "SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select(Site,decimal_dat,PEUK.mL)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(PEUK.mL)
str(south2)

south2$Site=as.numeric(south2$Site)


# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'PEUK.mL') 


# Finally we create our gridded result
svg("PEUK.mL_ALOHA_Contour.svg", width=4, height=4)
PEUK=ggplot(data = ctd_mba, aes(x =decimal_dat, y = Site)) +
  geom_raster(aes(fill = PEUK.mL)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = PEUK.mL), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = PEUK.mL), breaks = NULL, colour = "black", alpha = 0.2) +
### Activate to see which pixels are real and not interpolated
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Photosynthetic \n picoeukaryotes \n cells mL") +
  coord_cartesian(expand = 0)+  scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2017.5, 2018.0,2018.5, 2019.0, 2019.5, 2020.0, 2020.5, 2021.0, 2021.5), labels=c( "", "2018", "","2019", "", "2020", "", "2021", ""))


PEUK
dev.off()
ggsave("PEUK_ODV_KByT_ALOHA.pdf")
```

```{r}
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")

south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")

Sitelabs <- c("HP1","SB", "SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select(Site,decimal_dat,HBACT.mL)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(HBACT.mL)
str(south2)

south2$Site=as.numeric(south2$Site)


?mba.surf()

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'HBACT.mL') 



# Finally we create our gridded result
svg("HBACT.mL_ALOHA_Contour.svg", width=4, height=4)
HBACT=ggplot(data = ctd_mba, aes(x =decimal_dat, y = Site)) +
  geom_raster(aes(fill = HBACT.mL)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = HBACT.mL), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = HBACT.mL), breaks = NULL, colour = "black", alpha = 0.2) +
 geom_point(data = south2, aes( x = decimal_dat,y =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(x = "Date", y = NULL, fill = "Heterotrophic \n bacteria \n cells mL") +
  coord_cartesian(expand = 0)+scale_y_continuous(labels=Sitelabs)+scale_x_continuous(breaks=c(2017.5, 2018.0,2018.5, 2019.0, 2019.5, 2020.0, 2020.5, 2021.0, 2021.5), labels=c( "", "2018", "","2019", "", "2020", "", "2021", ""))
HBACT
dev.off()
ggsave("HBACT_ODV_KByT_ALOHA.pdf")
```



```{r}

##join together the HPLC and FCM plots 
library(cowplot)
svg("Contour_plots_HPLC_FCM_ALOHA.svg", height=7, width=14)
plot_grid(HBACT,SYN,PEUK,PRO,Tchla,Fuco,Peri,Allo,X19Hex,X19But, Zea, Dva, ncol=4, align="hv")
dev.off()
ggsave("Contour_plots_HPLC_FCM_ALOHA_v2.pdf", height=7, width=14)


```



######
Figure 3
######

```{r}


library(dendextend)
library(phyloseq)
library(microbiome)


##let's look at the distribution of phytoplankton across the samples with a clr transformation to account for differences in sequencing depth.
CLR_All_Phyto=microbiome::transform(All_phyto, transform="clr")
euc_dist <- dist(t(otu_table(CLR_All_Phyto)))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 
plot(d1)
##we see three main groups in the tree and by outputting the sample grouping from the tree you can define the tree groups. 
k234 <- cutree(d1, k = 2:4)
k234[,2]
write.csv(k234[,2], "Clusters.csv")
##It clear there are three main groups and the samples contained in each group are cluster samples based on space in the ocean timeseries- Nearshore, transition, and offshore. 
##note this grouping information has already been combined into our metadata as "Spatial" to keep things simple. Here we can color the different parts of the tree based on these designations for visualization. 

Site_type <- factor(sample_data(All_phyto)$Site, levels=c("WAI2","KAHO","AR","HP1","NB","SB","CB","SR8","SR2","NR2","STO1","NTO1"))
colors_Site <- c("#000000","#f94279","#967969","#ffd000","#ff8c00","#C47F2B","#ad0201","#D7B4F3", "#AAFF00", "#049861", '#00FFFF','#0045ff')
col_Site_type <- colors_Site[Site_type]
col_Site_type



Spatial_type <- factor(sample_data(All_phyto)$Spatial, levels=c("Nearshore","Transition","Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_spatial <- c("#B70D6A","#22C959","dodgerblue3")
col_Spatial_type <- colors_spatial[Spatial_type]

###export and take a peak
svg("new_Season_CLR_test_2.svg", height=10, width=20)
par(mar = c(2,2,2,20))
k234 <- cutree(d1, k = 2:4)
plot(d1,horiz = TRUE,main="")
colored_bars(cbind(col_Site_type, col_Spatial_type), d1, rowLabels = c( "St", "CT"), horiz = TRUE, y_shift=20)
dev.off()


```


############
Figure 3b
############
#b) Distribution of samples defined as nearshore, transition, and offshore community types across stations and sampling events (36 total between 2017-2021). Size of the circle is relative to chlorophyll a concentration. The assignment of each sample to the three community types across stations and sampling events. The size of circle represents the chlorophyll a concentration during the time of sampling and the color of the circle represents the community type: nearshore, transition, and offshore.  

```{r}

chla_over_time=no_dup_metadata_KBytNERR%>%dplyr::select(Date1,Site, Spatial, Season_Tucker2021, chla.ug.per.L)
library(reshape2)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Nearshore"="#B70D6A")


Site_order=factor(chla_over_time$Site,levels=c("WAI2", "KAHO",  "AR", "HP1","SB","CB", "NB","SR8", "SR2", "NR2", "STO1", "NTO1"))

break.vec <- c(seq(from = as.Date("2017-08-01"), to = as.Date("2021-07-01"), by = "1 month"))
labs=c("2017-08-01"="Aug-2017", "2017-09-01"="","2017-10-01"="","2017-11-01"="","2017-12-01"="","2018-01-01"="Jan-2018", "2018-02-01"="", "2018-03-01"="", "2018-04-01"="", "2018-05-01"="", "2018-06-01"="", "2018-07-01"="Jul-2018", "2018-08-01"="", "2018-09-01"="", "2018-10-01"="", "2018-11-01"="", "2018-12-01"="", "2019-01-01"="Jan-2019", "2019-02-01"="", "2019-03-01"="", "2019-04-01"="", "2019-05-01"="", "2019-06-01"="", "2019-07-01"="Jul-2019", "2019-08-01"="", "2019-09-01"="", "2019-10-01"="", "2019-11-01"="", "2019-12-01"="", "2020-01-01"="Jan-2020", "2020-02-01"="", "2020-03-01"="", "2020-04-01"="", "2020-05-01"="", "2020-06-01"="", "2020-07-01"="Jul-2020", "2020-08-01"="",  "2020-09-01"="",  "2020-10-01"="",  "2020-11-01"="",  "2020-12-01"="",  "2021-01-01"="Jan-2021",  "2021-02-01"="", "2021-03-01"="", "2021-04-01"="", "2021-05-01"="", "2021-06-01"="", "2021-07-01"="Jul-2021")


library(scales)


svg("time_series_chla_Oct2023.svg", height =3.5, width=6)
ggplot(chla_over_time, aes(x = Site_order, y = as.Date(Date1, format="%m/%d/%y"))) + 
      geom_point(aes(size = chla.ug.per.L, fill = Spatial),  shape = 21) + 
       scale_size_continuous(limits = c(0.1, 20), range = c(1,10), breaks = c(0.1,0.5,1,5,10)) +
    scale_fill_manual(values=colorspatial) + theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=9)) + coord_flip() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 9))+  theme(legend.position = "bottom") +# minimal theme
theme(legend.title = element_blank()) + theme(axis.text=element_text(size=12),axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size=12), axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+xlab("Station")+ylab("Date")+ scale_y_date(breaks = break.vec, labels=labs)
ggsave('Time_Series_Enviro_Chla_Oct2023.pdf', width=6, height=4)
dev.off()
```


#######
Figure 4
########
Figure 4a) Mean relative abundance of the top 10 phytoplankton groups across the nearshore (N), transition (T), and offshore (O) environments. 

```{r}
#######
#Figure 4a
########
##note a curated taxonomic grouping (called Curated) was added to the taxonomy table based on the lowest level identification possible for phytoplankton classes and genera with >1% average relative abundance in at least one community. 
#this allowed us to look at abundant phytoplankton groups with varying levels of taxonomic assignment (e.g. Cyanobacteria:Synechococcus, Stramenopiles:Bacillariophyta, Archaeplastida:Mamiellophyceae; Archaeplastida:Micromonas) and assign a larger taxonomic grouping to the less abundant ASVs that donʻt already fall into one of the pre-defined categories (e.g. Hacrobia:Other, Stramenopiles:Other, Eukaryota:Other, Archaeplastida:Other,Cyanobacteria:Other)
# this is valuable for assessing across these different groups that tend to be able to classified at the family level but maybe not at the genus level (or your can skip to the plotting Figure 4a) 


###identify the top 10 most abundant groups across all samples on average
rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID,Curated) %>% dplyr::summarise(sum = sum(Abundance)) 
rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Curated) %>% dplyr::summarise(mean = mean(sum)*100, sd=sd(sum)*100) 

top10=subset(rel2Phyto_ASV_Cluster, Curated=="Cyanobacteria:Synechococcus"|
Curated=="Archaeplastida:Mamiellophyceae"|
Curated=="Stramenopiles:Bacillariophyta"|
Curated=="Cyanobacteria:Cyanobium"|
Curated=="Hacrobia:Prymnesiophyceae"|
Curated=="Stramenopiles:Chaetoceros"|
Curated=="Archaeplastida:Micromonas"|
Curated=="Hacrobia:Cryptophyceae"|
Curated=="Archaeplastida:Ostreococcus"|
Curated=="Cyanobacteria:Prochlorococcus")

###okay but we want to to look at their distribution accross the three community types so now that we defined the top 10 groups, lets look at this at relative abundance based on 



###identify the top 10 most abundant groups across all samples on average
rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID,Spatial,Curated) %>% dplyr::summarise(sum = sum(Abundance)) 
rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Spatial, Curated) %>% dplyr::summarise(mean = mean(sum)*100, sd=sd(sum)*100) 

top10=subset(rel2Phyto_ASV_Cluster, Curated=="Cyanobacteria:Synechococcus"|
Curated=="Archaeplastida:Mamiellophyceae"|
Curated=="Stramenopiles:Bacillariophyta"|
Curated=="Cyanobacteria:Cyanobium"|
Curated=="Hacrobia:Prymnesiophyceae"|
Curated=="Stramenopiles:Chaetoceros"|
Curated=="Archaeplastida:Micromonas"|
Curated=="Hacrobia:Cryptophyceae"|
Curated=="Archaeplastida:Ostreococcus"|
Curated=="Cyanobacteria:Prochlorococcus")

#order then based on abudance 
top10$Curated=factor(top10$Curated, levels=c("Stramenopiles:Chaetoceros",
"Hacrobia:Cryptophyceae",
"Archaeplastida:Micromonas",
"Archaeplastida:Ostreococcus","Cyanobacteria:Cyanobium",
"Hacrobia:Prymnesiophyceae","Stramenopiles:Bacillariophyta","Archaeplastida:Mamiellophyceae","Cyanobacteria:Prochlorococcus","Cyanobacteria:Synechococcus"
))


top10$Spatial=factor(top10$Spatial, levels=c("Nearshore", "Transition", "Offshore"))


svg("heatmap_Spatial_Clusters_top10.svg", width=8, height=6)
bp=ggplot(top10, aes(x=as.factor(Spatial), y=as.factor(Curated), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) +
  scale_fill_viridis(option="turbo",trans = 'log10')  + labs(fill = "Mean Relative Abundance \n as Percent of \n Phytoplankton Abundance")+theme(axis.text=element_text(size=11), legend.text=element_text(size=11), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+xlab("")+ylab("")
bp 

dev.off()

ggsave("heatmap_Clusters_spatial_mean.pdf", width=8, height=6)


```

#######
#Figure 4b
########
4b)  Phytoplankton genera with significantly different distributions across the three environments. Colored boxes denote the peak in abundance for each genus. Phytoplankton groups that were classified at the family-level but unidentified at the genus-level are denoted with an asterisk.


##first step is to test which genera have significantly different abundances across the environments
```{r}

glom=tax_glom(All_phyto, "Genus")
ntaxa(glom)
dex = phyloseq::phyloseq_to_deseq2(glom, ~Spatial)

diagdds_pol = DESeq2::DESeq(dex, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(glom)[rownames(sigtab), ], "matrix"))
head(sigtab)



sigtab2=sigtab [,c(1:6,15)]
sigtab2$DeSeq_Spatial_Cluster=sigtab$Domain
sigtab2$DeSeq_Spatial_Cluster="DeSeq_Spatial_Variable"
sigtab2=as.data.frame(sigtab2)
taxonomy_Phyto_we = read.csv("taxonomy_table_clean_May2024.csv", sep=",") 
taxa_variable_spatial=left_join(taxonomy_Phyto_we,sigtab2,by="ASV")

##examine differential abundances of genera between each community type- this one is transition vs offshore 
res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Offshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Transition_Offshore_April2024.csv")

Genus_Transition_Offshore=sigtab

Genus_Transition_Offshore$Genus_Transition_Offshore="Genus_Transition_Offshore"

names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'log2FoldChange'] <- 'Genus_TO_log2FoldChange'
names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'padj'] <- 'Genus_TO_padj'
names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'baseMean'] <- 'Genus_TO_baseMean'

Genus_Transition_Offshore_v2=Genus_Transition_Offshore%>%dplyr::select(Genus, Genus_TO_log2FoldChange, Genus_TO_padj, Genus_TO_baseMean,Genus_Transition_Offshore)

names(Genus_Transition_Offshore)



##examine differential abundances of genera between each community type- this one is transition vs Nearshore
res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Nearshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Nearshore_Transition_April2024.csv")

Genus_Nearshore_Transition=sigtab
Genus_Nearshore_Transition$Genus_Nearshore_Transition="Genus_Nearshore_Transition"

names(Genus_Nearshore_Transition)[names(Genus_Nearshore_Transition) == 'log2FoldChange'] <- 'Genus_NT_log2FoldChange'
names(Genus_Nearshore_Transition)[names(Genus_Nearshore_Transition) == 'padj'] <- 'Genus_NT_padj'
names(Genus_Nearshore_Transition)[names(Genus_Nearshore_Transition) == 'baseMean'] <- 'Genus_NT_baseMean'

Genus_Nearshore_Transition_v2=Genus_Nearshore_Transition%>%dplyr::select(Genus, Genus_NT_log2FoldChange, Genus_NT_padj, Genus_NT_baseMean,Genus_Nearshore_Transition)



##examine differential abundances of genera between each community type- this one is Nearshore vs offshore

res <- results(diagdds_pol, contrast=c("Spatial", "Offshore","Nearshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Nearshore_Offshore_April2024.csv")

Genus_Nearshore_Offshore=sigtab
Genus_Nearshore_Offshore$Genus_Nearshore_Offshore="Genus_Nearshore_Offshore"

names(Genus_Nearshore_Offshore)[names(Genus_Nearshore_Offshore) == 'log2FoldChange'] <- 'Genus_NO_log2FoldChange'
names(Genus_Nearshore_Offshore)[names(Genus_Nearshore_Offshore) == 'padj'] <- 'Genus_NO_padj'
names(Genus_Nearshore_Offshore)[names(Genus_Nearshore_Offshore) == 'baseMean'] <- 'Genus_NO_baseMean'

Genus_Nearshore_Offshore_v2=Genus_Nearshore_Offshore%>%dplyr::select(Genus, Genus_NO_log2FoldChange, Genus_NO_padj, Genus_NO_baseMean,Genus_Nearshore_Offshore)



NOT=full_join(Genus_Nearshore_Offshore_v2,Genus_Nearshore_Transition_v2, by="Genus")
NOOT=full_join(NOT,Genus_Transition_Offshore_v2, by="Genus")


###left join with the most recent taxonomy table to get information on taxonomy
taxonomy_Phyto_we = read.csv("taxonomy_table_clean_May2024.csv", sep=",") 
output=left_join(NOOT,taxonomy_Phyto_we,by="Genus")
output2=output%>%distinct(Genus, .keep_all = TRUE)
write.csv(output2, "Genus_all_comparisons_NEWCLR_Spatial_April2024.csv")

###note differentially abundant taxa that are unidentified at higher than the class level were not included 
##all this info was included in Table S7 

```

##So now that we know which genera are differentially abundant we can look where these differentially abundant taxa peak in abundance across the three environments 
```{r}
relglom=tax_glom(All_phyto, "Genus")
p0.rel2 <- microbiome::transform(relglom, "clr")

relglomdiffer2=subset_taxa(p0.rel2 , Genus=="D_5__Prochlorococcus_MIT9313" |
Genus=="D_5__Cyanobium_PCC-6307" |
Genus=="D_5__Atelocyanobacterium_(UCYN-A)" |
Genus=="D_5__Crocosphaera" |
Genus=="D_5__Trichodesmium_IMS101" |
Genus=="Tetraselmis" |
Genus=="Chloropicon" |
Genus=="unclassified_Bathycoccaceae" |
Genus=="Ostreococcus" |
Genus=="Bathycoccus" |
Genus=="Micromonas" |
Genus=="unclassified_Mamiellaceae" |
Genus=="Mamiella" |
Genus=="Marsupiomonas" |
Genus=="Resultor" |
Genus=="Prasinoderma" |
Genus=="Eutreptiella" |
Genus=="Proteomonas" |
Genus=="unclassified_Hemiselmidaceae" |
Genus=="Braarudosphaera" |
Genus=="Phaeocystis" |
Genus=="unclassified_Prymnesiaceae" |
Genus=="Thalassionema" |
Genus=="Chaetoceros" |
Genus=="unclassified_Polar-centric-Mediophyceae" |
Genus=="unclassified_Radial-centric-basal-Coscinodiscophyceae" |
Genus=="unclassified_Raphid-pennate" |
Genus=="Ochromonas" |
Genus=="Mesopedinella" |
Genus=="unclassified_Pedinellales" |
Genus=="Helicopedinella" |
Genus=="unclassified_Pelagomonadaceae" |
Genus=="Aureococcus")



rel2Phyto_Genus_Spatial<-relglomdiffer2 %>% psmelt()%>% dplyr::group_by(Spatial, Genus) %>% dplyr::summarise(mean = mean(Abundance), sd=sd(Abundance)) 

##select only the Genera who had significantly different abundances 
rel2Phyto_Genus_Spatial$Genus=factor(rel2Phyto_Genus_Spatial$Genus, levels=c("unclassified_Pelagomonadaceae","Aureococcus","unclassified_Pedinellales", "Mesopedinella","Helicopedinella","Ochromonas","Thalassionema","unclassified_Raphid-pennate","unclassified_Radial-centric-basal-Coscinodiscophyceae","unclassified_Polar-centric-Mediophyceae","Chaetoceros","unclassified_Prymnesiaceae","Phaeocystis","Braarudosphaera","Proteomonas","unclassified_Hemiselmidaceae","Eutreptiella","Prasinoderma","Resultor","Marsupiomonas","unclassified_Mamiellaceae","unclassified_Bathycoccaceae","Ostreococcus" ,"Micromonas" ,"Mamiella","Bathycoccus" ,"Chloropicon","Tetraselmis", "D_5__Trichodesmium_IMS101","D_5__Prochlorococcus_MIT9313","D_5__Crocosphaera",  "D_5__Atelocyanobacterium_(UCYN-A)","D_5__Cyanobium_PCC-6307"))


rel2Phyto_Genus_Spatial$Spatial=factor(rel2Phyto_Genus_Spatial$Spatial, levels=c("Nearshore", "Transition", "Offshore"))
#https://stackoverflow.com/questions/48424682/how-do-i-limit-the-range-of-the-viridis-colour-scale
bp=ggplot(rel2Phyto_Genus_Spatial, aes(x=as.factor(Spatial), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + 
  scale_fill_gradientn(colors = viridis_pal()(9), limits=c(-6, 5), 
                       na.value = "#FDE725FF")+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
bp 
ggsave("CLR_genera_that_Differ_CLR_March2024.pdf", height=6, width=9)

##while this figure is fine, marking where the peak in abundance is per genus is a little more straightforward

###we can now identify the where maximum abundance occurs for each Genus across the three environments 
rel2Phyto_Genus_Spatial2=reshape2::dcast(rel2Phyto_Genus_Spatial,Genus~Spatial, value.var= "mean")
rel2Phyto_Genus_Spatial2

rel2Phyto_Genus_Spatial3=rel2Phyto_Genus_Spatial2 %>% 
  rowwise() %>% 
  mutate( x = max(c_across(2:4))) %>% 
  mutate(across(2:4, ~case_when(. == x ~ 1,
                                TRUE ~ 0))
         )

taxa= read.csv("taxonomy_table_clean_May2024.csv", sep=",") 
Genus_Class=taxa%>%dplyr::select("Genus", "Class", "Supergroup")%>%distinct(Genus, .keep_all = TRUE)

rel2Phyto_Genus_Spatial4=left_join(rel2Phyto_Genus_Spatial3, Genus_Class, by="Genus")

write.csv(rel2Phyto_Genus_Spatial4, "summary_genera.csv")

##I reordered the summary above and fixed some of the labels to get this final figure
genotype <- read.csv("summary_genera_ordered.csv", row.names=1)

unique(genotype$Genus)

pattern_df = data.frame(genotype)
rownames(pattern_df) = rownames(genotype)


svg("genus_Peak_CLR.svg", height=11, width=8.5)
ComplexHeatmap::pheatmap(pattern_df[,1:3], cluster_cols = F, cluster_row = F, annotation_row = pattern_df  , color = c("white","black"),cellwidth=20,cellheight=10, row_names_side = c("left"))
dev.off()

##from here I kept the heat map and updated the coloring from black and white to colors specific to genera in Inkscape
```


#######
Figure 5
########
Metagenomic read recruitment of cyanobacterial genomes from Prochlorococcus and Synechococcus across KByT and Station ALOHA. Recruitment of <0.5% not shown.


```{r}
library(tidyverse)

#Note: detection and mean coverage q2q3 data input comes from analyses conducted in metagenomic analyses code

#### look at all genomes for detection levels
detection=read.delim("detection.txt")
names(detection)

Taxonomy=read.delim("Cyano_Genera_genome_info_layer.txt")
names(Taxonomy)

Taxonomy_MinorClade_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "MinorClade")
names(Taxonomy_MinorClade_bins)

detection_MinorClade=left_join(detection, Taxonomy_MinorClade_bins, by="bins")

detection_MinorClade=detection_MinorClade%>%unite("Merged", c(bins,MinorClade), sep = "__")
?unite()
detection_MinorClade=as.data.frame(detection_MinorClade)

detection_MinorClade <- data.frame(detection_MinorClade[,-1], row.names=detection_MinorClade[,1])


detection_MinorClade2<- detection_MinorClade %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_MinorClade2)

detection_MinorClade3=detection_MinorClade2 %>% separate(rowname, c('bins', 'MinorClade'), sep='__')


##using a threshold of 0.25, which clades were detected in our samples:
detection_MinorClade4=detection_MinorClade3%>%filter(value >0.25)%>% group_by(MinorClade, bins) %>% tally(sort = TRUE)

unique(detection_MinorClade4$MinorClade)

```




```{r}
#### look at all genomes for detection levels
detection_cyano=read.delim("detection.txt")
names(detection)

Taxonomy=read.delim("Cyano_Genera_genome_info_layer.txt")
names(Taxonomy)

Taxonomy_MinorClade_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "MinorClade")
names(Taxonomy_MinorClade_bins)

detection_MinorClade=left_join(detection, Taxonomy_MinorClade_bins, by="bins")

detection_MinorClade=detection_MinorClade%>%unite("Merged", c(bins,MinorClade),sep="__")

write.csv(detection_MinorClade, "detection_MinorClade.csv")
detection_MinorClade=as.data.frame(detection_MinorClade)

detection_MinorClade <- data.frame(detection_MinorClade[,-1], row.names=detection_MinorClade[,1])


detection_MinorClade2<- detection_MinorClade %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(detection_MinorClade2)




detection_MinorClade2_genome_order <- factor(detection_MinorClade2$rowname,level=c(
"MED4__HL_I",
"MIT9515__HL_I",
"AS9601__HL_II",
"MIT9301__HL_II",
"MIT9215__HL_II",
"MIT9202__HL_II",
"MIT9107__HL_II",
"GP2__HL_II",
"MIT9201__HL_II",
"MIT9302__HL_II",
"MIT9311__HL_II",
"MIT9314__HL_II",
"MIT9321__HL_II",
"SB__HL_II",
"NATL2A__LL_I",
"NATL1A__LL_I",
"PAC1__LL_I",
"CCMP1375__LL_II",
"MIT9211__LL_III",
"MIT9303__LL_IV",
"MIT9313__LL_IV",
"HIMB2401__II",
"CC9605__II",
"WH8109__II",
"N32__II",
"N5__II",
"N26__II",
"N19__II",
"UW86__II",
"KORDI52__WPC2",
"UW106__XV",
"UW69__XV",
"CC9902__IV",
"BL107__IV",
"WH8102__III",
"CC9616__UC-A",
"KORDI100__UC-A",
"KORDI49__WPC1",
"MITS9509__CRD1",
"MITS9508__CRD1",
"GEYO__CRD1",
"UW179A__CRD1",
"WH7803__V",
"WH7805__VI",
"CC9311__I",
"WH8020__I",
"WH8016__I",
"UW179B__I",
"UW105__XVI",
"UW140__XVI",
"RS9916__IX",
"RS9917__VIII",
"PCC7001__SC_5.2",
"CB0101__SC_5.2",
"PCC6307__SC_5.2",
"RCC307__SC_5.3"))




detection_MinorClade2_sample_order <- factor(detection_MinorClade2$colname, level=c("KSe054HP",
"KSe055AR",
"KOc293R8",
"KSe058SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe387AR",
"KFe386HP",
"KFe389R8",
"KFe395NB",
"KMa160SB",
"KMa158CB",
"KMa156HP",
"KMa157AR",
"KMa159R8",
"KMa165NB",
"KOc299NB",
"KSe062NR",
"KSe063NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KFe393NT",
"KFe416ST",
"KMa162ST",
"KMa163NT",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))



svg('Cyano_detection_KByT_Aloha_April2024_aboveaquarter.svg', width=8.5, height=11)
ggplot(detection_MinorClade2, aes(x = detection_MinorClade2_sample_order, y = detection_MinorClade2_genome_order)) + 
      geom_point(aes(size = value, fill = value), alpha = 0.75, shape = 21) + 
       scale_size_continuous(limits = c(0.25, 10), range = c(1,10), breaks = c(0.25,0.5,0.75,1)) +
    scale_fill_gradient(low="#FBFEF9",high="#A63446") + scale_x_discrete(name="Metagenome") + scale_y_discrete(name="Genome", position = "right") + labs(fill = "Detection") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, vjust = 1, 
hjust = -0.05, size=5)) + coord_flip() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 1))

ggsave('Cyano_detection_KByT_Aloha_April2024_aboveaquarter.pdf', width=7, height=10)
dev.off()



```




```{r}

#### look at all genomes for read recruitment levels, however lets normalize for detection being greater than >0.25
meanq2q3=read.delim("mean_coverage_Q2Q3.txt")
names(detection)

Taxonomy=read.delim("Cyano_Genera_genome_info_layer.txt")
names(Taxonomy)

Taxonomy_MinorClade_bins=Taxonomy%>%rename_all(recode,"samples"="bins")%>%dplyr::select("bins", "MinorClade")
names(Taxonomy_MinorClade_bins)



meanq2q3_MinorClade=left_join(meanq2q3, Taxonomy_MinorClade_bins, by="bins")

meanq2q3_MinorClade=meanq2q3_MinorClade%>%unite("Merged", c(bins,MinorClade), sep="__")

meanq2q3_MinorClade=as.data.frame(meanq2q3_MinorClade)

meanq2q3_MinorClade <- data.frame(meanq2q3_MinorClade[,-1], row.names=meanq2q3_MinorClade[,1])


meanq2q3_MinorClade2<- meanq2q3_MinorClade %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(meanq2q3_MinorClade2)




meanq2q3_MinorClade2=meanq2q3_MinorClade2%>%dplyr::rename("q2q3" = "value")
detection_MinorClade3=detection_MinorClade2%>%dplyr::rename("rowname_detect" = "rowname", "colname_detect"="colname")
q2q3_detection=cbind(meanq2q3_MinorClade2, detection_MinorClade3)

q2q3_detection$q2q3_normalized=q2q3_detection$q2q3

q2q3_detection$q2q3_normalized[q2q3_detection$value < 0.25] <- 0

q2q3_detection2=q2q3_detection%>%dplyr::select(rowname, colname, q2q3_normalized)

test=reshape2::dcast(q2q3_detection2, rowname ~colname)

test.rownames <- data.frame(test[,-1], row.names=test[,1])


freqs2 <- apply(test.rownames, 2, function(i) i/sum(i))


freqs3=freqs2*100
freqs3=as.data.frame(freqs3)
write.csv(freqs3, "normalized_cyano_q2q3_matrix.csv")

read_recruit<-freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)





####this one 2 clusters
read_recruit_sample_order <- factor(read_recruit$colname, level=c("KSe054HP",
"KSe055AR",
"KSe063NB",
"KMa157AR",
"KMa159R8",
"KOc293R8",
"KOc299NB",
"KFe387AR",
"KSe058SB",
"KMa156HP",
"KMa158CB",
"KMa160SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe386HP",
"KFe389R8",
"KFe395NB",
"KSe062NR",
"KMa165NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KMa163NT",
"KMa162ST",
"KFe393NT",
"KFe416ST",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))




read_recruit_genome_order <- factor(read_recruit$rowname,level=c(
"MED4__HL_I",
"MIT9515__HL_I",
"AS9601__HL_II",
"MIT9301__HL_II",
"MIT9215__HL_II",
"MIT9202__HL_II",
"MIT9107__HL_II",
"GP2__HL_II",
"MIT9201__HL_II",
"MIT9302__HL_II",
"MIT9311__HL_II",
"MIT9314__HL_II",
"MIT9321__HL_II",
"SB__HL_II",
"NATL2A__LL_I",
"NATL1A__LL_I",
"PAC1__LL_I",
"CCMP1375__LL_II",
"MIT9211__LL_III",
"MIT9303__LL_IV",
"MIT9313__LL_IV",
"HIMB2401__II",
"CC9605__II",
"WH8109__II",
"N32__II",
"N5__II",
"N26__II",
"N19__II",
"UW86__II",
"KORDI52__WPC2",
"UW106__XV",
"UW69__XV",
"CC9902__IV",
"BL107__IV",
"WH8102__III",
"CC9616__UC-A",
"KORDI100__UC-A",
"KORDI49__WPC1",
"MITS9509__CRD1",
"MITS9508__CRD1",
"GEYO__CRD1",
"UW179A__CRD1",
"WH7803__V",
"WH7805__VI",
"CC9311__I",
"WH8020__I",
"WH8016__I",
"UW179B__I",
"UW105__XVI",
"UW140__XVI",
"RS9916__IX",
"RS9917__VIII",
"PCC7001__SC_5.2",
"CB0101__SC_5.2",
"PCC6307__SC_5.2",
"RCC307__SC_5.3"))

read_recruit2=read_recruit
read_recruit2[read_recruit2 == 0] <- NA


svg('cyano_q2q3_KByT_Aloha_March2024.svg', width=8.5, height=11)
geom.text.size = 2
recruit=ggplot(read_recruit2, aes(x = read_recruit_genome_order, y = read_recruit_sample_order, fill = value)) +
    geom_tile(aes(fill = value)) + geom_text(aes(label = round(value, 0)), size=geom.text.size) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) + 
  scale_fill_gradient2(low = "yellow", high="purple", mid = "red", midpoint = 10,na.value = "white", 
  guide = "colourbar", 
  aesthetics = "fill")  + scale_x_discrete(name="Genome") + scale_y_discrete(name="Metagenome", position = "right") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=5)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=6),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+ coord_flip()


recruit
ggsave('cyano_q2q3_KByT_Aloha_March2024.pdf', width=7, height=10)
dev.off()

```


```{r}
##look at which clades are abundant in samples 
stats=read_recruit%>%filter(value>0.5)%>% separate(rowname, c('Genome', 'Clade'), sep='__')

##look at abundance of HLII, II, and SC_5.3 across community types (nearshore, transition, offshore, open ocean)

stats=read_recruit%>%separate(rowname, c('Genome', 'Clade'), sep='__')

stats=stats%>%dplyr::rename("sample"="colname")
  
samp=read.delim("MG_sample_guide.txt")

stats2=left_join(stats, samp, by="sample")

by_clade_ra=stats2%>%group_by(Clade, sample, Community_Type)%>%summarise(sum=sum(value))
by_clade_ra2=by_clade_ra%>%group_by(Clade, Community_Type)%>%summarise(mean=mean(sum), sd=sd(sum))%>%subset(Clade=="HL_II" | Clade=="II" | Clade=="SC_5.3")


```



```{r}
##finally lets display the read recruitment data at the genome level for HLII, II, and S.C 5_3 and group everything else and make a pretty figure 

rr_grouped_clades=read_recruit
rr_grouped_clades=rr_grouped_clades%>% separate(rowname, c('Genome', 'Clade'), sep='__')


Taxonomy=read.delim("Cyano_Genera_genome_info_layer.txt")
names(Taxonomy)

Taxonomy_MajorClade=Taxonomy%>%rename_all(recode,"samples"="Genome")%>%dplyr::select("Genome", "MajorClade")


rr_grouped_clades_mj=left_join(rr_grouped_clades, Taxonomy_MajorClade)


rr_grouped_clades2=rr_grouped_clades_mj%>%dplyr::mutate(Clade = case_when(Clade=="HL_II" ~ "HL_II", Clade=="HL_I" ~ "OtherP", Clade=="LL_I" ~ "OtherP", Clade=="LL_II"~"OtherP", Clade=="LL_III"~"OtherP",Clade=="LL_IV"~"OtherP", Clade=="II"~ "II", Clade=="SC_5.3"~"SC_5.3",TRUE ~ "OtherS" )) 
unique(rr_grouped_clades2$Clade)   


rr_grouped_clades3=rr_grouped_clades2%>%dplyr::mutate(Clade = case_when(Genome=="AS9601" ~ "AS9601", Genome=="GP2"~ "GP2",Genome=="MIT9107"~ "MIT9107",Genome=="MIT9201"~ "MIT9201",Genome=="MIT9202"~ "MIT9202",Genome=="MIT9215"~ "MIT9215",Genome=="MIT9301"~ "MIT9301",Genome=="MIT9302"~ "MIT9302",Genome=="MIT9311"~ "MIT9311",Genome=="MIT9314"~ "MIT9314",Genome=="MIT9321"~ "MIT9321",Genome=="SB"~ "SB",Genome=="CC9605"~ "CC9605",Genome=="N19"~ "N19",Genome=="N26"~ "N26",Genome=="N32"~ "N32",Genome=="N5"~ "N5",Genome=="UW86"~ "UW86",Genome=="WH8109"~ "WH8109",Genome=="HIMB2401"~ "HIMB2401",Genome=="RCC307"~ "RCC307",TRUE ~ Clade )) 

unique(rr_grouped_clades3$Clade)  
names(rr_grouped_clades3)
rr_grouped_clades4=rr_grouped_clades3%>%dplyr::select("Clade", "colname", "value", "MajorClade")

rr_grouped_clades5=rr_grouped_clades4%>%group_by(Clade, colname, MajorClade)%>%summarise(value=sum(value))

#sanity check #make sure you get 100% read recruitment accross the samples
#test=rr_grouped_clades4%>%group_by(colname)%>%summarise(value=sum(value))

freqs4=rr_grouped_clades5

colnames(freqs4)[colnames(freqs4) == "colname"] ="sample"
freqs4

Clade_order<- factor(freqs4$Clade,level=c(
"AS9601",
"SB",
"MIT9314",
"MIT9301",
"MIT9201",
"MIT9215",
"MIT9202",
"MIT9107",
"GP2",
"MIT9302",
"MIT9311",
"MIT9321",
"OtherP",
"UW86",
"N19",
"N26",
"N32",
"N5",
"HIMB2401",
"CC9605",
"WH8109",
"OtherS",
"RCC307"))
freqs4$colname


sample_order <- factor(freqs4$sample, level=c("KSe054HP",
"KSe055AR",
"KOc293R8",
"KSe058SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe387AR",
"KFe386HP",
"KFe389R8",
"KFe395NB",
"KMa160SB",
"KMa158CB",
"KMa156HP",
"KMa157AR",
"KMa159R8",
"KMa165NB",
"KOc299NB",
"KSe062NR",
"KSe063NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KFe393NT",
"KFe416ST",
"KMa162ST",
"KMa163NT",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))
freqs4$value[freqs4$value == 0] <- NA
freqs5=freqs4
freqs5$value[freqs5$value<0.5]<- NA



svg("PRO_SYN_clade_isolate_March2024.svg", width=6, height=5.5)
geom.text.size = 3
recruit=ggplot(freqs5, aes(x = sample_order, y = Clade_order, fill = value, by=MajorClade)) +
    geom_tile(aes(fill = value)) + facet_wrap(~MajorClade, scales = "free")+
    #geom_text(aes(label = round(value, 0)), size=geom.text.size) +
  scale_fill_gradient2(low = "yellow", high="purple", mid = "red", midpoint = 10,na.value = "white") + scale_x_discrete(name="Metagenomes") + scale_y_discrete(name="Clade", position = "left") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, size=9, hjust=0.95,vjust=0.2)) +coord_flip()+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))

recruit
dev.off()

ggsave('Metagenomes_Pro_Syn_Cyano_March2024.pdf', width=9.0, height=8)


```





######
Figure 6
######

####
first lets look how beautifully these seasonal trends in environmental patterns are over months
```{r}

library(seecolor)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Nearshore"="#B70D6A")
print_color(colorspatial)
library(ggplot2)



no_dup_metadata_KBytNERR$Month_abb=factor(no_dup_metadata_KBytNERR$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

library(scales)


a=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SW.Temperature.at.site, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Temperature (C)")+xlab("")
a


f=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=HBACT.mL, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Heterotrophic prokaryotic cells mL"^-1))
f

g=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SYN.mL, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Synechococus cells mL"^-1))

g

```



####Okay okay lets get to the figure, with boxplots over seasons, letʻs contrast the most dramatic contrasts, winter and summer 
```{r}


library(seecolor)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Nearshore"="#B70D6A")

library(ggplot2)


SuWi_only=no_dup_metadata_KBytNERR%>%subset(Season_Tucker2021=="Winter"| Season_Tucker2021=="Summer")

SuWi_only$Spatial=factor(SuWi_only$Spatial, levels=c("Nearshore", "Transition", "Offshore"))


svg("Syn_Seasons.svg", height=2, width=6)
SYN<- ggplot(data=SuWi_only, aes(x=Season_Tucker2021, color=Spatial, y=SYN.mL))+geom_boxplot(outlier.shape = NA, color="black")+facet_wrap(~Spatial) 
SYN=SYN+ theme(legend.title = element_blank()) + ylab("SYN") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorspatial) +  theme(legend.position="none")
SYN
dev.off()
ggsave("Syn_Seasons.pdf", height=3, width=6)



svg("PEUK_Seasons.svg", height=2, width=6)
PEUK<- ggplot(data=SuWi_only, aes(x=Season_Tucker2021, y=PEUK.mL, color=Spatial))+geom_boxplot(outlier.shape = NA, color="black") +facet_wrap(~Spatial)
PEUK=PEUK+ theme(legend.title = element_blank()) + ylab("PEUK") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorspatial) +  theme(legend.position="none")
PEUK
dev.off()
ggsave("PEUK_Seasons.pdf", height=3, width=6)



svg("PRO_Seasons.svg", height=2, width=6)
PRO<- ggplot(data=SuWi_only, aes(x=Season_Tucker2021, y=PRO.mL, color=Spatial))+geom_boxplot(outlier.shape = NA, color="black")+facet_wrap(~Spatial)
PRO=PRO+ theme(legend.title = element_blank()) + ylab("PRO") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorspatial)  +  theme(legend.position="none") 
PRO
dev.off()
ggsave("PRO_Seasons.pdf", height=3, width=6)

names(SuWi_only)

svg("chla_Seasons.svg", height=2, width=6)
chla<- ggplot(data=SuWi_only, aes(x=Season_Tucker2021, y=chla.ug.per.L, color=Spatial))+geom_boxplot(outlier.shape = NA, color="black")+facet_wrap(~Spatial)
chla=chla+ theme(legend.title = element_blank()) + ylab("chla") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorspatial)  +  theme(legend.position="none") + scale_y_continuous(trans = log10_trans())
chla
dev.off()
ggsave("chla_Seasons.pdf", height=3, width=6)
#

names(SuWi_only)

svg("temp_Seasons.svg", height=2, width=6)
temp<- ggplot(data=SuWi_only, aes(x=Season_Tucker2021, y=SW.Temperature.at.site, color=Spatial))+geom_boxplot(outlier.shape = NA, color="black")+facet_wrap(~Spatial)
temp=temp+ theme(legend.title = element_blank()) + ylab("Temperature (C)") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorspatial)  +  theme(legend.position="none") 
temp
dev.off()
ggsave("temp_Seasons.pdf", height=3, width=6)
#+ scale_y_continuous(trans = log10_trans())

library(cowplot)
svg("Box_plot_Seasons.svg", width=3.5, height=8)
plot_grid(temp, SYN, PEUK, PRO, chla, align = 'hv',label_size = 12, ncol = 1, nrow=5)
dev.off()



```



#####
Figure 6b
#####


```{r}

##Correlation plot 
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)

enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Nearshore")%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL",
                                                       "Log_SYN.mL", 
                                                       "Log_PEUK.mL", 
                                                       "Log_HBACT.mL", 
                                                        "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day"))


no_dup_metadata_KBytNERR$Date1

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))



svg("Nearshore_Chla_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'circle', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'black', order = 'AOE',addgrid.col="black",col = rev(COL2('RdBu', 10)))

dev.off

##to get information about the p values can use this copy 
svg("numbers_Nearshore_Chla.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
```


#####
Figure 6c
#####

```{r}

#####
#### Correlation between nearshore HPLC pigments and water temperature

no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Nearshore")%>%dplyr::select(c("SampleID","Log_Chla", "SW.Temperature.at.site", "Fuco.Rel", "X19But.Rel", "X19Hex.Rel", "Allo.Rel", "Log_plus_Lut.Rel", "Log_Peri.Rel", "Prasino.Rel", "Zea.Rel", "Log_Dva.Rel"))


enviro_cor2=enviro_cor%>%drop_na()

sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
testRes$p
try=unlist(testRes)

## leave blank on non-significant coefficient
## add all correlation coefficients
svg("numbers.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
?corrplot()
??cor()


corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$p, 4))

svg("HPLC_Nearshore_Chla_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20',
 order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off


```






######
Figure 7
######


```{r}
##Use Lomb Scargle Periodograms to determine the seasonally significant taxa, within each of the community types
####This code was built from the work of https://github.com/adriaaulaICM/bbmo_niche_sea/
###repeat with subsetting for each environment


library(tidyverse)
library(dplyr)
glom_genus=tax_glom(All_phyto, "Genus")

##
bl.phy.asinh <- transform_sample_counts(glom_genus, function(x) asinh(x))
bl.phy.asinh <- subset_samples(bl.phy.asinh, Spatial=="Offshore")
nsamples(bl.phy.asinh)
ntaxa(bl.phy.asinh)
physeq.list <- NULL

psmelt_dplyr = function(physeq) {
  sd = data.frame(sample_data(physeq)) %>% 
    rownames_to_column("Sample")
  TT = data.frame(as(tax_table(physeq),'matrix')) %>%
    rownames_to_column("OTU")
  if(taxa_are_rows(physeq)){
    otu.table = data.frame(as(otu_table(physeq),"matrix"),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  } else {
    otu.table = data.frame(t(as(otu_table(physeq),"matrix")),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  }
  
  all <- otu.table %>%
    left_join(TT, by = "OTU") %>%
    gather_("Sample", "Abundance", setdiff(colnames(otu.table), "OTU")) %>%
    left_join(sd, by = "Sample") %>% 
    dplyr::select(Sample, Abundance, OTU, everything())
  
  return(all)
}


tsdf.0.1 <- bl.phy.asinh %>% 
  psmelt_dplyr() 


df.ts <- tsdf.0.1 %>% 
  arrange(decimal_dat) %>% 
  dplyr::select(decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)

library(lomb)

lomb.03  <-  df.ts %>% 
  map(~lsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5, from=0.16))


library(fdrtool)
lomb.sea.03.ofac <- tibble( asv = names(lomb.03),
                       pval = map_dbl(lomb.03, ~.x$p.value),
                       peak = map_dbl(lomb.03, ~.x$peak),
                       interval  = map(lomb.03, ~.x$peak.at),
                       int.min = map_dbl(interval, ~.[[2]]),
                       int.max = map_dbl(interval, ~.[[1]])) %>% 
  drop_na(int.max) %>%
  mutate(qval = fdrtool::fdrtool(pval, statistic = 'pvalue')$qval) %>% 
  filter(qval <= 0.01)%>%filter(int.max>0.75)%>%filter(int.max<1.25)

results.lomb03 <- lomb.03[lomb.sea.03.ofac  %>% pull(asv)]

write_rds(results.lomb03, 'Genus_Offshore_from016_lomb_v2.rds')


lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))

plotsPath = "Genus_periodoplots_from016_Offshore_SigASV.pdf"
pdf(file=plotsPath)  

periodoplots <- map(results.lomb03, ~tibble( scanned = .x$scanned,
                          power = .x$power)) %>% 
  bind_rows(.id = 'asv') %>% 
  split(.$asv) %>% 
  map(~ggplot(.x, aes(scanned, power)) + 
        geom_line(aes(group = asv)) + 
        facet_wrap(~asv) + 
        lil.strip)

periodoplots
dev.off()

##add information about the periodic ASVs to your tax table
tosubset=lomb.sea.03.ofac%>%dplyr::select(asv, qval)
tosubset$Genus_v2_Periodicity_Offshore="Genus_v2_periodic_Offshore"
names(tosubset)[1] <- "ASV"
names(tosubset)[2] <- "Genus_v2_Offshore_periodicity_qval"
tax12 = read.csv("taxonomy_table_clean_May2024.csv", sep=",") 


taxonomy_Phyto3=left_join(tax12,tosubset, by="ASV")
Genus_Offshore_Period=left_join(tosubset, tax12, by="ASV")

###note this code was re-ran for each each environment (e.g. Nearshore and Transition)

```




```{r}


colorsgenus=c("unclassified_Mamiellaceae"="#909F2D",
"Micromonas"="darkseagreen",
"Chloropicon"="#088F8F",
"Pyramimonas"="#3EB489", 
"Mamiella"="#64E986", 
"Prasinoderma"="chartreuse4",
"unclassified_Pycnococcaceae"="#D2DA9F", 
"Eutreptiella"="black",
"Teleaulax"="coral",
"Braarudosphaera"="#9E476C",
"Chrysochromulina"="red",
 "Isochrysis"="#DA70D6",
"Partenskyella"="#9D9997",
"Chaetoceros"="#CB9D06","unclassified_Polar-centric-Mediophyceae"="#FFCE03","Helicopedinella"="#B2AA93","Mesopedinella"="#714416","unclassified_Pelagomonadaceae"="#927E25",
"D_5__Prochlorococcus_MIT9313"="navy","D_5__Synechococcus_CC9902"="dodgerblue3","D_5__Cyanobium_PCC-6307"="deepskyblue","D_5__Atelocyanobacterium_(UCYN-A)"="#8D1CFF","D_5__Crocosphaera"="#94A2FA")

library(seecolor)
print_color(colorsgenus)

WiSum_only=subset_samples(rel2All_phyto, Season_Tucker2021=="Winter"| Season_Tucker2021=="Summer")
nsamples(WiSum_only)

WiSu_ASV=psmelt(WiSum_only)


WiSu_ASV$Date1 <- as.Date(WiSu_ASV$Date1,format = "%m/%d/%y")

WiSu_Genera_almost=WiSu_ASV%>% group_by(Spatial, Season_Tucker2021, SampleID,Genus) %>% summarise(SumMeanAbun=sum(Abundance))
WiSu_Genera=WiSu_Genera_almost%>% group_by(Spatial, Season_Tucker2021, Genus) %>% summarise(SumMeanAbun2=mean(SumMeanAbun))


#WiSu_Genera$Genus=factor(WiSu_Genera$Genus, levels=c("unclassified_Mamiellaceae","Micromonas","Chloropicon","Pyramimonas", "Mamiella", "Prasinoderma","unclassified_Pycnococcaceae", "Eutreptiella","Teleaulax","Braarudosphaera","Chrysochromulina", "Isochrysis","Partenskyella","Chaetoceros","unclassified_Polar-centric-Mediophyceae","Helicopedinella","Mesopedinella","unclassified_Pelagomonadaceae","D_5__Prochlorococcus_MIT9313","D_5__Synechococcus_CC9902","D_5__Cyanobium_PCC-6307","D_5__Atelocyanobacterium_(UCYN-A)","D_5__Crocosphaera"))

WiSu_Genera_Seasonal_all=WiSu_Genera%>%subset( Genus== "D_5__Atelocyanobacterium_(UCYN-A)" |
 Genus== "Micromonas" |
 Genus== "D_5__Crocosphaera" |
 Genus== "unclassified_Pelagomonadaceae" |
 Genus== "D_5__Synechococcus_CC9902" |
 Genus== "Braarudosphaera" |Genus== "Prasinoderma" |
 Genus== "Pyramimonas" |  Genus== "D_5__Synechococcus_CC9902" | Genus== "Braarudosphaera" | Genus== "D_5__Cyanobium_PCC-6307" | Genus=="Chaetoceros"|
Genus=="unclassified_Mamiellaceae"|
Genus=="unclassified_Polar-centric-Mediophyceae"|
Genus=="D_5__Cyanobium_PCC-6307"| Genus=="Micromonas"|Genus=="Chloropicon"| Genus=="Teleaulax"|Genus=="D_5__Prochlorococcus_MIT9313"| Genus=="Eutreptiella"| Genus=="unclassified_Pycnococcaceae"|
Genus=="Mesopedinella"|
Genus=="Isochrysis"|
Genus=="Partenskyella"|
Genus=="Helicopedinella"|
Genus=="Chrysochromulina"|
Genus=="Braarudosphaera"|
Genus=="Mamiella" | Genus== "D_5__Synechococcus_CC9902")

stats_WiSu_Genera_Seasonal_all=reshape2::dcast(WiSu_Genera_Seasonal_all, Spatial+Genus~Season_Tucker2021, value.var = "SumMeanAbun2")



Offshore=WiSu_Genera%>%subset( 
 Genus== "D_5__Atelocyanobacterium_(UCYN-A)" |
 Genus== "Micromonas" |
 Genus== "D_5__Crocosphaera" |
 Genus== "unclassified_Pelagomonadaceae" |
 Genus== "D_5__Synechococcus_CC9902" |
 Genus== "Braarudosphaera" )



Three_t=subset(Offshore, Spatial=="Offshore")
p_3 <- ggplot(data=Three_t, aes(x=Season_Tucker2021, y=SumMeanAbun2*100, fill=as.factor(Genus))) + geom_bar(aes(), stat="identity", position="stack")  + scale_fill_manual(values=colorsgenus)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean relative abundance (%) \n of seasonal genera") +ylim(0,100) 

p_3

ggsave("seasonal_taxa_only_Su_Wi_Offshore.pdf")



Transiton=WiSu_Genera%>%subset(  Genus== "Prasinoderma" |
 Genus== "Pyramimonas" |  Genus== "D_5__Synechococcus_CC9902" | Genus== "Braarudosphaera" | Genus== "D_5__Cyanobium_PCC-6307" )



Three_t=subset(Transiton, Spatial=="Transition")
p_2 <- ggplot(data=Three_t, aes(x=Season_Tucker2021, y=SumMeanAbun2*100, fill=as.factor(Genus))) + geom_bar(aes(), stat="identity", position="stack")  + scale_fill_manual(values=colorsgenus)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean relative abundance (%) \n of seasonal genera") +ylim(0,100) 

p_2

ggsave("seasonal_taxa_only_Su_Wi_Transition.pdf")


Nearshore=WiSu_Genera%>%subset( Genus=="Chaetoceros"|
Genus=="unclassified_Mamiellaceae"|
Genus=="unclassified_Polar-centric-Mediophyceae"|
Genus=="D_5__Cyanobium_PCC-6307"| Genus=="Micromonas"|Genus=="Chloropicon"| Genus=="Teleaulax"|Genus=="D_5__Prochlorococcus_MIT9313"| Genus=="Eutreptiella"| Genus=="unclassified_Pycnococcaceae"|
Genus=="Mesopedinella"|
Genus=="Isochrysis"|
Genus=="Partenskyella"|
Genus=="Helicopedinella"|
Genus=="Chrysochromulina"|
Genus=="Braarudosphaera"|
Genus=="Mamiella" | Genus== "D_5__Synechococcus_CC9902")



svg("seasonal_taxa_only_Su_Wi_Nearshore_legend.svg", height=5, width=13)
Three_t=subset(Nearshore, Spatial=="Nearshore")
p_1<- ggplot(data=Three_t, aes(x=Season_Tucker2021, y=SumMeanAbun2*100, fill=as.factor(Genus))) + geom_bar(aes(), stat="identity", position="stack") + scale_fill_manual(values=colorsgenus)+ theme(legend.position = "bottom")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean relative abundance (%) \n of seasonal genera")+ylim(0,100) 
p_1
dev.off
ggsave("seasonal_taxa_only_Su_Wi_Nearshore_legend.pdf", height=5, width=13)




library(cowplot)
svg("Seasonal_Taxa_Only_plots_SuWi_Spatial_clusters_v3.svg", height=3, width=5)
plot_grid(p_1,p_2, p_3, ncol=3, nrow=1)
dev.off()
ggsave("Seasonal_Taxa_Only_plots_SuWi_Spatial_clusters.pdf", width=5, height=3)



```


```{r}
## having identified the seasonally significant taxa above use the code below to graph

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 

rel_glom=tax_glom(rel2All_phyto, "Genus")
ntaxa(rel_glom)

colorsgenus=c("unclassified_Pycnococcaceae"="#D2DA9F", 
"D_5__Prochlorococcus_MIT9313"="navy",
"Mesopedinella"="#714416",
"Chaetoceros"="#CB9D06",
"Chloropicon"="#088F8F",
"Teleaulax"="coral",
"unclassified_Mamiellaceae"="#909F2D",
"Micromonas"="darkseagreen",
"Isochrysis"="#DA70D6",
"Partenskyella"="#9D9997",
"unclassified_Polar-centric-Mediophyceae"="#FFCE03",
"Helicopedinella"="#B2AA93",
"Eutreptiella"="black",
"D_5__Synechococcus_CC9902"="dodgerblue3",
"Chrysochromulina"="red",
"Braarudosphaera"="#9E476C",
"D_5__Cyanobium_PCC-6307"="deepskyblue",
"Mamiella"="#64E986","Prasinoderma"="chartreuse4", "Pyramimonas"="#3EB489", "D_5__Crocosphaera"="#94A2FA","unclassified_Pelagomonadaceae"="#927E25","D_5__Atelocyanobacterium_(UCYN-A)"="#8D1CFF", "Micromonas"="darkseagreen")

library(seecolor)
print_color(colorsgenus)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Nearshore"="#B70D6A")

#determing which taxa are abundant (>0.005 (0.5%) aveerage relative abundance in the nearshore)
Rel_glom_Phyto_Nearshore=subset_samples(rel_glom, Spatial=="Nearshore")
abund_Nearshore<-Rel_glom_Phyto_Nearshore %>% psmelt() %>% dplyr::group_by(Genus, Class, Supergroup)%>%summarise(mean=mean(Abundance))%>% dplyr::filter(mean >0.005)

##note of this list only some are seasonal in the nearshore, so this list is subsetted futher for only seasonal taxa in the nearshore and plotted below 

######

Rel_glom_Phyto_Nearshore=subset_samples(rel_glom, Spatial=="Nearshore")
Rel_glom_Phyto_Nearshore_2<-Rel_glom_Phyto_Nearshore %>% psmelt() %>% dplyr::group_by( Month_abb, Supergroup, Genus, Class)

########

Rel_glom_Phyto_Nearshore_3=Rel_glom_Phyto_Nearshore_2%>%subset( Genus=="Chaetoceros"|
Genus=="unclassified_Polar-centric-Mediophyceae")

Rel_glom_Phyto_Nearshore_3$Month_abb=factor(Rel_glom_Phyto_Nearshore_3$Month_abb, levels=c( "Jan", "Feb", "Mar", "Apr","May", "Jun", "Jul", "Aug","Sep", "Oct", "Nov", "Dec"))

strameno<-ggplot(data=Rel_glom_Phyto_Nearshore_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
strameno

#######

Rel_glom_Phyto_Nearshore_3=Rel_glom_Phyto_Nearshore_2%>%subset(
Genus=="unclassified_Mamiellaceae"| Genus=="Micromonas"|Genus=="Chloropicon")

Rel_glom_Phyto_Nearshore_3$Month_abb=factor(Rel_glom_Phyto_Nearshore_3$Month_abb, levels=c("Jan", "Feb", "Mar", "Apr","May", "Jun", "Jul", "Aug","Sep", "Oct", "Nov", "Dec"))

Archi<-ggplot(data=Rel_glom_Phyto_Nearshore_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
Archi


#######

Rel_glom_Phyto_Nearshore_3=Rel_glom_Phyto_Nearshore_2%>%subset( Genus=="D_5__Synechococcus_CC9902"|
Genus=="D_5__Cyanobium_PCC-6307"| Genus=="D_5__Prochlorococcus_MIT9313")


Rel_glom_Phyto_Nearshore_3$Month_abb=factor(Rel_glom_Phyto_Nearshore_3$Month_abb, levels=c( "Jan", "Feb", "Mar", "Apr","May", "Jun", "Jul", "Aug","Sep", "Oct", "Nov", "Dec"))

cyano<-ggplot(data=Rel_glom_Phyto_Nearshore_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
cyano

#####


svg("by_abundance_genus_v2.svg", width=2, height=8)
plot_grid(cyano, Archi,strameno, align = 'hv',axis = "bt",label_size = 12, ncol = 1, nrow=3)
ggsave("by_abundance_genus_v2.pdf", width=2, height=8)
dev.off()


```



###now lets looks at diversity 
```{r}

sample_data(All_phyto)$Season_Tucker2021
win_sum=subset_samples(All_phyto, Season_Tucker2021=="Winter" | Season_Tucker2021=="Summer")
nsamples(win_sum)
win_sum_Nearshore=subset_samples(win_sum, Spatial=="Nearshore")

Nearshore_All_phyto=subset_samples(All_phyto, Spatial=="Nearshore")



library(DivNet)
library(tidyverse)

divnet_Nearshore_Season_2 <- win_sum_Nearshore %>%
  divnet(X = "Season_Tucker2021", ncores = 4)
divnet_Nearshore_Season_2$shannon


p=plot(divnet_Nearshore_Season_2$shannon, 
    win_sum_Nearshore, 
     col = "Season_Tucker2021")
p  

save(list="divnet_Nearshore_Season_2", file="divnet_Nearshore_Season_win_sum_SEP2023.rdata")

dv=divnet_Nearshore_Season_2
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) 

df
meta=as.data.frame(sample_data(win_sum_Nearshore))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Season_Tucker2021)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

gg=df2%>%arrange(desc(Site), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=gg%>%dplyr::group_by(Season_Tucker2021)%>%dplyr::summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

gg_aveerage$Spatial="Nearshore"

###repeat with Transition


library(DivNet)
library(tidyverse)
win_sum_Transition=subset_samples(win_sum, Spatial=="Transition")
Transition_All_phyto=subset_samples(All_phyto, Spatial=="Transition")

Transition_All_phyto=subset_samples(Transition_All_phyto, Season_Tucker2021=="Winter" | Season_Tucker2021=="Summer")

divnet_Transition_Season_2 <- win_sum_Transition %>%
  divnet(X = "Season_Tucker2021", ncores = 4)
divnet_Transition_Season_2$shannon


p=plot(divnet_Transition_Season_2$shannon, 
    win_sum_Transition, 
     col = "Season_Tucker2021")
p  

save(list="divnet_Transition_Season_2", file="divnet_Transition_Season_win_sum_SEP2023.rdata")

dv=divnet_Transition_Season_2
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) 

df
meta=as.data.frame(sample_data(win_sum_Transition))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Season_Tucker2021)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

gg=df2%>%arrange(desc(Site), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)


Transition_gg_aveerage=gg%>%dplyr::group_by(Season_Tucker2021)%>%dplyr::summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

Transition_gg_aveerage$Spatial="Transition"
###repeat with Offshore
library(DivNet)
library(tidyverse)
win_sum_Offshore=subset_samples(win_sum, Spatial=="Offshore")
Offshore_All_phyto=subset_samples(All_phyto, Spatial=="Offshore")

divnet_Offshore_Season_2 <- win_sum_Offshore %>%
  divnet(X = "Season_Tucker2021", ncores = 4)
divnet_Offshore_Season_2$shannon


p=plot(divnet_Offshore_Season_2$shannon, 
    win_sum_Offshore, 
     col = "Season_Tucker2021")
p  

save(list="divnet_Offshore_Season_2", file="divnet_Offshore_Season_win_sum_SEP2023.rdata")

dv=divnet_Offshore_Season_2

ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) 

df
meta=as.data.frame(sample_data(win_sum_Offshore))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Season_Tucker2021)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

gg=df2%>%arrange(desc(Site), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)


Offshore_gg_aveerage=gg%>%dplyr::group_by(Season_Tucker2021)%>%dplyr::summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

Offshore_gg_aveerage$Spatial="Offshore"



#######
test=rbind(gg_aveerage, Offshore_gg_aveerage)
test2=rbind(test, Transition_gg_aveerage)
test3=test2%>%unite("Season_Spatial", Spatial,Season_Tucker2021, remove=FALSE)
write.csv(test3, "shannons_season_tog_Summer_Winter.csv")


test3=read.csv("shannons_season_tog_Summer_Winter_May2024.csv")
test3$Season_Spatial=factor(test3$Season_Spatial, levels=c("Nearshore_Summer","Nearshore_Winter", "Transition_Summer", "Transition_Winter", "Offshore_Summer", "Offshore_Winter"))
test3$Spatial=factor(test3$Spatial, levels=c("Nearshore", "Transition",  "Offshore"))


svg("Season_Sum_winter_alpha_shannons_all_Sep2023.svg", width=2, height=2.4)
shannons=test3%>% ggplot(aes(x = Season_Tucker2021, xend=Season_Tucker2021, color=Spatial))+
  geom_point(aes(x = Season_Tucker2021, y = h0)) + facet_wrap(~Spatial)+
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() + scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+ylab("Shannon Diversity Estimate")+theme(axis.title=element_text(size=8, face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
shannons
dev.off()
ggsave("Season_Sum_winter_alpha_shannons_all_Sep2023.pdf", width=4, height=2)



Nearshore_Shannon_Season_2=as.data.frame(testDiversity(divnet_Nearshore_Season_2, "shannon"))
write.csv(Nearshore_Shannon_Season_2, "Nearshore_Season_Shannon_Sep2023.csv")

Transition_Shannon_Season_2=as.data.frame(testDiversity(divnet_Transition_Season_2, "shannon"))
write.csv(Transition_Shannon_Season_2, "Transition_Season_Shannon_Sep2023.csv")

Offshore_Shannon_Season_2=as.data.frame(testDiversity(divnet_Offshore_Season_2, "shannon"))
write.csv(Offshore_Shannon_Season_2, "Offshore_Season_Shannon_Sep2023.csv")


```

```{r}


####ASV richness
ba <- breakaway(win_sum_Nearshore)
ba

Nearshore_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_Nearshore, "Season_Tucker2021"))
Nearshore_Season_Tucker_bet=as.data.frame(Nearshore_Season_Tucker$table)
write.csv(Nearshore_Season_Tucker_bet, "richness_Nearshore_Season_Tucker.csv")


ba <- breakaway(win_sum_Transition)
ba

Transition_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_Transition, "Season_Tucker2021"))
Transition_Season_Tucker_bet=as.data.frame(Transition_Season_Tucker$table)
write.csv(Transition_Season_Tucker_bet, "richness_Transition_Season_Tucker.csv")



ba <- breakaway(win_sum_Offshore)
ba

Offshore_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_Offshore, "Season_Tucker2021"))
Offshore_Season_Tucker_bet=as.data.frame(Offshore_Season_Tucker$table)
write.csv(Offshore_Season_Tucker_bet, "richness_Offshore_Season_Tucker.csv")



plot(ba, win_sum_Offshore, color = "Season_Tucker2021")
ggsave("estimate_richness.pdf")

##

richness=read.csv("richness_Spatial_Season_Tucker_May2024.csv")
richness$Season_Spatial=factor(richness$Season_Spatial, levels=c("Nearshore_Summer","Nearshore_Winter", "Transition_Summer", "Transition_Winter", "Offshore_Summer", "Offshore_Winter"))
richness$Spatial=factor(richness$Spatial, levels=c("Nearshore", "Transition", "Offshore"))


svg("Season_Spatial_ASV_richness_Tucker2021_sep2023.svg", width=2, height=2.4)
ASVrich=richness%>%ggplot(aes(x = Season_Tucker2021, xend=Season_Tucker2021, color=Spatial))+
  geom_point(aes(x = Season_Tucker2021, y = est)) + facet_wrap(~Spatial)+
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text())+  theme(legend.position="none")+ scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))
ASVrich
dev.off()
ggsave("Season_Spatial_ASV_richness_Tucker2021_sep2023.pdf",width=5, height=2)


```


```{r}

library(cowplot)
svg("Seasonal_Alphadiveristy.svg", height=4, width=3.25)
plot_grid(shannons, ASVrich, ncol=1, nrow=2)
dev.off()
ggsave("Seasonal_Alphadiveristy.pdf", width=4, height=3.25)

```



############
Figure S1
############

```{r}
ok=read.csv("Per_IME.csv")
library(tidyverse)
ok2=ok%>%reshape2::melt(measure.vars=c("AR.STO1","SR8.STO1","CB.STO1","HP1.STO1","NB.STO1"        ,"SB.STO1","KAHO.STO1","WAI2.STO1","WAI2.NTO1", "AR.NTO1", "SR8.NTO1" ,"CB.NTO1" , "HP1.NTO1"     ,"NB.NTO1", "SB.NTO1" ,  "KAHO.NTO1"))


ok3=ok2%>%drop_na()%>%summarise(mean=mean(value), sd=sd(value), max=max(value), min=min(value))
svg("IME_Prevalence_v2.svg", height=5, width=4)
ggplot(ok2, aes(x=as.Date(Date1, format="%m/%d/%y"), y=value))+geom_point()+xlab("Date")+ylab("Delta chlorophyll a (ug per L)")+ theme(axis.text=element_text(size=10),axis.title=element_text(size=10), legend.text=element_text(size=10), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+
  geom_hline(yintercept=0, linetype='dotted', col = 'red')+ggtitle("Prevalence of IME: comparisons of stations in Nearshore areas and those offshore")
dev.off()
ggsave("IME_Prevalence_v2.pdf")
```


############
Figure S2
############

```{r}
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)
#no_dup_metadata_KBytNERR2=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Nearshore")
#no_dup_metadata_KBytNERR2=no_dup_metadata_KBytNERR2%>%subset(Site!="KAHO")
#no_dup_metadata_KBytNERR2=no_dup_metadata_KBytNERR2%>%subset(Site!="WAI2")

enviro_cor=no_dup_metadata_KBytNERR2%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL",
                                                       "Log_SYN.mL", 
                                                       "Log_PEUK.mL", 
                                                       "Log_HBACT.mL", 
                                                        "SW.Temperature.at.site", "Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"))

###"Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))

svg("All_wnut_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off
```


###Fig. S3. The relative abundance of major phytoplankton groups differs across the three community types. Phytoplankton classes and genera with >1% average relative abundance in at least one community type are specified. Within each community type samples are grouped first by sampling event (36 total between 2017-2021), then by the station.
```{r}
##Identify genera with >1% average relative abundance in at least one community type are specified.
rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID, Spatial, Genus, Curated) %>% dplyr::summarise(sum = sum(Abundance)) 
rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Spatial, Genus, Curated) %>% dplyr::summarise(mean = mean(sum)*100, sd=sd(sum)*100) 

cure=rel2Phyto_ASV_Cluster%>%filter(mean >1)
write.csv(cure, "cure.csv")
##these were labelled so that they are easy to identify and plot using the Curated ID 

```



```{r}

##okay moving forward with our Curated taxonomy 

test=psmelt(rel2All_phyto)
ntaxa(rel2All_phyto)
nsamples(rel2All_phyto)


test$Curated=factor(test$Curated, levels=c("Archaeplastida:Tetraselmis",
"Archaeplastida:Mamiellophyceae",
"Archaeplastida:Ostreococcus",
"Archaeplastida:Micromonas",
"Archaeplastida:Chloropicon",
"Archaeplastida:Other",
"Hacrobia:Other",
"Hacrobia:Cryptophyceae",
"Hacrobia:Teleaulax",
"Hacrobia:Prymnesiophyceae",
"Stramenopiles:Bacillariophyta",
"Stramenopiles:Chaetoceros",
"Stramenopiles:Aureococcus",
"Stramenopiles:Other",
"Cyanobacteria:Prochlorococcus",
"Cyanobacteria:Synechococcus",
"Cyanobacteria:Cyanobium",
"Cyanobacteria:Other",'Eukaryota:Other'))




coloresbarplot2=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="#00A300",
"Archaeplastida:Ostreococcus"="#688873",
"Archaeplastida:Micromonas"="#0CBAA6",
"Archaeplastida:Chloropicon"="darkolivegreen3", 
"Archaeplastida:Other"="#bef4dc", 
"Hacrobia:Other"="#EDABEF",
"Hacrobia:Cryptophyceae"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiophyceae"="red",
"Stramenopiles:Bacillariophyta"="gold",
"Stramenopiles:Chaetoceros"="#fffdd0",
"Stramenopiles:Aureococcus"="#C3AD30",
"Stramenopiles:Other"="#714416",
'Eukaryota:Other'="black",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")



test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")

names(test)


Three_t=subset(test, Spatial=="Offshore")
p_3 <- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_text(angle=90)) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance (%)") +scale_x_discrete(breaks=c("AA_A_NTAu0041",
"AB_A_NT61_R2",
"AC_C_NRNo0082",
"AD_A_NTDe0101",
"AE_A_NTJa0123",
"AF_A_NTFe0143",
"AG_A_NTMa0163",
"AH_A_NTMy0183",
"AI_B_STJn0202",
"AJ_B_STJn0222",
"AK_A_NTJy0251",
"AO_B_STJa0360",
"AP_A_NTFe0393",
"AQ_A_NTMa0425",
"AR_B_STAp0480",
"AT_A_NTJn0504",
"AU_B_STAu0523",
"AW_A_NTOc0564",
"AX_A_NTDe0584",
"AY_A_NTJa0604",
"AY_C_NRJa0605",
"AZ_A_NTMa0624",
"BB_A_NTJy0664",
"BC_A_NTAu0684",
"BE_A_NTNo0724",
"BF_A_NTDe0744",
"BG_A_NTFe0764",
"BH_A_NTAp0784",
"BI_B_STMy0813",
"BJ_A_NTJu0834"), labels=c("AA_A_NTAu0041"="08/23/17",
"AB_A_NT61_R2"="09/20/17",
"AC_C_NRNo0082"="11/01/17",
"AD_A_NTDe0101"="12/08/17",
"AE_A_NTJa0123"="01/29/18",
"AF_A_NTFe0143"="02/21/18",
"AG_A_NTMa0163"="03/28/18",
"AH_A_NTMy0183"="05/01/18",
"AI_B_STJn0202"="06/01/18",
"AJ_B_STJn0222"="06/19/18",
"AK_A_NTJy0251"="07/23/18",
"AO_B_STJa0360"="01/11/19",
"AP_A_NTFe0393"="02/25/19",
"AQ_A_NTMa0425"="03/21/19",
"AR_B_STAp0480"="04/22/19",
"AT_A_NTJn0504"="06/14/19",
"AU_B_STAu0523"="08/13/19",
"AW_A_NTOc0564"="10/07/19",
"AX_A_NTDe0584"="12/09/19",
"AY_A_NTJa0604"="01/24/20",
"AY_C_NRJa0605"="01/24/20",
"AZ_A_NTMa0624"="03/20/20",
"BB_A_NTJy0664"="07/21/20",
"BC_A_NTAu0684"="08/28/20",
"BE_A_NTNo0724"="11/16/20",
"BF_A_NTDe0744"="12/23/20",
"BG_A_NTFe0764"="02/12/21",
"BH_A_NTAp0784"="04/16/21",
"BI_B_STMy0813"="05/24/21",
"BJ_A_NTJu0834"="06/21/21"))
p_3

ggsave("Cluster_3_all_Dates.pdf")


Three_t=subset(test, Spatial=="Transition")
p_2 <- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_text(angle=90)) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance (%)") +scale_x_discrete(breaks=c("AB_C_NRSe0062",
"AC_A_NTNo0081",
"AD_C_NRDe0102",
"AE_C_NRJa0124",
"AF_B_STFe0142",
"AG_C_NRMa0164",
"AH_C_NRMy0184",
"AI_A_NTJn0203",
"AJ_A_NTJn0223",
"AK_C_NRJy0252",
"AL_A_NTSe0271",
"AM_A_NT297_R2",
"AO_A_NTJa0361",
"AP_D_R2Fe0391",
"AR_A_NTAp0457",
"AS_B_STMy0483",
"AT_C_NRJn0505",
"AU_A_NTAu0524",
"AV_A_NTAu0544",
"AW_B_STOc0563",
"AX_D_R2De0582",
"AY_D_R2Ja0602",
"AZ_C_NRMa0625",
"BA_A_NTMy0644",
"BB_C_NRJy0665",
"BC_C_NRAu0685",
"BD_A_NTSe0704",
"BE_B_STNo0723",
"BF_C_NRDe0745",
"BH_B_STAp0783",
"BI_A_NTMy0814",
"BJ_B_STJu0833"), labels=c("AB_C_NRSe0062"="09/20/17",
"AC_A_NTNo0081"="11/01/17",
"AD_C_NRDe0102"="12/08/17",
"AE_C_NRJa0124"="01/29/18",
"AF_B_STFe0142"="02/21/18",
"AG_C_NRMa0164"="03/28/18",
"AH_C_NRMy0184"="05/01/18",
"AI_A_NTJn0203"="06/01/18",
"AJ_A_NTJn0223"="06/19/18",
"AK_C_NRJy0252"="07/23/18",
"AL_A_NTSe0271"="09/11/18",
"AM_A_NT297_R2"="10/11/18",
"AO_A_NTJa0361"="01/11/19",
"AP_D_R2Fe0391"="02/25/19",
"AR_A_NTAp0457"="04/22/19",
"AS_B_STMy0483"="05/21/19",
"AT_C_NRJn0505"="06/14/19",
"AU_A_NTAu0524"="08/13/19",
"AV_A_NTAu0544"="08/29/19",
"AW_B_STOc0563"="10/07/19",
"AX_D_R2De0582"="12/09/19",
"AY_D_R2Ja0602"="01/24/20",
"AZ_C_NRMa0625"="03/20/20",
"BA_A_NTMy0644"="05/14/20",
"BB_C_NRJy0665"="07/21/20",
"BC_C_NRAu0685"="08/28/20",
"BD_A_NTSe0704"="09/22/20",
"BE_B_STNo0723"="11/16/20",
"BF_C_NRDe0745"="12/23/20",
"BH_B_STAp0783"="04/16/21",
"BI_A_NTMy0814"="05/24/21",
"BJ_B_STJu0833"="06/21/21"))
p_2

ggsave("Cluster_2_all_Dates.pdf")

test$Date1
Three_t=subset(test, Spatial=="Nearshore")
p_1<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_text(angle=90)) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance(%)") +scale_x_discrete(breaks=c("AA_C_NRAu0042","AB_F_R857_R2","AC_F_R8No0077","AD_F_R8De0097","AE_E_NBJa0121","AF_D_R2Fe0141","AG_E_NBMa0165","AH_E_NBMy0185","AI_E_NBJn0205","AJ_C_NRJn0224","AK_E_NBJy0253","AL_E_NBSe0273","AM_D_R2Oc0295","AN_A_NTNo0329","AO_E_NBJa0363","AP_C_NRFe0394","AQ_C_NRMa0426","AR_E_NBAp0459","AS_A_NTMy0484","AT_E_NBJn0506","AU_E_NBAu0526","AV_E_NBAu0546","AW_F_R8Oc0570","AX_E_NBDe0586","AY_E_NBJa0606","AZ_B_STMa0623","BA_E_NBMy0646","BB_E_NBJy0666","BC_E_NBAu0686","BD_E_NBSe0706","BE_E_NBNo0726","BF_E_NBDe0746","BG_E_NBFe0766","BH_C_NRAp0785","BI_E_NBMy0816","BJ_C_NRJu0835"), labels=c("AA_C_NRAu0042"="08/23/17",
"AB_F_R857_R2"="09/20/17",
"AC_F_R8No0077"="11/01/17",
"AD_F_R8De0097"="12/08/17",
"AE_E_NBJa0121"="01/29/18",
"AF_D_R2Fe0141"="02/21/18",
"AG_E_NBMa0165"="03/28/18",
"AH_E_NBMy0185"="05/01/18",
"AI_E_NBJn0205"="06/01/18",
"AJ_C_NRJn0224"="06/19/18",
"AK_E_NBJy0253"="07/23/18",
"AL_E_NBSe0273"="09/11/18",
"AM_D_R2Oc0295"="10/11/18",
"AN_A_NTNo0329"="11/09/18",
"AO_E_NBJa0363"="01/11/19",
"AP_C_NRFe0394"="02/25/19",
"AQ_C_NRMa0426"="03/21/19",
"AR_E_NBAp0459"="04/22/19",
"AS_A_NTMy0484"="05/21/19",
"AT_E_NBJn0506"="06/14/19",
"AU_E_NBAu0526"="08/13/19",
"AV_E_NBAu0546"="08/29/19",
"AW_F_R8Oc0570"="10/07/19",
"AX_E_NBDe0586"="12/09/19",
"AY_E_NBJa0606"="01/24/20",
"AZ_B_STMa0623"="03/20/20",
"BA_E_NBMy0646"="05/14/20",
"BB_E_NBJy0666"="07/21/20",
"BC_E_NBAu0686"="08/28/20",
"BD_E_NBSe0706"="09/22/20",
"BE_E_NBNo0726"="11/16/20",
"BF_E_NBDe0746"="12/23/20",
"BG_E_NBFe0766"="02/12/21",
"BH_C_NRAp0785"="04/16/21",
"BI_E_NBMy0816"="05/24/21",
"BJ_C_NRJu0835"="06/21/21"))

#guide = guide_axis(n.dodge = 2)
p_1


ggsave("Cluster_1_all_Dates.pdf", width=20)


library(cowplot)
svg("plots_over_Time_Spatial_clusters_all_Dates_April2024.svg", height=7, width=7)
plot_grid(p_1,p_2, p_3, ncol=1, nrow=3)
dev.off()
ggsave("Clusters_color_Ordered_Site_Sample_all_Dates_April2024.pdf", width=6.8, height=5.5)
```






############
Figure S4. Average nucleotide identity (ANI) between closely related Prochlorococcus Clade HLII isolate genomes (a), as well as those for Synechococcus Clade II isolate genomes (b).
############


```{r}
###ANI input comes from metagenomic section 

ani=read.delim("ANIb_percentage_identity.txt", row.names = 1)
ani[ani >= 0.99] <- NA

ani2=ani
ani2$genome_name=rownames(ani)
long_ani=melt(ani2)


ani_cyano_min=long_ani%>%summarise(ani_min=round(min(value, na.rm = TRUE),3), ani_mean=round(mean(value, na.rm = TRUE),3), ani_max=round(max(value, na.rm = TRUE),3))

###Min ANI by MinorClade
Taxonomy=read.delim("Cyano_Genera_genome_info_layer.txt")
names(Taxonomy)


MinorClade_Tax=Taxonomy%>%dplyr::select("MinorClade", "samples")%>%dplyr::rename("variable"= "samples", "MinorClade_variable"="MinorClade")
long_ani2=left_join(long_ani, MinorClade_Tax)

names(long_ani2)

MinorClade_Tax2=Taxonomy%>%dplyr::select("MinorClade", "samples")%>%dplyr::rename(  "genome_name"="samples")

MinorClade_ani=left_join(long_ani2, MinorClade_Tax2)



min_ani_genera=MinorClade_ani%>%dplyr::filter(MinorClade_variable==MinorClade)%>%dplyr::group_by(MinorClade_variable)%>%dplyr::summarise(ani_min=round(min(value, na.rm = TRUE),3), ani_mean=round(mean(value, na.rm = TRUE),3))
names(MinorClade_ani)


order_genome=factor(MinorClade_ani$genome_name, level=c("MED4",
"MIT9515",
"AS9601",
"SB",
"MIT9301",
"MIT9314",
"MIT9215",
"MIT9202",
"GP2",
"MIT9201",
"MIT9321",
"MIT9302",
"MIT9311",
"MIT9107",
"NATL2A",
"NATL1A",
"PAC1",
"CCMP1375",
"MIT9211",
"MIT9303",
"MIT9313",
"CC9605",
"WH8109",
"N32",
"HIMB2401",
"N5",
"N26",
"N19",
"UW86",
"KORDI52",
"UW106",
"UW69",
"CC9902",
"BL107",
"WH8102",
"CC9616",
"KORDI100",
"KORDI49",
"MITS9509",
"MITS9508",
"GEYO",
"UW179A",
"WH7803",
"WH7805",
"CC9311",
"WH8020",
"WH8016",
"UW179B",
"UW105",
"UW140",
"RS9916",
"RS9917",
"PCC7001",
"CB0101",
"PCC6307",
"RCC307"))


order_variable=factor(MinorClade_ani$variable, level=c("MED4",
"MIT9515",
"AS9601",
"SB",
"MIT9301",
"MIT9314",
"MIT9215",
"MIT9202",
"GP2",
"MIT9201",
"MIT9321",
"MIT9302",
"MIT9311",
"MIT9107",
"NATL2A",
"NATL1A",
"PAC1",
"CCMP1375",
"MIT9211",
"MIT9303",
"MIT9313",
"CC9605",
"WH8109",
"N32",
"HIMB2401",
"N5",
"N26",
"N19",
"UW86",
"KORDI52",
"UW106",
"UW69",
"CC9902",
"BL107",
"WH8102",
"CC9616",
"KORDI100",
"KORDI49",
"MITS9509",
"MITS9508",
"GEYO",
"UW179A",
"WH7803",
"WH7805",
"CC9311",
"WH8020",
"WH8016",
"UW179B",
"UW105",
"UW140",
"RS9916",
"RS9917",
"PCC7001",
"CB0101",
"PCC6307",
"RCC307"))




MinorClade_ani2=MinorClade_ani
MinorClade_ani2[is.na(MinorClade_ani2)] <- 1

class(MinorClade_ani2$value)

svg("ani_cyano.svg", height=6.5, width=10)
geom.text.size.ani=2
ani_all=ggplot(MinorClade_ani2, aes(x = order_genome, y = order_variable, fill = value)) +
    geom_tile(aes(fill = value)) + geom_text(aes(label = round(value, 2)), size=geom.text.size.ani) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) +
  scale_fill_stepsn(colors =c("white","#999999","green","#6666FF", "#FFFF66","#FF6633"),
                    breaks = c(0.01,0.85, 0.90,0.95, 1.00))+ labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=7)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=7),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_blank(), legend.text=element_text(size=7), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")

ani_all

dev.off()

ggsave("ani_cyano.pdf", height=6.5, width=10)


###only HL_II
HLII_only=MinorClade_ani%>%subset(MinorClade=="HL_II" & MinorClade_variable=="HL_II")
HLII_only[is.na(HLII_only)] <- 1


HLII_genome=factor(HLII_only$genome_name, level=c("AS9601",
"SB",
"MIT9301",
"MIT9314",
"MIT9215",
"MIT9202",
"GP2",
"MIT9201",
"MIT9321",
"MIT9302",
"MIT9311",
"MIT9107"))


HLII_variable=factor(HLII_only$variable, level=c(
"AS9601",
"SB",
"MIT9301",
"MIT9314",
"MIT9215",
"MIT9202",
"GP2",
"MIT9201",
"MIT9321",
"MIT9302",
"MIT9311",
"MIT9107"))

geom.text.size.ani=2
HLII_ani=ggplot(HLII_only, aes(x = HLII_genome, y = HLII_variable, fill = value)) +
    geom_tile(aes(fill = value)) + geom_text(aes(label = round(value, 2)), size=geom.text.size.ani) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) +
  scale_fill_stepsn(colors =c("#6666FF", "#FFFF66","#FF6633"),
                    breaks = c(0.85, 0.92,0.95, 1.00))+ labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=7)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=7),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_blank(), legend.text=element_text(size=7), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+ggtitle("Prochlorococcus Clade HLII ANI")
HLII_ani




###only II
II_only=MinorClade_ani%>%subset(MinorClade=="II" & MinorClade_variable=="II")
II_only[is.na(II_only)] <- 1


II_genome=factor(II_only$genome_name, level=c("CC9605",
"WH8109",
"N32",
"HIMB2401",
"N5",
"N26",
"N19",
"UW86"))


II_variable=factor(II_only$variable, level=c(
"CC9605",
"WH8109",
"N32",
"HIMB2401",
"N5",
"N26",
"N19",
"UW86"))

geom.text.size.ani=2
II_ani=ggplot(II_only, aes(x = II_genome, y = II_variable, fill = value)) +
    geom_tile(aes(fill = value)) + geom_text(aes(label = round(value, 2)), size=geom.text.size.ani) +
  theme(axis.title=element_text(size=6,face="bold"), legend.text=element_text(size=6), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1), legend.position="none", axis.text.x=element_blank()) +
  scale_fill_stepsn(colors =c("#6666FF", "#FFFF66","#FF6633"),
                    breaks = c(0.90, 0.93, 0.95,1.00))+ labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=7)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=7),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_blank(), legend.text=element_text(size=7), panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+ggtitle("Synechococcus Clade II (SC 5.1) ANI")
II_ani


library(cowplot)
svg("ANI_CladeII_HLII.svg", width=5, height=8)
plot_grid(HLII_ani, II_ani, labels = c('a)', 'b)'), align = 'hv',label_size = 10, ncol = 1)
dev.off()
ggsave("ANI_CladeII_HLII.pdf", width=5, height=8)



```






############
Figure S5
############



```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)


test=no_dup_metadata_KBytNERR%>%subset(Site=="HP1")

rain=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), Rainfall.mm.7.day)) +
  geom_point()+ geom_line()+theme(axis.text.x=element_blank(), axis.title.x=element_blank())+ylab("Rainfall \n (mm)")+xlab("Date")
rain

temp=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), SW.Temperature.at.site)) +
  geom_point()+ geom_line()+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ylab("Temperature \n (C)")+xlab("Date")
temp

wind=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), Wind.Speed.m.s.7.day)) +
  geom_point()+ geom_line()+theme(axis.text.x=element_blank(), axis.title.x=element_blank())+ylab("Wind speed \n (m/s)")+xlab("Date")
wind
test=no_dup_metadata_KBytNERR%>%subset(Spatial=="Nearshore")
chla=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), chla.ug.per.L)) +
  geom_point(aes(color=Site))+theme(axis.text.x = element_blank(), axis.title.x=element_blank())+scale_y_continuous(trans = log10_trans())+scale_color_manual(values=colorssite2)+
  theme(legend.position="none")+ylab("Chlorophyll a \n (ug/L)")+xlab("Date")
chla


chla2=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), chla.ug.per.L)) +
  geom_point(aes(color=Site))+theme(axis.text.x = element_blank(), axis.title.x=element_blank())+ylim(c(0,4))+scale_color_manual(values=colorssite2)+
  theme(legend.position="none")+ylab("Chlorophyll a \n (ug/L)")+xlab("Date")
chla2

library(cowplot)
svg("chla_weather_correct.svg", width=5, height=8)
plot_grid(chla,chla2,wind,rain,temp, labels = c('A)', 'B)', "C)", "D)", "E)"), align = 'hv',label_size = 10, ncol = 1)
dev.off()
ggsave("chla_wind_v2.pdf", width=5, height=8)




```



####
Figure S6
#######

```{r}
###repeat with transition
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Offshore")%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL",
                                                       "Log_SYN.mL", 
                                                       "Log_PEUK.mL", 
                                                       "Log_HBACT.mL", 
                                                        "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day"))

###"Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"

no_dup_metadata_KBytNERR$Date1

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))

?corrplot()
??cor()

svg("Offshore_Chla_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off
```




####
Figure S7
#######

```{r}
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Nearshore")%>%dplyr::select(c("SampleID","Log_Chla", "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day", "Fuco.Rel", "X19But.Rel", "X19Hex.Rel", "Allo.Rel", "Log_plus_Lut.Rel", "Log_Peri.Rel", "Prasino.Rel", "Zea.Rel", "Log_Dva.Rel"))


enviro_cor2=enviro_cor%>%drop_na()

sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
try=unlist(testRes)

## leave blank on non-significant coefficient
## add all correlation coefficients
svg("numbers.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
?corrplot()
??cor()

svg("HPLC_Nearshore_Chla_09172023_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off


#############Transition
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)


enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Transition")%>%dplyr::select(c("SampleID","Log_Chla", "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day", "Fuco.Rel", "X19But.Rel", "X19Hex.Rel", "Allo.Rel", "Log_plus_Lut.Rel", "Log_Peri.Rel", "Prasino.Rel", "Zea.Rel", "Log_Dva.Rel"))


enviro_cor2=enviro_cor%>%drop_na()

sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
try=unlist(testRes)

## leave blank on non-significant coefficient
## add all correlation coefficients
svg("numbers_Transition.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
?corrplot()
??cor()

svg("HPLC_Transition_Chla_09172023_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off



########Offshore

no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Offshore")%>%dplyr::select(c("SampleID","Log_Chla", "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day", "Fuco.Rel", "X19But.Rel", "X19Hex.Rel", "Allo.Rel", "Log_plus_Lut.Rel", "Log_Peri.Rel", "Prasino.Rel", "Zea.Rel", "Log_Dva.Rel"))


enviro_cor2=enviro_cor%>%drop_na()

sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
try=unlist(testRes)
## leave blank on non-significant coefficient
## add all correlation coefficients
svg("numbers_Offshore.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
?corrplot()
??cor()

svg("HPLC_Offshore_Chla_09172023_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off
```


Summary Statistics 

###Table S1. Biogeochemical parameters across stations. Abbreviations: DIN:P: Dissolved inorganic nitrogen to phosphate.

```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)

Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Phosphate.n = n(), Phosphate.Mean=round(mean(Phosphate),2), Phosphate.sd=round(sd(Phosphate),2))
Phosphate_means

Nitrate_means=no_dup_metadata_KBytNERR%>% drop_na(NitrateNitrite)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(NitrateNitrite.n = n(), NitrateNitrite.Mean=round(mean(NitrateNitrite),2), NitrateNitrite.sd=round(sd(NitrateNitrite),2))
Nitrate_means

Silicate_means=no_dup_metadata_KBytNERR%>% drop_na(Silicate)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Silicate.n = n(), Silicate.Mean=round(mean(Silicate),2), Silicate.sd=round(sd(Silicate),2))
Silicate_means


Ammonia_means=no_dup_metadata_KBytNERR%>% drop_na(Ammonia)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Ammonia.n = n(), Ammonia.Mean=round(mean(Ammonia),2), Ammonia.sd=round(sd(Ammonia),2))
Ammonia_means


chla_means=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(chla.n = n(), chla.Mean=round(mean(chla.ug.per.L),1), chla.sd=round(sd(chla.ug.per.L),1))
chla_means

Salinity_means=no_dup_metadata_KBytNERR%>% drop_na(Salinity)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Salinity.n = n(), Salinity.Mean=round(mean(Salinity),1), Salinity.sd=round(sd(Salinity),1))
Salinity_means


Temp_means=no_dup_metadata_KBytNERR%>% drop_na(SW.Temperature.at.site)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(Temp.n = n(), Temp.Mean=round(mean(SW.Temperature.at.site),1), Temp.sd=round(sd(SW.Temperature.at.site),1))
Temp_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),1), pH.sd=round(sd(pH),1))
pH_means


PEUK_means=no_dup_metadata_KBytNERR%>% drop_na(PEUK.mL)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(PEUK.n = n(), PEUK.Mean=round(mean(PEUK.mL),0), PEUK.sd=round(sd(PEUK.mL),0))
PEUK_means

SYN_means=no_dup_metadata_KBytNERR%>% drop_na(SYN.mL)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(SYN.n = n(), SYN.Mean=round(mean(SYN.mL),0), SYN.sd=round(sd(SYN.mL),0))
SYN_means

PRO_means=no_dup_metadata_KBytNERR%>% drop_na(PRO.mL)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(PRO.n = n(), PRO.Mean=round(mean(PRO.mL),0), PRO.sd=round(sd(PRO.mL),0))
PRO_means


HBACT_means=no_dup_metadata_KBytNERR%>% drop_na(HBACT.mL)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(HBACT.n = n(), HBACT.Mean=round(mean(HBACT.mL),0), HBACT.sd=round(sd(HBACT.mL),0))
HBACT_means

names(no_dup_metadata_KBytNERR)

DIN_DIP_means=no_dup_metadata_KBytNERR%>% drop_na(DIN_DIP)%>%drop_na(Site)%>% dplyr::group_by(Site)%>%dplyr::summarise(DIN_DIP.n = n(), DIN_DIP.Mean=round(mean(DIN_DIP),2), DIN_DIP.sd=round(sd(DIN_DIP),2))
DIN_DIP_means


Site_mean_sd=do.call("cbind", list(Temp_means, pH_means,Salinity_means,chla_means,  PEUK_means,HBACT_means,SYN_means,PRO_means,Phosphate_means,Silicate_means, Nitrate_means))

write.csv(Site_mean_sd, "New_Clusters_UpdateFCM_Site_mean_sd.csv")
Site_mean_sd_10=do.call("cbind", list(Ammonia_means,DIN_DIP_means))
write.csv(Site_mean_sd_10, "Ammonia_DIN_Site_mean_sd.csv")



```


###Table S2.Isolate genomes used in metagenomic read recruitment analyses and clade identities. (No data analyses to report for this Table)


###Table S3 Comparisons of biogeochemical parameters across stations using one-way ANOVA with Tukey’s multiple comparisons test. Significance (Holm corrected p-values) are shown. Asterisks(*) denotes a log transformation to improve normality, while a carrot (^) denotes a log transformation plus 1.


```{r}
KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")
names(KBaloha)

Fuco=KBaloha%>%drop_na(Peri.Rel)
Fuco$Site=factor(Fuco$Site, levels=c("WAI2", "KAHO", "HP1", "AR", "NB", "SB", "CB", "SR8", "SR2", "NR2", "STO1", "NTO1", "ALOHA"))
#Fuco=Fuco%>%subset(Spatial=="Transition")


  amod <- aov(Peri.Rel~ Site, data =Fuco)
  amod
  summary(amod)


library("multcomp")
  g=glht(amod, linfct = mcp(Site= "Tukey"))
  summary(g, test = adjusted("holm"))

```


###Table S4 Phytoplankton pigments across stations. Abbreviations: Tchla: Total Chlorophyll a; Fuco: Fucoxanthin, Peri: Peridinin; Allo: Alloxanthin; Prasino: Prasinoxanthin; Hex: 19′- hexanoyloxyfucoxanthin; But: 19′-butanoyloxyfucoxanthin; Zea: zeaxanthin; DVchl: divinyl chlorophyll a
```{r}

no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)

Fuco=no_dup_metadata_KBytNERR2%>%drop_na(Fuco)%>%dplyr::group_by(Site)%>%dplyr::summarise(X19Butmean=round(mean(X19But.Rel),2),X19Butsd=round(sd(X19But.Rel),2),X19Hexmean=round(mean(X19Hex.Rel),2), X19Hexsd=round(sd(X19Hex.Rel),2),Allomean=round(mean(Allo.Rel),2), Allosd=round(sd(Allo.Rel),2),Fucomean=round(mean(Fuco.Rel),2), Fucosd=round(sd(Fuco.Rel),2),Lutmean=round(mean(Lut.Rel),2), Lutsd=round(sd(Lut.Rel),2),Perimean=round(mean(Peri.Rel),2), Perisd=round(sd(Peri.Rel),2),Prasinomean=round(mean(Prasino.Rel),2), Prasinosd=round(sd(Prasino.Rel),2),Zeamean=round(mean(Zea.Rel),2), Zeasd=round(sd(Zea.Rel),2),Dvamean=round(mean(Dva.Rel),2), Dvasd=round(sd(Dva.Rel),2))
write.csv(Fuco, "Update_enviro_Site_Rel_Sep2023.csv")
Fuco=no_dup_metadata_KBytNERR2%>%drop_na(Fuco)%>%dplyr::group_by(Site)%>%tally()

KBaloha=read.csv("KByT_Aloha_enviro_May2024.csv")
aloha=subset(KBaloha, Site=="ALOHA")
write.csv(aloha, "Aloha.csv")
Fuco=aloha%>%drop_na(Dva.Rel)%>%dplyr::group_by(Site)%>%dplyr::summarise(X19Butmean=round(mean(X19But.Rel),2),X19Butsd=round(sd(X19But.Rel),2),X19Hexmean=round(mean(X19Hex.Rel),2), X19Hexsd=round(sd(X19Hex.Rel),2),Allomean=round(mean(Allo.Rel),2), Allosd=round(sd(Allo.Rel),2),Fucomean=round(mean(Fuco.Rel),2), Fucosd=round(sd(Fuco.Rel),2),Lutmean=round(mean(Lut.Rel),2), Lutsd=round(sd(Lut.Rel),2),Perimean=round(mean(Peri.Rel),2), Perisd=round(sd(Peri.Rel),2),Prasinomean=round(mean(Prasino.Rel),2), Prasinosd=round(sd(Prasino.Rel),2),Zeamean=round(mean(Zea.Rel),2), Zeasd=round(sd(Zea.Rel),2),Dvamean=round(mean(Dva.Rel),2), Dvasd=round(sd(Dva.Rel),2))
write.csv(Fuco, "Update_enviro_Site_Rel_Sep2023_Aloha.csv")
Fuco=aloha%>%drop_na(Fuco)%>%dplyr::group_by(Site)%>%tally()



```


##Table S5 Differences in biogeochemical parameters among community types.
```{r}
##The following code was used to evaluate summary statistics of different parameters over time and space

library(multcomp)
no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)

  
Phosphate_means=no_dup_metadata_KBytNERR%>% drop_na(Phosphate)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(Phosphate.n = n(), Phosphate.Mean=round(mean(Phosphate),2), Phosphate.sd=round(sd(Phosphate),2))
Phosphate_means

Nitrate_means=no_dup_metadata_KBytNERR%>% drop_na(NitrateNitrite)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(NitrateNitrite.n = n(), NitrateNitrite.Mean=round(mean(NitrateNitrite),2), NitrateNitrite.sd=round(sd(NitrateNitrite),2))
Nitrate_means

Silicate_means=no_dup_metadata_KBytNERR%>% drop_na(Silicate)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(Silicate.n = n(), Silicate.Mean=round(mean(Silicate),2), Silicate.sd=round(sd(Silicate),2))
Silicate_means


Ammonia_means=no_dup_metadata_KBytNERR%>% drop_na(Ammonia)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(Ammonia.n = n(), Ammonia.Mean=round(mean(Ammonia),2), Ammonia.sd=round(sd(Ammonia),2))
Ammonia_means


chla_means=no_dup_metadata_KBytNERR%>% drop_na(chla.ug.per.L)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(chla.n = n(), chla.Mean=round(mean(chla.ug.per.L),2), chla.sd=round(sd(chla.ug.per.L),2))
chla_means

Salinity_means=no_dup_metadata_KBytNERR%>% drop_na(Salinity)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(Salinity.n = n(), Salinity.Mean=round(mean(Salinity),2), Salinity.sd=round(sd(Salinity),2))
Salinity_means


Temp_means=no_dup_metadata_KBytNERR%>% drop_na(SW.Temperature.at.site)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(Temp.n = n(), Temp.Mean=round(mean(SW.Temperature.at.site),2), Temp.sd=round(sd(SW.Temperature.at.site),2))
Temp_means


pH_means=no_dup_metadata_KBytNERR%>% drop_na(pH)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means


PEUK_means=no_dup_metadata_KBytNERR%>% drop_na(PEUK.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(PEUK.n = n(), PEUK.Mean=round(mean(PEUK.mL),2), PEUK.sd=round(sd(PEUK.mL),2))
PEUK_means

SYN_means=no_dup_metadata_KBytNERR%>% drop_na(SYN.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(SYN.n = n(), SYN.Mean=round(mean(SYN.mL),2), SYN.sd=round(sd(SYN.mL),2))
SYN_means

PRO_means=no_dup_metadata_KBytNERR%>% drop_na(PRO.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(PRO.n = n(), PRO.Mean=round(mean(PRO.mL),2), PRO.sd=round(sd(PRO.mL),2))
PRO_means


HBACT_means=no_dup_metadata_KBytNERR%>% drop_na(HBACT.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial)%>%dplyr::summarise(HBACT.n = n(), HBACT.Mean=round(mean(HBACT.mL),2), HBACT.sd=round(sd(HBACT.mL),2))
HBACT_means


Enviro_mean_sd=do.call("cbind", list(Temp_means, Salinity_means,pH_means,chla_means,HBACT_means, PRO_means, SYN_means, PEUK_means, Silicate_means, Phosphate_means, Nitrate_means,Ammonia_means))

write.csv(Enviro_mean_sd, "New_Clusters_UpdateFCM_DeSeq_Spatial_mean_sd.csv")

  
```


```{r}

no_dup_metadata_KBytNERR$Spatial=factor(no_dup_metadata_KBytNERR$Spatial, levels=c("Nearshore", "Transition", "Offshore"))

no_NA=no_dup_metadata_KBytNERR%>%drop_na(Spatial)%>%drop_na(Ammonia)
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(Ammonia~ Spatial, data = no_NA)
  amod

  g=glht(amod, linfct = mcp(Spatial= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
```



##Table S6 Alpha diversity metrics (Shannon’s and ASV richness) across community types
```{r}

library(DivNet)
library(tidyverse)



divnet_Spatial <-All_phyto %>%
  divnet(X = "Spatial", ncores = 4)
divnet_Spatial$shannon


p=plot(divnet_Spatial $shannon, 
     All_phyto, 
     col = "Spatial")
p  



save(list="divnet_Spatial", file="divnet_Spatial.rdata")


#load("divnet_Spatial.rdata")

divnet_Spatial$X
dv=divnet_Spatial
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$Spatial) 

df
meta=as.data.frame(sample_data(All_phyto))


df2=left_join(df, meta, by="SampleID")

gg_aveerage=df2%>%group_by(Spatial)%>%summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))
gg_aveerage$Spatial=factor(gg_aveerage$Spatial, levels=c("Nearshore", "Transition", "Offshore"))


svg("Spatial_alpha_shannons.svg", width=4, height=2)
Spatial_Shan=gg_aveerage%>% ggplot(aes(x = Spatial, xend = Spatial, color=Spatial))+
  geom_point(aes(x = Spatial, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+  theme(legend.position="none")+xlab("")+ scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+ylab("Shannon Diversity Estimate")+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
Spatial_Shan
dev.off()
ggsave("alpha_diversity_shannons_Spatial.pdf", width=4, height=2)

Shannon_Spatial=as.data.frame(testDiversity(divnet_Spatial, "shannon"))


write.csv(Shannon_Spatial, "Shannon_Spatial.csv")


ba <- breakaway(All_phyto)
ba

Spatial <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(All_phyto, "Spatial"))
Spatial_bet=as.data.frame(Spatial$table)
write.csv(Spatial_bet, "Spatial_bet.csv")

Spatial_bet$est

richness=read.csv("Spatial_richness.csv")
richness$Spatial=factor(richness$Spatial, levels=c("Nearshore", "Transition", "Offshore"))


svg("richness_Spatial.svg", width=4, height=2)
Spatial_Rich=richness%>%ggplot(aes(x = Spatial, xend=Spatial, color=Spatial))+
  geom_point(aes(x = Spatial, y = est)) +
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text())+  theme(legend.position="none")+ scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
Spatial_Rich
dev.off()
ggsave("Spatial_ASV_richness.pdf",width=4, height=2)


svg("Spatial_Alpha.svg", width=6, height=6)
plot_grid(Spatial_Shan,Spatial_Rich,labels = c('A', 'B'), align = 'hv',axis = "bt",label_size = 12, ncol = 1, nrow=2)
ggsave("Spatial_Alpha.pdf", width=6, height=6)
dev.off()


```


###Table S7. Significant differences in abundance across community types for phytoplankton classes. 
```{r}

glom=tax_glom(All_phyto, "Class")
ntaxa(glom)
dex = phyloseq::phyloseq_to_deseq2(glom, ~Spatial)

diagdds_pol = DESeq2::DESeq(dex, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(glom)[rownames(sigtab), ], "matrix"))
head(sigtab)



sigtab2=sigtab [,c(1:6,15)]
sigtab2$DeSeq_Spatial_Cluster=sigtab$Domain
sigtab2$DeSeq_Spatial_Cluster="DeSeq_Spatial_Variable"
sigtab2=as.data.frame(sigtab2)
taxonomy_Phyto_we = read.csv("taxonomy_table_clean_May2024.csv", sep=",") 
taxa_variable_spatial=left_join(taxonomy_Phyto_we,sigtab2,by="ASV")

##examine differential abundances of genera between each community type- this one is transition vs offshore 
res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Offshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Class_DESeq_All_Phyto_Clustering_NEWCLR_Transition_Offshore_April2024.csv")

Class_Transition_Offshore=sigtab

Class_Transition_Offshore$Class_Transition_Offshore="Class_Transition_Offshore"

names(Class_Transition_Offshore)[names(Class_Transition_Offshore) == 'log2FoldChange'] <- 'Class_TO_log2FoldChange'
names(Class_Transition_Offshore)[names(Class_Transition_Offshore) == 'padj'] <- 'Class_TO_padj'
names(Class_Transition_Offshore)[names(Class_Transition_Offshore) == 'baseMean'] <- 'Class_TO_baseMean'

Class_Transition_Offshore_v2=Class_Transition_Offshore%>%dplyr::select(Class, Class_TO_log2FoldChange, Class_TO_padj, Class_TO_baseMean,Class_Transition_Offshore)

names(Class_Transition_Offshore)



##examine differential abundances of genera between each community type- this one is transition vs Nearshore
res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Nearshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Class_DESeq_All_Phyto_Clustering_NEWCLR_Nearshore_Transition_April2024.csv")

Class_Nearshore_Transition=sigtab
Class_Nearshore_Transition$Class_Nearshore_Transition="Class_Nearshore_Transition"

names(Class_Nearshore_Transition)[names(Class_Nearshore_Transition) == 'log2FoldChange'] <- 'Class_CT_log2FoldChange'
names(Class_Nearshore_Transition)[names(Class_Nearshore_Transition) == 'padj'] <- 'Class_CT_padj'
names(Class_Nearshore_Transition)[names(Class_Nearshore_Transition) == 'baseMean'] <- 'Class_CT_baseMean'

Class_Nearshore_Transition_v2=Class_Nearshore_Transition%>%dplyr::select(Class, Class_CT_log2FoldChange, Class_CT_padj, Class_CT_baseMean,Class_Nearshore_Transition)



##examine differential abundances of genera between each community type- this one is Nearshore vs offshore

res <- results(diagdds_pol, contrast=c("Spatial", "Offshore","Nearshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Class_DESeq_All_Phyto_Clustering_NEWCLR_Nearshore_Offshore_April2024.csv")

Class_Nearshore_Offshore=sigtab
Class_Nearshore_Offshore$Class_Nearshore_Offshore="Class_Nearshore_Offshore"

names(Class_Nearshore_Offshore)[names(Class_Nearshore_Offshore) == 'log2FoldChange'] <- 'Class_CO_log2FoldChange'
names(Class_Nearshore_Offshore)[names(Class_Nearshore_Offshore) == 'padj'] <- 'Class_CO_padj'
names(Class_Nearshore_Offshore)[names(Class_Nearshore_Offshore) == 'baseMean'] <- 'Class_CO_baseMean'

Class_Nearshore_Offshore_v2=Class_Nearshore_Offshore%>%dplyr::select(Class, Class_CO_log2FoldChange, Class_CO_padj, Class_CO_baseMean,Class_Nearshore_Offshore)



COT=full_join(Class_Nearshore_Offshore_v2,Class_Nearshore_Transition_v2, by="Class")
COOT=full_join(COT,Class_Transition_Offshore_v2, by="Class")


###left join with the most recent taxonomy table to get information on taxonomy
taxonomy_Phyto_we = read.csv("taxonomy_table_clean_May2024.csv", sep=",") 
output=left_join(COOT,taxonomy_Phyto_we,by="Class")
output2=output%>%distinct(Class, .keep_all = TRUE)
write.csv(output2, "Class_all_comparisons_NEWCLR_Spatial_April2024.csv")

###note differentially abundant taxa that are unidentified at higher than the class level were not included 
##all this info was included in Table S7 

```


###Table S8. Significant differences in abundance across community types for phytoplankton genera. Analyses can be found in output of Figure 4.


####Table S9 Seasonal differences in biogeochemical parameters across community types. Comparisons of biogeochemical parameters across seasons using one-way ANOVA with Tukey’s multiple comparisons test. Significance (Holm corrected p-values) are shown. Asterisks(*) denotes a log transformation to improve normality, while a carrot (^) denotes a log transformation plus 1.

```{r}
no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERR_main_analyses.csv", header=TRUE, fill=TRUE)

win_sum=no_dup_metadata_KBytNERR%>%subset(Season_Tucker2021=="Winter"|Season_Tucker2021=="Summer")
dim(win_sum)

Phosphate_means=win_sum%>% drop_na(Phosphate)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(Phosphate.n = n(), Phosphate.Mean=round(mean(Phosphate),2), Phosphate.sd=round(sd(Phosphate),2))
Phosphate_means

Nitrate_means=win_sum%>% drop_na(NitrateNitrite)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(NitrateNitrite.n = n(), NitrateNitrite.Mean=round(mean(NitrateNitrite),2), NitrateNitrite.sd=round(sd(NitrateNitrite),2))
Nitrate_means

Silicate_means=win_sum%>% drop_na(Silicate)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(Silicate.n = n(), Silicate.Mean=round(mean(Silicate),2), Silicate.sd=round(sd(Silicate),2))
Silicate_means


Ammonia_means=win_sum%>% drop_na(Ammonia)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(Ammonia.n = n(), Ammonia.Mean=round(mean(Ammonia),2), Ammonia.sd=round(sd(Ammonia),2))
Ammonia_means


chla_means=win_sum%>% drop_na(chla.ug.per.L)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(chla.n = n(), chla.Mean=round(mean(chla.ug.per.L),2), chla.sd=round(sd(chla.ug.per.L),2))
chla_means

Salinity_means=win_sum%>% drop_na(Salinity)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(Salinity.n = n(), Salinity.Mean=round(mean(Salinity),2), Salinity.sd=round(sd(Salinity),2))
Salinity_means


Temp_means=win_sum%>% drop_na(SW.Temperature.at.site)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(Temp.n = n(), Temp.Mean=round(mean(SW.Temperature.at.site),2), Temp.sd=round(sd(SW.Temperature.at.site),2))
Temp_means


pH_means=win_sum%>% drop_na(pH)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(pH.n = n(), pH.Mean=round(mean(pH),2), pH.sd=round(sd(pH),2))
pH_means


PEUK_means=win_sum%>% drop_na(PEUK.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(PEUK.n = n(), PEUK.Mean=round(mean(PEUK.mL),2), PEUK.sd=round(sd(PEUK.mL),2))
PEUK_means

SYN_means=win_sum%>% drop_na(SYN.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(SYN.n = n(), SYN.Mean=round(mean(SYN.mL),2), SYN.sd=round(sd(SYN.mL),2))
SYN_means

PRO_means=win_sum%>% drop_na(PRO.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(PRO.n = n(), PRO.Mean=round(mean(PRO.mL),2), PRO.sd=round(sd(PRO.mL),2))
PRO_means


HBACT_means=win_sum%>% drop_na(HBACT.mL)%>%drop_na(Spatial)%>% dplyr::group_by(Spatial, Season_Tucker2021)%>%dplyr::summarise(HBACT.n = n(), HBACT.Mean=round(mean(HBACT.mL),2), HBACT.sd=round(sd(HBACT.mL),2))
HBACT_means

names(win_sum)

Wind_dg_means=win_sum%>% drop_na(Wind.Direction.dg.mean)%>%subset(Site=="HP1")%>% dplyr::group_by( Season_Tucker2021)%>%dplyr::summarise(Wind.Direction.dg.mean.n = n(), Wind.Direction.dg.mean.Mean=round(mean(Wind.Direction.dg.mean),2), Wind.Direction.dg.mean.sd=round(sd(Wind.Direction.dg.mean),2))


Wind_max_means=win_sum%>% drop_na(Wind.Speed.m.s.7.day)%>%subset(Site=="HP1")%>% dplyr::group_by(Season_Tucker2021)%>%dplyr::summarise(Max.Max.Windmean.n = n(), Max.Max.WindMean=round(mean(Wind.Speed.m.s.7.day),2), Max.Max.Windsd=round(sd(Wind.Speed.m.s.7.day),2))


Max.Rainfall=win_sum%>% drop_na(Rainfall.mm.7.day)%>%subset(Site=="HP1")%>% dplyr::group_by( Season_Tucker2021)%>%dplyr::summarise(Rainfallmean.n = n(), RainfallMean=round(mean(Rainfall.mm.7.day),2), Rainfallsd=round(sd(Rainfall.mm.7.day),2))

names(win_sum)


Enviro_mean_sd=do.call("cbind", list(Temp_means,pH_means, Salinity_means,chla_means,PEUK_means, HBACT_means,  SYN_means, PRO_means, Phosphate_means, Silicate_means, Nitrate_means,Ammonia_means))

write.csv(Enviro_mean_sd, "Oct_2023UpdateFCM_DeSeqSpatial_Season_TUCKER_WIN_SUM_mean_sd.csv")


Fuco=win_sum%>%dplyr::group_by(Spatial, Season_Tucker2021)%>%tally()
Nearshore_win_sum=subset(win_sum, Spatial=="Nearshore")
Transition_win_sum=subset(win_sum, Spatial=="Transition")
Offshore_win_sum=subset(win_sum, Spatial=="Offshore")


Fuco=Nearshore_win_sum%>%drop_na(Log_PEUK.mL)
Fuco$Season_Tucker2021=factor(Fuco$Season_Tucker2021, levels=c("Summer", "Winter"))
#Fuco=Fuco%>%subset(Spatial=="Transition")


  amod <- aov(Log_PEUK.mL~ Season_Tucker2021, data =Fuco)
  amod
  summary(amod)

  
library("multcomp")
  g=glht(amod, linfct = mcp(Season_Tucker2021= "Tukey"))
  summary(g, test = adjusted("holm"))

  
  
Fuco=Transition_win_sum%>%drop_na(Log_PEUK.mL)
Fuco$Season_Tucker2021=factor(Fuco$Season_Tucker2021, levels=c("Summer", "Winter"))
#Fuco=Fuco%>%subset(Spatial=="Transition")


  amod <- aov(Log_PEUK.mL~ Season_Tucker2021, data =Fuco)
  amod
  summary(amod)

 
library("multcomp")
  g=glht(amod, linfct = mcp(Season_Tucker2021= "Tukey"))
  summary(g, test = adjusted("holm"))

  
  
  
Fuco=Offshore_win_sum%>%drop_na(Log_PEUK.mL)
Fuco$Season_Tucker2021=factor(Fuco$Season_Tucker2021, levels=c("Summer", "Winter"))
#Fuco=Fuco%>%subset(Spatial=="Transition")


  amod <- aov(Log_PEUK.mL~ Season_Tucker2021, data =Fuco)
  amod
  summary(amod)

library("multcomp")
  g=glht(amod, linfct = mcp(Season_Tucker2021= "Tukey"))
  summary(g, test = adjusted("holm"))
 
  
  names(win_sum)
```

###Table S10. Alpha diversity metrics (Shannon’s and ASV richness) across community types and seasons.- for code see analyses for Figure 7.

###Table S11. Phytoplankton genera with significant seasonality as defined by Lomb-Scargle Periodicty test. FDR-corrected q-values reported.- see analyses for Figure 7






