---
title: "Qiime_2021_2017_final"
author: "Sarah Tucker"
date: "5/13/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

###There were two different MiSeq sequencing runs that contributed to this analysis: KByT Samples collected July 2019-June 2021 (published here) & KByT samples collected August 2017-June 2019 (published in Tucker et al., 2021 https://peerj.com/articles/12274/). In this file I conduct quality control and denoising via Qiime2 (Boylen et al., 2019) and Dada2 (Callahan et al., 2019) for each sequencing run and then merge output of the two datasets. I conduct taxonomy assignments with SILVA v138 (Quast et al., 2012) and for chloroplast and eukaryotic sequences I also classify with PR2 v4.14.0 (Guillou et al., 2013).



```{bash}
cd files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC-updated/BaseCalls/
for f in *R2_001.fastq.gz; do mv $f ${f%*R2_001.fastq.gz}L001_R2_001.fastq.gz; done

for f in *R1_001.fastq.gz; do mv $f ${f%*R1_001.fastq.gz}L001_R1_001.fastq.gz; done
```



```{python}

screen -S Miseq 
source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4
qiime tools import --show-importable-types
cd files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC/BaseCalls

qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC-updated/BaseCalls \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path demux-paired-end.qza 

```


```{bash, eval=FALSE}
qiime demux summarize \
  --i-data  demux-paired-end.qza   \
  --o-visualization demux-paired-end.qzv 
```



```{bash}
#double check there is no primer
qiime cutadapt trim-paired \
    --i-demultiplexed-sequences demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences demux-paired-end-trimmed.qza \
    --p-cores 20\
    --verbose
qiime demux summarize \
  --i-data demux-paired-end-trimmed.qza  \
  --o-visualization demux-paired-end-trimmed.qzv
```



```{bash}
qiime dada2 denoise-single \
  --i-demultiplexed-seqs demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences 251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table 251-2021-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats 251-2021-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 30
  
qiime metadata tabulate \
  --m-input-file 251-2021-single_stats-dada2_AllSeqs.qza \
  --o-visualization 251-2021-single_stats-dada2_AllSeqs.qzv
  
qiime feature-table summarize \
  --i-table 251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization 251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file metadata.tsv
  
qiime feature-table tabulate-seqs \
  --i-data 251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization 251-2021-single_rep-seqs-dada2_AllSeqs.qzv

```




```{bash}
qiime feature-classifier extract-reads \
  --i-sequences silva-138-99-seqs.qza \
  --p-f-primer GTGCCAGCMGCCGCGGTAA \
  --p-r-primer CCGYCAATTYMTTTRAGTTT \
  --p-trunc-len 251 \
  --p-min-length 100 \
  --p-max-length 500 \
  --o-reads 251-ref-seqs.qza

qiime feature-classifier fit-classifier-naive-bayes \
  --i-reference-reads Trim251-run2021/251-ref-seqs.qza \
  --i-reference-taxonomy silva-138-99-tax.qza \
  --o-classifier 251-classifier.qza

```



```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier 251-classifier.qza \
  --i-reads 251-2021-single_rep-seqs-dada2_AllSeqs.qza  \
  --o-classification 251-2021-single_rep-seqs-taxonomy.qza \
  --p-n-jobs 15

qiime metadata tabulate \
  --m-input-file 251-2021-single_rep-seqs-taxonomy.qza \
  --o-visualization 251-2021-single_rep-seqs-taxonomy.qzv
  
qiime taxa barplot --i-table  251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy 251-2021-single_rep-seqs-taxonomy.qza --o-visualization 251-2021-single_taxa-bar-plots.qzv
  
```





```{bash}

###renaming sample IDS
qiime feature-table group \
  --i-table 251-2021-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file metadata.tsv \
  --m-metadata-column Seq2 \
  --p-mode sum \
  --o-grouped-table Renamed-251-2021-single_table-dada2_AllSeqs.qza

qiime feature-table summarize \
  --i-table Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization Renamed-251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file New_metadata.tsv

qiime taxa barplot --i-table Renamed-251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy 251-2021-single_rep-seqs-taxonomy.qza --o-visualization Renamed-251-2021-single_taxa-bar-plots.qzv

```


```{bash}

##get rid of samples that were poorly sequenced under 1,000 quality control reads.

qiime feature-table filter-samples \
  --i-table Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --m-metadata-file New_metadata.tsv\
  --p-where "[keep]='keep'" \
  --o-filtered-table Quality-Samples-2021-251-filtered-table.qza

qiime taxa barplot --i-table Quality-Samples-2021-251-filtered-table.qza --i-taxonomy 251-2021-single_rep-seqs-taxonomy.qza --o-visualization Quality-Samples-251-2021-single_taxa-bar-plots.qzv

######have just KByT samples 
qiime feature-table filter-samples \
  --i-table Quality-Samples-2021-251-filtered-table.qza \
  --m-metadata-file New_metadata.tsv \
  --p-where "[Project]='KByT'" \
  --o-filtered-table KByT-2021-251-filtered-table.qza

qiime taxa barplot --i-table KByT-2021-251-filtered-table.qza --i-taxonomy 251-2021-single_rep-seqs-taxonomy.qza --o-visualization KByT-251-2021-single_taxa-bar-plots.qzv

```


####August 2017- June 2019 Sequencing run (previously published in Tucker et al., 2021 (PeerJ))

```{python}

screen -S Miseq 

source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4

qiime tools import --show-importable-types

cd /tank/tucker_data/Base_Calls_manuscript

qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path Base_Calls_manuscript \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza 

```


**(2) Generate a summary to evaluate the data, it will help determine how to truncate data in a future step.**

```{bash, eval=FALSE}
  
qiime demux summarize \
  --i-data /Redo_2019/redo2019-demux-paired-end.qza \
  --o-visualization /Redo_2019/redo2019-demux-paired-end.qzv

```



**(3) Use Cutadapt to remove primers**

https://forum.qiime2.org/t/trimming-primers-from-both-ends-of-paired-end-reads/9984
https://docs.qiime2.org/2017.12/plugins/available/cutadapt/demux-paired/


```{bash}
qiime cutadapt trim-paired \
    --i-demultiplexed-sequences /Redo_2019/redo2019-demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences /Redo_2019/redo2019-demux-paired-end-trimmed.qza \
    --p-cores 5\
    --verbose

qiime demux summarize \
  --i-data /Redo_2019/redo2019-demux-paired-end-trimmed.qza  \
  --o-visualization /Redo_2019/redo2019-demux-paired-end-trimmed.qzv
```




```{bash}
qiime dada2 denoise-single \
  --i-demultiplexed-seqs /Redo_2019/redo2019-demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences /Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table /Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats /Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 20

qiime metadata tabulate \
  --m-input-file /Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --o-visualization /Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qzv
  
  
qiime feature-table summarize \
  --i-table /Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /Redo_2019/redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv
  
  
```



```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier 251-classifier.qza \
  --i-reads /Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-classification /Redo_2019/redo-2019-single_read_taxonomy.qza

qiime metadata tabulate \
  --m-input-file /Redo_2019/redo-2019-single_read_taxonomy.qza \
  --o-visualization /Redo_2019/redo-2019-single_read_taxonomy.qzv
  

qiime taxa barplot --i-table /Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization /Redo_2019/redo-2019-single_taxa-bar-plots.qzv
  
```


```{bash}

###renaming sample IDS

qiime feature-table group \
  --i-table /Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file /Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv \
  --m-metadata-column Seq \
  --p-mode sum \
  --o-grouped-table /Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza

###start here for both

qiime feature-table summarize \
  --i-table /Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /Redo_2019/New_Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv

qiime taxa barplot --i-table /Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization /Redo_2019/Renamed-redo-2019-single_taxa-bar-plots.qzv
```


###merging
```{bash}

#######Time to merge!! 
qiime feature-table merge \
  --i-tables KByT-2021-251-filtered-table.qza \
  --i-tables /Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-merged-table MiSeq_merged_2021_2019_table.qza

qiime feature-table merge-seqs \
  --i-data 251-2021-single_rep-seqs-dada2_AllSeqs.qza  \
  --i-data /Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-merged-data MiSeq_merged_2021_2019-rep-seqs.qza

qiime feature-table summarize \
  --i-table MiSeq_merged_2021_2019_table.qza \
  --o-visualization MiSeq_merged_2021_2019_table.qzv \
  --m-sample-metadata-file 2021_2019_metadata.tsv

qiime feature-table tabulate-seqs \
  --i-data MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-visualization MiSeq_merged_2021_2019-rep-seqs.qzv

qiime feature-table merge-taxa \
--i-data 251-2021-single_rep-seqs-taxonomy.qza \
  --i-data /Redo_2019/redo-2019-single_read_taxonomy.qza \
  --o-merged-data MiSeq_merged_2021_2019_taxonomy.qza

```


```{bash}

#This filter can be applied to the feature axis to remove low abundance features from a table. For example, you can remove all features with a total abundance (summed across all samples) of less than 10 as follows.

qiime feature-table filter-features \
  --i-table MiSeq_merged_2021_2019_table.qza \
  --p-min-frequency 10 \
  --o-filtered-table MiSeq_merged_2021_2019-feature-frequency-filtered-table.qza
  
#This filtering is commonly used for filtering features that show up in only one or a few samples, based on the suspicion that these may not represent real biological diversity but rather PCR or sequencing errors (such as PCR chimeras). Features that are present in only a single sample could be filtered from a feature table as follows.

qiime feature-table filter-features \
  --i-table MiSeq_merged_2021_2019-feature-frequency-filtered-table.qza \
  --p-min-samples 2 \
  --o-filtered-table MiSeq_merged_2021_2019-final-filtered-table.qza

qiime feature-table summarize \
  --i-table MiSeq_merged_2021_2019-final-filtered-table.qza \
  --o-visualization MiSeq_merged_2021_2019-final-filtered-table.qzv \
  --m-sample-metadata-file 2021_2019_metadata.tsv

qiime taxa barplot --i-table MiSeq_merged_2021_2019-final-filtered-table.qza --i-taxonomy MiSeq_merged_2021_2019_taxonomy.qza --o-visualization Filtered-2019-2021-taxa-bar-plots.qzv

#https://docs.qiime2.org/2021.4/tutorials/filtering/

qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-alignment aligned_MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-masked-alignment masked_aligned_MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-tree MiSeq_merged_2021_2019_unrooted-tree.qza \
  --o-rooted-tree MiSeq_merged_2021_2019_rooted-tree.qza

 qiime tools export \
  --input-path  MiSeq_merged_2021_2019-rep-seqs.qza \
  --output-path exported-rep-seqs

##more prep
qiime tools export \
  --input-path MiSeq_merged_2021_2019_unrooted-tree.qza \
  --output-path unrooted-exported-tree

qiime tools export \
  --input-path MiSeq_merged_2021_2019-final-filtered-table.qza \
  --output-path exported-filtered-feature-table

cd exported-filtered-feature-table
biom convert -i feature-table.biom -o feature-table.tsv --to-tsv

qiime tools export \
  --input-path MiSeq_merged_2021_2019_rooted-tree.qza \
  --output-path rooted-exported-tree

qiime tools export \
  --input-path MiSeq_merged_2021_2019_taxonomy.qza \
  --output-path  exported-taxonomy
 
 qiime tools export \
  --input-path single_aligned-rep-seqs.qza \
  --output-path exported-rep-seqs
 
 qiime tools export \
  --input-path masked-single_aligned-rep-seqs.qza \
  --output-path exported-masked-rep-seqs

```




#########
Qiime2 to Phyloseq
##########


```{r, echo=FALSE, warning=FALSE, message=FALSE}

library("ggplot2") 
library("phyloseq") 

###now that you are in the right format you can bring things in as phyloseq objects

# read in otu table
otu_table_KBytNERR = read.csv("2021_2019_MiSeq_feature-table_Sep272021.csv", sep=",", row.names=1) 
otu_table_KBytNERR = as.matrix(otu_table_KBytNERR)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

#note this was the original taxonomy, before using Pr2
taxonomy_KBytNERR = read.csv("2021_2019_MiSeq_taxonomy_Sep272021.csv", sep=",", row.names=1) 


taxonomy_KBytNERR = as.matrix(taxonomy_KBytNERR)

#read in metadata
metadata_KBytNERR=read.csv("metadata_2019_2021_10222021.csv", header=TRUE, fill=TRUE)

rownames(metadata_KBytNERR) <- metadata_KBytNERR$SampleID
metadata_KBytNERR$SampleID
META_KBytNERR = sample_data(metadata_KBytNERR)

#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# import as phyloseq objects
OTU_KBytNERR = otu_table(otu_table_KBytNERR, taxa_are_rows = TRUE) 
TAX_KBytNERR = tax_table(taxonomy_KBytNERR)


physeq_KBytNERR = phyloseq(OTU_KBytNERR, TAX_KBytNERR, phy_treeKBytNERR)

physeq_KBytNERR=merge_phyloseq(physeq_KBytNERR, META_KBytNERR)
ntaxa(physeq_KBytNERR)
#4462
nsamples(physeq_KBytNERR)
write.csv(tax_table(physeq_KBytNERR), "tax_table_physeq_KByTNERR.csv")

```



######
CLASSIFYING CHLOROPLAST & EUKARYOTIC SEQUENCES
######


```{r}
test=subset_taxa(physeq_KBytNERR, Domain=="D_0__Eukaryota")
ntaxa(test)

head(tax_table(physeq_KBytNERR))

chloro_euks_unassigned=subset_taxa(physeq_KBytNERR, Order=="D_3__Chloroplast" | Domain=="D_0__Eukaryota"| Domain=="Unassigned")
ntaxa(chloro_euks_unassigned)
test=as.data.frame(tax_table(chloro_euks_unassigned))
tail(tax_table(chloro_euks_unassigned))

write.csv(taxa_names(chloro_euks_unassigned), "chloro_euks_unassigned.csv")

```


```{bash}
cd /Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq
grep -A 1 -f <(cut -d ',' -f 2 chloro_euks_unassigned.csv | tr -d "\"") 2021_2019_merged-dna-sequences.fasta | sed '/^--$/d' >> chloro_euk_unassigned_ASV.fasta


###get the seqeunces into tab form
awk 'BEGIN{RS=">"}{print "#"$1"\t"$2;}' chloro_euk_unassigned_ASV.fasta | tail -n+2 > chloro_euk_unassigned_ASV_reformat.txt

```


##The threshold of 60% is recommended at the default confidence threshold.

```{r}
#if (!requireNamespace("BiocManager", quietly=TRUE))
#   install.packages("BiocManager")
#BiocManager::install("DECIPHER")


#install.packages(devtools)
#devtools::install_github("pr2database/pr2database")

library(DECIPHER)
library(pr2database)

# Read training set 

#available here https://github.com/pr2database/pr2database
trainingSet <- readRDS("/Phyloseq_2019_2021_MiSeq/pr2_version_4.14.0_SSU.decipher.trained.rds")
# Read sequences to assign
seq <- readDNAStringSet("/Phyloseq_2019_2021_MiSeq/chloro_euk_unassigned_ASV.fasta")
# Get the taxonomy from the training set
  ids <- IdTaxa(seq,
                trainingSet,
                type="extended",
                strand="top",
                threshold=60)
#ids threshold is 60 now, before did not set a threshold
  
  #n_seq <- length(ids)
  #df_rows <- list()
taxo_levels <- c("kingdom", "supergroup", "division", "class", "order", "family", "genus", "species")

###Code I used 
output <- sapply(ids,
function (id) {
seq_name <- names(ids)
paste(id$taxon,
";",
round(id$confidence, digits=1),
sep="",
collapse="; ")
})
tail(output)

idk=as.data.frame(output)
write.csv(idk, "pr2_assigment_chloro_euk_unassigned.csv")

pr2_taxonomy=read.csv("pr2_assigment_chloro_euk_unassigned_edited.csv")


silva_taxonomy=read.csv("tax_table_physeq_KByTNERR_edited.csv")


combined_taxonomy=left_join(silva_taxonomy, pr2_taxonomy, by="OTUID")

write.csv(combined_taxonomy, "all_taxonomy_pr2_euk_chloro_unassigned.csv")


  
```


######
Following the reidentification of the chloroplast and eukaryotic sequences, duplicate samples and samples from April Fishpond sampling were removed and ASVs were filtered to remove ASVs that summed to less than 10 reads across all samples, and did not show up in at least two unique samples, reducing the total otu table down to 4117 ASVs

```{r}

#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# read in otu table with duplicate samples removed
no_dup_otu_table_KBytNERR = read.csv("no_duplicates_2021_2019_MiSeq_feature-table_Sep282021.csv", sep=",", row.names=1) 
no_dup_otu_table_KBytNERR = as.matrix(no_dup_otu_table_KBytNERR)

# import as phyloseq objects
no_dup_OTU_KBytNERR = otu_table(no_dup_otu_table_KBytNERR, taxa_are_rows = TRUE) 

taxonomy_Phyto = read.csv("all_taxonomy_pr2_euk_chloro_unassigned_correct_v7.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)
ntaxa(TAX_Phyto)

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv3.csv", header=TRUE, fill=TRUE)

rownames(no_dup_metadata_KBytNERR) <- no_dup_metadata_KBytNERR$SampleID
no_dup_metadata_KBytNERR$SampleID
META_no_dup_KBytNERR = sample_data(no_dup_metadata_KBytNERR)
nsamples(META_no_dup_KBytNERR)



physeq_Dav_no_dup = phyloseq(no_dup_OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav_no_dup=merge_phyloseq(physeq_Dav_no_dup, META_no_dup_KBytNERR)
ntaxa(physeq_Dav_no_dup)
#4462

#4462 tax with the 366 final samples
taxonomy_Phyto_binary <- transform_sample_counts(physeq_Dav_no_dup, function(abund) 1*(abund>0))
test=as.data.frame(otu_table(taxonomy_Phyto_binary))
test$SUM=rowSums(test)
library(tibble)
gg2 <- tibble::rownames_to_column(test, "ASV")
test2=gg2%>%select(ASV, SUM)


count10=as.data.frame(otu_table(physeq_Dav_no_dup))
count10$occur=rowSums(count10)
gg3 <- tibble::rownames_to_column(count10, "ASV")

okay=left_join(gg3,test2, by="ASV")
write.csv(okay,"filtering_otu_table.csv")

#ASVs were removed that were less than 10 total across all samples (count10<10) and that didn't occur in at least 2 or more samples (SUM<2), leaving 4117 ASVs
#Final data tables: 
"otu_final_tucker.csv"
"unrooted-tree.nwk"
"taxonomy_table_clean_May2024.csv"
"no_dup_metadata_KBytNERR_main_analyses.csv"

```






