---
title: "Metagenomics_LO_Submission"
author: "Sarah Tucker"
date: "5/6/2023"
output: html_document
---

These analyses were completed in Anvi'o v7.0 (Eren et al., 2021) with built in programs to quality filter using ‘iu-filterquality-minoche’ a illumina-utils library (Eren et al. 2013) v1.4.1, competitively mapping to genomes with Bowtie2 v2.3.5 (Langmead and Salzberg 2012), and calculation of Average Nucleotide Identity (ANI) using FastANI (Jain et al. 2018). 

```{python}

iu-gen-configs samples.txt -o 01_QC


anvi-gen-contigs-database -f fixed-names-Cyano_PRO_SYN_ISOLATES_EDITED_w3B_v4.fa -o cyano-CONTIGS.db -n "Cyano isolate genomes" -T 20


anvi-run-hmms -c cyano-CONTIGS.db --num-threads 20

anvi-run-ncbi-cogs --cog-data-dir /ANVIO_NCBI_COGS_2020 -c cyano-CONTIGS.db --num-threads 20

anvi-run-kegg-kofams -c cyano-CONTIGS.db --kegg-data-dir /tank/tucker_data/KEGG -T 20



bowtie2-build fixed-names-Cyano_PRO_SYN_ISOLATES_EDITED_w3B_v4.fa Cyano-KbyT-Aloha25m --threads 20

for sample in `awk '{print $1}' /KByT_Mende/samples_Aloha_KByT_mende_v2.txt`
do
 if [ "$sample" == "sample" ]; then continue; fi
    # do the bowtie mapping to get the SAM file:
    bowtie2 --threads 20 \
            -x Cyano-KbyT-Aloha25m \
            -1 /01_QC/$sample-QUALITY_PASSED_R1.fastq.gz \
            -2 /01_QC/$sample-QUALITY_PASSED_R2.fastq.gz \
            --no-unal \
            -S $sample.sam
    # covert the resulting SAM file to a BAM file:
    samtools view -F 4 -bS $sample.sam > $sample-RAW.bam

    # sort and index the BAM file:
    samtools sort $sample-RAW.bam -o $sample.bam
    samtools index $sample.bam

    # remove temporary files:
    rm $sample.sam $sample-RAW.bam
done


for sample in `awk '{print $1}' /KByT_Mende/samples_Aloha_KByT_mende_v2.txt`
do
    if [ "$sample" == "sample" ]; then continue; fi

    anvi-profile -c cyano-CONTIGS.db \
                 -i $sample.bam \
                 -M 100 \
                 --profile-SCVs \
                 --num-threads 20 \
                 -o $sample
done


anvi-merge */PROFILE.db -o cyano-MERGED -c cyano-CONTIGS.db



for split_name in `sqlite3 cyano-CONTIGS.db 'select split from splits_basic_info;'`
do
    # in this loop $split_name looks like this AS9601-00000001_split_00001, form which
    # we can extract the genome name the split belongs to:
    GENOME=`echo $split_name | awk 'BEGIN{FS="_"}{print $1}'`

    # print it out with a TAB character
    echo -e "$split_name\t$GENOME"
done > cyano-GENOME-COLLECTION.txt


anvi-import-collection cyano-GENOME-COLLECTION.txt -c cyano-CONTIGS.db -p cyano-MERGED/PROFILE.db -C Genomes


anvi-summarize -c cyano-CONTIGS.db -p cyano-MERGED/PROFILE.db -C Genomes --init-gene-coverages -o cyano-Feb2023-SUMMARY

anvi-gen-genomes-storage -i cyanobium_internal.txt -o cyano-PAN-GENOMES.db


anvi-pan-genome -g cyano-PAN-GENOMES.db --use-ncbi-blast --minbit 0.5 --mcl-inflation 2 --project-name cyano-Feb2023-PAN --num-threads 20


anvi-meta-pan-genome -i cyanobium_internal.txt -p cyano-Feb2023-PAN/cyano-Feb2023-PAN-PAN.db -g cyano-PAN-GENOMES.db --fraction-of-median-coverage 0.25


anvi-display-pan -p cyano-Feb2023-PAN/cyano-Feb2023-PAN-PAN.db -g cyano-PAN-GENOMES.db

anvi-compute-genome-similarity --internal-genomes cyanobium_internal.txt --program fastANI --output-dir CYANO-ANI --num-threads 30 -p cyano-Feb2023-PAN/cyano-Feb2023-PAN-PAN.db
```


