---
title: "Qiime_2021_final"
author: "Sarah Tucker"
date: "5/13/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown



```{bash}
cd /tank/tucker_data/2021_MiSeqFinal/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC-updated/BaseCalls/
for f in *R2_001.fastq.gz; do mv $f ${f%*R2_001.fastq.gz}L001_R2_001.fastq.gz; done

for f in *R1_001.fastq.gz; do mv $f ${f%*R1_001.fastq.gz}L001_R1_001.fastq.gz; done
```



```{python}

screen -S Miseq 
source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4
qiime tools import --show-importable-types
cd /tank/tucker_data/MiSeq/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC/BaseCalls

qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path /tank/tucker_data/2021_MiSeqFinal/files.cgrb.oregonstate.edu/Illumina/210816_M01498_0823_000000000-JPJTC-updated/BaseCalls \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end.qza 

```


```{bash, eval=FALSE}
qiime demux summarize \
  --i-data  /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end.qza   \
  --o-visualization  /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end.qzv 
```



```{bash}
qiime cutadapt trim-paired \
    --i-demultiplexed-sequences /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end-trimmed.qza \
    --p-cores 20\
    --verbose
qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end-trimmed.qza  \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end-trimmed.qzv
```



```{bash}
qiime dada2 denoise-single \
  --i-demultiplexed-seqs /tank/tucker_data/MiSeq_Analysis_2021_Final/demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 30
  
qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_stats-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_stats-dada2_AllSeqs.qzv
  
qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/metadata.tsv
  
qiime feature-table tabulate-seqs \
  --i-data /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-dada2_AllSeqs.qzv

```




```{bash}
qiime feature-classifier extract-reads \
  --i-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/silva-138-99-seqs.qza \
  --p-f-primer GTGCCAGCMGCCGCGGTAA \
  --p-r-primer CCGYCAATTYMTTTRAGTTT \
  --p-trunc-len 251 \
  --p-min-length 100 \
  --p-max-length 500 \
  --o-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-ref-seqs.qza

qiime feature-classifier fit-classifier-naive-bayes \
  --i-reference-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-ref-seqs.qza \
  --i-reference-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/silva-138-99-tax.qza \
  --o-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza

```



```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza \
  --i-reads /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-dada2_AllSeqs.qza  \
  --o-classification /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza \
  --p-n-jobs 15

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qzv
  
qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_taxa-bar-plots.qzv
  
```





```{bash}

###renaming sample IDS
qiime feature-table group \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/metadata.tsv \
  --m-metadata-column Seq2 \
  --p-mode sum \
  --o-grouped-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_table-dada2_AllSeqs.qza

qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --o-visualization  /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/New_metadata.tsv

qiime taxa barplot --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_taxa-bar-plots.qzv

```


```{bash}

##get rid of samples that were poorly sequenced I got rid of all samples under 1,000 quality control reads. This included all the negatives, but also a number of JdFR samples, 3 KByT samples (NBJu0836, R2De0742, R2Ap0782), and a mangrove sample (Mg020bot). It looks like people are not recommending repeating the denoising step and instead recommending just a filtering base on sample: https://forum.qiime2.org/t/remove-blank-samples/9119


qiime feature-table filter-samples \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Renamed-251-2021-single_table-dada2_AllSeqs.qza \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/New_metadata.tsv\
  --p-where "[keep]='keep'" \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Quality-Samples-2021-251-filtered-table.qza

qiime taxa barplot --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Quality-Samples-2021-251-filtered-table.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/Quality-Samples-251-2021-single_taxa-bar-plots.qzv

######have just KByT samples 
qiime feature-table filter-samples \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/Quality-Samples-2021-251-filtered-table.qza \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/New_metadata.tsv \
  --p-where "[Project]='KByT'" \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_2021_Final/KByT-2021-251-filtered-table.qza

qiime taxa barplot --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/KByT-2021-251-filtered-table.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/KByT-251-2021-single_taxa-bar-plots.qzv

```


####2019 

##############redoing all of 2019
#This is technically already how DADA2 works. It makes the error model using all the samples from the same run, then uses that model to denoise each read separately and independently, this is why it is nicely parallelizable.


```{python}

screen -S Miseq 

source /opt/anvio_conda/miniconda3/bin/activate bio-gen
conda activate qiime2-2021.4

qiime tools import --show-importable-types

cd /tank/tucker_data/Base_Calls_manuscript

qiime tools import \
  --type 'SampleData[PairedEndSequencesWithQuality]' \
  --input-path /tank/tucker_data/Base_Calls_manuscript \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza 

```


**(2) Generate a summary to evaluate the data, it will help determine how to truncate data in a future step.**

```{bash, eval=FALSE}
  
qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qzv

```



**(3) Use Cutadapt to remove primers**

https://forum.qiime2.org/t/trimming-primers-from-both-ends-of-paired-end-reads/9984
https://docs.qiime2.org/2017.12/plugins/available/cutadapt/demux-paired/


```{bash}
qiime cutadapt trim-paired \
    --i-demultiplexed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end.qza \
    --p-adapter-f 'GTGYCAGCMGCCGCGGTAA$' \
    --p-adapter-r 'CCGYCAATTYMTTTRAGTTT$' \
    --o-trimmed-sequences /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza \
    --p-cores 5\
    --verbose

qiime demux summarize \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza  \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qzv
```




```{bash}
qiime dada2 denoise-single \
  --i-demultiplexed-seqs /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo2019-demux-paired-end-trimmed.qza \
  --p-trunc-len 251 \
  --o-representative-sequences  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-denoising-stats /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --p-n-threads 20

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_stats-dada2_AllSeqs.qzv
  
  
qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv
  
  
qiime feature-table tabulate-seqs \
  --i-data single_rep-seqs-dada2_AllSeqs.qza \
  --o-visualization single_rep-seqs-dada2_AllSeqs.qzv
  
```



```{bash}
qiime feature-classifier classify-sklearn \
  --i-classifier /tank/tucker_data/MiSeq_Analysis_Sep082021/Trim251-run2021/251-classifier.qza \
  --i-reads /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-classification /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza

qiime metadata tabulate \
  --m-input-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qzv
  

qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_taxa-bar-plots.qzv
  
```


```{bash}

###renaming sample IDS

qiime feature-table group \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_table-dada2_AllSeqs.qza \
  --p-axis sample \
  --m-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv \
  --m-metadata-column Seq \
  --p-mode sum \
  --o-grouped-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza

###start here for both

qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/New_Redo_run2019_New_Seasons_No_repsMetaData_KByt08192020.tsv

qiime taxa barplot --i-table  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza --o-visualization  /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_taxa-bar-plots.qzv
```


###merging
```{bash}

#######Time to merge!! 
qiime feature-table merge \
  --i-tables /tank/tucker_data/MiSeq_Analysis_2021_Final/KByT-2021-251-filtered-table.qza \
  --i-tables /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/Renamed-redo-2019-single_table-dada2_AllSeqs.qza \
  --o-merged-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_table.qza

qiime feature-table merge-seqs \
  --i-data /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-dada2_AllSeqs.qza  \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_rep-seqs-dada2_AllSeqs.qza \
  --o-merged-data /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-rep-seqs.qza

qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_table.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_table.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/2021_2019_metadata.tsv

qiime feature-table tabulate-seqs \
  --i-data /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-rep-seqs.qzv

###https://forum.qiime2.org/t/merging-feature-tables-should-i-also-merge-taxonomy-or-does-it-need-to-be-re-run/15678/2

qiime feature-table merge-taxa \
--i-data /tank/tucker_data/MiSeq_Analysis_2021_Final/251-2021-single_rep-seqs-taxonomy.qza \
  --i-data /tank/tucker_data/MiSeq_Analysis_Sep082021/Redo_2019/redo-2019-single_read_taxonomy.qza \
  --o-merged-data /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_taxonomy.qza

```


```{bash}

#This filter can be applied to the feature axis to remove low abundance features from a table. For example, you can remove all features with a total abundance (summed across all samples) of less than 10 as follows.

qiime feature-table filter-features \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_table.qza \
  --p-min-frequency 10 \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-feature-frequency-filtered-table.qza
  
#This filtering is commonly used for filtering features that show up in only one or a few samples, based on the suspicion that these may not represent real biological diversity but rather PCR or sequencing errors (such as PCR chimeras). Features that are present in only a single sample could be filtered from a feature table as follows.

qiime feature-table filter-features \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-feature-frequency-filtered-table.qza \
  --p-min-samples 2 \
  --o-filtered-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-final-filtered-table.qza

qiime feature-table summarize \
  --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-final-filtered-table.qza \
  --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-final-filtered-table.qzv \
  --m-sample-metadata-file /tank/tucker_data/MiSeq_Analysis_2021_Final/2021_2019_metadata.tsv

qiime taxa barplot --i-table /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-final-filtered-table.qza --i-taxonomy /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_taxonomy.qza --o-visualization /tank/tucker_data/MiSeq_Analysis_2021_Final/Filtered-2019-2021-taxa-bar-plots.qzv

#https://docs.qiime2.org/2021.4/tutorials/filtering/

qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-alignment /tank/tucker_data/MiSeq_Analysis_2021_Final/aligned_MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-masked-alignment /tank/tucker_data/MiSeq_Analysis_2021_Final/masked_aligned_MiSeq_merged_2021_2019-rep-seqs.qza \
  --o-tree /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_unrooted-tree.qza \
  --o-rooted-tree /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_rooted-tree.qza

 qiime tools export \
  --input-path  /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-rep-seqs.qza \
  --output-path /tank/tucker_data/MiSeq_Analysis_2021_Final/exported-rep-seqs

##more prep
qiime tools export \
  --input-path /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_unrooted-tree.qza \
  --output-path /tank/tucker_data/MiSeq_Analysis_2021_Final/unrooted-exported-tree

qiime tools export \
  --input-path /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019-final-filtered-table.qza \
  --output-path /tank/tucker_data/MiSeq_Analysis_2021_Final/exported-filtered-feature-table

cd /tank/tucker_data/MiSeq_Analysis_2021_Final/exported-filtered-feature-table
biom convert -i feature-table.biom -o feature-table.tsv --to-tsv

qiime tools export \
  --input-path /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_rooted-tree.qza \
  --output-path /tank/tucker_data/MiSeq_Analysis_2021_Final/rooted-exported-tree

qiime tools export \
  --input-path /tank/tucker_data/MiSeq_Analysis_2021_Final/MiSeq_merged_2021_2019_taxonomy.qza \
  --output-path  /tank/tucker_data/MiSeq_Analysis_2021_Final/exported-taxonomy
 
 qiime tools export \
  --input-path single_aligned-rep-seqs.qza \
  --output-path exported-rep-seqs
 
 qiime tools export \
  --input-path masked-single_aligned-rep-seqs.qza \
  --output-path exported-masked-rep-seqs

```




##MOCK COMMUNITY ANALYSES 


```{r}

# load libraries
library("ggplot2") 
library("phyloseq") 
library(ape)
library("knitr")
library(dada2)
library(methods)
library(plotly)
library(tidyverse)
library(ggpubr)

###this is from the R tutorial document"https://github.com/joey711/phyloseq/files/1795523/QIIME2_to_Phyloseq.pdf"


#setwd("/Users/sarahtucker/Documents/Dissertation/chapter 1/miseq_run_82019") 

#getwd()


###now that you are in the right format you can bring things in as phyloseq objects


# read in otu table
otu_table_Mock = read.csv("MiSeq2021-Mock-table.csv", sep=",", row.names=1) 
otu_table_Mock = as.matrix(otu_table_Mock)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

taxonomy_2021 = read.csv("taxonomy-table-Miseq2021.csv", sep=",", row.names=1) 
taxonomy_2021 = as.matrix(taxonomy_2021)

#read in metadata
#make sure when you are saving from excel that you use a tab delineated format
####also see my R file making_metadata.Rmd in my Chapter_2 folder for a quick and easy way to integrate genetic metadata with environmental!
##except I did need to remember add metadata for R2 samples (R857R2, SB58R2, NT61R2)
##i put in an average of the month before and month after for FCM data of SBoC0294 
metadata_Mock= read.csv("Mocks_only_metadata.csv", header=TRUE, fill=TRUE)

rownames(metadata_Mock) <- metadata_Mock$SampleID
metadata_Mock$SampleID

#read in tree
phy_tree = read_tree("MiSeq_2021_tree.nwk")

# import as phyloseq objects
OTU_MOCK = otu_table(otu_table_Mock, taxa_are_rows = TRUE) 
TAX_2021= tax_table(taxonomy_2021)
#check that the names are correct 
#rank_names(TAX3)
#ncol(taxonomy_phylo3)
#ncol(TAX3)

#bring in your metadata 

META_MOCK = sample_data(metadata_Mock)
sample_names(META_MOCK)
#colnames(META)

#first make sure your sample names match 
#sample_names(META)
#sample_names(OTU)

#now create your phyloseq object
#not sure why, but if you make 2 objects and add the metadata on the merge step it keeps the column names 

physeq_Mock = phyloseq(OTU_MOCK, TAX_2021, phy_tree)

physeq_Mock2=merge_phyloseq(physeq_Mock, META_MOCK)
ntaxa(physeq_Mock)
```


#http://evomics.org/wp-content/uploads/2016/01/phyloseq-Lab-01-Answers.html

```{r}
write.csv(otu_table(physeq_Mock), "physeq_Mock_otu.csv")
write.csv(tax_table(physeq_Mock), "physeq_Mock_tax.csv")

physeq_Mock2 = prune_taxa(taxa_sums(physeq_Mock)>10, physeq_Mock)
ntaxa(physeq_Mock2)

physeq_Mock2_rel  = transform_sample_counts(physeq_Mock2, function(x) x / sum(x) )

write.csv(otu_table(physeq_Mock2_rel), "physeq_Mock_rel_otu2.csv")
write.csv(tax_table(physeq_Mock2), "physeq_Mock_tax2.csv")


```


```{r}

Mock_2021 = read.csv("mock_analysis_2021.csv", sep=",", row.names=1) 
library(dplyr)
library(plotly)
library(scales)
library(ggplot2)

getwd()

Mockdata <- read.csv('mock_analysis_2021.csv')

Everything = subset(Mockdata, select= c("Sample", "MGII", "Thaumarchaeota","OCS115","Flavo", "Prochlorococcus", "MGA", "Plancto", "SAR11", "SAR86", "AEGEAN169", "SAR116", "SAR202","Porticoccaceae", "Oceanospirillales", "Puniceicoccaceae"))

xform <- list(font = list(
    family = "sans-serif",
    size = 16),
  categoryorder = "array",
              categoryarray = c("Staggered_Expected","MoSTOrg2","MoSTOrg1","MoST30c1","MoST25c1","STOrig_SE_2019","Even_Expected","MoEvOrg2","MoEV20c1", "EVOrig_SE_2019"
))

data.table::melt(Everything, id.vars='Sample') %>%
plot_ly(x = ~Sample, y = ~value, type = 'bar', 
                name = ~variable, color = ~variable) %>%
      layout(yaxis = list(title = 'Count'), xaxis=xform, title= "Mock Communities Comparison Expected to 2021 MiSeq Run ", barmode = 'stack')

```







