---
title: "Code_LO_Submission"
author: "Sarah Tucker"
date: "3/5/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{r}
# load libraries
library("ggplot2") 
library("phyloseq") 
library(ape)
library("knitr")
library(methods)
library(plotly)
library(tidyverse)
library(ggpubr)
library(compositions)
library(reshape2)
library(fpc)
library(pracma)
library(cluster)
library(ade4)
library(vegan)
library(DESeq2)
library(RColorBrewer)
library(randomcoloR)
library(purrr)
library(lomb)
library(lubridate)
library(tidyverse)
library(colorspace)
library(dendextend)
library(DESeq2)
library(DivNet)
library(breakaway)
library(corncob)
library(seecolor)
library(microbiome)
library(viridis)
library(microbiomeutilities)
library("pheatmap")
library(viridis)
```

#########
Qiime2 to Phyloseq
##########


```{r, echo=FALSE, warning=FALSE, message=FALSE}

###now that you are in the right format you can bring things in as phyloseq objects

# read in otu table
otu_table_KBytNERR = read.csv("2021_2019_MiSeq_feature-table_Sep272021.csv", sep=",", row.names=1) 
otu_table_KBytNERR = as.matrix(otu_table_KBytNERR)

# read in taxonomy
# seperated by kingdom phylum class order family genus species 

#you also need to edit your taxonomy table by separating each column by a comma so that you don't have just one column with all the taxonomic levels. Relabel to be  "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species" as the column headers 

#note this was the original taxonomy, before using Pr2
taxonomy_KBytNERR = read.csv("2021_2019_MiSeq_taxonomy_Sep272021.csv", sep=",", row.names=1) 


taxonomy_KBytNERR = as.matrix(taxonomy_KBytNERR)

#poop=subset(taxonomy_KBytNERR, Order=="D_3__Chloroplast")

#read in metadata
##note that there is not seq data for all metadata samples 
###when analyzing metadata strictly, need to edit
metadata_KBytNERR=read.csv("metadata_2019_2021_10222021.csv", header=TRUE, fill=TRUE)

rownames(metadata_KBytNERR) <- metadata_KBytNERR$SampleID
metadata_KBytNERR$SampleID
META_KBytNERR = sample_data(metadata_KBytNERR)

#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# import as phyloseq objects
OTU_KBytNERR = otu_table(otu_table_KBytNERR, taxa_are_rows = TRUE) 
TAX_KBytNERR = tax_table(taxonomy_KBytNERR)


physeq_KBytNERR = phyloseq(OTU_KBytNERR, TAX_KBytNERR, phy_treeKBytNERR)

physeq_KBytNERR=merge_phyloseq(physeq_KBytNERR, META_KBytNERR)
ntaxa(physeq_KBytNERR)
#4462

write.csv(tax_table(physeq_KBytNERR), "tax_table_physeq_KByTNERR.csv")

```



######
CLASSIFYING CHLOROPLAST & EUKARYOTIC SEQUENCES
######


```{r}
test=subset_taxa(physeq_KBytNERR, Domain=="D_0__Eukaryota")
ntaxa(test)

head(tax_table(physeq_KBytNERR))

chloro_euks_unassigned=subset_taxa(physeq_KBytNERR, Order=="D_3__Chloroplast" | Domain=="D_0__Eukaryota"| Domain=="Unassigned")
ntaxa(chloro_euks_unassigned)
test=as.data.frame(tax_table(chloro_euks_unassigned))
tail(tax_table(chloro_euks_unassigned))

write.csv(taxa_names(chloro_euks_unassigned), "chloro_euks_unassigned.csv")

```


```{bash}
cd /Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq
grep -A 1 -f <(cut -d ',' -f 2 chloro_euks_unassigned.csv | tr -d "\"") 2021_2019_merged-dna-sequences.fasta | sed '/^--$/d' >> chloro_euk_unassigned_ASV.fasta


###get the seqeunces into tab form
awk 'BEGIN{RS=">"}{print "#"$1"\t"$2;}' chloro_euk_unassigned_ASV.fasta | tail -n+2 > chloro_euk_unassigned_ASV_reformat.txt

```


##The threshold of 60% is recommended at the default confidence threshold.


```{r}
#if (!requireNamespace("BiocManager", quietly=TRUE))
#   install.packages("BiocManager")
#BiocManager::install("DECIPHER")


#install.packages(devtools)
#devtools::install_github("pr2database/pr2database")

library(DECIPHER)
library(pr2database)

# Read training set 
trainingSet <- readRDS("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq/pr2_version_4.14.0_SSU.decipher.trained.rds")
# Read sequences to assign
seq <- readDNAStringSet("/Users/sarahtucker/Documents/KByt/Phyloseq_2019_2021_MiSeq/chloro_euk_unassigned_ASV.fasta")
# Get the taxonomy from the training set
  ids <- IdTaxa(seq,
                trainingSet,
                type="extended",
                strand="top",
                threshold=60)
#ids threshold is 60 now, before did not set a threshold
  
  #n_seq <- length(ids)
  #df_rows <- list()
taxo_levels <- c("kingdom", "supergroup", "division", "class", "order", "family", "genus", "species")

###Code I used 
output <- sapply(ids,
function (id) {
seq_name <- names(ids)
paste(id$taxon,
";",
round(id$confidence, digits=1),
sep="",
collapse="; ")
})
tail(output)

idk=as.data.frame(output)
write.csv(idk, "pr2_assigment_chloro_euk_unassigned.csv")

pr2_taxonomy=read.csv("pr2_assigment_chloro_euk_unassigned_edited.csv")


silva_taxonomy=read.csv("tax_table_physeq_KByTNERR_edited.csv")


combined_taxonomy=left_join(silva_taxonomy, pr2_taxonomy, by="OTUID")

write.csv(combined_taxonomy, "all_taxonomy_pr2_euk_chloro_unassigned.csv")


  
```


######
Note following this, duplicate samples and samples from April were removed
######



#########
Current phyloseq objects
##########

```{r}
#read in tree
phy_treeKBytNERR = read_tree("unrooted-tree.nwk")

# read in otu table
no_dup_otu_table_KBytNERR = read.csv("otu_final_tucker.csv", sep=",", row.names=1) 
no_dup_otu_table_KBytNERR = as.matrix(no_dup_otu_table_KBytNERR)

# import as phyloseq objects
no_dup_OTU_KBytNERR = otu_table(no_dup_otu_table_KBytNERR, taxa_are_rows = TRUE) 


taxonomy_Phyto = read.csv("taxonomy_table_v23.csv", sep=",", row.names=1) 
taxonomy_Phyto = as.matrix(taxonomy_Phyto)
TAX_Phyto = tax_table(taxonomy_Phyto)
ntaxa(TAX_Phyto)

#v5_nodups_wo_AprFP_2019_2021_04212022.csv
#no_dup_metadata_KBytNERRv3.csv
#v6_nodups_wo_AprFP_2019_2021_07062022.csv

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv14.csv", header=TRUE, fill=TRUE)

rownames(no_dup_metadata_KBytNERR) <- no_dup_metadata_KBytNERR$SampleID
no_dup_metadata_KBytNERR$SampleID
META_no_dup_KBytNERR = sample_data(no_dup_metadata_KBytNERR)
nsamples(META_no_dup_KBytNERR)



physeq_Dav_no_dup = phyloseq(no_dup_OTU_KBytNERR, TAX_Phyto, phy_treeKBytNERR)

physeq_Dav_no_dup=merge_phyloseq(physeq_Dav_no_dup, META_no_dup_KBytNERR)
ntaxa(physeq_Dav_no_dup)
#4117
```



```{r}

##Removing unwanted sequences
table(tax_table(physeq_Dav_no_dup)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean <- subset_taxa(physeq_Dav_no_dup, !is.na(Domain) & !Domain %in% c("", "unclassified_Root"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean)
ntaxa(physeq_Dav_no_dup_clean)
#4343
table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#check to see all of the uncharacterized phylum
table(tax_table(physeq_Dav_no_dup_clean)[, "Phylum"], exclude = NULL)
##Removing unwanted sequences

table(tax_table(physeq_Dav_no_dup_clean)[, "Domain"], exclude = NULL)

#see how many unassiged ASVs (to a domain) there are and remove them
physeq_Dav_no_dup_clean_no18S <- subset_taxa(physeq_Dav_no_dup_clean, !is.na(Domain) & !Domain %in% c("", "Eukaryota"))
##note comma before Unassigned is an artifact of my conversion of the taxonomy table
nsamples(physeq_Dav_no_dup_clean_no18S)
ntaxa(physeq_Dav_no_dup_clean_no18S)
#3987
table(tax_table(physeq_Dav_no_dup_clean_no18S)[, "Domain"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)

head(tax_table(rel.abun))
ntaxa(rel.abun)
#3987
#3680
nsamples(rel.abun)
#366


#Bac_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Bac_16S= subset_taxa(physeq_Dav_no_dup_clean_no18S,Domain == "D_0__Bacteria")
ntaxa(Bac_16S)

Arc_16S_rel.abun= subset_taxa(rel.abun,Domain == "D_0__Bacteria")
Arc_16S= subset_taxa(physeq_Dav_no_dup_clean_no18S,Domain == "D_0__Bacteria")
ntaxa(Bac_16S)


rel2Bac_16S <- transform_sample_counts(Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 


Just_Het_Bac_16S= subset_taxa(Bac_16S,Phylum != "D_1__Cyanobacteria")
ntaxa(Just_Het_Bac_16S)

rel2_Het_Bac_16S <- transform_sample_counts(Just_Het_Bac_16S, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 
ntaxa(rel2_Het_Bac_16S)


Cyano = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria")
ntaxa(Cyano)


All_phyto = subset_taxa(physeq_Dav_no_dup_clean_no18S,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto)
nsamples(All_phyto)

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 

All_phyto_Rel = subset_taxa(rel.abun,Phylum == "D_1__Cyanobacteria"| Domain == "Eukaryota_plas")
ntaxa(All_phyto_Rel)
table(tax_table(All_phyto)[, "Phylum"], exclude = NULL)

rel.abun <- transform_sample_counts(physeq_Dav_no_dup_clean_no18S, function(OTU) OTU/sum(OTU))
ntaxa(rel.abun)
#3680
nsamples(rel.abun)
#366


```





######
Figure 1
######


```{r}

colorssite2 <- c("WAI2"='#7F0000', "KAHO"='#f94279',"AR"="#c78100" ,"HP1"="#ffd000", "NB"="#ff7e00","SB"="#c46210","CB"="#ffa812","SR8"="#ff9966", "SR2"="#00cc95", "NR2"="#00cc2f", "STO1"='#0085ff', "NTO1"='#0045ff')


KBaloha=read.csv("KByT_Aloha_enviro.csv")


KBaloha$Site

KBaloha=read.csv("KByT_Aloha_enviro.csv")
aloha=subset(KBaloha, Site=="ALOHA")
summary(aloha$chla.ug.per.L)

blueberry=factor(KBaloha$Site,levels=c("WAI2", "KAHO", "AR", "HP1","NB","SB","CB", "SR8", "SR2", "NR2", "STO1", "NTO1", "ALOHA"))

svg("Silicate_Box_Site_v2.svg", height=2, width=6)
Silicate<- ggplot(data=KBaloha, aes(x=blueberry, y=Silicate, color=Site))+geom_boxplot(outlier.shape = NA) 
Silicate=Silicate + theme(legend.title = element_blank()) + ylab("Silicate (uM)") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
Silicate
dev.off()
ggsave("Silicate_clusters_v2.pdf", height=3, width=6)


svg("Salinity_Box_Site_v2.svg", height=2, width=6)
Salinity<- ggplot(data=no_dup_metadata_KBytNERR, aes(x=blueberry, y=Salinity, color=Site))+geom_boxplot() 
Salinity=Salinity + theme(legend.title = element_blank()) + ylab("Salinity (ppt)") +xlab("") +geom_jitter()+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  
Salinity
dev.off()


svg("chla_Box_Site.svg", height=2, width=6)
Chla<- ggplot(data=KBaloha, aes(x=blueberry, y=chla.ug.per.L, color=Site))+geom_boxplot(outlier.shape = NA) 
Chla=Chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a (ug/L)") +xlab("") +geom_jitter(size=0.6)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) + scale_color_manual(values=colorssite2)  + scale_y_continuous(trans = log10_trans())+  theme(legend.position="none")
Chla
dev.off()

ggsave("chla_sites_v2.pdf", height=3, width=6)



library(cowplot)
svg("Chla_Silicate_Aloha.svg", width=4, height=4)
plot_grid(Chla,Silicate,labels = c('B)', 'C)'), align = 'hv',label_size = 12, ncol = 1)
dev.off()
ggsave("chla_silicate_legend.svg", width=7, height=11)

library(svglite)

svg("DIN_P_4.svg", height=3, width=2)
p=ggplot(no_dup_metadata_KBytNERR, aes(x = Phosphate, y = DIN, color=Site)) + geom_point(size=1.5) + geom_abline(slope = 16, yintercept=0)+ xlim(0,0.5)+ scale_color_manual(values=colorssite2)+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), axis.text.x = element_text(angle = 90), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1)) +ylab("DIN (uM)")+xlab("Phosphate (uM)")+theme(legend.position="none")
p
ggsave("DIN_v2.pdf")
dev.off()

```

######
Figure 2
######
```{r}
# Load libraries
#library(ggplot2)
#library(marmap)
#library(ocedata)
#library(RColorBrewer)
#library(dplyr)
#library(magrittr)
#library(tibble)
#library(ggthemes)

library(tidyverse)
library(lubridate)
library(reshape2)
library(MBA)
library(mgcv)

ODV_colours <- c("#feb483", "#d31f2a", "#ffc000", "#27ab19", "#0db5e6", "#7139fe", "#d16cfa")

###the following can be repeated with all HPLC pigments

KBaloha=read.csv("KByT_Aloha_enviro.csv")


south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
south=south%>%filter(decimal_dat<2019.2)%>%filter(decimal_dat>2017.8)
Sitelabs <- c( "SB","HP1", "SR8", "SR2", "STO1", "ALOHA")


south2=south%>%dplyr::select( Site,decimal_dat,Fuco.Rel)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(Fuco.Rel)
str(south2)

south2$Site=as.numeric(south2$Site)
#write.csv(south2, "South2.csv")
#south2=read.csv("South2.csv", row.names=1)
# The date column must then be converted to numeric values

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'Fuco.Rel') 
 # filter(depth < 0) %>% 
 # mutate(temp = round(temp, 1))


# Finally we create our gridded result
Fuco=ggplot(data = ctd_mba, aes(y =decimal_dat, x = Site)) +
  geom_raster(aes(fill = Fuco.Rel)) +
  scale_fill_gradientn(colours = rev(ODV_colours)) +
  geom_contour(aes(z = Fuco.Rel), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = Fuco.Rel), breaks = 4, colour = "black") +
### Activate to see which pixels are real and not interpolated
 geom_point(data = south2, aes( y = decimal_dat,x =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(y = "Date", x = NULL, fill = "Diatoms \n Fucoxanthin:Tchla") +
  coord_cartesian(expand = 0)+ scale_x_continuous(labels=Sitelabs)+scale_y_continuous(breaks=c(2018.0,2018.5, 2019.0), labels=c("Jan 2018", "Jul 2018", "Jan 2019"))

Fuco
ggsave("Fuco_ODV_KByT_SB.pdf")


##################FCM- the following can be repeated with all FCM

KBaloha=read.csv("KByT_Aloha_enviro.csv")

south=subset(KBaloha,Site=="SB"|Site=="HP1"|Site=="SR8"|Site=="SR2"|Site=="STO1"|Site=="ALOHA")
#south=south%>%filter(decimal_dat<2020.0)
Sitelabs <- c( "SB","HP1", "SR8", "SR2", "STO1", "ALOHA")

south2=south%>%dplyr::select(Site,decimal_dat,PRO.mL2)%>%dplyr::mutate_at(c('Site'),funs(str_replace(., "SB", "1")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "HP1", "2")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR8", "3")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "SR2", "4")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "STO1", "5")))%>%
  dplyr::mutate_at(c('Site'),funs(str_replace(., "ALOHA", "6")))%>%drop_na(PRO.mL2)
str(south2)
type(south2$Site)
south2$Site=as.numeric(south2$Site)
#write.csv(south2, "South2.csv")
#south2=read.csv("South2.csv", row.names=1)
# The date column must then be converted to numeric values

# Now we may interpolate the data
ctd_mba <- mba.surf(south2, no.X = 300, no.Y = 300, extend = T)
dimnames(ctd_mba$xyz.est$z) <- list(ctd_mba$xyz.est$x, ctd_mba$xyz.est$y)
ctd_mba <- melt(ctd_mba$xyz.est$z, varnames = c( 'Site','decimal_dat'), value.name = 'PRO.mL2') 
 # filter(depth < 0) %>% 
 # mutate(temp = round(temp, 1))


# Finally we create our gridded result
svg("PRO.mL2_ALOHA_Contour.svg", width=4, height=4)
PRO=ggplot(data = ctd_mba, aes(y =decimal_dat, x = Site)) +
  geom_raster(aes(fill = PRO.mL2)) +
  scale_fill_gradientn(colours = rev(ODV_colours),limits = c(0,300000)) +
  geom_contour(aes(z = PRO.mL2), binwidth = NULL, colour = "black", alpha = 0.2) +
  geom_contour(aes(z = PRO.mL2), breaks = 40, colour = "black") +
### Activate to see which pixels are real and not interpolated
 geom_point(data = south2, aes( y = decimal_dat,x =Site),
     color = 'black', size = 0.2, alpha = 0.4, shape = 8)+
###
  labs(y = "Date", x = NULL, fill = "Prochlorococcus \n cells mL") +
  coord_cartesian(expand = 0)+ scale_x_continuous(labels=Sitelabs)

PRO
dev.off()
ggsave("PRO_ODV_KByT_ALOHA.pdf")



library(cowplot)
svg("Contour_plots_HPLC_FCM_ALOHA.svg", height=9, width=12)
plot_grid(SYN, PRO, PEUK,HBACT,Tchla,Allo, Fuco,Peri,X19But,X19Hex, Zea, Dva, ncol=3, align="hv")
dev.off()
ggsave("Contour_plots_HPLC_FCM_ALOHA.pdf", height=9, width=13)



```



######
Figure 3
######

```{r}


library(magrittr)
library(dendextend)
library(phyloseq)
library(microbiome)


colorssite2 <- c("WAI2"='#000000', "KAHO"='#f94279',"AR"="#967969" ,"HP1"="#ffd000", "NB"="#ff8c00","SB"="#C47F2B","CB"="#ad0201ff","SR8"="#D7B4F3", "SR2"="#AAFF00", "NR2"="#049861", "STO1"='#00FFFF', "NTO1"='#0045ff')

Site_type <- factor(sample_data(All_phyto)$Site, levels=c("WAI2","KAHO","AR","HP1","NB","SB","CB","SR8","SR2","NR2","STO1","NTO1"))
colors_Site <- c("#000000","#f94279","#967969","#ffd000","#ff8c00","#C47F2B","#ad0201","#D7B4F3", "#AAFF00", "#049861", '#00FFFF','#0045ff')
col_Site_type <- colors_Site[Site_type]
col_Site_type



Spatial_type <- factor(sample_data(All_phyto)$Spatial, levels=c("Coastal","Transition","Offshore"))
#n_Enviro_types <- length(unique(Enviro_type))
#Enviro_cols <- colorspace::rainbow_hcl(n_Enviro_types, c = 70, l  = 50)
colors_spatial <- c("gold","darkseagreen","dodgerblue3")
col_Spatial_type <- colors_spatial[Spatial_type]


CLR_All_Phyto=transform(All_phyto, transform="clr")
euc_dist <- dist(t(otu_table(CLR_All_Phyto)))
euc_clust <- hclust(euc_dist, method="ward.D2")
d1=euc_clust%>% as.dendrogram() 
plot(d1)
k234 <- cutree(d1, k = 2:4)
svg("new_Season_CLR_check_v2.svg", height=10, width=20)
par(mar = c(2,2,2,20))
d1%>% 
    set("labels_cex",0.2) %>% 
    plot(main="", horiz = TRUE)
colored_bars(cbind(col_Site_type, col_Spatial_type), d1, rowLabels = c( "Site", "Environment"), horiz = TRUE, y_shift=20)
dev.off()
k234[,2]
write.csv(k234[,2], "maybce.csv")
write.csv(labels(d1), "new_order.csv")


test=psmelt(rel2All_phyto)
ntaxa(rel2All_phyto)
nsamples(rel2All_phyto)


coloresbarplot2=c("Archaeplastida:Tetraselmis"="greenyellow",
"Archaeplastida:Mamiellophyceae" ="#00A300",
"Archaeplastida:Ostreococcus"="#688873",
"Archaeplastida:Micromonas"="#0CBAA6",
"Archaeplastida:Chloropicon"="darkolivegreen3", 
"Archaeplastida:Other"="#bef4dc", 
"Hacrobia:Other"="#EDABEF",
"Hacrobia:Cryptophyceae"="coral", 
"Hacrobia:Teleaulax"="maroon1",
"Hacrobia:Prymnesiophyceae"="red",
"Stramenopiles:Bacillariophyta"="gold",
"Stramenopiles:Chaetoceros"="#fffdd0",
"Stramenopiles:Aureococcus"="#C3AD30",
"Stramenopiles:Other"="#714416",
'Eukaryota:Other'="black",
"Cyanobacteria:Prochlorococcus"="navy",
"Cyanobacteria:Synechococcus"="dodgerblue3",
"Cyanobacteria:Cyanobium"="deepskyblue",
"Cyanobacteria:Other"="lightblue2")


test$Date1 <- as.Date(test$Date1,format = "%m/%d/%y")


test2=test%>%subset(Season_Tucker2021=="Summer" | Season_Tucker2021=="Winter")

Three_t=subset(test2, Spatial=="Offshore")
p_3 <- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated_v4))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance (%)")+facet_wrap(~Season_Tucker2021,scales = "free_x") +
  scale_x_discrete()

p_3

ggsave("Offshore_Seasons.pdf")


Three_t=subset(test2, Spatial=="Transition")
p_2 <- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated_v4))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance (%)") +facet_wrap(~Season_Tucker2021,scales = "free_x") +
  scale_x_discrete()
p_2

ggsave("Transition_Seasons.pdf")


Three_t=subset(test2, Spatial=="Coastal")
p_1<- ggplot(data=Three_t, aes(x=Ordered_Site_Sample, y=Abundance*100, fill=as.factor(Curated_v4))) + geom_bar(aes(), stat="identity", position="stack") +theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + scale_fill_manual(values=coloresbarplot2)+ theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Relative abundance(%)")+ facet_wrap(~Season_Tucker2021,scales = "free_x") +  geom_col(width = 1)+
  scale_x_discrete()
p_1


ggsave("Coastal_season.pdf")


library(cowplot)
svg("plots_over_Time_Spatial_clusters_v2.svg", height=5.5, width=4)
plot_grid(p_1,p_2, p_3, ncol=1, nrow=3)
dev.off()

```

#######
Figure 4
########

```{r}
#######
#Figure 4a
########
rel2Phyto_ASV_total <-rel2All_phyto %>% psmelt() %>% dplyr::group_by(SampleID, Spatial, Curated_v4) %>% dplyr::summarise(sum = sum(Abundance)) 


rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Spatial, Curated_v4) %>% dplyr::summarise(mean = mean(sum)*100, sd=sd(sum)*100) 





ugh=dcast(rel2Phyto_ASV_Cluster, Curated_v4~Spatial, value.var ="mean")
ugh$sum=rowSums(ugh[2:4])
ugh$mean=rowMeans(ugh[2:4])

write.csv(ugh, "top_finding.csv")
top10=subset(rel2Phyto_ASV_Cluster, Curated_v4=="Cyanobacteria:Synechococcus"|
Curated_v4=="Archaeplastida:Mamiellophyceae"|
Curated_v4=="Stramenopiles:Bacillariophyta"|
Curated_v4=="Cyanobacteria:Cyanobium"|
Curated_v4=="Hacrobia:Prymnesiophyceae"|
Curated_v4=="Stramenopiles:Chaetoceros"|
Curated_v4=="Archaeplastida:Micromonas"|
Curated_v4=="Hacrobia:Cryptophyceae"|
Curated_v4=="Archaeplastida:Ostreococcus"|
Curated_v4=="Cyanobacteria:Prochlorococcus")




test=psmelt(rel2All_phyto)
top10$Curated_v4=factor(top10$Curated_v4, levels=c("Stramenopiles:Chaetoceros",
"Hacrobia:Cryptophyceae",
"Archaeplastida:Micromonas",
"Archaeplastida:Ostreococcus","Cyanobacteria:Cyanobium",
"Hacrobia:Prymnesiophyceae","Stramenopiles:Bacillariophyta","Archaeplastida:Mamiellophyceae","Cyanobacteria:Prochlorococcus","Cyanobacteria:Synechococcus"
))


top10$Spatial=factor(top10$Spatial, levels=c("Coastal", "Transition", "Offshore"))


svg("heatmap_Spatial_Clusters_top10.svg", width=8, height=6)
bp=ggplot(top10, aes(x=as.factor(Spatial), y=as.factor(Curated_v4), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) +
  scale_fill_viridis(option="turbo",trans = 'log10')  + labs(fill = "Mean Relative Abundance \n as Percent of \n Phytoplankton Abundance")+theme(axis.text=element_text(size=11), legend.text=element_text(size=11), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+xlab("")+ylab("")
bp 

dev.off()

ggsave("heatmap_Clusters_spatial_mean.pdf", width=8, height=6)


```




```{r}
#######
#Figure 4b
########
glom=tax_glom(All_phyto, "Genus")
ntaxa(glom)
dex = phyloseq::phyloseq_to_deseq2(glom, ~Spatial)

dex = phyloseq::phyloseq_to_deseq2(glom, ~Spatial)

diagdds_pol = DESeq2::DESeq(dex, test="Wald", fitType="parametric")
res = results(diagdds_pol)
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(glom)[rownames(sigtab), ], "matrix"))
head(sigtab)



sigtab2=sigtab [,c(1:6,15)]
sigtab2$DeSeq_Spatial_Cluster=sigtab$Domain
sigtab2$DeSeq_Spatial_Cluster="DeSeq_Spatial_Variable"
sigtab2=as.data.frame(sigtab2)
paper=left_join(taxonomy_Phyto_we,sigtab2,by="ASV")

write.csv(paper,"taxonmy_all_Deseqdata.csv")
left_join()


res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Offshore"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Transition_Offshore.csv")

Genus_Transition_Offshore=sigtab

Genus_Transition_Offshore$Genus_Transition_Offshore="Genus_Transition_Offshore"

names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'log2FoldChange'] <- 'Genus_TO_log2FoldChange'
names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'padj'] <- 'Genus_TO_padj'
names(Genus_Transition_Offshore)[names(Genus_Transition_Offshore) == 'baseMean'] <- 'Genus_TO_baseMean'

Genus_Transition_Offshore_v2=Genus_Transition_Offshore%>%dplyr::select(Genus, Genus_TO_log2FoldChange, Genus_TO_padj, Genus_TO_baseMean,Genus_Transition_Offshore)

names(Genus_Transition_Offshore)


#####################3
#####################

res <- results(diagdds_pol, contrast=c("Spatial", "Transition","Coastal"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Coastal_Transition.csv")

Genus_Coastal_Transition=sigtab
Genus_Coastal_Transition$Genus_Coastal_Transition="Genus_Coastal_Transition"

names(Genus_Coastal_Transition)[names(Genus_Coastal_Transition) == 'log2FoldChange'] <- 'Genus_CT_log2FoldChange'
names(Genus_Coastal_Transition)[names(Genus_Coastal_Transition) == 'padj'] <- 'Genus_CT_padj'
names(Genus_Coastal_Transition)[names(Genus_Coastal_Transition) == 'baseMean'] <- 'Genus_CT_baseMean'

Genus_Coastal_Transition_v2=Genus_Coastal_Transition%>%dplyr::select(Genus, Genus_CT_log2FoldChange, Genus_CT_padj, Genus_CT_baseMean,Genus_Coastal_Transition)



#####################3
#####################

res <- results(diagdds_pol, contrast=c("Spatial", "Offshore","Coastal"))
alpha = 0.05
sigtab = res[which(res$padj < alpha), ]

sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(All_phyto)[rownames(sigtab), ], "matrix"))
head(sigtab)
write.csv(sigtab, "Genus_DESeq_All_Phyto_Clustering_NEWCLR_Coastal_Offshore.csv")

Genus_Coastal_Offshore=sigtab
Genus_Coastal_Offshore$Genus_Coastal_Offshore="Genus_Coastal_Offshore"

names(Genus_Coastal_Offshore)[names(Genus_Coastal_Offshore) == 'log2FoldChange'] <- 'Genus_CO_log2FoldChange'
names(Genus_Coastal_Offshore)[names(Genus_Coastal_Offshore) == 'padj'] <- 'Genus_CO_padj'
names(Genus_Coastal_Offshore)[names(Genus_Coastal_Offshore) == 'baseMean'] <- 'Genus_CO_baseMean'

Genus_Coastal_Offshore_v2=Genus_Coastal_Offshore%>%dplyr::select(Genus, Genus_CO_log2FoldChange, Genus_CO_padj, Genus_CO_baseMean,Genus_Coastal_Offshore)



COT=full_join(Genus_Coastal_Offshore_v2,Genus_Coastal_Transition_v2, by="Genus")
COOT=full_join(COT,Genus_Transition_Offshore_v2, by="Genus")



taxonomy_Phyto_we = read.csv("taxonomy_table_v18.csv", sep=",") 
maybe=left_join(COOT,taxonomy_Phyto_we,by="Genus")
maybe2=maybe%>%distinct(Genus, .keep_all = TRUE)

write.csv(maybe2, "Genus_all_comparisons_NEWCLR_Spatial.csv")

###note differentially abundant taxa that are unidentified at higher than the class level were removed 
## I subsetted the data table below based on the differentially abundant genera

```

```{r}

p0.rel <- transform(All_phyto, "clr")

relglom=tax_glom(p0.rel, "Genus")
relglomdiffer=subset_taxa(relglom, Genus=="D_5__Prochlorococcus_MIT9313" |
Genus=="D_5__Cyanobium_PCC-6307" |
Genus=="D_5__Atelocyanobacterium_(UCYN-A)" |
Genus=="D_5__Crocosphaera" |
Genus=="D_5__Trichodesmium_IMS101" |
Genus=="Tetraselmis" |
Genus=="Chloropicon" |
Genus=="unclassified_Bathycoccaceae" |
Genus=="Ostreococcus" |
Genus=="Bathycoccus" |
Genus=="Micromonas" |
Genus=="unclassified_Mamiellaceae" |
Genus=="Mamiella" |
Genus=="Marsupiomonas" |
Genus=="Resultor" |
Genus=="Prasino-Clade-9_XXX" |
Genus=="Prasinoderma" |
Genus=="Eutreptiella" |
Genus=="Proteomonas" |
Genus=="unclassified_Hemiselmidaceae" |
Genus=="Braarudosphaera" |
Genus=="Phaeocystis" |
Genus=="unclassified_Prymnesiaceae" |
Genus=="Thalassionema" |
Genus=="Chaetoceros" |
Genus=="unclassified_Polar-centric-Mediophyceae" |
Genus=="unclassified_Radial-centric-basal-Coscinodiscophyceae" |
Genus=="unclassified_Raphid-pennate" |
Genus=="Ochromonas" |
Genus=="Mesopedinella" |
Genus=="unclassified_Pedinellales" |
Genus=="Helicopedinella" |
Genus=="unclassified_Pelagomonadaceae" |
Genus=="Aureococcus")

######
v2 <-relglomdiffer %>% psmelt() %>% dplyr::group_by(SampleID, Cluster_DeSeq_Spatial, Genus) %>% dplyr::summarise(sum = sum(Abundance)) 


rel2Phyto_ASV_total <-relglomdiffer %>% psmelt() %>% dplyr::group_by(SampleID, Cluster_DeSeq_Spatial, Genus) %>% dplyr::summarise(sum = sum(Abundance)) 


rel2Phyto_ASV_Cluster <-rel2Phyto_ASV_total %>% dplyr::group_by(Cluster_DeSeq_Spatial, Genus) %>% dplyr::summarise(mean = mean(sum), sd=sd(sum)) 


rel2Phyto_ASV_Cluster$Genus=factor(rel2Phyto_ASV_Cluster$Genus, levels=c("D_5__Prochlorococcus_MIT9313","D_5__Cyanobium_PCC-6307","D_5__Atelocyanobacterium_(UCYN-A)","D_5__Crocosphaera" ,"D_5__Trichodesmium_IMS101" ,"Tetraselmis" ,"Chloropicon" ,"unclassified_Bathycoccaceae" ,"Ostreococcus" ,"Bathycoccus" ,"Micromonas" ,"unclassified_Mamiellaceae" ,"Mamiella" ,"Marsupiomonas" ,"Resultor" ,"Prasino-Clade-9_XXX","Prasinoderma","Eutreptiella","Proteomonas","unclassified_Hemiselmidaceae","Braarudosphaera","Phaeocystis","unclassified_Prymnesiaceae","Thalassionema","Chaetoceros","unclassified_Polar-centric-Mediophyceae","unclassified_Radial-centric-basal-Coscinodiscophyceae","unclassified_Raphid-pennate" ,"Ochromonas" ,"Mesopedinella","unclassified_Pedinellales","Helicopedinella" ,"unclassified_Pelagomonadaceae","Aureococcus"))

library(rstatix)
library(scales)
library(viridis)
?log10_trans()

names(rel2Phyto_ASV_Cluster)

rel2Phyto_ASV_Cluster$Cluster_DeSeq_Spatial=factor(rel2Phyto_ASV_Cluster$Cluster_DeSeq_Spatial, levels=c("Coastal", "Transition", "Offshore"))
#https://stackoverflow.com/questions/48424682/how-do-i-limit-the-range-of-the-viridis-colour-scale
bp=ggplot(rel2Phyto_ASV_Cluster, aes(x=as.factor(Cluster_DeSeq_Spatial), y=as.factor(Genus), fill = mean)) +
    geom_tile(aes(fill = mean)) + 
    geom_text(aes(label = round(mean, 1))) + 
  scale_fill_gradientn(colors = viridis_pal()(9), limits=c(-6, 5), 
                       na.value = "#FDE725FF")+theme(axis.text=element_text(size=14), legend.text=element_text(size=14), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
bp 

ggsave("CLR_genera_that_Differ_CLR.pdf", height=6, width=9)


###based on the peaks of CLR abundance I made the following csv 
genotype <- read.csv("summary_genera.csv", row.names=1)

unique(genotype$Genus)

pattern_df = data.frame(genotype)
rownames(pattern_df) = rownames(genotype)


svg("genus_Peak.svg", height=11, width=8.5)
ComplexHeatmap::pheatmap(pattern_df[,1:3], cluster_cols = F, cluster_row = F, annotation_row = pattern_df  , color = c("white","black"),cellwidth=20,cellheight=10, row_names_side = c("left"))
dev.off()
```



#######
Figure 5
########

```{r}
library(data.table)

readrecruit=read.csv("mean_coverage_Q2Q3_cyano_isolate_clade_v2.csv", header=TRUE, row.names = 1)
library(tidyverse)
library(reshape2)
library(dplyr)
library(stats)


woof=aggregate(.~ clade, data = readrecruit, FUN = "sum")
help(aggregate)

samp.with.rownames <- data.frame(woof[,-1], row.names=woof[,1])

#samp.with.rownames2=samp.with.rownames[,-1]

freqs2 <- apply(samp.with.rownames , 2, function(i) i/sum(i))

freqs3=freqs2*100
freqs3=as.data.frame(freqs3)

write.csv(freqs3,"test.csv")

freqs4<- freqs3 %>%
  rownames_to_column() %>%
  gather(colname, value, -rowname)
head(freqs4)




#freqs3=read.csv("SCG_normalized_all_depths.csv", row.name=1)


sample_order <- factor(freqs4$rowname,level=c(
"AS9601",
"SB",
"MIT9314",
"MIT9301",
"MIT9201",
"MIT9215",
"MIT9202",
"MIT9107",
"GP2",
"MIT9302",
"MIT9311",
"MIT9321",
"Other_PRO",
"UW86",
"N19",
"N26",
"N32",
"N5",
"cyano3B",
"CC9605",
"WH8109",
"Other_SYN",
"RCC307"))
freqs4$colname


####this one is right???
level_order <- factor(freqs4$colname, level=c("KSe054HP",
"KSe055AR",
"KOc293R8",
"KSe058SB",
"KOc290HP",
"KOc291AR",
"KSe056CB",
"KSe057R8",
"KFe387AR",
"KFe386HP",
"KFe389R8",
"KFe395NB",
"KMa160SB",
"KMa158CB",
"KMa156HP",
"KMa157AR",
"KMa159R8",
"KMa165NB",
"KOc299NB",
"KSe062NR",
"KSe063NB",
"KMa161R2",
"KMa164NR",
"KOc297NT",
"KOc321ST",
"KSe059R2",
"KFe393NT",
"KFe416ST",
"KMa162ST",
"KMa163NT",
"KSe060ST",
"KSe061NT",
"HOT_224_25M",
"HOT_225_25M",
"HOT_226_25M",
"HOT_227_25M",
"HOT_229_25M",
"HOT_231_25M",
"HOT_232_25M",
"HOT_233_25M",
"HOT_234_25M",
"HOT_236_25M",
"HOT_237_25M",
"HOT_238_25M"))
freqs4[freqs4 == 0] <- NA
freqs4[freqs4<0.5]<- NA


svg("PRO_SYN_clade_isolate_v3.svg", width=6, height=5)
geom.text.size = 3
recruit=ggplot(freqs4, aes(x = level_order, y = sample_order, fill = value)) +
    geom_tile(aes(fill = value)) + 
    #geom_text(aes(label = round(value, 0)), size=geom.text.size) +
  scale_fill_gradient2(low = "yellow", high="purple", mid = "red", midpoint = 10,na.value = "white") + scale_x_discrete(name="Metagenomes") + scale_y_discrete(name="Clade", position = "left") + labs(fill = "Summed\npercent\nrecruitment") +  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 90, size=9)) +coord_flip()+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))

recruit
dev.off()

ggsave('Metagenomes_Pro_Syn_Cyano_subclade.pdf', width=9.0, height=8)

```


#######
Figure 6
########

```{r}

?month.abb
no_dup_metadata_KBytNERR$Month
no_dup_metadata_KBytNERR$Date2
no_dup_metadata_KBytNERR$Month = factor(no_dup_metadata_KBytNERR$Month, levels==month.name)
your_data$month = factor(your_data$month, levels = month.abb)
list(month.abb)

library(seecolor)
print_color(colorspatial)
colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Coastal"="#B70D6A")

library(ggplot2)



no_dup_metadata_KBytNERR$Month_abb=factor(no_dup_metadata_KBytNERR$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

library(scales)


a=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SW.Temperature.at.site, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) +ylab("Temperature (C)")+xlab("")
a


f=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=HBACT.mL2, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Heterotrophic prokaryotic cells mL"^-1))
f

g=ggplot(data = no_dup_metadata_KBytNERR, aes(x=Month_abb, y=SYN.mL2, group=Spatial, color=Spatial))+
  geom_smooth() +
  geom_point()+ theme(legend.position="none")+
  scale_color_manual(values=colorspatial)+theme(axis.text=element_text(size=12), legend.text=element_text(size=12), axis.title=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1)) + xlab("")+ylab(bquote("Synechococus cells mL"^-1))

g

names(no_dup_metadata_KBytNERR)
test=no_dup_metadata_KBytNERR%>%drop_na(chla.ug.per.L)%>%subset(Spatial=="Coastal")%>%subset(Season_Tucker2021=="Summer" | Season_Tucker2021=="Winter")

svg("Chlorophyll_Season_Tucker_2021_help.svg",width=5, height=3)
chla<-ggplot(data=test, aes(x=Season_Tucker2021, y=chla.ug.per.L, fill=Season_Tucker2021), color=Sampling.Order)+geom_boxplot(outlier.shape = NA) +geom_jitter(color="#B70D6A")
chla=chla + theme(legend.title = element_blank()) + ylab("Chlorophyll a (ug L)") +xlab("")+ theme(axis.text=element_text(size=12),axis.title=element_text(size=12),  legend.text=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))   + scale_y_continuous(trans = log10_trans())+
  theme(legend.position="none")
chla
dev.off()
ggsave("chla_coastal_only_season.pdf", width=3, height=3)


library(cowplot)
svg("v2_Enviro_sup_figure_spatial_month_based.svg", width=7, height=9)
plot_grid(chla, a, f,g,labels = c('a', 'b','c', 'd','e', 'f','g', 'h','i', 'j','k', 'l'), align = 'hv',label_size = 12, ncol = 2, nrow=3)

dev.off()
ggsave("v2_Enviro_sup_figure_spatial_month_based.pdf", width=19, height=12)

dev.off()


##Correlation plot 
no_dup_metadata_KBytNERR2=read.csv("no_dup_metadata_KBytNERRv15.csv", header=TRUE, fill=TRUE)

enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Coastal")%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL2",
                                                       "Log_SYN.mL2", 
                                                       "Log_PEUK.mL2", 
                                                       "Log_HBACT.mL2", 
                                                        "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day"))

###"Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"

no_dup_metadata_KBytNERR$Date1

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))

?corrplot()
??cor()

svg("Coastal_Chla_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off

```




######
Figure 7
######


```{r}
##Use Lomb Scargle Periodograms to determine the seasonally significant taxa
####This code was built from the work of https://github.com/adriaaulaICM/bbmo_niche_sea/
###repeat with subsetting for each environment
library(tidyverse)
library(dplyr)
glom_genus=tax_glom(All_phyto, "Genus")

bl.phy.asinh <- transform_sample_counts(glom_genus, function(x) asinh(x))
bl.phy.asinh <- subset_samples(bl.phy.asinh, Spatial=="Offshore")
nsamples(bl.phy.asinh)
ntaxa(bl.phy.asinh)
physeq.list <- NULL

psmelt_dplyr = function(physeq) {
  
  sd = data.frame(sample_data(physeq)) %>% 
    rownames_to_column("Sample")
  TT = data.frame(as(tax_table(physeq),'matrix')) %>%
    rownames_to_column("OTU")
  if(taxa_are_rows(physeq)){
    otu.table = data.frame(as(otu_table(physeq),"matrix"),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  } else {
    otu.table = data.frame(t(as(otu_table(physeq),"matrix")),
                           check.names = FALSE) %>%
      rownames_to_column("OTU")
  }
  
  all <- otu.table %>%
    left_join(TT, by = "OTU") %>%
    gather_("Sample", "Abundance", setdiff(colnames(otu.table), "OTU")) %>%
    left_join(sd, by = "Sample") %>% 
    dplyr::select(Sample, Abundance, OTU, everything())
  
  return(all)
}


tsdf.0.1 <- bl.phy.asinh %>% 
  psmelt_dplyr() 


df.ts <- tsdf.0.1 %>% 
  arrange(decimal_dat) %>% 
  dplyr::select(decimal_dat, Abundance, OTU) %>% 
  split(.$OTU, drop = T)

library(lomb)

lomb.03  <-  df.ts %>% 
  map(~lsp( x =.x$Abundance,
                times = .x$decimal_dat,
                type = 'period',
                plot = T, ofac=5, from=0.16))


library(fdrtool)
lomb.sea.03.ofac <- tibble( asv = names(lomb.03),
                       pval = map_dbl(lomb.03, ~.x$p.value),
                       peak = map_dbl(lomb.03, ~.x$peak),
                       interval  = map(lomb.03, ~.x$peak.at),
                       int.min = map_dbl(interval, ~.[[2]]),
                       int.max = map_dbl(interval, ~.[[1]])) %>% 
  drop_na(int.max) %>%
  mutate(qval = fdrtool::fdrtool(pval, statistic = 'pvalue')$qval) %>% 
  filter(qval <= 0.01)%>%filter(int.max>0.75)%>%filter(int.max<1.25)

results.lomb03 <- lomb.03[lomb.sea.03.ofac  %>% pull(asv)]

write_rds(results.lomb03, 'Genus_Offshore_from016_lomb_v2.rds')

# Strip problems
lil.strip <- theme(strip.background = element_blank(),
                   strip.text.x =element_text(margin = margin(.05, 0, .1, 0, "cm")))

plotsPath = "Genus_periodoplots_from016_Offshore_SigASV.pdf"
pdf(file=plotsPath)  

periodoplots <- map(results.lomb03, ~tibble( scanned = .x$scanned,
                          power = .x$power)) %>% 
  bind_rows(.id = 'asv') %>% 
  split(.$asv) %>% 
  map(~ggplot(.x, aes(scanned, power)) + 
        geom_line(aes(group = asv)) + 
        facet_wrap(~asv) + 
        lil.strip)

periodoplots
dev.off()
```



```{r}
## having identified the seasonally significant taxa above use the code below to graph

rel2All_phyto <- transform_sample_counts(All_phyto, function(OTU) OTU/sum(OTU))
#write.csv(as.data.frame(otu_table(rel2All_phyto)), "otu_rel2All_phyto.csv") 

rel_glom=tax_glom(rel2All_phyto, "Genus")
ntaxa(rel_glom)

colorsgenus=c("unclassified_Pycnococcaceae"="#D2DA9F", 
"D_5__Prochlorococcus_MIT9313"="navy",
"Mesopedinella"="#714416",
"Chaetoceros"="#CB9D06",
"Chloropicon"="#088F8F",
"Teleaulax"="coral",
"unclassified_Mamiellaceae"="#909F2D",
"Micromonas"="darkseagreen",
"Isochrysis"="#DA70D6",
"Partenskyella"="#9D9997",
"unclassified_Polar-centric-Mediophyceae"="#FFCE03",
"Helicopedinella"="#B2AA93",
"Eutreptiella"="black",
"D_5__Synechococcus_CC9902"="dodgerblue3",
"Chrysochromulina"="red",
"Braarudosphaera"="#9E476C",
"D_5__Cyanobium_PCC-6307"="deepskyblue",
"Mamiella"="#64E986","Prasinoderma"="chartreuse4", "Pyramimonas"="#3EB489", "D_5__Crocosphaera"="#94A2FA","unclassified_Pelagomonadaceae"="#927E25","D_5__Atelocyanobacterium_(UCYN-A)"="#8D1CFF", "Micromonas"="darkseagreen")

library(seecolor)
print_color(colorsgenus)

colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Coastal"="#B70D6A")

Rel_glom_Phyto_Coastal=subset_samples(rel_glom, Spatial=="Coastal")
Rel_glom_Phyto_Coastal_2<-Rel_glom_Phyto_Coastal %>% psmelt() %>% dplyr::group_by( Month_abb, Supergroup, Genus, Class)
#%>% dplyr::summarise(mean = mean(Abundance)*100) 

Rel_glom_Phyto_Coastal_3=Rel_glom_Phyto_Coastal_2%>%subset(Genus=="D_5__Synechococcus_CC9902")

sum<-Rel_glom_Phyto_Coastal_3%>% dplyr::group_by(Season)%>%summarise(mean=mean(Abundance),sd=sd(Abundance))
Rel_glom_Phyto_Coastal_3$Month_abb=factor(Rel_glom_Phyto_Coastal_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar", "Apr"))


#+facet_wrap(~Supergroup, scales="free_y", ncol=6)

Rel_glom_Phyto_Coastal_3=Rel_glom_Phyto_Coastal_2%>%subset( Genus=="Chaetoceros"|
Genus=="unclassified_Mamiellaceae"|
Genus=="unclassified_Polar-centric-Mediophyceae"|
Genus=="D_5__Cyanobium_PCC-6307"| Genus=="Micromonas"|Genus=="Chloropicon"| Genus=="Teleaulax"|Genus=="D_5__Prochlorococcus_MIT9313")

Rel_glom_Phyto_Coastal_3$Month_abb=factor(Rel_glom_Phyto_Coastal_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar", "Apr"))

cb<-ggplot(data=Rel_glom_Phyto_Coastal_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
cb



Rel_glom_Phyto_Coastal_3=Rel_glom_Phyto_Coastal_2%>%subset( Genus=="Eutreptiella"| Genus=="unclassified_Pycnococcaceae"|
Genus=="Mesopedinella"|
Genus=="Isochrysis"|
Genus=="Partenskyella"|
Genus=="Helicopedinella"|
Genus=="Chrysochromulina"|
Genus=="Braarudosphaera"|
Genus=="Mamiella")
Rel_glom_Phyto_Coastal_3$Month_abb=factor(Rel_glom_Phyto_Coastal_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar", "Apr"))
svg("legend.svg", width=6, height=5)
cc<-ggplot(data=Rel_glom_Phyto_Coastal_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "right")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)+ylim(0,1)
cc
dev.off()





Rel_glom_Phyto_Offshore=subset_samples(rel_glom, Spatial=="Offshore")
Rel_glom_Phyto_Offshore_2<-Rel_glom_Phyto_Offshore %>% psmelt() %>% dplyr::group_by( Month_abb, Supergroup, Genus, Class) 
#%>% dplyr::summarise(mean = mean(Abundance)*100) 

Rel_glom_Phyto_Offshore_3=Rel_glom_Phyto_Offshore_2%>%subset( 
 Genus== "D_5__Atelocyanobacterium_(UCYN-A)" |
 Genus== "Micromonas" |
 Genus== "D_5__Crocosphaera" |
 Genus== "unclassified_Pelagomonadaceae" |
 Genus== "D_5__Synechococcus_CC9902" |
 Genus== "Braarudosphaera" )


sum<-Rel_glom_Phyto_Offshore_3 %>% dplyr::group_by(SampleID)%>%summarise(sum=sum(Abundance))
mean=sum%>%summarise(mean=mean(sum), sd=sd(sum))



Rel_glom_Phyto_Offshore_3$Month_abb=factor(Rel_glom_Phyto_Offshore_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

Rel_glom_Phyto_Offshore_3=Rel_glom_Phyto_Offshore_2%>%subset( 
 Genus== "D_5__Synechococcus_CC9902"  )


Syn=Rel_glom_Phyto_Offshore_2%>%subset( 
 Genus== "D_5__Synechococcus_CC9902"  )

sum<-Syn%>% dplyr::group_by(Season)%>%summarise(mean=mean(Abundance),sd=sd(Abundance))


Rel_glom_Phyto_Offshore_3$Month_abb=factor(Rel_glom_Phyto_Offshore_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

oa<-ggplot(data=Rel_glom_Phyto_Offshore_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
oa


Rel_glom_Phyto_Offshore_3=Rel_glom_Phyto_Offshore_2%>%subset( 
 Genus== "Micromonas" |
 Genus== "Braarudosphaera" )


Rel_glom_Phyto_Offshore_3$Month_abb=factor(Rel_glom_Phyto_Offshore_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))


ob<-ggplot(data=Rel_glom_Phyto_Offshore_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
ob



Rel_glom_Phyto_Offshore_3=Rel_glom_Phyto_Offshore_2%>%subset( 
 Genus== "D_5__Atelocyanobacterium_(UCYN-A)" |
 Genus== "D_5__Crocosphaera" |
 Genus== "unclassified_Pelagomonadaceae"  )


Rel_glom_Phyto_Offshore_3$Month_abb=factor(Rel_glom_Phyto_Offshore_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))


oc<-ggplot(data=Rel_glom_Phyto_Offshore_3, aes(x=Month_abb, y=Abundance, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
oc




Rel_glom_Phyto_Transition=subset_samples(rel_glom,Spatial=="Transition")
Rel_glom_Phyto_Transition_2<-Rel_glom_Phyto_Transition %>% psmelt() %>% dplyr::group_by( Month_abb, Supergroup, Genus, Class) 
#%>% dplyr::summarise(mean = mean(Abundance)*100) 

Rel_glom_Phyto_Transition_3=Rel_glom_Phyto_Transition_2%>%subset(  Genus== "Prasinoderma" |
 Genus== "Pyramimonas" |  Genus== "D_5__Synechococcus_CC9902" | Genus== "Braarudosphaera" | Genus== "D_5__Cyanobium_PCC-6307" )

Syn=Rel_glom_Phyto_Transition_2%>%subset( Genus== "D_5__Synechococcus_CC9902" )

sum<-Syn%>% dplyr::group_by(Season)%>%dplyr::summarise(mean=mean(Abundance),sd=sd(Abundance))



Rel_glom_Phyto_Transition_3$Month_abb=factor(Rel_glom_Phyto_Transition_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))


Rel_glom_Phyto_Transition_3=Rel_glom_Phyto_Transition_2%>%subset(
 Genus== "D_5__Synechococcus_CC9902" )

Rel_glom_Phyto_Transition_3$Month_abb=factor(Rel_glom_Phyto_Transition_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

ta<-ggplot(data=Rel_glom_Phyto_Transition_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
ta



Rel_glom_Phyto_Transition_3=Rel_glom_Phyto_Transition_2%>%subset(
 Genus== "Pyramimonas" | Genus== "D_5__Cyanobium_PCC-6307" )

Rel_glom_Phyto_Transition_3$Month_abb=factor(Rel_glom_Phyto_Transition_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

tb<-ggplot(data=Rel_glom_Phyto_Transition_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
tb


Rel_glom_Phyto_Transition_3=Rel_glom_Phyto_Transition_2%>%subset(  Genus== "Prasinoderma" | Genus== "Braarudosphaera" )

Rel_glom_Phyto_Transition_3$Month_abb=factor(Rel_glom_Phyto_Transition_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar","Apr"))

tc<-ggplot(data=Rel_glom_Phyto_Transition_3, aes(x=Month_abb, y=Abundance*100, color=Genus, group=Genus)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorsgenus)
tc





Rel_glom_Phyto_Coastal_2<-rel_glom %>% psmelt() %>% dplyr::group_by( Month_abb, Supergroup, Genus, Class, Spatial)
#%>% dplyr::summarise(mean = mean(Abundance)*100) 

Rel_glom_Phyto_Coastal_3=Rel_glom_Phyto_Coastal_2%>%subset(Genus=="D_5__Synechococcus_CC9902")

Rel_glom_Phyto_Coastal_3$Month_abb=factor(Rel_glom_Phyto_Coastal_3$Month_abb, levels=c( "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec", "Jan", "Feb", "Mar", "Apr"))


ca<-ggplot(data=Rel_glom_Phyto_Coastal_3, aes(x=Month_abb, y=Abundance*100, color=Spatial, group=Spatial)) + geom_point(alpha=0.5)+geom_smooth(se=FALSE)+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text(angle = 90, vjust = 1, hjust=1)) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))+ xlab("") + ylab("Mean Relative abundance (%)") +ggtitle("")+scale_color_manual(values=colorspatial)
ca


library(cowplot)

svg("by_abundance_seasonal_genera.svg", width=7, height=9)
plot_grid(ca, cb,cc,ta,tb,tc,oa,ob,oc,labels = c('a', 'b', 'c', 'd','e','f','g','h','i'), align = 'hv',axis = "bt",label_size = 12, ncol = 3, nrow=3)
ggsave("by_abundance_seasonal_genera.pdf", width=7, height=9)
dev.off()

All_phyto$Season_Tucker2021
sample_data(All_phyto)$Season_Tucker2021
win_sum=subset_samples(All_phyto, Season_Tucker2021=="Winter" | Season_Tucker2021=="Summer")
nsamples(win_sum)
win_sum_coastal=subset_samples(win_sum, Spatial=="Coastal")

Coastal_All_phyto=subset_samples(All_phyto, Spatial=="Coastal")
library(DivNet)
library(tidyverse)

divnet_Coastal_Season_2 <- win_sum_coastal %>%
  divnet(X = "Season_Tucker2021", ncores = 4)
divnet_Coastal_Season_2$shannon


p=plot(divnet_Coastal_Season_2$shannon, 
    win_sum_coastal, 
     col = "Season")
p  

save(list="divnet_Coastal_Season_2", file="divnet_Coastal_Season_win_sum.rdata")

dv=divnet_Coastal_Season_2
# my DivNet object - replace yours here :) 
ests <- sapply(dv[["shannon"]], function(x) x$estimate)
lci <- sapply(dv[["shannon"]], function(x) x$interval[1])
uci <- sapply(dv[["shannon"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 dv$X) # the X variables I used in my model

df
meta=as.data.frame(sample_data(win_sum_coastal))


df2=left_join(df, meta, by="SampleID")

ggplot(df2, aes(x = SampleID, xend = SampleID, color=Season_Tucker2021)) +
  geom_point(aes(x = SampleID, y = ests)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

gg=df2%>% arrange(desc(Site_Code), SampleID)

gg$SampleID <- factor(gg$SampleID, levels = gg$SampleID)

gg_aveerage=gg%>%dplyr::group_by(Season_Tucker2021)%>%dplyr::summarise(h0=mean(h0), lci=mean(lci), uci=mean(uci))

gg_aveerage$Spatial="Coastal"
test=rbind(gg_aveerage, Offshore_gg_aveerage)
test2=rbind(test, Transition_gg_aveerage)
test3=test2%>%unite("Season_Spatial", Spatial,Season_Tucker2021, remove=FALSE)
write.csv(test3, "shannons_season_tog_Summer_Winter.csv")
test3$Season_Spatial=factor(test3$Season_Spatial, levels=c("Coastal_Summer","Coastal_Winter", "Transition_Summer", "Transition_Winter", "Offshore_Summer", "Offshore_Winter"))

svg("Season_Sum_winter_alpha_shannons_all.svg", width=2, height=2.4)
shannons=test3%>% ggplot(aes(x = Season_Spatial, xend = Season_Spatial, color=Spatial))+
  geom_point(aes(x = Season_Spatial, y = h0)) +
  geom_segment(aes(y = lci, yend = uci)) +
  ylab(paste("Shannon estimate")) +
  theme_bw() + scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+ylab("Shannon Diversity Estimate")+theme(axis.title=element_text(size=8, face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",
                                    fill = NA,size = 1))
shannons
dev.off()
ggsave("Season_Sum_winter_alpha_shannons_all.pdf", width=4, height=2)

Coastal_Shannon_Season_2=as.data.frame(testDiversity(divnet_Coastal_Season_2, "shannon"))
write.csv(Coastal_Shannon_Season_2, "Coastal_Season_Shannon_2.csv")
Coastal_Season <- divnet_Coastal_Season$`bray-curtis`


Transition_Shannon_Season_2=as.data.frame(testDiversity(divnet_Transition_Season_2, "shannon"))
write.csv(Transition_Shannon_Season_2, "Transition_Season_Shannon_2.csv")

Offshore_Shannon_Season_2=as.data.frame(testDiversity(divnet_Offshore_Season_2, "shannon"))
write.csv(Offshore_Shannon_Season_2, "Offshore_Season_Shannon_2.csv")



####ASV richness
ba <- breakaway(win_sum_coastal)
ba

Coastal_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_coastal, "Season_Tucker2021"))
Coastal_Season_Tucker_bet=as.data.frame(Coastal_Season_Tucker$table)
write.csv(Coastal_Season_Tucker_bet, "richness_Coastal_Season_Tucker.csv")



ba <- breakaway(win_sum_Transition)
ba

Transition_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_Transition, "Season_Tucker2021"))
Transition_Season_Tucker_bet=as.data.frame(Transition_Season_Tucker$table)
write.csv(Transition_Season_Tucker_bet, "richness_Transition_Season_Tucker.csv")



ba <- breakaway(win_sum_Offshore)
ba

Offshore_Season_Tucker <- betta(summary(ba)$estimate,
            summary(ba)$error,
            make_design_matrix(win_sum_Offshore, "Season_Tucker2021"))
Offshore_Season_Tucker_bet=as.data.frame(Offshore_Season_Tucker$table)
write.csv(Offshore_Season_Tucker_bet, "richness_Offshore_Season_Tucker.csv")


ests <- sapply(ba[["richness"]], function(x) x$estimate)

lci <- sapply(ba[["richness"]], function(x) x$interval[1])
uci <- sapply(ba[["richness"]], function(x) x$interval[2])

df <- data.frame("SampleID" = names(ests),
                 "h0" = ests, lci, uci, 
                 ba$X) # the X variables I used in my model

df

plot(ba, Offshore_All_phyto, color = "Season")
ggsave("estimate_richness.pdf")



richness=read.csv("richness_Spatial_Season_Tucker.csv")
richness$Season_Spatial=factor(richness$Season_Spatial, levels=c("Coastal_Summer","Coastal_Winter", "Transition_Summer", "Transition_Winter", "Offshore_Summer", "Offshore_Winter"))


svg("Season_Spatial_ASV_richness_Tucker2021.svg", width=2, height=2.4)
ASVrich=richness%>%ggplot(aes(x = Season_Spatial, xend=Season_Spatial, color=Spatial))+
  geom_point(aes(x = Season_Spatial, y = est)) +
  geom_segment(aes(y = est-lci, yend = est+uci)) +
  ylab(paste("ASV Richness Estimate")) +
  theme_bw() +
  theme(axis.text.x = element_text())+  theme(legend.position="none")+ scale_color_manual(values=colorspatial)+
  theme(axis.text.x = element_text())+  theme(legend.position="none")+xlab("")+theme(axis.title=element_text(size=8,face="bold"), axis.text.x = element_text()) + theme(legend.position = "none")+ theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))
ASVrich
dev.off()
ggsave("Season_Spatial_ASV_richness_Tucker2021.pdf",width=5, height=2)



svg("by_abundance_seasonal_genera_with_diversity.svg", width=6, height=8)
plot_grid(shannons,ASVrich, ca, cb,tb,ob,cc,tc,oc,labels = c('a', 'b', 'c', 'd','e','f','g','h','i'), align = 'hv',axis = "bt",label_size = 12, ncol = 3, nrow=3)
ggsave("by_abundance_seasonal_genera.pdf", width=5, height=8)
dev.off()

```




############
Figure S1
############

```{r}
ok=read.csv("Per_IME.csv")
library(tidyverse)
ok2=ok%>%reshape2::melt(measure.vars=c("AR.STO1","SR8.STO1","CB.STO1","HP1.STO1","NB.STO1"        ,"SB.STO1","KAHO.STO1","WAI2.STO1","WAI2.NTO1", "AR.NTO1", "SR8.NTO1" ,"CB.NTO1" , "HP1.NTO1"     ,"NB.NTO1", "SB.NTO1" ,  "KAHO.NTO1"))


ok3=ok2%>%drop_na()%>%summarise(mean=mean(value), sd=sd(value), max=max(value), min=min(value))
svg("IME_Prevalence_v2.svg")
ggplot(ok2, aes(x=as.Date(Date1, format="%m/%d/%y"), y=value))+geom_point()+xlab("Date")+ylab("Delta chlorophyll a (ug per L)")+ theme(axis.text=element_text(size=10),axis.title=element_text(size=10), legend.text=element_text(size=10), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1))+
  geom_hline(yintercept=0, linetype='dotted', col = 'red')+ggtitle("Prevalence of IME: comparisons of stations in coastal areas and those offshore")
dev.off()
ggsave("IME_Prevalence_v2.pdf")
```


############
Figure S2
############

```{r}

enviro_cor=no_dup_metadata_KBytNERR2%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL2",
                                                       "Log_SYN.mL2", 
                                                       "Log_PEUK.mL2", 
                                                       "Log_HBACT.mL2", 
                                                        "SW.Temperature.at.site", "Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"))

###"Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"

no_dup_metadata_KBytNERR$Date1

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))

?corrplot()
??cor()

svg("All_wnut_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off
```


############
Figure S3
############
```{r}

igh=no_dup_metadata_KBytNERR%>%dplyr::select(Date1,Site, Spatial, Season_Tucker2021, chla.ug.per.L)
library(reshape2)
colorssite2 <- c("WAI2"='#000000', "KAHO"='#f94279',"AR"="#967969" ,"HP1"="#ffd000", "NB"="#ff8c00","SB"="#C47F2B","CB"="#ad0201ff","SR8"="#D7B4F3", "SR2"="#AAFF00", "NR2"="#049861", "STO1"='#00FFFF', "NTO1"='#0045ff')


colorspatial=c("Offshore"="dodgerblue3",
"Transition" ="#22C959",
"Coastal"="#B70D6A")

Site_order=factor(igh$Site,levels=c("WAI2", "KAHO",  "AR", "HP1","SB","CB", "NB","SR8", "SR2", "NR2", "STO1", "NTO1"))

svg("time_series_chla.svg", height=3.5, width=6.5)
ggplot(igh, aes(x = Site_order, y = as.Date(Date1, format="%m/%d/%y"))) + 
      geom_point(aes(size = chla.ug.per.L, fill = Spatial),  shape = 21) + 
       scale_size_continuous(limits = c(0.1, 20), range = c(1,10), breaks = c(0.1,0.5,1,5,10)) +
    scale_fill_manual(values=colorspatial) + theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=9)) + coord_flip() +
  theme(panel.border = element_rect(color = "black",
                                    fill = NA,
                                    size = 9))+  theme(legend.position = "bottom") +# minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1, 
hjust = -0.05, size=12)) + theme(legend.title = element_blank()) + theme(axis.text=element_text(size=12),axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title=element_text(size=12,face="bold"), legend.text=element_text(size=12), panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
panel.background = element_blank(),panel.border = element_rect(color = "black",fill = NA,size = 1),legend.position="bottom")+xlab("Station")+ylab("Date")
dev.off()
ggsave('Time_Series_Enviro_Chla_v4.pdf', width=8.5, height=4)

```


############
Figure S4
Was completed in word & analyses in the metagenomics section
############


############
Figure S5
############

```{r}

no_dup_metadata_KBytNERR=read.csv("no_dup_metadata_KBytNERRv14.csv", header=TRUE, fill=TRUE)


test=no_dup_metadata_KBytNERR%>%subset(Spatial=="Coastal")
ggplot(test, aes(Max.Max.Wind, chla.ug.per.L))+geom_point()

#"Rainfall.mm.7.day","Wind.Speed.m.s.7.day",
test=no_dup_metadata_KBytNERR%>%subset(Spatial=="Coastal")
ggplot(test, aes(SW.Temperature.at.site, chla.ug.per.L))+geom_point()+ylim(0,5)+geom_smooth()


test=no_dup_metadata_KBytNERR%>%subset(Spatial=="Coastal")
ggplot(test, aes(Max.Max.Wind, chla.ug.per.L))+geom_point()

test=no_dup_metadata_KBytNERR%>%subset(Site=="HP1")

rain=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), Rainfall.mm.7.day)) +
  geom_point()+ geom_line()+theme(axis.text.x=element_blank(), axis.title.x=element_blank())+ylab("Rainfall \n (mm)")+xlab("Date")
rain

temp=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), SW.Temperature.at.site)) +
  geom_point()+ geom_line()+theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), axis.title.x=element_blank())+ylab("Temperature \n (C)")+xlab("Date")
temp

wind=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), Wind.Speed.m.s.7.day)) +
  geom_point()+ geom_line()+theme(axis.text.x=element_blank(), axis.title.x=element_blank())+ylab("Wind speed \n (m/s)")+xlab("Date")
wind
test=no_dup_metadata_KBytNERR%>%subset(Spatial=="Coastal")
chla=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), chla.ug.per.L)) +
  geom_point(aes(color=Site))+theme(axis.text.x = element_blank(), axis.title.x=element_blank())+scale_y_continuous(trans = log10_trans())+scale_color_manual(values=colorssite2)+
  theme(legend.position="none")+ylab("Chlorophyll a \n (ug/L)")+xlab("Date")
chla


chla2=ggplot(test, aes(x=as.Date(Date1, format="%m/%d/%y"), chla.ug.per.L)) +
  geom_point(aes(color=Site))+theme(axis.text.x = element_blank(), axis.title.x=element_blank())+ylim(c(0,4))+scale_color_manual(values=colorssite2)+
  theme(legend.position="none")+ylab("Chlorophyll a \n (ug/L)")+xlab("Date")
chla2

library(cowplot)
svg("chla_weather_correct.svg", width=5, height=8)
plot_grid(chla,chla2,wind,rain,temp, labels = c('A)', 'B)', "C)", "D)", "E)"), align = 'hv',label_size = 10, ncol = 1)
dev.off()
ggsave("chla_wind_v2_nofeb.pdf", width=5, height=8)

```

####
Figure S6
#######

```{r}
###repeat with transition
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Offshore")%>%dplyr::select(c("SampleID", "Salinity","Log_Chla","Log_plus_PRO.mL2",
                                                       "Log_SYN.mL2", 
                                                       "Log_PEUK.mL2", 
                                                       "Log_HBACT.mL2", 
                                                        "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day"))

###"Log_Silicate", "Log_Phosphate", "Log_NitrateNitrite"

no_dup_metadata_KBytNERR$Date1

enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
## leave blank on non-significant coefficient
## add all correlation coefficients
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))

?corrplot()
??cor()

svg("Offshore_Chla_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off
```

####
Figure S7
#######

```{r}
enviro_cor=no_dup_metadata_KBytNERR2%>%subset(Spatial=="Coastal")%>%dplyr::select(c("SampleID","Log_Chla", "SW.Temperature.at.site", "Log_plus_Wind.Direction.dg.mean", "Log_plus_Rainfall.mm.7.day","Wind.Speed.m.s.7.day", "Fuco.Rel", "X19But.Rel", "X19Hex.Rel", "Allo.Rel", "Log_plus_Lut.Rel", "Log_Peri.Rel", "Prasino.Rel", "Zea.Rel", "Log_Dva.Rel"))



enviro_cor2=enviro_cor%>%drop_na()


sapply(enviro_cor2, class)
check_cor=enviro_cor2 %>% mutate_if(is.factor,as.numeric)
check_cor=enviro_cor2%>% mutate_if(is.integer,as.numeric)
sapply(check_cor, class)
check_cor2=check_cor[,-1]


library(corrplot)
M = cor(check_cor2, method="p")
testRes = cor.mtest(check_cor2, conf.level = 0.95)
try=unlist(testRes)
write.csv()
## leave blank on non-significant coefficient
## add all correlation coefficients
svg("numbers.svg", height=10, width=10)
corrplot(M, p.mat = testRes$p, method = 'circle', type = 'lower', insig='blank',col = rev(COL2('RdBu', 10)),
         order = 'AOE', diag = FALSE)$corrPos -> p1
text(p1$x, p1$y, round(p1$corr, 2))
dev.off()
?corrplot()
??cor()

svg("HPLC_Coastal_Chla_11252022_log.svg", height=6, width=6)
corrplot(M, p.mat = testRes$p, method = 'color', diag = FALSE, type = 'upper',
         sig.level = c(0.001, 0.01, 0.05), pch.cex = 0.9,
         insig = 'label_sig', pch.col = 'grey20', order = 'AOE', col = rev(COL2('RdBu', 10)))

dev.off

```


Summary Statistics 

```{r}

##The following code was used to evaluate summary statisitcs of different parameters over time and space

library(multcomp)

no_dup_metadata_KBytNERR$Cluster_DeSeq_Spatial=as.factor(no_dup_metadata_KBytNERR$Cluster_DeSeq_Spatial)

no_NA=no_dup_metadata_KBytNERR%>%drop_na(Cluster_DeSeq_Spatial)%>%drop_na(Ammonia)
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(Ammonia~ Cluster_DeSeq_Spatial, data = no_NA)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Cluster_DeSeq_Spatial= "Tukey"))
  summary(g, test = adjusted("holm"))
  
  
  
Transition=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Transition")
Coastal=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Coastal")
Offshore=no_dup_metadata_KBytNERR%>%subset(Cluster_DeSeq_Spatial=="Offshore")


Coastal$Season_Aka=as.factor(Coastal$Season_Aka)
Transition$Season_Aka=as.factor(Transition$Season_Aka)
Offshore$Season_Aka=as.factor(Offshore$Season_Aka)

no_NA=Offshore%>%drop_na(Cluster_DeSeq_Spatial)%>%drop_na(Ammonia)
### multiple comparison procedures
  ### set up a one-way ANOVA
  amod <- aov(Ammonia~ Season_Aka, data = no_NA)
  amod

  ### set up all-pair comparisons for factor `tension'
  ### using a symbolic description (`type' argument 
  ### to `contrMat()')
  g=glht(amod, linfct = mcp(Season_Aka= "Tukey"))
  summary(g, test = adjusted("holm"))
```

